---
title: "Complete Guide for Seven Bridges API R Client"
output:
  BiocStyle::html_document:
    toc: true
    number_sections: true
    highlight: haddock
    css: style.css
---

<!--
%\VignetteIndexEntry{Complete Guide for API R Client}
%\VignettePackage{sevenbridges}
%\VignetteEngine{knitr::rmarkdown}
-->

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown(css.files = "custom.css")
```

<!-- override white background for classless code chunks -->
<style type="text/css" scoped>
  pre:not([class]) { background-color: #F8F8F8; }
  pre code { background-color: transparent; }
</style>

```{r code, echo = FALSE}
code <- function(...) {
    cat(paste(..., sep = "\n"))
}

code2 <- function(...) {
    cat(paste("```markdown", ..., "\n", "```", sep = "\n"))
}
```

```{r global_options, include=FALSE}
library(BiocStyle)
knitr::opts_chunk$set(eval = FALSE)
```

# Introduction

This package is designed to support as many Seven Bridges supported platforms 
as possible, including the NCI Cancer Genomic Cloud Pilot developed by Seven Bridges, as long as you provide the correct API URL, it should work normally. 

Currently tested platform including

- [Cancer Genomics Cloud Platform](http://www.cancergenomicscloud.org/)
- [Seven Bridgs US platform on AWS](https://www.sbgenomics.com/)
- [Seven Bridgs US platform on Google Cloud](https://gcp.sbgenomics.com/)


It will be helpful to read complete platform documentation 

- [Full doc for Cancer Genomics Cloud Platform](http://docs.cancergenomicscloud.org/docs)
- [Full doc for Seven Bridgs US platform](https://docs.sbgenomics.com/display/developerhub/Seven+Bridges+Genomics+Developer+Hub)

And it's API part if you want to understand details and use its full power:

- [API doc for Cancer Genomics Cloud Platform](http://docs.cancergenomicscloud.org/docs/the-cgc-api)
- [API doc for Seven Bridgs US platform](https://docs.sbgenomics.com/display/developerhub/API)

## API V1

Plesae note, `sevenbridges` package doesn't support old API, please use `sbgr` package instead if you need to operate on old project type. `sbgr` package will be under maintainance mode. 

## API V2 and beyond

Current public v2 API

- Cancer Genomics Cloud : https://cgc-api.sbgenomics.com/v2/
- Seven Bridges on AWS: https://api.sbgenomics.com/v2/
- Seven Bridges on Google: https://gcp-api.sbgenomics.com/v2/


`sevenbbridges` package only support V2 and later API, which support CWL compatible project. 

## Installation

This package is in devel branch on bioconductor now, to install the package.

```{r}
source("http://bioconductor.org/biocLite.R")
useDevel(devel = TRUE)
biocLite("sevenbridges")
```

To install the package from github

```{r}
# install.packages("devtools") if devtools was not installed
library("devtools")
install_github("tengfei/sevenbridges")
```

This package is also availabe in docker image `tengfei/sevenbridges`, which the latest is build on `bioconductor/devel_base` on dockerhub.

Checkout he _Dockerfile_ at 

```{r}
system.file("docker", "Dockerfile", package = "sevenbridges")
```



# Quickstart

If you want to understand details how to use the API client in R, please go for the second section for complete guide. This section, I am going to use a simple example for a quick start.


## Create Auth object

Set up the `Auth` object, it remembers your auth token and url, every action started from this object. 

You have three different ways to setup the token. 

1. configuration file in home folder
2. option in R session
3. direct setup

The first one is most safe and easy way, you don't save your token in your script. Please check out the complete guide section for more details. 

when you provide url, please don't forget `/` in the end. 

```{r}
library(sevenbridges)
## direct setup
a <- Auth()
a <- Auth(token = "aef7e9e3f6c54fb1b338ac4ecddf1a56")
a
a <- Auth(token = "1c0e6e202b544030870ccc147092c257",
          url = "https://cgc-api.sbgenomics.com/v2/")
## or load default, deafult url is cgc  API, more safe
a <- Auth(platform = "cgc", username = "tengfei")
a
```

## About a user

```{r}
a$user()
```

## Rate limit

```{r}
a$rate_limit()
```

## Show billing information
Billing information, every project is associted with a billing group

```{r}
## check your billing info
a$billing()
## check invoices
a$invoice()
```

## Create a project
Create a new project called "api testing", with the billing group id. 

```{r}
## get billing group id
bid <- a$billing()[[1]]$id
## create new project
a$project_new(name = "api testing", bid, description = "Just a testing")
## confirm the new created project
(p <- a$project("api testing", detail = TRUE))
```

## Import CWL app and run a task
Then we want to run a task, and you want to upload your json file that describe your tool, here we use an randome number generator example saved in this package. 

Note: alternatively you can directly describe your CWL tool in R with this pacakge, please read another vignettes on "Describe CWL Tools/Workflows in R and Execute it on Seven Bridges Supported Platform"

```{r}
## Add an CWL file to your project
fl.runif <- system.file("extdata", "runif.json", package = "sevenbridges")
## name your app 
p$app_add("runif_draft", fl.runif)
## check it out and get app id
(aid <- p$app("runif")$id)
```

Draft your task, you need to specify

- name of the task
- descfiption
- app id
- inputs of your task: you need to know about the app you are running, in this case, the cwl app accept 4 parameters (number, min, max, seed). 


I want to generate 1 number (default) between 1 and 10. 

```{r}
(tsk <- p$task_add(name = "Draft runif 3", 
           description = "Description for runif", 
           app = aid,
           inputs = list(min = 1, max = 10)))

## confirm, show all task status is draft
p$task(status = "draft")
```

Now you have your draft task, you can delete your draft task or update it.

```{r}
## not run
## tsk$delete()
```

looks like there is only a single number, oops, I want 100 numbers, so 
let me update my draft task 

```{r}
tsk <- p$task(status = "draft")
tsk <- p$task(id = "504de3b7-4a8f-4755-a039-e1ec12ddd3fa")
tsk$update()
tsk$getInputs()
## missing number input, only update number
tsk$update(inputs = list(number = 300))
```

Or we just want to run it in the cloud!

```{r}
## Run your task
tsk$run()
```
To monitor the task, you can always call `update` on task object to check the status. 

```{r}
tsk$update()
```

Or more fun, you can monitor a running task with hook function, so trigger a function when that status is "compeleted", "running" etc, please check the 
details in section about hook of task. 

By default it just show message when the task is completed. 

```{r}
## Monitor your task (skip this part)
## tsk$monitor()
```

To about running task just  all

```
## not run
## tsk$abort()
```

To download all files from a completed tasks

```{r}
tsk$download("~/Downloads")
```

More fun to set task hook, so when it's complete download the files

```{r}
setTaskHook("completed", function(){
    tsk$download("~/Downloads")
})
tsk$monitor()
```





# Low level API 

- You can always use "httr"" package to explore API at full capabiity. This pacakge provides functions like `POST`, `GET`, `DELETE`, `PUT`, `PATCH` and other utitlies, I recommend for advanced R users. This pacakge is also "sevenbridges" package dpends on.
- you can use single `api()` function in "sevenbridges".
- you can use `Auth$api()` call to save your time retype url and token. 

__However__ I recommend you use easy API that is introduced for rest of the guide. 

# User-friendly API

This is what the package try to help, and a much more user-friendly interface that we suggest our users to use, so you don't have to combine several `api()` calls and refer to the API documentation all the times to finish a simple task. 


## Authentification

### Set up default token for different platforms

You can create a file called '.sbg.auth.yml' in your home folder, and maintain multiple account for a list of platforms, including private or public ones. 

```
us:
  url: https://api.sbgenomics.com/v2/
  user:
    tengfei:
      token: fake_token
    yintengfei:
      token: fake_token
cgc:
  url: https://cgc-api.sbgenomics.com/
  user:
    tengfei:
      token: fake_token
gcp:
  url: https://gcp-api.sbgenomics.com/v2/
  user:
    tengfei:
      token: fake_token
```

When you load sevenbridges package, it will first try to parse your token configuration file first into an options list. 

```{r}
## show all 
getToken()
## show all pre-set user token for platform 
getToken("cgc")
## show individual token for a user
getToken(platform = "cgc", username = "tengfei")
```



### Create Auth object

First thing first, you need to construct an Auth object, everything begins with
this object, it stores

- The authentification token
- The API URL
- The platform (US platform, Cancer Genomics Cloud etc), this is optional, will translate into API url.

The logic is like this

1. If you didn't pass url or token, we think you are loading from config file
2. if no platform or user provided, will use the first item in your token config file, __this is not recommended, at least provide platform/username set.__ 


```{r}
library(sevenbridges)
## direct setup
a <- Auth(token = "1c0e6e202b544030870ccc147092c257",
          url = "https://cgc-api.sbgenomics.com/v2/")
## or load default, deafult url is cgc  API, more safe
Auth()
Auth(platform = "us")
Auth(platform = "us", username = "yintengfei")

```

By default it points to Cancer Genomics Cloud platform, unless you specify

- API URL (more flexible)
- or Platform (currently support 'CGC', 'US')


```{r}
## NOT RUN
## specify platform
a <- Auth("4b6068016ab746c6bf6401ae69b80e1f", platform = "us")
## or specify the url
a <- Auth("4b6068016ab746c6bf6401ae69b80e1f",
    url = "https://cgc-api.sbgenomics.com/v2/")

```

Note: when you construct the Auth object, make sure you input the correct platform or 
API url for your authentifiation, where you get it from. 

For the tutorial about how to get your authentifiation, please check

- [Get token from US platform](https://docs.sbgenomics.com/display/developerhub/Authentication+Token)
- [Get token from Cancer Genomics Cloud Platform](http://docs.cancergenomicscloud.org/docs/the-cgc-api)

## List All API calls

If we didn't pass any parameters to api() from Auth, it will list all API calls,
and anything parameter we provided will pass on to api() function, but you don't 
need to input token and url again! The Auth object will know that information already.\
And this call from Auth object will check the response too. 

```{r}
a$api()
```

## Rate Limits

This call returns information about your current rate limit. 
This is the number of API calls you can make in one hour.

```{r}
a$rate_limit()

```

## Users

This call returns a list of the resources, such as projects, billing groups, and 
organizations, that are accessible to you. If you are not an administrator, this call will only return a successful response if {username} is replaced with your own username. If you are an administrator, you can replace {username} with the username of any CGC user, to return information on their resources.

_Case sensitivity_: Don't forget to capitalize your username in the same way as you set it when you registered on the CGC.

If you don't provide a username, your user information will be shown.

```{r}
## return your information
a$user()
## return user 'tengfei''s information
a$user("tengfei")
```


## Billing Group and Invoices

### For billing

if no id provided, This call returns a list of paths used to access billing information via the API. else, This call lists all your billing groups, including groups that are pending or have been disabled. if `breakdown = TRUE`, This call returns a breakdown of spending per-project for the billing group specified by billing_group. For each project that the billing group is associated with, information is shown on the tasks run, including their initiating user (the runner), start and end times, and cost.


```{r}
## return a BillingList object
b <- a$billing()
a$billing(id = b[[1]]$id)
a$billing(id = b[[1]]$id, breakdown = TRUE)
```

### For invoices

If no id provided, This call returns a list of invoices, with information about each, including whether or not the invoice is pending and the billing period it covers. The call returns information about all your available invoices, unless you use the query parameter bg_id to specify the ID of a particular billing group, in which case it will return the invoice incurred by that billing group only. if id provided, This call retrieves information about a selected invoice, including the costs for analysis and storage, and the invoice period.

```{r}
a$invoice()
a$invoice(id = "fake_id")
```


Note (TODO): Invoice is not a object yet, it currently just return a list.



## Project Operation

Project is the basic unit to organize different entities: files, tasks, apps, etc. So lots actions comes from this `Project' object. 

### List All Projects 

This call returns a list of all projects you are a member of. Each project's project_id and URL on the CGC will be returned.

```{r}
a$project()
```

Then if you want to list the projects owned by and accessible to a particular user, specify the `owner` argument. Each project's ID and URL will be returned.

```{r}
a$project(owner = "yintengfei")

## unfixed bugs
sevenbridges:::status_check(api(token = a$token, base_url = a$url, path = "projects/Rfranklin", method = "GET"))
```

To get details about a project, use `detail = TRUE`

```{r}
a$project(detail = TRUE)
```

### Partial Match Project Name

For more friendly interface and conventient search, we support partial name match in this 
interface. The first argument for the call is "name", 
users can provide part of the name and we do a search for you automatically. 

```{r}
## want to return a project called
a$project("hello")
```

### Create a New Project

To create a new project, user need to specify

- name (required)
- billing_group_id (required)
- description (optional)
- tags (optional): this has to be a list(), only if you are "TCGA" user, you can 
create TCGA project by passing tags list("tcga")
- type (optional): by default, we are creating a cwl project "v2"



```{r}
a$project_new("api_testing", a$billing()[[1]]$id)
a$project_new("api_testing_tcga", a$billing()[[1]]$id,
              description = "Test for API", tags = c("tcga"))
```

### Delete a Project

Next we delete what we created for testing, only *single* project could be deleted now, so
please pay attention to the returned object from `a$project()`, sometimes if you are using partial matching by name, it will return a list and we don't support this operation on a list of project yet. 

```{r}
## remove it
a$project("api_testing")$delete()
## check
a$project("api_testing")
```

### Update/Edit a Project

You can update information about an existing project, including

- name
- description
- billing_group

```{r}
a$project(id = "tengfei/helloworld")
a$project(id = "tengfei/helloworld")$update(name = "Hello World Update", 
                                                    description = "Update description")

```

### Project Member

#### List members

This call returns a list of the members of the specified project. For each member, the response lists:

- The member's username on the CGC
- The member's permissions in the project specified

```{r}
a$project(id = "tengfei/demo-project")$member()
```

#### Add a member
This call adds a new user to a specified project. It can only be successfully made by a user who has admin permissions in the project.

Requests to add a project member must include the key permissions. However, if you do not include a value for some permission, it will be set to false by default.

Set permission by passing: copy, write, execute, admin, read argument. 

Note: read is implicit and set by default, you can not be project member without having read permission

```{r}
a$project(id = "tengfei/demo-project")$member_add(username = "Controlled_Data_User")
```

### Update a member

This call edits a user's permissions in a specified project. It can only be successfully made by a user who has admin permissions in the project.

```{r}
a$project(id = "tengfei/demo-project")$member(username = "Controlled_Data_User")
a$project(id = "tengfei/demo-project")$
    member(username = "Controlled_Data_User")$
    update(copy = TRUE)
a$project(id = "tengfei/demo-project")$member(username = "Controlled_Data_User")
```

#### Delete a member

To delete an existing member, just to call `delete()` action on `Member` object.

```{r}
a$project(id = "tengfei/demo-project")$member(username = "Controlled_Data_User")$delete()
## confirm
a$project(id = "tengfei/demo-project")$member()
```

### List all Files

To list all files belongs to a project simple use

```{r}
a$project(id = "tengfei/demo-project")$file()
```
## Files and Metadata

### List all files

This call returns a list of all files in a specified project that you can access. For each file, the call returns:

- Its ID
- Its filename

The project is specified as a query parameter in the call.

```{r}
a$file(project = a$project(name = "hello")$id)
a$file("omni", project = a$project(name = "hello")$id, detail = TRUE)
```

However we recommend user use cascading way to list files. It's the 'style' we 
are using through all the tutorial. 

```{r}
a$project("hello")$file()
```

To get details about files, please use `detail = TRUE` in the call.

```{r}
a$project("hello")$file(detail = TRUE)
```

### Copy a file or group of files

This call copies the specified file to a new project. Files retain their metadata when copied, but may be assigned new names in their target project.

Note that Controlled Data files may not be copied to Open Data projects. To make this call, you should have copy permission within the project you are copying from.

Let's try to copy a file from CGC public files, the id you can tell from the url is 
"561e1b33e4b0aa6ec48167d7"

You must provide 

- `id` file id, or list/vector of files ids. 
- `project` parameter: project id. 
- `name` is optional, if omitted, use the same.

```{r}
## 1000G_omni2.5.b37.vcf
fid <- "561e1b33e4b0aa6ec48167d7"
fid2 <- "561e1b33e4b0aa6ec48167d3"
pid <- a$project("demo")$id
a$copyFile(c(fid, fid2), project = pid)
a$project(id = pid)$file()
```

NOTE: to copy a group of files, you need `Auth$copyFile()` interface. The id of those fies in your project iwll be different from public id. 

Alternatively you can do single file copy 

```{r}
a$project("hello")$file(id = fid)$copyTo(pid)
```

### Delete a file

Note: the `delete` action only work for single file now, make sure your `file` call return 
a single file not a file list.

```{r}
a$project("demo")$file(id = fid)$delete()
## confirm the deletion
a$project("demo")$file()
```

You can also delete a group of files or `FilesList` object, __be careful__ with this function! 

```{r}
## return 5 files
a$project("demo")$file("phase1")
## delete all of them
delete(a$project("demo")$file("phase1"))
a$project("demo")$file("phase1")
```

### Download files

To get the download information, basically a url, please use

```{r}
a$project("demo")$file()[[1]]$download_url()
```

To download directly from R, use `download` call directly from single File object.

```{r}
fid3 <- a$project("demo")$file()[[1]]$id
a$project("demo")$file(id = fid3)$download("~/Downloads/")
```

I also created `download` function for `FilesList` object to save your time

```{r}
fid3 <- a$project("demo")$file()[[1]]$id
a$project("demo")$file(id = fid3)$download("~/Downloads/")
```

To download all files from a project.

```{r}
a$project("demo")$download("~/Downloads")
```

### Upload files

TODO: currently only v1 supported.

Seven Brdiges platforms provide couple different ways for data import

- command line uploader
- graphic UI uploader
- from ftp, http etc from interface directly

Here an upload function is implemented directly in R for easy usage. 

You need to specify the project you need first. 

```{r}
fl <- system.file("extdata", "sample1.fastq", package = "sevenbridges")
fl
a$project("demo")$upload(fl)
library(httr)
ufo <- upload_file(fl)
POST()

```

### Public files

The only way you can list public files now via API is from `Auth$file()` call, and for now, 
the project id is "admin/sbg-public-data". This may be updated later. Alternative, just click the file from our GUI, you will see the id in the url link.

```
a$file(project = "admin/sbg-public-data")
```

### Update a file

You can call `update()` function from Files object, following things could be updated

- name
- metadata (list): this is going to overwrite all meta for the file, so please provide the full list. For more flexible operation, please check next section about Metadata. 

If no parameters provided, will just get detail for the same file and update the object
itself. 

```
fl <- a$project("demo")$file()[[1]]
## show metadata
fl$meta()
fl$update(name = "newb37.vcf", metadata = list(new_item1 = "item1", new_item2 = "item2"))
## check it out
fl
```


### Metadata Operation

A full list of metadata fields and their permissible values on the CGC is available on the page [TCGA Metadata](http://docs.cancergenomicscloud.org/v1.0/docs/tcga-metadata-on-the-cgc).
 
Note that the file name is not the same as its ID. The ID is a hexadecimal string, automatically assigned to a file in a project. The file's name is a human-readable string. For information, please see the API overview.


To get metdata for a file call `meta()`. 

```{r}
## meta is pulling the latest information via API
fl$meta()
## field meta data saved the previous saved one
fl$metadata
```


Although CGC defined a set of meta schema, which is visible on the UI of the platform, but you can pass any free form of meta for the file, it's just not visible on UI, but it's stored with the data. 

Only the value being specfied stored with files, to set metadata please call `setMeta()` from
Files object. 

__Important__:

- By default, we are not overwriting the meta field using `setMeta` call, unless you pass
the `overwrite = TRUE` argument

```{r}
fl$setMeta(new_item3 = "item3")
fl
## oops it removed rest of the meta
fl$setMeta(new_item4 = "item4", overwrite = TRUE)
fl
```

Let's keep playing with meta, if you are really interested in the default schema that shon on the UI, you can use `Metadata()` constructor and check details of each meta; Simply call the function (name of meta), it will show description and enumerated items. Please pay attention to `suggested_values` field.

```{r}
## check which schema we have
Metadata()$show(full = TRUE)
## check details for each, play with it
platform()
paired_end()
quality_scale()
```

To construct the Metadata, we encourage you use `Metadata()` directly, pass metadata directly into the call, it will do the validation. 

```{r}
m <- Metadata(platform = "nanopore", quality_scale = "solexa", new_item = "newnew")
m
```

## App

From now on we are going to have fun with Apps! The CWL(Common Workflow Language) based 
approach. It gets more and more popular and really designed for reproducible pipeline description and execution. All Seven Bridges platforms support cwl natively in the cloud. 
So in this section, I will introduce how we are going to do this via API and inside R. 

### List all apps 

This call lists all the apps available to you.

```{r}
a$app()
## or show details
## FIXME: detail is long
a$app(detail = TRUE)
```

To search a name, please pass a pattern for the `name` argument; or provide a unique `id`.
```{r}
## pattern match
a$app(name = "STAR")
## unique id
aid <- a$app()[[1]]$id
aid
a$app(id = aid)
## get a specific revision from an app
a$app(id = aid, revision = 0)
``` 



To list all apps belong to one project use `project` argument

```{r}
## my favorite, always
a$project("demo")$app()

## or alternatviely
pid <- a$project("demo")$id
a$app(project = pid)
```

To list all public apps, use `visibility` argument

```{r}
## show 100 items from public
a$app(visibility = "public", limit = 100)
```

TODO: To search an app cross all published apps (this may take a while)
when limit is NULL, show everything. 

```{r}
a$app(visibility = "public")
```

### Copy an App

This call copies the specified app to the specified project. The app should be one in a project that you can access; this could be an app that has been uploaded to the CGC by a project member, or a publicly available app that has been copied to the project.

Need two arguments

- project: id character
- name: optional, to re-name your app

```{r}
aid <- a$app(visibility = "public")[[1]]$id
aid
a$copyApp(aid, project = pid, name = "copy-rename-test")
## check it is copied
a$app(project = pid)
```

### Get CWL from an App

This call returns information about the specified app, as raw CWL. The call differs from the call to GET details of an app by returning a JSON object that is the CWL.

The app should be one in a project that you can access; this could be an app that has been uploaded to the CGC by a project member, or a publicly available app that has been copied to the project.

To get a specific revision, pass `revision` argument. 

```{r}
ap <- a$app(visibility = "public")[[1]]
a$project("demo")$app("index")
## get a specific revision
a$project("demo")$app("index", revision = 0)

```

TODO: convert it to an CWL object

### Add CWL as an APP

Use `app_add` function call from a `Project` object, two parameteres requried

- short_name: a short id for your app, alphanumeric character, no spacings; this not name field. 
- filename: you json file for cwl.

```{r}
cwl.fl <- system.file("extdata", "bam_index.json", package = "sevenbridges")
a$project("demo")$app_add(short_name = "new_bam_index_app", filename = cwl.fl)
a$project("demo")$app_add(short_name = "new_bam_index_app", revision = 2, filename = cwl.fl)
```



### Directly Describe CWL in R

This is fun and is introduced in another vignette. 


## Task Operation

### List tasks

This call returns a list of tasks that you can access. You are able to filter by status

```{r}
## all tasks
a$task()
## filter
a$task(status = "completed")
a$task(status = "running")
```

To list all tasks in a project

```{r}
## FIXME;
a$project("demo")$task()
pid <- a$project("demo")$id
pid
a$task(project = pid)
```

### Create a draft task

To create a draft, you need to call the `task_add` function from Project object. And you need to pass following arguments

- name: name for this task
- description: description for this task
- app: app id you have access to
- inputs: inputs list for this task

```{r}
## push an app first
fl.runif <- system.file("extdata", "runif.json", package = "sbgr")
a$project("demo")$app_add("runif_draft", fl.runif)
runif_id <- "tengfei/demo-project/runif_draft"
## create a draft task
a$project("demo")$task_add(name = "Draft runif 3", 
                           description = "Description for runif 3", 
                           app = runif_id,
                           inputs = list(min = 1, max = 10))
## confirm
a$project("demo")$task(status = "draft")
```

### Modify a task

Call `update` function fro a Task object, you can update

- name
- description
- inputs list (only update items you provided.)

```{r}
## get the single task you want to update
tsk <- a$project("demo")$task("Draft runif 3")
tsk
tsk$update(name = "Draft runif update", description = "draft 2", 
           inputs = list(max = 100))
## alternative way to check all inputs
tsk$getInputs()
```

### Run a task

This call runs (executes) the specified task. Only tasks whose status is "DRAFT" may be run.

```{r}
tsk$run()
## run update without information just return latest information
tsk$update()
```

### Monitor a running task and set function hook

To monitor a running task, call `monitor` from a task object

- first arugent set interval time to check the status
- rest arugments might be used for hook function



```{r}
tsk$monitor()
```

get and set default hook function for task status, currently failed, completed taskss
will break the monitoring. 

Note: Hook functio has to return `TRUE` (break monitoring) or `FALSE` (continuing) in the end. 

```{r}
getTaskHook("completed")
getTaskHook("draft")
setTaskHook("draft", function(){message("never happens"); return(TRUE)})
getTaskHook("draft")
```

### Abort a runing task

This call aborts the specified task. Only tasks whose status is "RUNNING" may be aborted.

```{r}
## abort
tsk$abort()
## check
tsk$update()
```



### Delete a task

Note that you can only delete draft tasks, not running tasks.

```{r}
tsklst <- a$task(status = "draft")
## delete a single task
tsklst[[1]]$delete()
## confirm
a$task(status = "draft")
## delete a list of tasks
delete(tsklst)
```

### Download all files from a completed task

```
tsk$download("~/Downloads")
```


## Get Raw Response from httr

In easy API, we return an object which contains the raw response from httr as a field,
you can either call `response()` on that object or just get the field out of it

## Batch operation on project/files/tasks

Right now,  users have to use `lapply` to do those operations themsevles. It's simple implementation. 

In this package, we impelemnt `delete` and `download` for some object like task and project or file. 

# Cheatsheet

Quick cheet sheet (in progress)

```{r}
## Authentification
getToken()
a <- Auth(token = token)
a <- Auth(token = token, 
          url = "https://cgc-api.sbgenomics.com/v2/")
a <- Auth(platform = "us", username = "tengfei")

## list API
a$api()

## Rate limits
a$rate_limit()

## Users
a$user()
a$user("tengfei")

## billing
a$billing()
a$billing(id = , breakdown = TRUE)
a$invoice()
a$invoice(id = "fake_id")

## Project
### create new project
a$project_new(name = , billing_group_id = , description = )
### list all project owned by you
a$project()
a$project(owner = "yintengfei")
### partial match
p <-  a$project(name = , id = , exact = TRUE)
### delete
p$delete()
### update
p$update(name = , description = )
### members
p$member()
p$member_add(username = )
p$member(username = )$update(write = , copy = , execute = )
p$memeber(usrname = )$delete()

## file
### list all files in this project
p$file()
### list all public files
a$file(visibility = "public")
### copy
a$copyFile(c(fid, fid2), project = pid)
### delete
p$file(id = fid)$delete()
### download
p$file()[[1]]$download_url()
p$file(id = fid3)$download("~/Downloads/")
### download all
download(p$file())
### update a file
fl$update(name = , metadata = list(a =  ,b = , ...))
### meta
fl$meta()
fl$setMeta()
fl$setMeta(..., overwrite = TRUE)

## App
a$app()
### apps in a project
p$app()
p$app(name, id, revision = )
a$copyApp(aid, project = pid, name = )
### add
p$app_add(short_name = , filename =)

## Task
a$task()
a$task(name = , id = )
a$task(status = )

p$task()
p$task(name = , id = )
p$task(status = )

tsk <- p$task(name = , id = )
tsk$update()
tsk$abort()
tsk$run()
tsk$download()
tsk$detele()
tsk$getInputs()
tsk$monitor()

getTaskHook()
setTaskHook(statis = , fun  =)


```



