---
title: "Tutorial: Make the Best of Bioconductor Workflows"
output:
  BiocStyle::html_document:
    toc: true
    number_sections: true
    highlight: haddock
---

<!--
%\VignetteIndexEntry{Tutorial: make the best of bioconductor workflows}
%\VignettePackage{sevenbridges}
%\VignetteEngine{knitr::rmarkdown}
-->

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown(css.files = "custom.css")
```



```{r}
library("airway")
dir <- system.file("extdata", package="airway", mustWork=TRUE)
set.seed(params$seed)
sampleTable <- read.csv(params$design,row.names=1)
head(sampleTable)
library("Rsamtools")
bamfiles <- BamFileList(params$bam, yieldSize=2000000)

library("GenomicFeatures")
gtffile <- file.path(dir,"Homo_sapiens.GRCh37.75_subset.gtf")
txdb <- makeTxDbFromGFF(gtffile, format="gtf", circ_seqs=character())

(ebg <- exonsBy(txdb, by="gene"))

library("GenomicAlignments")
library("BiocParallel")
register(SerialParam())

se <- summarizeOverlaps(features=ebg, reads=bamfiles,
                        mode="Union",
                        singleEnd=FALSE,
                        ignore.strand=TRUE,
                        fragments=TRUE )

(colData(se) <- DataFrame(sampleTable))

library("DESeq2")

dds <- DESeqDataSet(se, design = ~ cell + dex)

countdata <- assay(se)
head(countdata, 3)
coldata <- colData(se)
(ddsMat <- DESeqDataSetFromMatrix(countData = countdata,
                                  colData = coldata,
                                  design = ~ cell + dex))

nrow(dds)
dds <- dds[ rowSums(counts(dds)) > 1, ]
nrow(dds)

rld <- rlog(dds, blind=FALSE)
head(assay(rld), 3)
par( mfrow = c( 1, 2 ) )
dds <- estimateSizeFactors(dds)
plot(log2(counts(dds, normalized=TRUE)[,1:2] + 1),
     pch=16, cex=0.3)
plot(assay(rld)[,1:2],
     pch=16, cex=0.3)
```

