<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Convert App or a CWL JSON file to Tool or Flow object"><title>Convert App or a CWL JSON file to Tool or Flow object — convert_app • sevenbridges</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Convert App or a CWL JSON file to Tool or Flow object — convert_app"><meta property="og:description" content="Convert App or a CWL JSON file to Tool or Flow object"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">sevenbridges</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.19.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/api.html">Complete Guide for Seven Bridges API R Client</a>
    <a class="dropdown-item" href="../articles/apps.html">Describe and Execute CWL Tools/Workflows in R</a>
    <a class="dropdown-item" href="../articles/bioc-workflow.html">Master Tutorial: Use R for Cancer Genomics Cloud</a>
    <a class="dropdown-item" href="../articles/cgc-datasets.html">Find Data on CGC via Data Browser and Datasets API</a>
    <a class="dropdown-item" href="../articles/docker.html">Creating Your Docker Container and Command Line Interface (with docopt)</a>
    <a class="dropdown-item" href="../articles/rstudio.html">IDE Container: RStudio Server, Shiny Server, and More</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/sbg/sevenbridges-r/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Convert App or a CWL JSON file to Tool or Flow object</h1>
      <small class="dont-index">Source: <a href="https://github.com/sbg/sevenbridges-r/blob/HEAD/R/class-app.R" class="external-link"><code>R/class-app.R</code></a></small>
      <div class="d-none name"><code>convert_app.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Convert App or a CWL JSON file to Tool or Flow object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">convert_app</span><span class="op">(</span><span class="va">from</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">appType</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>from</dt>
<dd><p>an App object or a CWL JSON</p></dd>


<dt>x</dt>
<dd><p>a App object</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Tool or Flow object depends on CWL type.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function import CWL JSON file, based on its class: CommandLineTool
or Worklfow to relevant object in R, Tool object or Flow object.</p>
    </div>
    <div class="section level2">
    <h2 id="apptype">appType<a class="anchor" aria-label="anchor" href="#apptype"></a></h2>
    

<dl>
 this function return class of a App object.
</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">tool.in</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/app"</span>, <span class="st">"tool_star.json"</span>, package <span class="op">=</span> <span class="st">"sevenbridges"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">flow.in</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/app"</span>, <span class="st">"flow_star.json"</span>, package <span class="op">=</span> <span class="st">"sevenbridges"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># convert to Tool object</span></span></span>
<span class="r-in"><span><span class="fu">convert_app</span><span class="op">(</span><span class="va">tool.in</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:validationErrors: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:sbgMaintained: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:latestRevision: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:job:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   allocatedResources:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     mem: 60000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     cpu: 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignWindowsPerReadNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMheaderPG: outSAMheaderPG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     GENOME_DIR_NAME: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMatchNminOverLread: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rg_platform_unit_id: rg_platform_unit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     readMapNumber: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignSplicedMateMapLminOverLmate: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignMatesGapMax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMultimapNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     clip5pNbases:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMstrandField: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     readMatesLengthsIn: NotEqual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMattributes: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedMultimapNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rg_mfl: rg_mfl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimSegmentMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     winAnchorDistNbins: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSortingType: SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMultimapScoreRange: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbInsertSave: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     clip3pAfterAdapterNbases:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     twopassMode: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMatchNmin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     twopass1readsN: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMunmapped: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     genome:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       path: genome.ext</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbGTFtagExonParentTranscript: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     limitBAMsortRAM: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignEndsType: Local</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedNoneLociPerWindow: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rg_sample_id: rg_sample</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbGTFtagExonParentGene: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimScoreMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSJfilterIntronMaxVsReadN:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignSplicedMateMapLmin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedSearchStartLmaxOverLread: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSJfilterCountTotalMin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMorder: Paired</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMflagAND: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreDelBase: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignIntronMax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignSJoverhangMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterScoreMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMtype: BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedSearchStartLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreGapGCAG: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreGenomicLengthLog2scale: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterIntronMotifs: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     quantTranscriptomeBan: IndelSoftclipSingleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     reads:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       metadata:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         format: fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         paired_end: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seq_center: illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       path: /test-data/mate_1.fastq.bz2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreGap: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSJfilterOverhangMin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMflagOR: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMmode: Full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbScore: ~</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     winFlankNbins: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimScoreJunctionNonGTAG: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     clip3pAdapterSeq:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - clip3pAdapterSeq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimScoreDropMax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreGapATAC: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rg_platform: Ion Torrent PGM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     clip3pAdapterMMp:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbGTFfeatureExon: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outQSconversionAdd: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignSoftClipAtReferenceEnds: 'Yes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     quantMode: TranscriptomeSAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignIntronMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreInsBase: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMreadID: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreGapNoncan: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedSearchLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSJfilterDistToOtherSJmin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterScoreMinOverLread: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignSJDBoverhangMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     limitOutSJcollapsed: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     winAnchorMultimapNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMismatchNoverLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rg_seq_center: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMheaderHD: outSAMheaderHD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimOutType: Within</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMismatchNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     limitOutSJoneRead: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignTranscriptsPerWindowNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbOverhang: ~</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outReadsUnmapped: Fastx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreStitchSJshift: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedPerWindowNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSJfilterCountUniqueMin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreDelOpen: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbGTFfile:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - path: /demo/test-data/chr20.gtf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     clip3pNbases:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     winBinNbits: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterType: Normal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSJfilterReads: All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimScoreSeparation: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     seedPerReadNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outFilterMismatchNoverReadLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     rg_library_id: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sjdbGTFchrPrefix: chrPrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMprimaryFlag: OneBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     alignTranscriptsPerReadNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     scoreInsOpen: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     chimJunctionOverhangMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     limitSjdbInsertNsj: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outSAMmapqUnique: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:toolAuthor: Alexander Dobin/CSHL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:createdOn: 1453799778</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:categories:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - Alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:contributors:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:links:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: https://github.com/alexdobin/STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Homepage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: https://github.com/alexdobin/STAR/releases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Releases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Manual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: https://groups.google.com/forum/#!forum/rna-star</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Support</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Publication</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:project: sevenbridges/public-apps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:createdBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:toolkitVersion: 2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:id: https://api.sbgenomics.com/v2/apps/sevenbridges/public-apps/star/4/raw/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:license: GNU General Public License v3.0 only</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:revision: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix --sjdbInsertSave</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline ID:1 CN:illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode TranscriptomeSAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp; mv</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:modifiedOn: 1462904151</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:modifiedBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:revisionsInfo:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1453799778</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1453799780</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1453799782</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1462904151</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: sevenbridges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1462904151</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:toolkit: STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> id: https://api.sbgenomics.com/v2/apps/sevenbridges/public-apps/star/4/raw/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Read sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Read sequence.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         != null){\n        if (list[index].metadata.paired_end == 1){\n          left</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \         right = list[index].path\n        }\n      }\n    }\n    \n    if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (list[index].metadata != null){\n        if (list[index].metadata.paired_end</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         \"))\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: readMatesLengthsIn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - NotEqual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Equal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Reads lengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Equal/Not equal - lengths of names, sequences, qualities for both mates</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     are the same/not the same. "Not equal" is safe in all situations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#readMatesLengthsIn'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --readMatesLengthsIn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: NotEqual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Reads to map</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Number of reads to map from the beginning of the file.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#readMapNumber'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --readMapNumber</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Junctions max number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of junctions for one read (including all multi-mappers).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#limitOutSJoneRead'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --limitOutSJoneRead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Collapsed junctions max number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of collapsed junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#limitOutSJcollapsed'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --limitOutSJcollapsed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1000000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Fastx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Output unmapped reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Output of unmapped reads (besides SAM). None: no output; Fastx: output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     in separate fasta/fastq files, Unmapped.out.mate1/2.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outReadsUnmapped'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Quality conversion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Add this number to the quality score (e.g. to convert from Illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     to Sanger, use -31).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outQSconversionAdd'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outQSconversionAdd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMtype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Output format</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Format of output alignments.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMtype'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           SAM_type = $job.inputs.outSAMtype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           SORT_type = $job.inputs.outSortingType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           if (SAM_type &amp;&amp; SORT_type) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             return "--outSAMtype ".concat(SAM_type, " ", SORT_type)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSortingType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Unsorted</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Unsorted SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Output sorting type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Type of output sorting.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSortingType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMmode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - NoQS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: SAM mode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but without</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     quality scores.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMmode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMmode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMstrandField</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - intronMotif</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Strand field flag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Cufflinks-like strand field flag. None: not used; intronMotif: strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     derived from the intron motif. Reads with inconsistent and/or non-canonical introns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     are filtered out.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMstrandField'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMstrandField</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMattributes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - NH</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: SAM attributes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Desired SAM attributes, in the order desired for the output SAM. NH:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM NM MD jM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     jI; None: no attributes.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMattributes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMattributes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMunmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Within</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Write unmapped in SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Output of unmapped reads in the SAM format. None: no output Within:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMunmapped'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMunmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Paired</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - PairedKeepInputOrder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Sorting in SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Type of sorting for the SAM output. Paired: one mate after the other</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for all paired alignments; PairedKeepInputOrder: one mate after the other for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     all paired alignments, the order is kept the same as in the input FASTQ files.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMorder'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Paired</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMprimaryFlag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - OneBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - AllBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Primary alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Which alignments are considered primary - all others will be marked</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     is primary; AllBestScore: all alignments with the best score are primary.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMprimaryFlag'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMprimaryFlag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: OneBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSAMreadID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Read ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Read ID record type. Standard: first word (until space) from the FASTx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     read ID line, removing /1,/2 from the end; Number: read number (index) in the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     FASTx file.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMreadID'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMreadID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: MAPQ value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: MAPQ value for unique mappers (0 to 255).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMmapqUnique'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMmapqUnique</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '255'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: OR SAM flag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Set specific bits of the SAM FLAG.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMflagOR'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMflagOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: AND SAM flag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Set specific bits of the SAM FLAG.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMflagAND'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMflagAND</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '65535'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: SAM header @HD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: '@HD (header) line of the SAM header.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMheaderHD'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMheaderHD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: SAM header @PG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Extra @PG (software) line of the SAM header (in addition to STAR).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSAMheaderPG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSAMheaderPG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Sequencing center</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Specify the sequencing center for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#rg_seq_center'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Library ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Specify the library ID for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#rg_library_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Median fragment length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Specify the median fragment length for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#rg_mfl'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: rg_platform</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - LS 454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Helicos</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - ABI SOLiD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Ion Torrent PGM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - PacBio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Platform</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Specify the version of the technology that was used for sequencing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     or assaying.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#rg_platform'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Platform unit ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Specify the platform unit ID for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#rg_platform_unit_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Sample ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Specify the sample ID for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#rg_sample_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outFilterType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Normal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - BySJout</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Filtering type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Type of filtering. Normal: standard filtering using only current alignment;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     BySJout: keep only those reads that contain junctions that passed filtering into</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     SJ.out.tab.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Normal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Multimapping score range</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: The score range below the maximum score for multimapping alignments.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMultimapScoreRange'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMultimapScoreRange</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max number of mappings</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Read alignments will be output only if the read maps fewer than this</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     value, otherwise no alignments will be output.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMultimapNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max number of mismatches</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Alignment will be output only if it has fewer mismatches than this</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMismatchNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMismatchNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Mismatches to *mapped* length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Alignment will be output only if its ratio of mismatches to *mapped*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     length is less than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMismatchNoverLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMismatchNoverLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0.3'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Mismatches to *read* length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Alignment will be output only if its ratio of mismatches to *read*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     length is less than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMismatchNoverReadLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMismatchNoverReadLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Alignment will be output only if its score is higher than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterScoreMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterScoreMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min score normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: '''Minimum score'' normalized to read length (sum of mates'' lengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for paired-end reads).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterScoreMinOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterScoreMinOverLread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0.66'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min matched bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Alignment will be output only if the number of matched bases is higher</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMatchNmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMatchNmin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min matched bases normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: '''Minimum matched bases'' normalized to read length (sum of mates</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     lengths for paired-end reads).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterMatchNminOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterMatchNminOverLread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0.66'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outFilterIntronMotifs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - RemoveNoncanonical</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - RemoveNoncanonicalUnannotated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Motifs filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     filter out alignments that contain non-canonical unannotated junctions when using</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     annotated splice junctions database. The annotated non-canonical junctions will</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     be kept.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outFilterIntronMotifs'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outFilterIntronMotifs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: outSJfilterReads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Unique</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Collapsed junctions reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Which reads to consider for collapsed splice junctions output. All:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     all reads, unique- and multi-mappers; Unique: uniquely mapping reads only.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSJfilterReads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSJfilterReads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min overhang SJ</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum overhang length for splice junctions on both sides for each</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     of the motifs. To set no output for desired motif, assign -1 to the corresponding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     field. Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSJfilterOverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSJfilterOverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: 30 12 12 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min unique count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum uniquely mapping read count per junction for each of the motifs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     To set no output for desired motif, assign -1 to the corresponding field. Junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     conditions are satisfied. Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSJfilterCountUniqueMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSJfilterCountUniqueMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: 3 1 1 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min total count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum total (multi-mapping+unique) read count per junction for each</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     of the motifs. To set no output for desired motif, assign -1 to the corresponding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     conditions are satisfied. Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSJfilterCountTotalMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSJfilterCountTotalMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: 3 1 1 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min distance to other donor/acceptor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum allowed distance to other junctions' donor/acceptor for each</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     of the motifs (int &gt;= 0). Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSJfilterDistToOtherSJmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSJfilterDistToOtherSJmin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: 10 0 5 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max gap allowed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads (int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b, by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Does not apply to annotated junctions.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#outSJfilterIntronMaxVsReadN'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --outSJfilterIntronMaxVsReadN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: 50000 100000 200000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Gap open penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Gap open penalty.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreGap'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreGap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Non-canonical gap open</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Non-canonical gap open penalty (in addition to --scoreGap).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreGapNoncan'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreGapNoncan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-8'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: GC/AG and CT/GC gap open</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreGapGCAG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreGapGCAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-4'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: AT/AC and GT/AT gap open</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreGapATAC'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreGapATAC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-8'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Log scaled score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Extra score logarithmically scaled with genomic length of the alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;int&gt;*log2(genomicLength).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreGenomicLengthLog2scale'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreGenomicLengthLog2scale</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-0.25'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Deletion open penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Deletion open penalty.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreDelOpen'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreDelOpen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Deletion extension penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Deletion extension penalty per base (in addition to --scoreDelOpen).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreDelBase'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreDelBase</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Insertion Open Penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Insertion open penalty.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreInsOpen'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreInsOpen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Insertion extension penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Insertion extension penalty per base (in addition to --scoreInsOpen).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreInsBase'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreInsBase</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max score reduction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Maximum score reduction while searching for SJ boundaries in the stitching</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     step.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#scoreStitchSJshift'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --scoreStitchSJshift</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Search start point</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Defines the search start point through the read - the read is split</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     into pieces no longer than this value (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedSearchStartLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedSearchStartLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Search start point normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: seedSearchStartLmax normalized to read length (sum of mates' lengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     for paired-end reads).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedSearchStartLmaxOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedSearchStartLmaxOverLread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1.0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max seed length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Defines the maximum length of the seeds, if =0 max seed length is infinite</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedSearchLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedSearchLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Filter pieces for stitching</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Only pieces that map fewer than this value are utilized in the stitching</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     procedure (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedMultimapNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max seeds per read</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of seeds per read (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedPerReadNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max seeds per window</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of seeds per window (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedPerWindowNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedPerWindowNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max one-seed loci per window</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of one seed loci per window (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#seedNoneLociPerWindow'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --seedNoneLociPerWindow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min intron size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Minimum intron size: genomic gap is considered intron if its length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignIntronMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignIntronMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '21'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max intron size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Maximum intron size, if 0, max intron size will be determined by (2^winBinNbits)*winAnchorDistNbins.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignIntronMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignIntronMax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max mates gap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Maximum gap between two mates, if 0, max intron gap will be determined</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     by (2^winBinNbits)*winAnchorDistNbins.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignMatesGapMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignMatesGapMax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min overhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignSJoverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignSJoverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '5'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: 'Min overhang: annotated'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignSJDBoverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignSJDBoverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '3'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min mapped length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum mapped length for a read mate that is spliced (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignSplicedMateMapLmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignSplicedMateMapLmin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min mapped length normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignSplicedMateMapLminOverLmate'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignSplicedMateMapLminOverLmate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0.66'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max windows per read</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of windows per read (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignWindowsPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignWindowsPerReadNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max transcripts per window</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of transcripts per window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignTranscriptsPerWindowNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignTranscriptsPerWindowNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '100'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max transcripts per read</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of different alignments per read to consider (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignTranscriptsPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignTranscriptsPerReadNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: alignEndsType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Local</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - EndToEnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Alignment type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Type of read ends alignment. Local: standard local alignment with</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignEndsType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignEndsType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: Local</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: alignSoftClipAtReferenceEnds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 'Yes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - 'No'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Soft clipping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Option which allows soft clipping of alignments at the reference (chromosome)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ends. Can be disabled for compatibility with Cufflinks/Cuffmerge. Yes: Enables</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     soft clipping; No: Disables soft clipping.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#alignSoftClipAtReferenceEnds'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --alignSoftClipAtReferenceEnds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: 'Yes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max loci anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of loci anchors are allowed to map to (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#winAnchorMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --winAnchorMultimapNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Bin size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     each window will occupy an integer number of bins (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#winBinNbits'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --winBinNbits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '16'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max bins between anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of bins between two anchors that allows aggregation of anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     into one window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#winAnchorDistNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --winAnchorDistNbins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '9'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Flanking regions size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: =log2(winFlank), where win Flank is the size of the left and right</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     flanking regions for each window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#winFlankNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --winFlankNbins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '4'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min segment length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum length of chimeric segment length, if =0, no chimeric output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimSegmentMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --chimSegmentMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '15'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min total score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum total (summed) score of the chimeric segments (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimScoreMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --chimScoreMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max drop score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max drop (difference) of chimeric score (the sum of scores of all chimeric</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     segements) from the read length (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimScoreDropMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --chimScoreDropMax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '20'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min separation score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum difference (separation) between the best chimeric score and</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     the next one (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimScoreSeparation'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --chimScoreSeparation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Non-GT/AG penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Penalty for a non-GT/AG chimeric junction.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimScoreJunctionNonGTAG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --chimScoreJunctionNonGTAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Min junction overhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Minimum overhang for a chimeric junction (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimJunctionOverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --chimJunctionOverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '20'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: quantMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - TranscriptomeSAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - GeneCounts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Quantification mode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Types of quantification requested. 'TranscriptomeSAM' option outputs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts' option,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     STAR will count number of reads per gene while mapping.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#quantMode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Quantification of Annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Reads to process in 1st step</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Number of reads to process for the 1st step. 0: 1-step only, no 2nd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     pass; use very large number to map all reads in the first step (int&gt;0).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#twopass1readsN'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 2-pass mapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: twopassMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Two-pass mode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass mapping,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     with all 1st pass junctions inserted into the genome indices on the fly.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#twopassMode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --twopassMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: 2-pass mapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Genome dir name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Name of the directory which contains genome files (when genome.tar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     is uncompressed).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#genomeDirName'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: $job.inputs.genomeDirName || "genomeDir"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: sjdbInsertSave</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Save junction files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Which files to save when sjdb junctions are inserted on the fly at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     the mapping step. None: not saving files at all; Basic: only small junction/transcript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     files; All: all files including big Genome, SA and SAindex. These files are output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     as archive.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbInsertSave'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using ENSMEBL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     annotations with UCSC geneomes).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFchrPrefix'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Set exons feature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Feature type in GTF file to be used as exons for building transcripts.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFfeatureExon'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: exon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Exons' parents name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Tag name to be used as exons's transcript-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Gene name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Tag name to be used as exons's gene-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: '"Overhang" length'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Length of the donor/acceptor sequence on each side of the junctions,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database is</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     not used.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbOverhang'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '100'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Extra alignment score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Extra alignment score for alignments that cross database junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbScore'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Splice junction file</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Gene model annotations and/or known transcripts. No need to include</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     this input, except in case of using "on the fly" annotations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: GTF, GFF, TXT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Clip 3p bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Number of bases to clip from 3p of each mate. In case only one value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#clip3pNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --clip3pNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Clip 5p bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Number of bases to clip from 5p of each mate. In case only one value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#clip5pNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --clip5pNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Clip 3p adapter sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Adapter sequence to clip from 3p of each mate. In case only one value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#clip3pAdapterSeq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --clip3pAdapterSeq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max mismatches proportions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max proportion of mismatches for 3p adapter clipping for each mate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In case only one value is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#clip3pAdapterMMp'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --clip3pAdapterMMp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0.1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Clip 3p after adapter seq.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Number of bases to clip from 3p of each mate after the adapter clipping.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In case only one value is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#clip3pAfterAdapterNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --clip3pAfterAdapterNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: chimOutType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - SeparateSAMold</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Within</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Chimeric output type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimOutType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: SeparateSAMold</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Genome files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Genome files created using STAR Genome Generate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: TAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max insert junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Maximum number of junction to be inserted to the genome on the fly</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     at the mapping stage, including those from annotations and those detected in the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     1st step of the 2-pass run.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#limitSjdbInsertNsj'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --limitSjdbInsertNsj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '1000000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - name: quantTranscriptomeBan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - IndelSoftclipSingleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Singleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Prohibit alignment type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prohibit single-end alignments.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#quantTranscriptomeBan'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --quantTranscriptomeBan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Quantification of Annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: IndelSoftclipSingleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Limit BAM sorting memory</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Maximum available RAM for sorting BAM. If set to 0, it will be set</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     to the genome index size.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#limitBAMsortRAM'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     prefix: --limitBAMsortRAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Aligned SAM/BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Aligned sequence in SAM/BAM format.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           if ($job.inputs.outSortingType == 'SortedByCoordinate') {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sort_name = '.sortedByCoord'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sort_name = ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           if ($job.inputs.outSAMtype == 'BAM') {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sam_name = "*.Aligned".concat( sort_name, '.out.bam')</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sam_name = "*.Aligned.out.sam"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return sam_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: SAM, BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Transcriptome alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Alignments translated into transcript coordinates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#transcriptome_aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*Transcriptome*'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Reads per gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: File with number of reads per gene. A read is counted if it overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     (1nt or more) one and only one gene.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#reads_per_gene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*ReadsPerGene*'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: TAB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Log files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Log files produced during alignment.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#log_files'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*Log*.out'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: OUT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Splice junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: High confidence collapsed splice junctions in tab-delimited format.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Only junctions supported by uniquely mapping reads are reported.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#splice_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*SJ.out.tab'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: TAB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Chimeric junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Junctions' won't be output.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimeric_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*Chimeric.out.junction'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: JUNCTION</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Unmapped reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Output of unmapped reads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#unmapped_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*Unmapped.out*'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: FASTQ</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Intermediate genome files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Archive with genome files produced when annotations are included on</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     the fly (in the mapping step).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#intermediate_genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*_STARgenome.tar'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: TAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Chimeric alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Alignment SAM and Chimeric Junctions outputs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimeric_alignments'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     glob: '*.Chimeric.out.sam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:fileTypes: SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: ExpressionEngineRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     dockerPull: rabix/js-engine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hints:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   dockerImageId: a4b0ad2c3cae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: sbg:MemRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   value: 60000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: sbg:CPURequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   value: 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> label: STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> description: STAR is an ultrafast universal RNA-seq aligner. It has very high mapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   speed, accurate alignment of contiguous and spliced reads, detection of polyA-tails,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   non-canonical splices and chimeric (fusion) junctions. It works with reads starting</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   from lengths ~15 bases up to ~300 bases. In case of having longer reads, use of</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   STAR Long is recommended.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: CommandLineTool</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> baseCommand:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - tar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - -xvf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   script: $job.inputs.genome.path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - '&amp;&amp;'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - /opt/STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - --runThreadN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       return $job.allocatedResources.cpu</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> arguments:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         file = [].concat($job.inputs.reads)[0].path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         extension = /(?:\.([^.]+))?$/.exec(file)[1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if (extension == "gz") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return "--readFilesCommand zcat"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         } else if (extension == "bz2") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return "--readFilesCommand bzcat"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n  var</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths =</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext == \"gtf\")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n      }\n      if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       (ext == \"txt\") {\n        sjFormat = \"True\"\n        return sjFormat\n      }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \   })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\") {\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \       return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n  if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\", $job.inputs.sjdbGTFfeatureExon]\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript) {\n    c = [\"--sjdbGTFtagExonParentTranscript\",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       $job.inputs.sjdbGTFtagExonParentTranscript]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentGene)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    d = [\"--sjdbGTFtagExonParentGene\", $job.inputs.sjdbGTFtagExonParentGene]\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.sjdbOverhang) {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.sjdbScore) {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.sjdbInsertSave) {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  \n  \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\" \")\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if ($job.inputs.twopassMode == "Basic") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return "--twopass1readsN ".concat($job.inputs.twopass1readsN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if ($job.inputs.chimOutType == "Within") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return "--chimOutType SeparateSAMold"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n    if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       (value == \"\") {\n      return\n    }\n    else {\n      return param_list.push(key.concat(\":\",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value))\n    }\n  }\n  \n  add_param('ID', \"1\")\n  if ($job.inputs.rg_seq_center)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    add_param('CN', $job.inputs.rg_seq_center)\n  } else if ([].concat($job.inputs.reads)[0].metadata.seq_center)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    add_param('CN', [].concat($job.inputs.reads)[0].metadata.seq_center)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.rg_library_id) {\n    add_param('LB', $job.inputs.rg_library_id)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ } else if ([].concat($job.inputs.reads)[0].metadata.library_id) {\n    add_param('LB',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [].concat($job.inputs.reads)[0].metadata.library_id)\n  }\n  if ($job.inputs.rg_mfl)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    add_param('PI', $job.inputs.rg_mfl)\n  } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \".concat(param_list.join(\" \"))\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return "--quantMode ".concat($job.inputs.quantMode)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  intermediate =</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       common_prefix.replace( /\\-$|\\_$|\\.$/, '' ).concat(\"._STARgenome\")\n  source</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       = \"./\".concat(intermediate)\n  destination = intermediate.concat(\".tar\")\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       != \"None\") {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   prefix: --outFileNamePrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /\\-$|\\_$|\\.$/, '' ), \".\")\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - position: 101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  mate1 = common_prefix.replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /\\-$|\\_$|\\.$/, '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /\\-$|\\_$|\\.$/, '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped ==</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \" \", mate1fq,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \" \",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       mate1fq)\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> stdin: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> stdout: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> successCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> temporaryFailCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="co"># convert to Flow object</span></span></span>
<span class="r-in"><span><span class="fu">convert_app</span><span class="op">(</span><span class="va">flow.in</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:validationErrors: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:sbgMaintained: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:latestRevision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:toolAuthor: Seven Bridges Genomics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:createdOn: 1463601910</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:categories:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - Alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - RNA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:contributors:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - tengfei</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:project: tengfei/quickstart</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:createdBy: tengfei</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:toolkitVersion: 2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:license: Apache License 2.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:modifiedOn: 1463601974</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:modifiedBy: tengfei</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:revisionsInfo:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: tengfei</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1463601910</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: tengfei</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1463601952</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - sbg:modifiedBy: tengfei</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:modifiedOn: 1463601974</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:toolkit: STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: sjdbGTFfile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 160.4999759</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 195.0833106</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#fastq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 164.2499914</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 323.7499502</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: genomeFastaFiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#genomeFastaFiles'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 167.7499601</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 469.9999106</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Exons' parents name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Tag name to be used as exons' transcript-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 350.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Gene name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Tag name to be used as exons' gene-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 400.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max loci anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of loci anchors are allowed to map to (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#winAnchorMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 450.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: Max bins between anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   description: Max number of bins between two anchors that allows aggregation of anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     into one window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#winAnchorDistNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 500.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:toolDefaultValue: '9'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: unmapped_reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#unmapped_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.unmapped_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 766.2497863</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 159.5833091</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: transcriptome_aligned_reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#transcriptome_aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.transcriptome_aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1118.9998003</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 86.5833216</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: splice_junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#splice_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.splice_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1282.3330177</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 167.499976</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: reads_per_gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#reads_per_gene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.reads_per_gene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1394.4163557</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 245.749964</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: log_files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#log_files'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.log_files'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1505.0830269</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 322.9999518</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: chimeric_junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimeric_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.chimeric_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1278.7498062</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 446.7499567</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: intermediate_genome</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#intermediate_genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.intermediate_genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1408.9164783</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 386.0832876</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: chimeric_alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#chimeric_alignments'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#STAR.chimeric_alignments'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1147.5831348</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 503.2499285</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: sorted_bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#sorted_bam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#Picard_SortSam.sorted_bam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 934.2498228</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 557.2498436</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   label: result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   id: '#result'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   source: '#SBG_FASTQ_Quality_Detector.result'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 1431.6666548</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 644.9999898</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: CreateFileRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   fileDef: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> hints:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - class: sbg:AWSInstanceType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   value: c3.8xlarge</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> label: RNA-seq Alignment - STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> description: "Alignment to a reference genome and transcriptome presents the first</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   capable of mapping full length RNA sequences and detecting de novo canonical junctions,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sequence reads, but fine tuning of its parameters enables customization to satisfy</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   \ \nSplice junction annotations can optionally be collected from splice junction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   databases. Set the \"Overhang length\" parameter to a value larger than zero in</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   order to use splice junction databases. For constant read length, this value should</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   (ideally) be equal to mate length decreased by 1; for long reads with non-constant</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   files reveals read length distribution. STAR can detect chimeric transcripts, but</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   to a desired minimum chimeric segment length. Aligned reads are reported in BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   format and can be viewed in a genome browser (such as IGV). A file containing detected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   unmapped in SAM\" parameters enable unmapped output type selection."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: Workflow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> steps:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: '#STAR_Genome_Generate'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbScore'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbOverhang'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.genomeSAsparseD'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.genomeSAindexNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.genomeFastaFiles'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#genomeFastaFiles'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.genomeChrBinNbits'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR_Genome_Generate.genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hints: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   run:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:validationErrors: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:sbgMaintained: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:latestRevision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:job:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       allocatedResources:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mem: 60000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cpu: 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbScore: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFfeatureExon: sjdbGTFfeatureExon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbOverhang: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         genomeChrBinNbits: genomeChrBinNbits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         genomeSAsparseD: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFfile:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path: /demo/test-files/chr20.gtf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         genomeFastaFiles:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path: /sbgenomics/test-data/chr20.fa</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFchrPrefix: sjdbGTFchrPrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         genomeSAindexNbases: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolAuthor: Alexander Dobin/CSHL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdOn: 1450911469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:categories:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:contributors:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:links:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/alexdobin/STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Homepage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/alexdobin/STAR/releases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Releases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Manual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://groups.google.com/forum/#!forum/rna-star</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Support</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Publication</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:project: bix-demo/star-2-4-2a-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkitVersion: 2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:id: sevenbridges/public-apps/star-genome-generate/1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:license: GNU General Public License v3.0 only</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedOn: 1450911470</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revisionsInfo:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911470</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkit: STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     id: sevenbridges/public-apps/star-genome-generate/1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Extra alignment score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Extra alignment score for alignments that cross database junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbScore'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --sjdbScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: '"Overhang" length'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Length of the donor/acceptor sequence on each side of the junctions,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         is not used.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbOverhang'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --sjdbOverhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '100'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Exons' parents name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Tag name to be used as exons' transcript-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --sjdbGTFtagExonParentTranscript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Gene name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Tag name to be used as exons' gene-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --sjdbGTFtagExonParentGene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Splice junction file</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Gene model annotations and/or known transcripts.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: GTF, GFF, TXT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Set exons feature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Feature type in GTF file to be used as exons for building transcripts.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFfeatureExon'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --sjdbGTFfeatureExon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: exon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ENSMEBL annotations with UCSC geneomes).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFchrPrefix'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --sjdbGTFchrPrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions db parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Suffux array sparsity</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Distance between indices: use bigger numbers to decrease needed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         RAM at the cost of mapping speed reduction (int&gt;0).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genomeSAsparseD'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --genomeSAsparseD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Genome generation parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Pre-indexing string length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Length (bases) of the SA pre-indexing string. Typically between</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         10 and 15. Longer strings will use much more memory, but allow faster searches.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         For small genomes, this number needs to be scaled down, with a typical value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         this is equal to 9, for 100 kiloBase genome, this is equal to 7.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genomeSAindexNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --genomeSAindexNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Genome generation parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '14'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Genome fasta files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Reference sequence to which to align the reads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genomeFastaFiles'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --genomeFastaFiles</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: FASTA, FA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Bins size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for genome storage: each chromosome will occupy an integer number of bins.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         you may need to reduce this number to reduce RAM consumption. The following</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         is equal to 15.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genomeChrBinNbits'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --genomeChrBinNbits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Genome generation parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '18'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Genome Files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Genome files comprise binary genome sequence, suffix arrays, text</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chromosome names/lengths, splice junctions coordinates, and transcripts/genes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         information.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*.tar'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: TAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: ExpressionEngineRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dockerPull: rabix/js-engine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hints:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerImageId: a4b0ad2c3cae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:CPURequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:MemRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 60000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     label: STAR Genome Generate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     description: STAR Genome Generate is a tool that generates genome index files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       One set of files should be generated per each genome/annotation combination.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       Once produced, these files could be used as long as genome/annotation combination</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       stays the same. Also, STAR Genome Generate which produced these files and STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       aligner using them must be the same toolkit version.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: CommandLineTool</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     baseCommand:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - mkdir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - '&amp;&amp;'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - /opt/STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - --runMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - genomeGenerate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - --genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - ./genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - --runThreadN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return $job.allocatedResources.cpu</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     arguments:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 99</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdin: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdout: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     successCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temporaryFailCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     x: 384.0832266</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     'y': 446.4998957</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 100.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: '#SBG_FASTQ_Quality_Detector'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#SBG_FASTQ_Quality_Detector.fastq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#fastq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#SBG_FASTQ_Quality_Detector.result'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hints: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   run:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:validationErrors: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:sbgMaintained: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:latestRevision: 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:job:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       allocatedResources:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mem: 1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cpu: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         fastq:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path: /path/to/fastq.ext</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolAuthor: Seven Bridges Genomics</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdOn: 1450911312</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:categories:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - FASTQ-Processing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:contributors:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:project: bix-demo/sbgtools-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:license: Apache License 2.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revision: 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       /path/to/fastq.ext</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedOn: 1450911314</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revisionsInfo:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911312</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911314</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911313</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911313</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkit: SBGTools</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: FASTQ file.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#fastq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Result</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Source FASTQ file with updated metadata.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#result'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*.fastq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: FASTQ</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: CreateFileRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       fileDef: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hints:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerImageId: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:CPURequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:MemRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     label: SBG FASTQ Quality Detector</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     description: FASTQ Quality Scale Detector detects which quality encoding scheme</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       was used in your reads and automatically enters the proper value in the "Quality</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       Scale" metadata field.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: CommandLineTool</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     baseCommand:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - python</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - /opt/sbg_fastq_quality_scale_detector.py</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     arguments: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdin: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdout: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     successCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temporaryFailCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     x: 375.3333179</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     'y': 323.5833156</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 300.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: '#Picard_SortSam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.validation_stringency'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: SILENT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.sort_order'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: Coordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.quiet'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.output_type'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.memory_per_job'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.max_records_in_ram'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.input_bam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#STAR.aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.create_index'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: 'True'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.compression_level'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#Picard_SortSam.sorted_bam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hints: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   run:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:validationErrors: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:sbgMaintained: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:latestRevision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:job:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       allocatedResources:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mem: 2048</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cpu: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sort_order: Coordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         input_bam:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path: /root/dir/example.tested.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         memory_per_job: 2048</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         output_type: ~</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         create_index: ~</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolAuthor: Broad Institute</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdOn: 1450911168</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:categories:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - SAM/BAM-Processing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:contributors:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:links:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: http://broadinstitute.github.io/picard/index.html</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Homepage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/broadinstitute/picard/releases/tag/1.138</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Source Code</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: http://broadinstitute.github.io/picard/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Wiki</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/broadinstitute/picard/zipball/master</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Download</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: http://broadinstitute.github.io/picard/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Publication</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:project: bix-demo/picard-1-140-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkitVersion: '1.140'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:license: MIT License, Apache 2.0 Licence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       SORT_ORDER=coordinate  /root/dir/example.tested.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedOn: 1450911170</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revisionsInfo:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911168</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911169</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911170</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkit: Picard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     id: sevenbridges/public-apps/picard-sortsam-1-140/2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: validation_stringency</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - STRICT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - LENIENT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - SILENT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Validation stringency</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Validation stringency for all SAM files read by this program.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Setting stringency to SILENT can improve performance when processing a BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         file in which variable-length data (read, qualities, tags) do not otherwise</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         need to be decoded. This option can be set to ''null'' to clear the default</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value. Possible values: {STRICT, LENIENT, SILENT}.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#validation_stringency'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: VALIDATION_STRINGENCY=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               if ($job.inputs.validation_stringency)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 return $job.inputs.validation_stringency</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               else</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 return "SILENT"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: SILENT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: sort_order</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Unsorted</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Queryname</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Coordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Sort order</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Sort order of the output file. Possible values: {unsorted, queryname,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         coordinate}.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sort_order'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: SORT_ORDER=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               p = $job.inputs.sort_order.toLowerCase()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return p</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Coordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:altPrefix: SO</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: quiet</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 'True'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 'False'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Quiet</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'This parameter indicates whether to suppress job-summary info</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         on System.err. This option can be set to ''null'' to clear the default value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Possible values: {true, false}.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#quiet'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: QUIET=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 'False'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: output_type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - SAME AS INPUT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Output format</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Since Picard tools can output both SAM and BAM files, user can</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         choose the format of the output file.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#output_type'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: SAME AS INPUT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Memory per job</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         single threaded jobs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#memory_per_job'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '2048'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max records in RAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: When writing SAM files that need to be sorted, this parameter will</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         specify the number of records stored in RAM before spilling to disk. Increasing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         this number reduces the number of file handles needed to sort a SAM file,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         and increases the amount of RAM needed. This option can be set to 'null' to</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clear the default value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#max_records_in_ram'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: MAX_RECORDS_IN_RAM=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '500000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Input BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: The BAM or SAM file to sort.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#input_bam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: INPUT=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: File inputs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: BAM, SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:altPrefix: I</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: create_index</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 'True'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 'False'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Create index</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'This parameter indicates whether to create a BAM index when writing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         a coordinate-sorted BAM file. This option can be set to ''null'' to clear</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         the default value. Possible values: {true, false}.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#create_index'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: CREATE_INDEX=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 'False'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Compression level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Compression level for all compressed files created (e.g. BAM and</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         GELI). This option can be set to 'null' to clear the default value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#compression_level'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: COMPRESSION_LEVEL=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Other input types</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '5'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Sorted BAM/SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Sorted BAM or SAM file.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sorted_bam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*.sorted.?am'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: BAM, SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: ExpressionEngineRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dockerPull: rabix/js-engine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engineCommand: cwl-engine.js</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hints:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerImageId: eab0e70b6629</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:CPURequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:MemRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n  \treturn 2048\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     label: Picard SortSam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     description: Picard SortSam sorts the input SAM or BAM. Input and output formats</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       are determined by the file extension.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: CommandLineTool</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     baseCommand:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - java</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - -jar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - /opt/picard-tools-1.140/picard.jar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - SortSam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     arguments:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       prefix: OUTPUT=</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '')\n}\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 1000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdin: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdout: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     successCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temporaryFailCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     x: 773.0831807</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     'y': 470.9165939</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 500.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> - id: '#STAR'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.winFlankNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.winBinNbits'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.winAnchorMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#winAnchorMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.winAnchorDistNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#winAnchorDistNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.twopassMode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.twopass1readsN'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbScore'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbOverhang'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbInsertSave'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbGTFfeatureExon'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.sjdbGTFchrPrefix'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedSearchStartLmaxOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedSearchStartLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedSearchLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedPerWindowNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedNoneLociPerWindow'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.seedMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreStitchSJshift'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreInsOpen'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreInsBase'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreGenomicLengthLog2scale'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreGapNoncan'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreGapGCAG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreGapATAC'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreGap'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreDelOpen'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.scoreDelBase'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.rg_seq_center'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.rg_sample_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.rg_platform_unit_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.rg_platform'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.rg_mfl'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.rg_library_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#SBG_FASTQ_Quality_Detector.result'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.readMatesLengthsIn'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.readMapNumber'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.quantTranscriptomeBan'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.quantMode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: TranscriptomeSAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSortingType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSJfilterReads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSJfilterOverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSJfilterIntronMaxVsReadN'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSJfilterDistToOtherSJmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSJfilterCountUniqueMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSJfilterCountTotalMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMunmapped'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMtype'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMstrandField'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMreadID'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMprimaryFlag'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMorder'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMmode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMmapqUnique'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMheaderPG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMheaderHD'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMflagOR'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMflagAND'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outSAMattributes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outReadsUnmapped'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     default: Fastx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outQSconversionAdd'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterScoreMinOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterScoreMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMultimapScoreRange'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMismatchNoverReadLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMismatchNoverLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMismatchNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMatchNminOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterMatchNmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.outFilterIntronMotifs'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.limitSjdbInsertNsj'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.limitOutSJoneRead'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.limitOutSJcollapsed'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.limitBAMsortRAM'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.genomeDirName'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     source: '#STAR_Genome_Generate.genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.clip5pNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.clip3pNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.clip3pAfterAdapterNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.clip3pAdapterSeq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.clip3pAdapterMMp'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimSegmentMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimScoreSeparation'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimScoreMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimScoreJunctionNonGTAG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimScoreDropMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimOutType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimJunctionOverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignWindowsPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignTranscriptsPerWindowNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignTranscriptsPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignSplicedMateMapLminOverLmate'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignSplicedMateMapLmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignSoftClipAtReferenceEnds'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignSJoverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignSJDBoverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignMatesGapMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignIntronMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignIntronMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.alignEndsType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.unmapped_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.transcriptome_aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.splice_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.reads_per_gene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.log_files'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.intermediate_genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimeric_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.chimeric_alignments'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   - id: '#STAR.aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   hints: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   run:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:validationErrors: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:sbgMaintained: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:latestRevision: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:job:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       allocatedResources:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mem: 60000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         cpu: 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignWindowsPerReadNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMheaderPG: outSAMheaderPG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         GENOME_DIR_NAME: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMatchNminOverLread: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rg_platform_unit_id: rg_platform_unit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignTranscriptsPerReadNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         readMapNumber: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignSplicedMateMapLminOverLmate: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignMatesGapMax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMultimapNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clip5pNbases:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMstrandField: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         readMatesLengthsIn: NotEqual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMattributes: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedMultimapNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rg_mfl: rg_mfl</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimSegmentMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         winAnchorDistNbins: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSortingType: SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMultimapScoreRange: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbInsertSave: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clip3pAfterAdapterNbases:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreDelBase: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMatchNmin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         twopass1readsN: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMunmapped: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         genome:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path: genome.ext</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFtagExonParentTranscript: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         limitBAMsortRAM: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignEndsType: Local</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedNoneLociPerWindow: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rg_sample_id: rg_sample</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFtagExonParentGene: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimScoreMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSJfilterIntronMaxVsReadN:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         twopassMode: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignSplicedMateMapLmin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSJfilterReads: All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMprimaryFlag: OneBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSJfilterCountTotalMin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMorder: Paired</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMflagAND: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimScoreSeparation: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignSJoverhangMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterScoreMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedSearchStartLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreGapGCAG: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreGenomicLengthLog2scale: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterIntronMotifs: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMismatchNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         reads:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - size: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           secondaryFiles: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           metadata:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             format: fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             paired_end: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             seq_center: illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path: /test-data/mate_1.fastq.bz2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreGap: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSJfilterOverhangMin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMflagOR: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMmode: Full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rg_library_id: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimScoreJunctionNonGTAG: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreInsOpen: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clip3pAdapterSeq:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - clip3pAdapterSeq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimScoreDropMax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterType: Normal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreGapATAC: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rg_platform: Ion Torrent PGM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clip3pAdapterMMp:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFfeatureExon: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outQSconversionAdd: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         quantMode: TranscriptomeSAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignIntronMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreInsBase: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreGapNoncan: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedSearchLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSJfilterDistToOtherSJmin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterScoreMinOverLread: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignSJDBoverhangMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         limitOutSJcollapsed: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         winAnchorMultimapNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMismatchNoverLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         rg_seq_center: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMheaderHD: outSAMheaderHD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimOutType: Within</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         quantTranscriptomeBan: IndelSoftclipSingleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         limitOutSJoneRead: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignTranscriptsPerWindowNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbOverhang: ~</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outReadsUnmapped: Fastx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreStitchSJshift: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedPerWindowNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSJfilterCountUniqueMin:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         scoreDelOpen: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFfile:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - path: /demo/test-data/chr20.gtf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clip3pNbases:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         winBinNbits: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbScore: ~</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedSearchStartLmaxOverLread: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignIntronMax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seedPerReadNmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outFilterMismatchNoverReadLmax: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         winFlankNbins: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sjdbGTFchrPrefix: chrPrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignSoftClipAtReferenceEnds: 'Yes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMreadID: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMtype: BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimJunctionOverhangMin: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         limitSjdbInsertNsj: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         outSAMmapqUnique: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolAuthor: Alexander Dobin/CSHL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdOn: 1450911471</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:categories:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - Alignment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:contributors:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - ana_d</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - uros_sipetic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:links:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/alexdobin/STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Homepage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/alexdobin/STAR/releases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Releases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Manual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: https://groups.google.com/forum/#!forum/rna-star</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Support</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Publication</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:project: bix-demo/star-2-4-2a-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:createdBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkitVersion: 2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:id: sevenbridges/public-apps/star/4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:license: GNU General Public License v3.0 only</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revision: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedOn: 1462889222</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:modifiedBy: ana_d</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:revisionsInfo:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911471</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911473</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: bix-demo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1450911475</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: uros_sipetic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1462878528</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - sbg:modifiedBy: ana_d</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:modifiedOn: 1462889222</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:revision: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     sbg:toolkit: STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     id: sevenbridges/public-apps/star/4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     inputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Flanking regions size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: =log2(winFlank), where win Flank is the size of the left and right</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         flanking regions for each window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#winFlankNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --winFlankNbins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '4'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Bin size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         each window will occupy an integer number of bins (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#winBinNbits'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --winBinNbits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:includeInPorts: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '16'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max loci anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of loci anchors are allowed to map to (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#winAnchorMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --winAnchorMultimapNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max bins between anchors</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of bins between two anchors that allows aggregation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         of anchors into one window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#winAnchorDistNbins'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --winAnchorDistNbins</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Windows, Anchors, Binning</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '9'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: twopassMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Two-pass mode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mapping, with all 1st pass junctions inserted into the genome indices on the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         fly.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#twopassMode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --twopassMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 2-pass mapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Reads to process in 1st step</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Number of reads to process for the 1st step. 0: 1-step only, no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         2nd pass; use very large number to map all reads in the first step (int&gt;0).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#twopass1readsN'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 2-pass mapping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Extra alignment score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Extra alignment score for alignments that cross database junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbScore'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: '"Overhang" length'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Length of the donor/acceptor sequence on each side of the junctions,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         is not used.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbOverhang'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '100'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: sjdbInsertSave</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Save junction files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Which files to save when sjdb junctions are inserted on the fly</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         at the mapping step. None: not saving files at all; Basic: only small junction/transcript</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         files; All: all files including big Genome, SA and SAindex. These files are</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         output as archive.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbInsertSave'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Exons' parents name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Tag name to be used as exons' transcript-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFtagExonParentTranscript'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Gene name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Tag name to be used as exons' gene-parents.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFtagExonParentGene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: gene_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Splice junction file</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Gene model annotations and/or known transcripts. No need to include</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         this input, except in case of using "on the fly" annotations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFfile'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: GTF, GFF, TXT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Set exons feature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Feature type in GTF file to be used as exons for building transcripts.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFfeatureExon'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: exon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Chromosome names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ENSMEBL annotations with UCSC geneomes).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#sjdbGTFchrPrefix'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Splice junctions database</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Search start point normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: seedSearchStartLmax normalized to read length (sum of mates' lengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for paired-end reads).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedSearchStartLmaxOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedSearchStartLmaxOverLread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1.0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Search start point</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Defines the search start point through the read - the read is split</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         into pieces no longer than this value (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedSearchStartLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedSearchStartLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max seed length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Defines the maximum length of the seeds, if =0 max seed length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         is infinite (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedSearchLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedSearchLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max seeds per window</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of seeds per window (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedPerWindowNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedPerWindowNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '50'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max seeds per read</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of seeds per read (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedPerReadNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max one-seed loci per window</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of one seed loci per window (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedNoneLociPerWindow'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedNoneLociPerWindow</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Filter pieces for stitching</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Only pieces that map fewer than this value are utilized in the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stitching procedure (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#seedMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --seedMultimapNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max score reduction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Maximum score reduction while searching for SJ boundaries in the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         stitching step.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreStitchSJshift'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreStitchSJshift</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Insertion Open Penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Insertion open penalty.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreInsOpen'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreInsOpen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Insertion extension penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Insertion extension penalty per base (in addition to --scoreInsOpen).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreInsBase'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreInsBase</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Log scaled score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Extra score logarithmically scaled with genomic length of the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignment: &lt;int&gt;*log2(genomicLength).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreGenomicLengthLog2scale'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreGenomicLengthLog2scale</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-0.25'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Non-canonical gap open</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Non-canonical gap open penalty (in addition to --scoreGap).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreGapNoncan'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreGapNoncan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-8'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: GC/AG and CT/GC gap open</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreGapGCAG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreGapGCAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-4'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: AT/AC and GT/AT gap open</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreGapATAC'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreGapATAC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-8'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Gap open penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Gap open penalty.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreGap'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreGap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Deletion open penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Deletion open penalty.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreDelOpen'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreDelOpen</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Deletion extension penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Deletion extension penalty per base (in addition to --scoreDelOpen).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#scoreDelBase'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --scoreDelBase</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Scoring</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-2'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Sequencing center</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Specify the sequencing center for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#rg_seq_center'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Sample ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Specify the sample ID for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#rg_sample_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Platform unit ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Specify the platform unit ID for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#rg_platform_unit_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: rg_platform</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - LS 454</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Helicos</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - ABI SOLiD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Ion Torrent PGM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - PacBio</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Platform</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Specify the version of the technology that was used for sequencing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         or assaying.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#rg_platform'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Median fragment length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Specify the median fragment length for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#rg_mfl'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Library ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Specify the library ID for RG line.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#rg_library_id'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Inferred from metadata</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Read sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Read sequence.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             == 2){\n          right = list[index].path\n        }\n      }\n    }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ++index) {\n      \n      if (list[index].metadata != null){\n        if</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \       }\n      }\n    }\n    left_join = left.join()\n    right_join</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: readMatesLengthsIn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - NotEqual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Equal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Reads lengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Equal/Not equal - lengths of names, sequences, qualities for both</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         mates are the same/not the same. "Not equal" is safe in all situations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#readMatesLengthsIn'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --readMatesLengthsIn</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: NotEqual</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Reads to map</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Number of reads to map from the beginning of the file.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#readMapNumber'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --readMapNumber</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: quantTranscriptomeBan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - IndelSoftclipSingleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Singleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Prohibit alignment type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prohibit single-end alignments.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#quantTranscriptomeBan'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --quantTranscriptomeBan</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Quantification of Annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: IndelSoftclipSingleend</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: quantMode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - TranscriptomeSAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - GeneCounts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Quantification mode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Types of quantification requested. 'TranscriptomeSAM' option outputs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         option, STAR will count number of reads per gene while mapping.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#quantMode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Quantification of Annotations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSortingType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Unsorted</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Unsorted SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Output sorting type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Type of output sorting.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSortingType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: SortedByCoordinate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSJfilterReads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Unique</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Collapsed junctions reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Which reads to consider for collapsed splice junctions output.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         only.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSJfilterReads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSJfilterReads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min overhang SJ</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum overhang length for splice junctions on both sides for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         each of the motifs. To set no output for desired motif, assign -1 to the corresponding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         field. Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSJfilterOverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSJfilterOverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 30 12 12 12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max gap allowed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Does not apply to annotated junctions.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSJfilterIntronMaxVsReadN'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSJfilterIntronMaxVsReadN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 50000 100000 200000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min distance to other donor/acceptor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum allowed distance to other junctions' donor/acceptor for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSJfilterDistToOtherSJmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSJfilterDistToOtherSJmin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 10 0 5 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min unique count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum uniquely mapping read count per junction for each of the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         motifs. To set no output for desired motif, assign -1 to the corresponding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         conditions are satisfied. Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSJfilterCountUniqueMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSJfilterCountUniqueMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 3 1 1 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min total count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum total (multi-mapping+unique) read count per junction for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         each of the motifs. To set no output for desired motif, assign -1 to the corresponding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         conditions are satisfied. Does not apply to annotated junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSJfilterCountTotalMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSJfilterCountTotalMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: 'Output filtering: splice junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 3 1 1 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMunmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Within</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Write unmapped in SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Output of unmapped reads in the SAM format. None: no output Within:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMunmapped'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMunmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMtype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Output format</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Format of output alignments.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMtype'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               SAM_type = $job.inputs.outSAMtype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               SORT_type = $job.inputs.outSortingType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               if (SAM_type &amp;&amp; SORT_type) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMstrandField</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - intronMotif</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Strand field flag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         strand derived from the intron motif. Reads with inconsistent and/or non-canonical</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         introns are filtered out.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMstrandField'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMstrandField</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMreadID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Read ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Read ID record type. Standard: first word (until space) from the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         in the FASTx file.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMreadID'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMreadID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMprimaryFlag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - OneBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - AllBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Primary alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Which alignments are considered primary - all others will be marked</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         score is primary; AllBestScore: all alignments with the best score are primary.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMprimaryFlag'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMprimaryFlag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: OneBestScore</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Paired</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - PairedKeepInputOrder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Sorting in SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Type of sorting for the SAM output. Paired: one mate after the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         other for all paired alignments; PairedKeepInputOrder: one mate after the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         other for all paired alignments, the order is kept the same as in the input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         FASTQ files.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMorder'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMorder</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Paired</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMmode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - NoQS</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: SAM mode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         without quality scores.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMmode'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMmode</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Full</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: MAPQ value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: MAPQ value for unique mappers (0 to 255).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMmapqUnique'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMmapqUnique</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '255'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: SAM header @PG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Extra @PG (software) line of the SAM header (in addition to STAR).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMheaderPG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMheaderPG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: SAM header @HD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: '@HD (header) line of the SAM header.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMheaderHD'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMheaderHD</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: OR SAM flag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Set specific bits of the SAM FLAG.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMflagOR'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMflagOR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: AND SAM flag</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Set specific bits of the SAM FLAG.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMflagAND'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMflagAND</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '65535'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outSAMattributes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - NH</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - All</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: SAM attributes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Desired SAM attributes, in the order desired for the output SAM.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         NM MD jM jI; None: no attributes.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outSAMattributes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outSAMattributes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Standard</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Fastx</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Output unmapped reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         output in separate fasta/fastq files, Unmapped.out.mate1/2.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outReadsUnmapped'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Quality conversion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Add this number to the quality score (e.g. to convert from Illumina</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         to Sanger, use -31).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outQSconversionAdd'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outQSconversionAdd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outFilterType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Normal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - BySJout</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Filtering type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Type of filtering. Normal: standard filtering using only current</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignment; BySJout: keep only those reads that contain junctions that passed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         filtering into SJ.out.tab.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Normal</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min score normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: '''Minimum score'' normalized to read length (sum of mates'' lengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         for paired-end reads).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterScoreMinOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterScoreMinOverLread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0.66'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Alignment will be output only if its score is higher than this</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterScoreMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterScoreMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Multimapping score range</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: The score range below the maximum score for multimapping alignments.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMultimapScoreRange'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMultimapScoreRange</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max number of mappings</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Read alignments will be output only if the read maps fewer than</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         this value, otherwise no alignments will be output.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMultimapNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMultimapNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Mismatches to *read* length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Alignment will be output only if its ratio of mismatches to *read*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         length is less than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMismatchNoverReadLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMismatchNoverReadLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Mismatches to *mapped* length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Alignment will be output only if its ratio of mismatches to *mapped*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         length is less than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMismatchNoverLmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMismatchNoverLmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0.3'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max number of mismatches</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Alignment will be output only if it has fewer mismatches than this</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMismatchNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMismatchNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min matched bases normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: '''Minimum matched bases'' normalized to read length (sum of mates</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         lengths for paired-end reads).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMatchNminOverLread'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMatchNminOverLread</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0.66'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min matched bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Alignment will be output only if the number of matched bases is</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         higher than this value.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterMatchNmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterMatchNmin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: outFilterIntronMotifs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - RemoveNoncanonical</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - RemoveNoncanonicalUnannotated</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Motifs filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         filter out alignments that contain non-canonical unannotated junctions when</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         using annotated splice junctions database. The annotated non-canonical junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         will be kept.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#outFilterIntronMotifs'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --outFilterIntronMotifs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Output filtering</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: None</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max insert junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Maximum number of junction to be inserted to the genome on the</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         fly at the mapping stage, including those from annotations and those detected</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         in the 1st step of the 2-pass run.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#limitSjdbInsertNsj'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --limitSjdbInsertNsj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1000000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Junctions max number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of junctions for one read (including all multi-mappers).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#limitOutSJoneRead'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --limitOutSJoneRead</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Collapsed junctions max number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of collapsed junctions.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#limitOutSJcollapsed'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --limitOutSJcollapsed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '1000000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Limit BAM sorting memory</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Maximum available RAM for sorting BAM. If set to 0, it will be</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         set to the genome index size.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#limitBAMsortRAM'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --limitBAMsortRAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Limits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Genome dir name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Name of the directory which contains genome files (when genome.tar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         is uncompressed).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genomeDirName'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           script: $job.inputs.genomeDirName || "genomeDir"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: genomeDir</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Genome files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Genome files created using STAR Genome Generate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Basic</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: TAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Clip 5p bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Number of bases to clip from 5p of each mate. In case only one</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#clip5pNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --clip5pNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Clip 3p bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Number of bases to clip from 3p of each mate. In case only one</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#clip3pNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --clip3pNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Clip 3p after adapter seq.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Number of bases to clip from 3p of each mate after the adapter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         clipping. In case only one value is given, it will be assumed the same for</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#clip3pAfterAdapterNbases'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --clip3pAfterAdapterNbases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: string</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Clip 3p adapter sequence</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Adapter sequence to clip from 3p of each mate. In case only one</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         value is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#clip3pAdapterSeq'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --clip3pAdapterSeq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max mismatches proportions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max proportion of mismatches for 3p adapter clipping for each mate.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         In case only one value is given, it will be assumed the same for both mates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#clip3pAdapterMMp'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --clip3pAdapterMMp</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         itemSeparator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Read parameters</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0.1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min segment length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum length of chimeric segment length, if =0, no chimeric output</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimSegmentMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --chimSegmentMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '15'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min separation score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum difference (separation) between the best chimeric score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         and the next one (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimScoreSeparation'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --chimScoreSeparation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min total score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimScoreMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --chimScoreMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Non-GT/AG penalty</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Penalty for a non-GT/AG chimeric junction.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimScoreJunctionNonGTAG'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --chimScoreJunctionNonGTAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '-1'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max drop score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max drop (difference) of chimeric score (the sum of scores of all</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         chimeric segements) from the read length (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimScoreDropMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --chimScoreDropMax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '20'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: chimOutType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - SeparateSAMold</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Within</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Chimeric output type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimOutType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: SeparateSAMold</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min junction overhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum overhang for a chimeric junction (int&gt;=0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimJunctionOverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --chimJunctionOverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Chimeric Alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '20'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max windows per read</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of windows per read (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignWindowsPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignWindowsPerReadNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max transcripts per window</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of transcripts per window (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignTranscriptsPerWindowNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignTranscriptsPerWindowNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '100'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max transcripts per read</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Max number of different alignments per read to consider (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignTranscriptsPerReadNmax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignTranscriptsPerReadNmax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '10000'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - float</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min mapped length normalized</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignSplicedMateMapLminOverLmate'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignSplicedMateMapLminOverLmate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0.66'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min mapped length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum mapped length for a read mate that is spliced (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignSplicedMateMapLmin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignSplicedMateMapLmin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: alignSoftClipAtReferenceEnds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 'Yes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - 'No'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Soft clipping</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Option which allows soft clipping of alignments at the reference</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Yes: Enables soft clipping; No: Disables soft clipping.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignSoftClipAtReferenceEnds'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignSoftClipAtReferenceEnds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: 'Yes'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min overhang</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignSJoverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignSJoverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '5'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: 'Min overhang: annotated'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         alignments (int&gt;0).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignSJDBoverhangMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignSJDBoverhangMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '3'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max mates gap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Maximum gap between two mates, if 0, max intron gap will be determined</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         by (2^winBinNbits)*winAnchorDistNbins.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignMatesGapMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignMatesGapMax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Min intron size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Minimum intron size: genomic gap is considered intron if its length</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignIntronMin'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignIntronMin</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '21'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - int</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Max intron size</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Maximum intron size, if 0, max intron size will be determined by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (2^winBinNbits)*winAnchorDistNbins.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignIntronMax'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignIntronMax</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: '0'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - name: alignEndsType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         symbols:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - Local</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         - EndToEnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: enum</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Alignment type</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: 'Type of read ends alignment. Local: standard local alignment with</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#alignEndsType'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       inputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         prefix: --alignEndsType</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         shellQuote: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         sbg:cmdInclude: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         separator: ' '</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:category: Alignments and Seeding</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:toolDefaultValue: Local</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       required: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     outputs:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Unmapped reads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Output of unmapped reads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#unmapped_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*Unmapped.out*'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: FASTQ</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Transcriptome alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Alignments translated into transcript coordinates.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#transcriptome_aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*Transcriptome*'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Splice junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: High confidence collapsed splice junctions in tab-delimited format.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Only junctions supported by uniquely mapping reads are reported.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#splice_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*SJ.out.tab'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: TAB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Reads per gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: File with number of reads per gene. A read is counted if it overlaps</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         (1nt or more) one and only one gene.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#reads_per_gene'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*ReadsPerGene*'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: TAB</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - items: File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         type: array</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Log files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Log files produced during alignment.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#log_files'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*Log*.out'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: OUT</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Intermediate genome files</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Archive with genome files produced when annotations are included</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         on the fly (in the mapping step).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#intermediate_genome'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*_STARgenome.tar'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: TAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Chimeric junctions</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         'Chimeric Junctions' won't be output.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimeric_junctions'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*Chimeric.out.junction'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: JUNCTION</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Chimeric alignments</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Alignment SAM and Chimeric Junctions outputs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#chimeric_alignments'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob: '*.Chimeric.out.sam'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: SAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - type:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - 'null'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - File</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       label: Aligned SAM/BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       description: Aligned sequence in SAM/BAM format.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       streamable: no</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#aligned_reads'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       outputBinding:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         glob:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               if ($job.inputs.outSortingType == 'SortedByCoordinate') {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 sort_name = '.sortedByCoord'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 sort_name = ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               if ($job.inputs.outSAMtype == 'BAM') {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 sam_name = "*.Aligned.out.sam"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return sam_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       sbg:fileTypes: SAM, BAM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: ExpressionEngineRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       id: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       requirements:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         dockerPull: rabix/js-engine</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     hints:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: DockerRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       dockerImageId: a4b0ad2c3cae</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:MemRequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 60000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: sbg:CPURequirement</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       value: 15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     label: STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     description: STAR is an ultrafast universal RNA-seq aligner. It has very high</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       mapping speed, accurate alignment of contiguous and spliced reads, detection</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       with reads starting from lengths ~15 bases up to ~300 bases. In case of having</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       longer reads, use of STAR Long is recommended.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     class: CommandLineTool</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     baseCommand:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - tar</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - -xvf</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: $job.inputs.genome.path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - '&amp;&amp;'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - /opt/STAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - --runThreadN</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           return $job.allocatedResources.cpu</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     arguments:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             file = [].concat($job.inputs.reads)[0].path</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             extension = /(?:\.([^.]+))?$/.exec(file)[1]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if (extension == "gz") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return "--readFilesCommand zcat"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             } else if (extension == "bz2") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return "--readFilesCommand bzcat"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \")\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if ($job.inputs.twopassMode == "Basic") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if ($job.inputs.chimOutType == "Within") {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             else {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return "--chimOutType SeparateSAMold"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \   if (value == \"\") {\n      return\n    }\n    else {\n      return</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \".concat(param_list.join(\" \"))\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: |-</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               return "--quantMode ".concat($job.inputs.quantMode)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           }</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 100</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       prefix: --outFileNamePrefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     - position: 101</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       separate: yes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       valueFrom:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         engine: '#cwl-js-engine'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           \" \", mate1fq)\n  }\n}"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         class: Expression</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdin: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     stdout: ''</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     successCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     temporaryFailCodes: []</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     x: 624.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     'y': 323</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:x: 700.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   sbg:y: 200.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:canvas_zoom: 0.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:canvas_y: -16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sbg:canvas_x: -41</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Phil Webster, Soner Koc, Nan Xiao, Tengfei Yin, Velsera.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

    </footer></div>

  

  

  </body></html>

