<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Build workflow — SBGWorkflow-class • sevenbridges-r</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Build workflow — SBGWorkflow-class" />
<meta property="og:description" content="Build workflow" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-87185344-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-87185344-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sevenbridges-r</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.19.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/api.html">Complete Guide for Seven Bridges API R Client</a>
    </li>
    <li>
      <a href="../articles/apps.html">Describe and Execute CWL Tools/Workflows in R</a>
    </li>
    <li>
      <a href="../articles/bioc-workflow.html">Master Tutorial: Use R for Cancer Genomics Cloud</a>
    </li>
    <li>
      <a href="../articles/cgc-datasets.html">Find Data on CGC via Data Browser and Datasets API</a>
    </li>
    <li>
      <a href="../articles/docker.html">Creating Your Docker Container and Command Line Interface (with docopt)</a>
    </li>
    <li>
      <a href="../articles/rstudio.html">IDE Container: RStudio Server, Shiny Server, and More</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sbg/sevenbridges-r/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Build workflow</h1>
    <small class="dont-index">Source: <a href='https://github.com/sbg/sevenbridges-r/blob/master/R/class-flow.R'><code>R/class-flow.R</code></a></small>
    <div class="hidden name"><code>Flow.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Build workflow</p>
    </div>

    <pre class="usage">Flow(
  ...,
  graph = TRUE,
  x.width = 1000,
  y.width = 400,
  x.start = 100,
  y.start = 200,
  canvas_zoom = 1,
  canvas_x = 40,
  canvas_y = 130
)

# S4 method for Tool,Tool
+(e1, e2)

# S4 method for WorkflowStepList,Tool
+(e1, e2)

# S4 method for WorkflowStepList,WorkflowStepList
+(e1, e2)

# S4 method for App,App
+(e1, e2)

# S4 method for WorkflowStepList,App
+(e1, e2)

e1 %&gt;&gt;% e2

# S4 method for Tool,Tool
%&amp;gt;&amp;gt;%(e1, e2)

# S4 method for Workflow,Tool
%&amp;gt;&amp;gt;%(e1, e2)

# S4 method for Workflow,Workflow
%&amp;gt;&amp;gt;%(e1, e2)

# S4 method for App,App
%&amp;gt;&amp;gt;%(e1, e2)

# S4 method for Workflow,App
%&amp;gt;&amp;gt;%(e1, e2)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>...</th>
      <td><p>extra arguments passed to SBGWorkflow</p></td>
    </tr>
    <tr>
      <th>graph</th>
      <td><p>if add graph coordinates or not,
used for flow visualization on Seven Bridges platforms.</p></td>
    </tr>
    <tr>
      <th>x.width</th>
      <td><p>x scale width</p></td>
    </tr>
    <tr>
      <th>y.width</th>
      <td><p>y scale width</p></td>
    </tr>
    <tr>
      <th>x.start</th>
      <td><p>node x start point for a flow</p></td>
    </tr>
    <tr>
      <th>y.start</th>
      <td><p>node y start point for a flow</p></td>
    </tr>
    <tr>
      <th>canvas_zoom</th>
      <td><p>zoom factor</p></td>
    </tr>
    <tr>
      <th>canvas_x</th>
      <td><p>canvas x</p></td>
    </tr>
    <tr>
      <th>canvas_y</th>
      <td><p>canvas y</p></td>
    </tr>
    <tr>
      <th>e1</th>
      <td><p>either Tool App or Workflow object</p></td>
    </tr>
    <tr>
      <th>e2</th>
      <td><p>either Tool App or Workflow object</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a SBGWorkflow object.</p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    

<dl>
<dt><code>copy_obj()</code></dt><dd><p>this is a hack to make copy of reference cwl object</p></dd>

<dt><code>get_input(ids, force = FALSE)</code></dt><dd><p>get input by pure input id from all steps</p></dd>

<dt><code>get_input_exposed()</code></dt><dd><p>exposed input id other than file</p></dd>

<dt><code>get_input_node()</code></dt><dd><p>get input file nodes id</p></dd>

<dt><code>get_input_port()</code></dt><dd><p>show included port of all inputs</p></dd>

<dt><code>get_output(ids, force = FALSE)</code></dt><dd><p>get output by pure output id from all steps</p></dd>

<dt><code>get_output_node()</code></dt><dd><p>get output file nodes id</p></dd>

<dt><code>get_required()</code></dt><dd><p>show flow required input id and types</p></dd>

<dt><code>get_step(name = NULL, id = NULL)</code></dt><dd><p>get step object by name or id, name support pattern match</p></dd>

<dt><code>get_tool(name = NULL, id = NULL)</code></dt><dd><p>get a tool object by name or id, name support pattern match</p></dd>

<dt><code>input_id()</code></dt><dd><p>show input id</p></dd>

<dt><code><a href='input_matrix.html'>input_matrix(
  new.order = c("id", "label", "type", "required", "prefix", "fileTypes"),
  required = NULL
)</a></code></dt><dd><p>This return a matrix of input parameters, by default, following the order id, label, type, required, prefix, fileTypes. new.order accept names of column you want to print, but it has to be a field of inputs. When its set to NULL, it prints all fields. When required = TRUE, only print required field.</p></dd>

<dt><code>input_type()</code></dt><dd><p>Show a vector of flow input type, names of them are input id.</p></dd>

<dt><code>link_map()</code></dt><dd><p>show a table of all linked nodes</p></dd>

<dt><code>linked_input_id()</code></dt><dd><p>input id that linked to an output</p></dd>

<dt><code>linked_output_id()</code></dt><dd><p>output id that linked to an input</p></dd>

<dt><code>list_tool()</code></dt><dd><p>list all tools included in this flow</p></dd>

<dt><code>output_id()</code></dt><dd><p>show output id</p></dd>

<dt><code><a href='input_matrix.html'>output_matrix(new.order = c("id", "label", "type", "fileTypes"))</a></code></dt><dd><p>This return a matrix of output parameters, by default, following the order id, label, type, fileTypes. new.order accept names of column you want to print, but it has to be a field of outputs. When its set to NULL, it prints all fields. When required = TRUE, only print required field.</p></dd>

<dt><code>output_type()</code></dt><dd><p>Show a vector of flow output type, names of them are output id.</p></dd>

<dt><code>run(run_inputs = list(), engine = c("bunny", "rabix", "cwlrun"))</code></dt><dd><p>Run this tool with inputs locally. Engines supported: bunny, rabix, cwlrun. Inputs accept list or JSON.</p></dd>

<dt><code>set_batch(input = NULL, criteria = NULL, type = c("ITEM", "CRITERIA"))</code></dt><dd><p>Set a flow input node into a batch mode, this is now required before you execute a batch task on a batch-not-enabled flow.</p></dd>

<dt><code>set_flow_input(iid = NULL, add = TRUE)</code></dt><dd><p>Expose tool input node as flow input, default is additative, if add = FALSE, this will overwrite and only made provided id inputs of flow.</p></dd>

<dt><code>set_flow_output(oid = NULL, add = TRUE)</code></dt><dd><p>Expose tool output node as flow output, default is additative, if add = FALSE, this will overwrite and only made provided id outputs of flow.</p></dd>

<dt><code>set_input_port(ids, include = TRUE)</code></dt><dd><p>set included port for provided input id(s)</p></dd>

<dt><code>set_required(ids, required = TRUE)</code></dt><dd><p>Set a input node required (TRUE) or not required (FALSE) this require full input id (with tool id prefix) such as #STAR.alignIntronMax</p></dd>

<dt><code>step_input_id(full = FALSE)</code></dt><dd><p>Show step input id, default names of them is tool id. When full = TRUE, show full name then names of vector is type.</p></dd>

<dt><code>step_output_id(full = FALSE)</code></dt><dd><p>Show step output id, default names of them is tool id. when full = TRUE, show full name then names of vector is type.</p></dd>

</dl>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>f1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata/app"</span>, <span class='st'>"flow_star.json"</span>, package <span class='op'>=</span> <span class='st'>"sevenbridges"</span><span class='op'>)</span>
<span class='va'>f1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='convert_app.html'>convert_app</a></span><span class='op'>(</span><span class='va'>f1</span><span class='op'>)</span>
<span class='co'># input matrix</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_matrix</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                id                    label    type required
#&gt; 1                    #sjdbGTFfile              sjdbGTFfile File...    FALSE
#&gt; 2                          #fastq                    fastq File...     TRUE
#&gt; 3               #genomeFastaFiles         genomeFastaFiles    File     TRUE
#&gt; 4 #sjdbGTFtagExonParentTranscript      Exons' parents name  string    FALSE
#&gt; 5       #sjdbGTFtagExonParentGene                Gene name  string    FALSE
#&gt; 6          #winAnchorMultimapNmax         Max loci anchors     int    FALSE
#&gt; 7             #winAnchorDistNbins Max bins between anchors     int    FALSE
#&gt;   fileTypes
#&gt; 1      null
#&gt; 2      null
#&gt; 3      null
#&gt; 4      null
#&gt; 5      null
#&gt; 6      null
#&gt; 7      null</div><div class='input'><span class='co'># by name</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_matrix</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"id"</span>, <span class='st'>"type"</span>, <span class='st'>"required"</span>, <span class='st'>"link_to"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                id    type required
#&gt; 1                    #sjdbGTFfile File...    FALSE
#&gt; 2                          #fastq File...     TRUE
#&gt; 3               #genomeFastaFiles    File     TRUE
#&gt; 4 #sjdbGTFtagExonParentTranscript  string    FALSE
#&gt; 5       #sjdbGTFtagExonParentGene  string    FALSE
#&gt; 6          #winAnchorMultimapNmax     int    FALSE
#&gt; 7             #winAnchorDistNbins     int    FALSE
#&gt;                                                 link_to
#&gt; 1 #STAR_Genome_Generate.sjdbGTFfile | #STAR.sjdbGTFfile
#&gt; 2                     #SBG_FASTQ_Quality_Detector.fastq
#&gt; 3                #STAR_Genome_Generate.genomeFastaFiles
#&gt; 4  #STAR_Genome_Generate.sjdbGTFtagExonParentTranscript
#&gt; 5        #STAR_Genome_Generate.sjdbGTFtagExonParentGene
#&gt; 6                           #STAR.winAnchorMultimapNmax
#&gt; 7                              #STAR.winAnchorDistNbins</div><div class='input'><span class='co'># return only required</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_matrix</span><span class='op'>(</span>required <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                  id            label    type required fileTypes
#&gt; 2            #fastq            fastq File...     TRUE      null
#&gt; 3 #genomeFastaFiles genomeFastaFiles    File     TRUE      null</div><div class='input'><span class='co'># return everything</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_matrix</span><span class='op'>(</span><span class='cn'>NULL</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                id    type required fileTypes
#&gt; 1                    #sjdbGTFfile File...    FALSE      null
#&gt; 2                          #fastq File...     TRUE      null
#&gt; 3               #genomeFastaFiles    File     TRUE      null
#&gt; 4 #sjdbGTFtagExonParentTranscript  string    FALSE      null
#&gt; 5       #sjdbGTFtagExonParentGene  string    FALSE      null
#&gt; 6          #winAnchorMultimapNmax     int    FALSE      null
#&gt; 7             #winAnchorDistNbins     int    FALSE      null
#&gt;                      label                       category stageInput streamable
#&gt; 1              sjdbGTFfile                           null       null      FALSE
#&gt; 2                    fastq                           null       null      FALSE
#&gt; 3         genomeFastaFiles                           null       null      FALSE
#&gt; 4      Exons' parents name Splice junctions db parameters       null      FALSE
#&gt; 5                Gene name Splice junctions db parameters       null      FALSE
#&gt; 6         Max loci anchors      Windows, Anchors, Binning       null      FALSE
#&gt; 7 Max bins between anchors      Windows, Anchors, Binning       null      FALSE
#&gt;    sbg.x    sbg.y sbg.includeInPorts
#&gt; 1 160.50 195.0833                 NA
#&gt; 2 164.25 323.7500               TRUE
#&gt; 3 167.75 469.9999                 NA
#&gt; 4 200.00 350.0000                 NA
#&gt; 5 200.00 400.0000                 NA
#&gt; 6 200.00 450.0000                 NA
#&gt; 7 200.00 500.0000                 NA
#&gt;                                                                                          description
#&gt; 1                                                                                               &lt;NA&gt;
#&gt; 2                                                                                               &lt;NA&gt;
#&gt; 3                                                                                               &lt;NA&gt;
#&gt; 4                                                  Tag name to be used as exons' transcript-parents.
#&gt; 5                                                        Tag name to be used as exons' gene-parents.
#&gt; 6                                          Max number of loci anchors are allowed to map to (int&gt;0).
#&gt; 7 Max number of bins between two anchors that allows aggregation of anchors into one window (int&gt;0).
#&gt;   sbg.toolDefaultValue                                               link_to
#&gt; 1                 &lt;NA&gt; #STAR_Genome_Generate.sjdbGTFfile | #STAR.sjdbGTFfile
#&gt; 2                 &lt;NA&gt;                     #SBG_FASTQ_Quality_Detector.fastq
#&gt; 3                 &lt;NA&gt;                #STAR_Genome_Generate.genomeFastaFiles
#&gt; 4        transcript_id  #STAR_Genome_Generate.sjdbGTFtagExonParentTranscript
#&gt; 5              gene_id        #STAR_Genome_Generate.sjdbGTFtagExonParentGene
#&gt; 6                   50                           #STAR.winAnchorMultimapNmax
#&gt; 7                    9                              #STAR.winAnchorDistNbins</div><div class='input'><span class='co'># return a output matrix with more informtion</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>output_matrix</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                              id                       label    type fileTypes
#&gt; 1               #unmapped_reads              unmapped_reads File...      null
#&gt; 2  #transcriptome_aligned_reads transcriptome_aligned_reads    File      null
#&gt; 3             #splice_junctions            splice_junctions    File      null
#&gt; 4               #reads_per_gene              reads_per_gene    File      null
#&gt; 5                    #log_files                   log_files File...      null
#&gt; 6           #chimeric_junctions          chimeric_junctions    File      null
#&gt; 7          #intermediate_genome         intermediate_genome    File      null
#&gt; 8          #chimeric_alignments         chimeric_alignments    File      null
#&gt; 9                   #sorted_bam                  sorted_bam    File      null
#&gt; 10                      #result                      result    File      null</div><div class='input'><span class='co'># return only a few fields</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>output_matrix</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"id"</span>, <span class='st'>"type"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                              id    type
#&gt; 1               #unmapped_reads File...
#&gt; 2  #transcriptome_aligned_reads    File
#&gt; 3             #splice_junctions    File
#&gt; 4               #reads_per_gene    File
#&gt; 5                    #log_files File...
#&gt; 6           #chimeric_junctions    File
#&gt; 7          #intermediate_genome    File
#&gt; 8          #chimeric_alignments    File
#&gt; 9                   #sorted_bam    File
#&gt; 10                      #result    File</div><div class='input'><span class='co'># return everything</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>output_matrix</span><span class='op'>(</span><span class='cn'>NULL</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                              id                       label    type fileTypes
#&gt; 1               #unmapped_reads              unmapped_reads File...      null
#&gt; 2  #transcriptome_aligned_reads transcriptome_aligned_reads    File      null
#&gt; 3             #splice_junctions            splice_junctions    File      null
#&gt; 4               #reads_per_gene              reads_per_gene    File      null
#&gt; 5                    #log_files                   log_files File...      null
#&gt; 6           #chimeric_junctions          chimeric_junctions    File      null
#&gt; 7          #intermediate_genome         intermediate_genome    File      null
#&gt; 8          #chimeric_alignments         chimeric_alignments    File      null
#&gt; 9                   #sorted_bam                  sorted_bam    File      null
#&gt; 10                      #result                      result    File      null
#&gt;    required                             source streamable sbg.includeInPorts
#&gt; 1     FALSE               #STAR.unmapped_reads      FALSE               TRUE
#&gt; 2     FALSE  #STAR.transcriptome_aligned_reads      FALSE               TRUE
#&gt; 3     FALSE             #STAR.splice_junctions      FALSE               TRUE
#&gt; 4     FALSE               #STAR.reads_per_gene      FALSE               TRUE
#&gt; 5     FALSE                    #STAR.log_files      FALSE               TRUE
#&gt; 6     FALSE           #STAR.chimeric_junctions      FALSE               TRUE
#&gt; 7     FALSE          #STAR.intermediate_genome      FALSE               TRUE
#&gt; 8     FALSE          #STAR.chimeric_alignments      FALSE               TRUE
#&gt; 9     FALSE         #Picard_SortSam.sorted_bam      FALSE               TRUE
#&gt; 10    FALSE #SBG_FASTQ_Quality_Detector.result      FALSE               TRUE
#&gt;        sbg.x     sbg.y                            link_to
#&gt; 1   766.2498 159.58331               #STAR.unmapped_reads
#&gt; 2  1118.9998  86.58332  #STAR.transcriptome_aligned_reads
#&gt; 3  1282.3330 167.49998             #STAR.splice_junctions
#&gt; 4  1394.4164 245.74996               #STAR.reads_per_gene
#&gt; 5  1505.0830 322.99995                    #STAR.log_files
#&gt; 6  1278.7498 446.74996           #STAR.chimeric_junctions
#&gt; 7  1408.9165 386.08329          #STAR.intermediate_genome
#&gt; 8  1147.5831 503.24993          #STAR.chimeric_alignments
#&gt; 9   934.2498 557.24984         #Picard_SortSam.sorted_bam
#&gt; 10 1431.6667 644.99999 #SBG_FASTQ_Quality_Detector.result</div><div class='input'><span class='co'># flow inputs</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_type</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                    sjdbGTFfile                          fastq 
#&gt;                      "File..."                      "File..." 
#&gt;               genomeFastaFiles sjdbGTFtagExonParentTranscript 
#&gt;                         "File"                       "string" 
#&gt;       sjdbGTFtagExonParentGene          winAnchorMultimapNmax 
#&gt;                       "string"                          "int" 
#&gt;             winAnchorDistNbins 
#&gt;                          "int" </div><div class='input'><span class='co'># flow outouts</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>output_type</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;              unmapped_reads transcriptome_aligned_reads 
#&gt;                   "File..."                      "File" 
#&gt;            splice_junctions              reads_per_gene 
#&gt;                      "File"                      "File" 
#&gt;                   log_files          chimeric_junctions 
#&gt;                   "File..."                      "File" 
#&gt;         intermediate_genome         chimeric_alignments 
#&gt;                      "File"                      "File" 
#&gt;                  sorted_bam                      result 
#&gt;                      "File"                      "File" </div><div class='input'><span class='co'># flow input id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "#sjdbGTFfile"                    "#fastq"                         
#&gt; [3] "#genomeFastaFiles"               "#sjdbGTFtagExonParentTranscript"
#&gt; [5] "#sjdbGTFtagExonParentGene"       "#winAnchorMultimapNmax"         
#&gt; [7] "#winAnchorDistNbins"            </div><div class='input'><span class='co'># linked input id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>linked_input_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript"
#&gt;  [2] "#STAR_Genome_Generate.sjdbGTFtagExonParentGene"      
#&gt;  [3] "#STAR_Genome_Generate.sjdbGTFfile"                   
#&gt;  [4] "#STAR_Genome_Generate.genomeFastaFiles"              
#&gt;  [5] "#SBG_FASTQ_Quality_Detector.fastq"                   
#&gt;  [6] "#Picard_SortSam.input_bam"                           
#&gt;  [7] "#STAR.winAnchorMultimapNmax"                         
#&gt;  [8] "#STAR.winAnchorDistNbins"                            
#&gt;  [9] "#STAR.sjdbGTFfile"                                   
#&gt; [10] "#STAR.reads"                                         
#&gt; [11] "#STAR.genome"                                        </div><div class='input'><span class='co'># flow output id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>output_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  [1] "#unmapped_reads"              "#transcriptome_aligned_reads"
#&gt;  [3] "#splice_junctions"            "#reads_per_gene"             
#&gt;  [5] "#log_files"                   "#chimeric_junctions"         
#&gt;  [7] "#intermediate_genome"         "#chimeric_alignments"        
#&gt;  [9] "#sorted_bam"                  "#result"                     </div><div class='input'><span class='co'># linked output id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>linked_output_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "#STAR.aligned_reads"                "#SBG_FASTQ_Quality_Detector.result"
#&gt; [3] "#STAR_Genome_Generate.genome"      </div><div class='input'><span class='co'># link_map</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>link_map</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                                      id
#&gt; 1  #STAR_Genome_Generate.sjdbGTFtagExonParentTranscript
#&gt; 2        #STAR_Genome_Generate.sjdbGTFtagExonParentGene
#&gt; 3                     #STAR_Genome_Generate.sjdbGTFfile
#&gt; 4                #STAR_Genome_Generate.genomeFastaFiles
#&gt; 5                     #SBG_FASTQ_Quality_Detector.fastq
#&gt; 6                             #Picard_SortSam.input_bam
#&gt; 7                           #STAR.winAnchorMultimapNmax
#&gt; 8                              #STAR.winAnchorDistNbins
#&gt; 9                                     #STAR.sjdbGTFfile
#&gt; 10                                          #STAR.reads
#&gt; 11                                         #STAR.genome
#&gt; 12                                      #unmapped_reads
#&gt; 13                         #transcriptome_aligned_reads
#&gt; 14                                    #splice_junctions
#&gt; 15                                      #reads_per_gene
#&gt; 16                                           #log_files
#&gt; 17                                  #chimeric_junctions
#&gt; 18                                 #intermediate_genome
#&gt; 19                                 #chimeric_alignments
#&gt; 20                                          #sorted_bam
#&gt; 21                                              #result
#&gt;                                source   type
#&gt; 1     #sjdbGTFtagExonParentTranscript  input
#&gt; 2           #sjdbGTFtagExonParentGene  input
#&gt; 3                        #sjdbGTFfile  input
#&gt; 4                   #genomeFastaFiles  input
#&gt; 5                              #fastq  input
#&gt; 6                 #STAR.aligned_reads  input
#&gt; 7              #winAnchorMultimapNmax  input
#&gt; 8                 #winAnchorDistNbins  input
#&gt; 9                        #sjdbGTFfile  input
#&gt; 10 #SBG_FASTQ_Quality_Detector.result  input
#&gt; 11       #STAR_Genome_Generate.genome  input
#&gt; 12               #STAR.unmapped_reads output
#&gt; 13  #STAR.transcriptome_aligned_reads output
#&gt; 14             #STAR.splice_junctions output
#&gt; 15               #STAR.reads_per_gene output
#&gt; 16                    #STAR.log_files output
#&gt; 17           #STAR.chimeric_junctions output
#&gt; 18          #STAR.intermediate_genome output
#&gt; 19          #STAR.chimeric_alignments output
#&gt; 20         #Picard_SortSam.sorted_bam output
#&gt; 21 #SBG_FASTQ_Quality_Detector.result output</div><div class='input'><span class='co'># all step input id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>step_input_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;               #STAR_Genome_Generate               #STAR_Genome_Generate 
#&gt;                        "#sjdbScore"                     "#sjdbOverhang" 
#&gt;               #STAR_Genome_Generate               #STAR_Genome_Generate 
#&gt;   "#sjdbGTFtagExonParentTranscript"         "#sjdbGTFtagExonParentGene" 
#&gt;               #STAR_Genome_Generate               #STAR_Genome_Generate 
#&gt;                      "#sjdbGTFfile"               "#sjdbGTFfeatureExon" 
#&gt;               #STAR_Genome_Generate               #STAR_Genome_Generate 
#&gt;                 "#sjdbGTFchrPrefix"                  "#genomeSAsparseD" 
#&gt;               #STAR_Genome_Generate               #STAR_Genome_Generate 
#&gt;              "#genomeSAindexNbases"                 "#genomeFastaFiles" 
#&gt;               #STAR_Genome_Generate         #SBG_FASTQ_Quality_Detector 
#&gt;                "#genomeChrBinNbits"                            "#fastq" 
#&gt;                     #Picard_SortSam                     #Picard_SortSam 
#&gt;            "#validation_stringency"                       "#sort_order" 
#&gt;                     #Picard_SortSam                     #Picard_SortSam 
#&gt;                            "#quiet"                      "#output_type" 
#&gt;                     #Picard_SortSam                     #Picard_SortSam 
#&gt;                   "#memory_per_job"               "#max_records_in_ram" 
#&gt;                     #Picard_SortSam                     #Picard_SortSam 
#&gt;                        "#input_bam"                     "#create_index" 
#&gt;                     #Picard_SortSam                               #STAR 
#&gt;                "#compression_level"                    "#winFlankNbins" 
#&gt;                               #STAR                               #STAR 
#&gt;                      "#winBinNbits"            "#winAnchorMultimapNmax" 
#&gt;                               #STAR                               #STAR 
#&gt;               "#winAnchorDistNbins"                      "#twopassMode" 
#&gt;                               #STAR                               #STAR 
#&gt;                   "#twopass1readsN"                        "#sjdbScore" 
#&gt;                               #STAR                               #STAR 
#&gt;                     "#sjdbOverhang"                   "#sjdbInsertSave" 
#&gt;                               #STAR                               #STAR 
#&gt;   "#sjdbGTFtagExonParentTranscript"         "#sjdbGTFtagExonParentGene" 
#&gt;                               #STAR                               #STAR 
#&gt;                      "#sjdbGTFfile"               "#sjdbGTFfeatureExon" 
#&gt;                               #STAR                               #STAR 
#&gt;                 "#sjdbGTFchrPrefix"     "#seedSearchStartLmaxOverLread" 
#&gt;                               #STAR                               #STAR 
#&gt;              "#seedSearchStartLmax"                   "#seedSearchLmax" 
#&gt;                               #STAR                               #STAR 
#&gt;                "#seedPerWindowNmax"                  "#seedPerReadNmax" 
#&gt;                               #STAR                               #STAR 
#&gt;            "#seedNoneLociPerWindow"                 "#seedMultimapNmax" 
#&gt;                               #STAR                               #STAR 
#&gt;               "#scoreStitchSJshift"                     "#scoreInsOpen" 
#&gt;                               #STAR                               #STAR 
#&gt;                     "#scoreInsBase"      "#scoreGenomicLengthLog2scale" 
#&gt;                               #STAR                               #STAR 
#&gt;                   "#scoreGapNoncan"                     "#scoreGapGCAG" 
#&gt;                               #STAR                               #STAR 
#&gt;                     "#scoreGapATAC"                         "#scoreGap" 
#&gt;                               #STAR                               #STAR 
#&gt;                     "#scoreDelOpen"                     "#scoreDelBase" 
#&gt;                               #STAR                               #STAR 
#&gt;                    "#rg_seq_center"                     "#rg_sample_id" 
#&gt;                               #STAR                               #STAR 
#&gt;              "#rg_platform_unit_id"                      "#rg_platform" 
#&gt;                               #STAR                               #STAR 
#&gt;                           "#rg_mfl"                    "#rg_library_id" 
#&gt;                               #STAR                               #STAR 
#&gt;                            "#reads"               "#readMatesLengthsIn" 
#&gt;                               #STAR                               #STAR 
#&gt;                    "#readMapNumber"            "#quantTranscriptomeBan" 
#&gt;                               #STAR                               #STAR 
#&gt;                        "#quantMode"                   "#outSortingType" 
#&gt;                               #STAR                               #STAR 
#&gt;                 "#outSJfilterReads"           "#outSJfilterOverhangMin" 
#&gt;                               #STAR                               #STAR 
#&gt;      "#outSJfilterIntronMaxVsReadN"      "#outSJfilterDistToOtherSJmin" 
#&gt;                               #STAR                               #STAR 
#&gt;        "#outSJfilterCountUniqueMin"         "#outSJfilterCountTotalMin" 
#&gt;                               #STAR                               #STAR 
#&gt;                   "#outSAMunmapped"                       "#outSAMtype" 
#&gt;                               #STAR                               #STAR 
#&gt;                "#outSAMstrandField"                     "#outSAMreadID" 
#&gt;                               #STAR                               #STAR 
#&gt;                "#outSAMprimaryFlag"                      "#outSAMorder" 
#&gt;                               #STAR                               #STAR 
#&gt;                       "#outSAMmode"                 "#outSAMmapqUnique" 
#&gt;                               #STAR                               #STAR 
#&gt;                   "#outSAMheaderPG"                   "#outSAMheaderHD" 
#&gt;                               #STAR                               #STAR 
#&gt;                     "#outSAMflagOR"                    "#outSAMflagAND" 
#&gt;                               #STAR                               #STAR 
#&gt;                 "#outSAMattributes"                 "#outReadsUnmapped" 
#&gt;                               #STAR                               #STAR 
#&gt;               "#outQSconversionAdd"                    "#outFilterType" 
#&gt;                               #STAR                               #STAR 
#&gt;       "#outFilterScoreMinOverLread"                "#outFilterScoreMin" 
#&gt;                               #STAR                               #STAR 
#&gt;      "#outFilterMultimapScoreRange"            "#outFilterMultimapNmax" 
#&gt;                               #STAR                               #STAR 
#&gt;   "#outFilterMismatchNoverReadLmax"       "#outFilterMismatchNoverLmax" 
#&gt;                               #STAR                               #STAR 
#&gt;            "#outFilterMismatchNmax"      "#outFilterMatchNminOverLread" 
#&gt;                               #STAR                               #STAR 
#&gt;               "#outFilterMatchNmin"            "#outFilterIntronMotifs" 
#&gt;                               #STAR                               #STAR 
#&gt;               "#limitSjdbInsertNsj"                "#limitOutSJoneRead" 
#&gt;                               #STAR                               #STAR 
#&gt;              "#limitOutSJcollapsed"                  "#limitBAMsortRAM" 
#&gt;                               #STAR                               #STAR 
#&gt;                    "#genomeDirName"                           "#genome" 
#&gt;                               #STAR                               #STAR 
#&gt;                     "#clip5pNbases"                     "#clip3pNbases" 
#&gt;                               #STAR                               #STAR 
#&gt;         "#clip3pAfterAdapterNbases"                 "#clip3pAdapterSeq" 
#&gt;                               #STAR                               #STAR 
#&gt;                 "#clip3pAdapterMMp"                   "#chimSegmentMin" 
#&gt;                               #STAR                               #STAR 
#&gt;              "#chimScoreSeparation"                     "#chimScoreMin" 
#&gt;                               #STAR                               #STAR 
#&gt;         "#chimScoreJunctionNonGTAG"                 "#chimScoreDropMax" 
#&gt;                               #STAR                               #STAR 
#&gt;                      "#chimOutType"          "#chimJunctionOverhangMin" 
#&gt;                               #STAR                               #STAR 
#&gt;          "#alignWindowsPerReadNmax"    "#alignTranscriptsPerWindowNmax" 
#&gt;                               #STAR                               #STAR 
#&gt;      "#alignTranscriptsPerReadNmax" "#alignSplicedMateMapLminOverLmate" 
#&gt;                               #STAR                               #STAR 
#&gt;          "#alignSplicedMateMapLmin"     "#alignSoftClipAtReferenceEnds" 
#&gt;                               #STAR                               #STAR 
#&gt;               "#alignSJoverhangMin"             "#alignSJDBoverhangMin" 
#&gt;                               #STAR                               #STAR 
#&gt;                 "#alignMatesGapMax"                   "#alignIntronMin" 
#&gt;                               #STAR                               #STAR 
#&gt;                   "#alignIntronMax"                    "#alignEndsType" </div><div class='input'><span class='co'># all step input full id with type</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>step_input_id</span><span class='op'>(</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                                    int 
#&gt;                      "#STAR_Genome_Generate.sjdbScore" 
#&gt;                                                    int 
#&gt;                   "#STAR_Genome_Generate.sjdbOverhang" 
#&gt;                                                 string 
#&gt; "#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript" 
#&gt;                                                 string 
#&gt;       "#STAR_Genome_Generate.sjdbGTFtagExonParentGene" 
#&gt;                                                File... 
#&gt;                    "#STAR_Genome_Generate.sjdbGTFfile" 
#&gt;                                                 string 
#&gt;             "#STAR_Genome_Generate.sjdbGTFfeatureExon" 
#&gt;                                                 string 
#&gt;               "#STAR_Genome_Generate.sjdbGTFchrPrefix" 
#&gt;                                                    int 
#&gt;                "#STAR_Genome_Generate.genomeSAsparseD" 
#&gt;                                                    int 
#&gt;            "#STAR_Genome_Generate.genomeSAindexNbases" 
#&gt;                                                   File 
#&gt;               "#STAR_Genome_Generate.genomeFastaFiles" 
#&gt;                                                 string 
#&gt;              "#STAR_Genome_Generate.genomeChrBinNbits" 
#&gt;                                                   File 
#&gt;                    "#SBG_FASTQ_Quality_Detector.fastq" 
#&gt;                                                   enum 
#&gt;                "#Picard_SortSam.validation_stringency" 
#&gt;                                                   enum 
#&gt;                           "#Picard_SortSam.sort_order" 
#&gt;                                                   enum 
#&gt;                                "#Picard_SortSam.quiet" 
#&gt;                                                   enum 
#&gt;                          "#Picard_SortSam.output_type" 
#&gt;                                                    int 
#&gt;                       "#Picard_SortSam.memory_per_job" 
#&gt;                                                    int 
#&gt;                   "#Picard_SortSam.max_records_in_ram" 
#&gt;                                                   File 
#&gt;                            "#Picard_SortSam.input_bam" 
#&gt;                                                   enum 
#&gt;                         "#Picard_SortSam.create_index" 
#&gt;                                                    int 
#&gt;                    "#Picard_SortSam.compression_level" 
#&gt;                                                    int 
#&gt;                                  "#STAR.winFlankNbins" 
#&gt;                                                    int 
#&gt;                                    "#STAR.winBinNbits" 
#&gt;                                                    int 
#&gt;                          "#STAR.winAnchorMultimapNmax" 
#&gt;                                                    int 
#&gt;                             "#STAR.winAnchorDistNbins" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.twopassMode" 
#&gt;                                                    int 
#&gt;                                 "#STAR.twopass1readsN" 
#&gt;                                                    int 
#&gt;                                      "#STAR.sjdbScore" 
#&gt;                                                    int 
#&gt;                                   "#STAR.sjdbOverhang" 
#&gt;                                                   enum 
#&gt;                                 "#STAR.sjdbInsertSave" 
#&gt;                                                 string 
#&gt;                 "#STAR.sjdbGTFtagExonParentTranscript" 
#&gt;                                                 string 
#&gt;                       "#STAR.sjdbGTFtagExonParentGene" 
#&gt;                                                File... 
#&gt;                                    "#STAR.sjdbGTFfile" 
#&gt;                                                 string 
#&gt;                             "#STAR.sjdbGTFfeatureExon" 
#&gt;                                                 string 
#&gt;                               "#STAR.sjdbGTFchrPrefix" 
#&gt;                                                  float 
#&gt;                   "#STAR.seedSearchStartLmaxOverLread" 
#&gt;                                                    int 
#&gt;                            "#STAR.seedSearchStartLmax" 
#&gt;                                                    int 
#&gt;                                 "#STAR.seedSearchLmax" 
#&gt;                                                    int 
#&gt;                              "#STAR.seedPerWindowNmax" 
#&gt;                                                    int 
#&gt;                                "#STAR.seedPerReadNmax" 
#&gt;                                                    int 
#&gt;                          "#STAR.seedNoneLociPerWindow" 
#&gt;                                                    int 
#&gt;                               "#STAR.seedMultimapNmax" 
#&gt;                                                    int 
#&gt;                             "#STAR.scoreStitchSJshift" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreInsOpen" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreInsBase" 
#&gt;                                                  float 
#&gt;                    "#STAR.scoreGenomicLengthLog2scale" 
#&gt;                                                    int 
#&gt;                                 "#STAR.scoreGapNoncan" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreGapGCAG" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreGapATAC" 
#&gt;                                                    int 
#&gt;                                       "#STAR.scoreGap" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreDelOpen" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreDelBase" 
#&gt;                                                 string 
#&gt;                                  "#STAR.rg_seq_center" 
#&gt;                                                 string 
#&gt;                                   "#STAR.rg_sample_id" 
#&gt;                                                 string 
#&gt;                            "#STAR.rg_platform_unit_id" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.rg_platform" 
#&gt;                                                 string 
#&gt;                                         "#STAR.rg_mfl" 
#&gt;                                                 string 
#&gt;                                  "#STAR.rg_library_id" 
#&gt;                                                File... 
#&gt;                                          "#STAR.reads" 
#&gt;                                                   enum 
#&gt;                             "#STAR.readMatesLengthsIn" 
#&gt;                                                    int 
#&gt;                                  "#STAR.readMapNumber" 
#&gt;                                                   enum 
#&gt;                          "#STAR.quantTranscriptomeBan" 
#&gt;                                                   enum 
#&gt;                                      "#STAR.quantMode" 
#&gt;                                                   enum 
#&gt;                                 "#STAR.outSortingType" 
#&gt;                                                   enum 
#&gt;                               "#STAR.outSJfilterReads" 
#&gt;                                                 int... 
#&gt;                         "#STAR.outSJfilterOverhangMin" 
#&gt;                                                 int... 
#&gt;                    "#STAR.outSJfilterIntronMaxVsReadN" 
#&gt;                                                 int... 
#&gt;                    "#STAR.outSJfilterDistToOtherSJmin" 
#&gt;                                                 int... 
#&gt;                      "#STAR.outSJfilterCountUniqueMin" 
#&gt;                                                 int... 
#&gt;                       "#STAR.outSJfilterCountTotalMin" 
#&gt;                                                   enum 
#&gt;                                 "#STAR.outSAMunmapped" 
#&gt;                                                   enum 
#&gt;                                     "#STAR.outSAMtype" 
#&gt;                                                   enum 
#&gt;                              "#STAR.outSAMstrandField" 
#&gt;                                                   enum 
#&gt;                                   "#STAR.outSAMreadID" 
#&gt;                                                   enum 
#&gt;                              "#STAR.outSAMprimaryFlag" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.outSAMorder" 
#&gt;                                                   enum 
#&gt;                                     "#STAR.outSAMmode" 
#&gt;                                                    int 
#&gt;                               "#STAR.outSAMmapqUnique" 
#&gt;                                                 string 
#&gt;                                 "#STAR.outSAMheaderPG" 
#&gt;                                                 string 
#&gt;                                 "#STAR.outSAMheaderHD" 
#&gt;                                                    int 
#&gt;                                   "#STAR.outSAMflagOR" 
#&gt;                                                    int 
#&gt;                                  "#STAR.outSAMflagAND" 
#&gt;                                                   enum 
#&gt;                               "#STAR.outSAMattributes" 
#&gt;                                                   enum 
#&gt;                               "#STAR.outReadsUnmapped" 
#&gt;                                                    int 
#&gt;                             "#STAR.outQSconversionAdd" 
#&gt;                                                   enum 
#&gt;                                  "#STAR.outFilterType" 
#&gt;                                                  float 
#&gt;                     "#STAR.outFilterScoreMinOverLread" 
#&gt;                                                    int 
#&gt;                              "#STAR.outFilterScoreMin" 
#&gt;                                                    int 
#&gt;                    "#STAR.outFilterMultimapScoreRange" 
#&gt;                                                    int 
#&gt;                          "#STAR.outFilterMultimapNmax" 
#&gt;                                                  float 
#&gt;                 "#STAR.outFilterMismatchNoverReadLmax" 
#&gt;                                                  float 
#&gt;                     "#STAR.outFilterMismatchNoverLmax" 
#&gt;                                                    int 
#&gt;                          "#STAR.outFilterMismatchNmax" 
#&gt;                                                  float 
#&gt;                    "#STAR.outFilterMatchNminOverLread" 
#&gt;                                                    int 
#&gt;                             "#STAR.outFilterMatchNmin" 
#&gt;                                                   enum 
#&gt;                          "#STAR.outFilterIntronMotifs" 
#&gt;                                                    int 
#&gt;                             "#STAR.limitSjdbInsertNsj" 
#&gt;                                                    int 
#&gt;                              "#STAR.limitOutSJoneRead" 
#&gt;                                                    int 
#&gt;                            "#STAR.limitOutSJcollapsed" 
#&gt;                                                    int 
#&gt;                                "#STAR.limitBAMsortRAM" 
#&gt;                                                 string 
#&gt;                                  "#STAR.genomeDirName" 
#&gt;                                                   File 
#&gt;                                         "#STAR.genome" 
#&gt;                                                 int... 
#&gt;                                   "#STAR.clip5pNbases" 
#&gt;                                                 int... 
#&gt;                                   "#STAR.clip3pNbases" 
#&gt;                                                 int... 
#&gt;                       "#STAR.clip3pAfterAdapterNbases" 
#&gt;                                              string... 
#&gt;                               "#STAR.clip3pAdapterSeq" 
#&gt;                                               float... 
#&gt;                               "#STAR.clip3pAdapterMMp" 
#&gt;                                                    int 
#&gt;                                 "#STAR.chimSegmentMin" 
#&gt;                                                    int 
#&gt;                            "#STAR.chimScoreSeparation" 
#&gt;                                                    int 
#&gt;                                   "#STAR.chimScoreMin" 
#&gt;                                                    int 
#&gt;                       "#STAR.chimScoreJunctionNonGTAG" 
#&gt;                                                    int 
#&gt;                               "#STAR.chimScoreDropMax" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.chimOutType" 
#&gt;                                                    int 
#&gt;                        "#STAR.chimJunctionOverhangMin" 
#&gt;                                                  float 
#&gt;                        "#STAR.alignWindowsPerReadNmax" 
#&gt;                                                    int 
#&gt;                  "#STAR.alignTranscriptsPerWindowNmax" 
#&gt;                                                    int 
#&gt;                    "#STAR.alignTranscriptsPerReadNmax" 
#&gt;                                                  float 
#&gt;               "#STAR.alignSplicedMateMapLminOverLmate" 
#&gt;                                                    int 
#&gt;                        "#STAR.alignSplicedMateMapLmin" 
#&gt;                                                   enum 
#&gt;                   "#STAR.alignSoftClipAtReferenceEnds" 
#&gt;                                                    int 
#&gt;                             "#STAR.alignSJoverhangMin" 
#&gt;                                                    int 
#&gt;                           "#STAR.alignSJDBoverhangMin" 
#&gt;                                                    int 
#&gt;                               "#STAR.alignMatesGapMax" 
#&gt;                                                    int 
#&gt;                                 "#STAR.alignIntronMin" 
#&gt;                                                    int 
#&gt;                                 "#STAR.alignIntronMax" 
#&gt;                                                   enum 
#&gt;                                  "#STAR.alignEndsType" </div><div class='input'><span class='co'># all step output id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>step_output_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;          #STAR_Genome_Generate    #SBG_FASTQ_Quality_Detector 
#&gt;                      "#genome"                      "#result" 
#&gt;                #Picard_SortSam                          #STAR 
#&gt;                  "#sorted_bam"              "#unmapped_reads" 
#&gt;                          #STAR                          #STAR 
#&gt; "#transcriptome_aligned_reads"            "#splice_junctions" 
#&gt;                          #STAR                          #STAR 
#&gt;              "#reads_per_gene"                   "#log_files" 
#&gt;                          #STAR                          #STAR 
#&gt;         "#intermediate_genome"          "#chimeric_junctions" 
#&gt;                          #STAR                          #STAR 
#&gt;         "#chimeric_alignments"               "#aligned_reads" </div><div class='input'><span class='co'># all step output full id with type</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>step_output_id</span><span class='op'>(</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                 File                                 File 
#&gt;       "#STAR_Genome_Generate.genome" "#SBG_FASTQ_Quality_Detector.result" 
#&gt;                                 File                              File... 
#&gt;         "#Picard_SortSam.sorted_bam"               "#STAR.unmapped_reads" 
#&gt;                                 File                                 File 
#&gt;  "#STAR.transcriptome_aligned_reads"             "#STAR.splice_junctions" 
#&gt;                                 File                              File... 
#&gt;               "#STAR.reads_per_gene"                    "#STAR.log_files" 
#&gt;                                 File                                 File 
#&gt;          "#STAR.intermediate_genome"           "#STAR.chimeric_junctions" 
#&gt;                                 File                                 File 
#&gt;          "#STAR.chimeric_alignments"                "#STAR.aligned_reads" </div><div class='input'><span class='co'># get inputs objects</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input</span><span class='op'>(</span><span class='st'>"#clip3pNbases"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; type:
#&gt; - 'null'
#&gt; - items: int
#&gt;   type: array
#&gt; label: Clip 3p bases
#&gt; description: Number of bases to clip from 3p of each mate. In case only one value
#&gt;   is given, it will be assumed the same for both mates.
#&gt; streamable: no
#&gt; id: '#clip3pNbases'
#&gt; inputBinding:
#&gt;   position: 0
#&gt;   prefix: --clip3pNbases
#&gt;   separate: yes
#&gt;   itemSeparator: ' '
#&gt;   shellQuote: no
#&gt;   sbg:cmdInclude: yes
#&gt;   streamable: no
#&gt;   separator: ' '
#&gt; sbg:category: Read parameters
#&gt; sbg:toolDefaultValue: '0'
#&gt; required: no
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#clip3pNbases"</span>, <span class='st'>"#chimScoreMin"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [[1]]
#&gt; type:
#&gt; - 'null'
#&gt; - items: int
#&gt;   type: array
#&gt; label: Clip 3p bases
#&gt; description: Number of bases to clip from 3p of each mate. In case only one value
#&gt;   is given, it will be assumed the same for both mates.
#&gt; streamable: no
#&gt; id: '#clip3pNbases'
#&gt; inputBinding:
#&gt;   position: 0
#&gt;   prefix: --clip3pNbases
#&gt;   separate: yes
#&gt;   itemSeparator: ' '
#&gt;   shellQuote: no
#&gt;   sbg:cmdInclude: yes
#&gt;   streamable: no
#&gt;   separator: ' '
#&gt; sbg:category: Read parameters
#&gt; sbg:toolDefaultValue: '0'
#&gt; required: no
#&gt; 
#&gt; [[2]]
#&gt; type:
#&gt; - 'null'
#&gt; - int
#&gt; label: Min total score
#&gt; description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt; streamable: no
#&gt; id: '#chimScoreMin'
#&gt; inputBinding:
#&gt;   position: 0
#&gt;   prefix: --chimScoreMin
#&gt;   separate: yes
#&gt;   shellQuote: no
#&gt;   sbg:cmdInclude: yes
#&gt;   streamable: no
#&gt;   separator: ' '
#&gt; sbg:category: Chimeric Alignments
#&gt; sbg:toolDefaultValue: '0'
#&gt; required: no
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#clip3pNbases"</span>, <span class='st'>"#chimScoreMin"</span>, <span class='st'>"#STAR.outFilterMismatchNoverLmax"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [[1]]
#&gt; type:
#&gt; - 'null'
#&gt; - float
#&gt; label: Mismatches to *mapped* length
#&gt; description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;   length is less than this value.
#&gt; streamable: no
#&gt; id: '#outFilterMismatchNoverLmax'
#&gt; inputBinding:
#&gt;   position: 0
#&gt;   prefix: --outFilterMismatchNoverLmax
#&gt;   separate: yes
#&gt;   shellQuote: no
#&gt;   sbg:cmdInclude: yes
#&gt;   streamable: no
#&gt;   separator: ' '
#&gt; sbg:category: Output filtering
#&gt; sbg:toolDefaultValue: '0.3'
#&gt; required: no
#&gt; 
#&gt; [[2]]
#&gt; type:
#&gt; - 'null'
#&gt; - items: int
#&gt;   type: array
#&gt; label: Clip 3p bases
#&gt; description: Number of bases to clip from 3p of each mate. In case only one value
#&gt;   is given, it will be assumed the same for both mates.
#&gt; streamable: no
#&gt; id: '#clip3pNbases'
#&gt; inputBinding:
#&gt;   position: 0
#&gt;   prefix: --clip3pNbases
#&gt;   separate: yes
#&gt;   itemSeparator: ' '
#&gt;   shellQuote: no
#&gt;   sbg:cmdInclude: yes
#&gt;   streamable: no
#&gt;   separator: ' '
#&gt; sbg:category: Read parameters
#&gt; sbg:toolDefaultValue: '0'
#&gt; required: no
#&gt; 
#&gt; [[3]]
#&gt; type:
#&gt; - 'null'
#&gt; - int
#&gt; label: Min total score
#&gt; description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt; streamable: no
#&gt; id: '#chimScoreMin'
#&gt; inputBinding:
#&gt;   position: 0
#&gt;   prefix: --chimScoreMin
#&gt;   separate: yes
#&gt;   shellQuote: no
#&gt;   sbg:cmdInclude: yes
#&gt;   streamable: no
#&gt;   separator: ' '
#&gt; sbg:category: Chimeric Alignments
#&gt; sbg:toolDefaultValue: '0'
#&gt; required: no
#&gt; </div><div class='input'><span class='co'># get outputs objects</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_output</span><span class='op'>(</span><span class='st'>"#log_files"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; type:
#&gt; - 'null'
#&gt; - items: File
#&gt;   type: array
#&gt; label: Log files
#&gt; description: Log files produced during alignment.
#&gt; streamable: no
#&gt; id: '#log_files'
#&gt; outputBinding:
#&gt;   glob: '*Log*.out'
#&gt; sbg:fileTypes: OUT
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_output</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#log_files"</span>, <span class='st'>"intermediate_genome"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [[1]]
#&gt; type:
#&gt; - 'null'
#&gt; - items: File
#&gt;   type: array
#&gt; label: Log files
#&gt; description: Log files produced during alignment.
#&gt; streamable: no
#&gt; id: '#log_files'
#&gt; outputBinding:
#&gt;   glob: '*Log*.out'
#&gt; sbg:fileTypes: OUT
#&gt; 
#&gt; [[2]]
#&gt; type:
#&gt; - 'null'
#&gt; - File
#&gt; label: Intermediate genome files
#&gt; description: Archive with genome files produced when annotations are included on the
#&gt;   fly (in the mapping step).
#&gt; streamable: no
#&gt; id: '#intermediate_genome'
#&gt; outputBinding:
#&gt;   glob: '*_STARgenome.tar'
#&gt; sbg:fileTypes: TAR
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_output</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#log_files"</span>, <span class='st'>"intermediate_genome"</span>, <span class='st'>"#STAR.unmapped_reads"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [[1]]
#&gt; type:
#&gt; - 'null'
#&gt; - items: File
#&gt;   type: array
#&gt; label: Unmapped reads
#&gt; description: Output of unmapped reads.
#&gt; streamable: no
#&gt; id: '#unmapped_reads'
#&gt; outputBinding:
#&gt;   glob: '*Unmapped.out*'
#&gt; sbg:fileTypes: FASTQ
#&gt; 
#&gt; [[2]]
#&gt; type:
#&gt; - 'null'
#&gt; - items: File
#&gt;   type: array
#&gt; label: Log files
#&gt; description: Log files produced during alignment.
#&gt; streamable: no
#&gt; id: '#log_files'
#&gt; outputBinding:
#&gt;   glob: '*Log*.out'
#&gt; sbg:fileTypes: OUT
#&gt; 
#&gt; [[3]]
#&gt; type:
#&gt; - 'null'
#&gt; - File
#&gt; label: Intermediate genome files
#&gt; description: Archive with genome files produced when annotations are included on the
#&gt;   fly (in the mapping step).
#&gt; streamable: no
#&gt; id: '#intermediate_genome'
#&gt; outputBinding:
#&gt;   glob: '*_STARgenome.tar'
#&gt; sbg:fileTypes: TAR
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_output</span><span class='op'>(</span><span class='st'>"#log_files"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; type:
#&gt; - 'null'
#&gt; - items: File
#&gt;   type: array
#&gt; label: Log files
#&gt; description: Log files produced during alignment.
#&gt; streamable: no
#&gt; id: '#log_files'
#&gt; outputBinding:
#&gt;   glob: '*Log*.out'
#&gt; sbg:fileTypes: OUT
#&gt; </div><div class='input'><span class='co'># set flow input</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_flow_input</span><span class='op'>(</span><span class='st'>"#SBG_FASTQ_Quality_Detector.fastq"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 2
#&gt; sbg:toolAuthor: Seven Bridges Genomics
#&gt; sbg:createdOn: 1463601910
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; - RNA
#&gt; sbg:contributors:
#&gt; - tengfei
#&gt; sbg:project: tengfei/quickstart
#&gt; sbg:createdBy: tengfei
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2
#&gt; sbg:license: Apache License 2.0
#&gt; sbg:revision: 2
#&gt; sbg:modifiedOn: 1463601974
#&gt; sbg:modifiedBy: tengfei
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601910
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601952
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601974
#&gt;   sbg:revision: 2
#&gt; sbg:toolkit: STAR
#&gt; id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'
#&gt; inputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: sjdbGTFfile
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:x: 160.4999759
#&gt;   sbg:y: 195.0833106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: fastq
#&gt;   streamable: no
#&gt;   id: '#fastq'
#&gt;   sbg:x: 164.2499914
#&gt;   sbg:y: 323.7499502
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - File
#&gt;   label: genomeFastaFiles
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   sbg:x: 167.7499601
#&gt;   sbg:y: 469.9999106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 350.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 450.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: unmapped_reads
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   source: '#STAR.unmapped_reads'
#&gt;   sbg:x: 766.2497863
#&gt;   sbg:y: 159.5833091
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: transcriptome_aligned_reads
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   source: '#STAR.transcriptome_aligned_reads'
#&gt;   sbg:x: 1118.9998003
#&gt;   sbg:y: 86.5833216
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: splice_junctions
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   source: '#STAR.splice_junctions'
#&gt;   sbg:x: 1282.3330177
#&gt;   sbg:y: 167.499976
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: reads_per_gene
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   source: '#STAR.reads_per_gene'
#&gt;   sbg:x: 1394.4163557
#&gt;   sbg:y: 245.749964
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: log_files
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   source: '#STAR.log_files'
#&gt;   sbg:x: 1505.0830269
#&gt;   sbg:y: 322.9999518
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: chimeric_junctions
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   source: '#STAR.chimeric_junctions'
#&gt;   sbg:x: 1278.7498062
#&gt;   sbg:y: 446.7499567
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: intermediate_genome
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source: '#STAR.intermediate_genome'
#&gt;   sbg:x: 1408.9164783
#&gt;   sbg:y: 386.0832876
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: chimeric_alignments
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   source: '#STAR.chimeric_alignments'
#&gt;   sbg:x: 1147.5831348
#&gt;   sbg:y: 503.2499285
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: sorted_bam
#&gt;   streamable: no
#&gt;   id: '#sorted_bam'
#&gt;   source: '#Picard_SortSam.sorted_bam'
#&gt;   sbg:x: 934.2498228
#&gt;   sbg:y: 557.2498436
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: result
#&gt;   streamable: no
#&gt;   id: '#result'
#&gt;   source: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   sbg:x: 1431.6666548
#&gt;   sbg:y: 644.9999898
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; requirements:
#&gt; - class: CreateFileRequirement
#&gt;   fileDef: []
#&gt; hints:
#&gt; - class: sbg:AWSInstanceType
#&gt;   value: c3.8xlarge
#&gt; label: RNA-seq Alignment - STAR
#&gt; description: "Alignment to a reference genome and transcriptome presents the first
#&gt;   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner
#&gt;   capable of mapping full length RNA sequences and detecting de novo canonical junctions,
#&gt;   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian
#&gt;   sequence reads, but fine tuning of its parameters enables customization to satisfy
#&gt;   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).
#&gt;   \ \nSplice junction annotations can optionally be collected from splice junction
#&gt;   databases. Set the \"Overhang length\" parameter to a value larger than zero in
#&gt;   order to use splice junction databases. For constant read length, this value should
#&gt;   (ideally) be equal to mate length decreased by 1; for long reads with non-constant
#&gt;   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ
#&gt;   files reveals read length distribution. STAR can detect chimeric transcripts, but
#&gt;   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted
#&gt;   to a desired minimum chimeric segment length. Aligned reads are reported in BAM
#&gt;   format and can be viewed in a genome browser (such as IGV). A file containing detected
#&gt;   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format
#&gt;   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write
#&gt;   unmapped in SAM\" parameters enable unmapped output type selection."
#&gt; class: Workflow
#&gt; steps:
#&gt; - id: '#STAR_Genome_Generate'
#&gt;   inputs:
#&gt;   - id: '#STAR_Genome_Generate.sjdbScore'
#&gt;   - id: '#STAR_Genome_Generate.sjdbOverhang'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'
#&gt;     source: '#sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'
#&gt;     source: '#sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAsparseD'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAindexNbases'
#&gt;   - id: '#STAR_Genome_Generate.genomeFastaFiles'
#&gt;     source: '#genomeFastaFiles'
#&gt;   - id: '#STAR_Genome_Generate.genomeChrBinNbits'
#&gt;   outputs:
#&gt;   - id: '#STAR_Genome_Generate.genome'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 1
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         sjdbScore: 0
#&gt;         sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;         sjdbOverhang: 0
#&gt;         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;         genomeChrBinNbits: genomeChrBinNbits
#&gt;         genomeSAsparseD: 0
#&gt;         sjdbGTFfile:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /demo/test-files/chr20.gtf
#&gt;         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;         genomeFastaFiles:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /sbgenomics/test-data/chr20.fa
#&gt;         sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;         genomeSAindexNbases: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911469
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 1
#&gt;     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa
#&gt;       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD
#&gt;       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt;     sbg:modifiedOn: 1450911470
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911469
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911470
#&gt;       sbg:revision: 1
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbScore
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbOverhang
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentTranscript
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentGene
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFfeatureExon
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFchrPrefix
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Suffux array sparsity
#&gt;       description: 'Distance between indices: use bigger numbers to decrease needed
#&gt;         RAM at the cost of mapping speed reduction (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#genomeSAsparseD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAsparseD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Pre-indexing string length
#&gt;       description: Length (bases) of the SA pre-indexing string. Typically between
#&gt;         10 and 15. Longer strings will use much more memory, but allow faster searches.
#&gt;         For small genomes, this number needs to be scaled down, with a typical value
#&gt;         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,
#&gt;         this is equal to 9, for 100 kiloBase genome, this is equal to 7.
#&gt;       streamable: no
#&gt;       id: '#genomeSAindexNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAindexNbases
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '14'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome fasta files
#&gt;       description: Reference sequence to which to align the reads.
#&gt;       streamable: no
#&gt;       id: '#genomeFastaFiles'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeFastaFiles
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FA
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Bins size
#&gt;       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins
#&gt;         for genome storage: each chromosome will occupy an integer number of bins.
#&gt;         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,
#&gt;         you may need to reduce this number to reduce RAM consumption. The following
#&gt;         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this
#&gt;         is equal to 15.'
#&gt;       streamable: no
#&gt;       id: '#genomeChrBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeChrBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '18'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Genome Files
#&gt;       description: Genome files comprise binary genome sequence, suffix arrays, text
#&gt;         chromosome names/lengths, splice junctions coordinates, and transcripts/genes
#&gt;         information.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       outputBinding:
#&gt;         glob: '*.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     label: STAR Genome Generate
#&gt;     description: STAR Genome Generate is a tool that generates genome index files.
#&gt;       One set of files should be generated per each genome/annotation combination.
#&gt;       Once produced, these files could be used as long as genome/annotation combination
#&gt;       stays the same. Also, STAR Genome Generate which produced these files and STAR
#&gt;       aligner using them must be the same toolkit version.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - mkdir
#&gt;     - genomeDir
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runMode
#&gt;     - genomeGenerate
#&gt;     - --genomeDir
#&gt;     - ./genomeDir
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 99
#&gt;       separate: yes
#&gt;       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 384.0832266
#&gt;     'y': 446.4998957
#&gt;   sbg:x: 100.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#SBG_FASTQ_Quality_Detector'
#&gt;   inputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.fastq'
#&gt;     source: '#fastq'
#&gt;   outputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 3
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 1000
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         fastq:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /path/to/fastq.ext
#&gt;     sbg:toolAuthor: Seven Bridges Genomics
#&gt;     sbg:createdOn: 1450911312
#&gt;     sbg:categories:
#&gt;     - FASTQ-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:project: bix-demo/sbgtools-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     sbg:license: Apache License 2.0
#&gt;     sbg:revision: 3
#&gt;     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext
#&gt;       /path/to/fastq.ext
#&gt;     sbg:modifiedOn: 1450911314
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911312
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911314
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: SBGTools
#&gt;     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     inputs:
#&gt;     - type:
#&gt;       - File
#&gt;       label: Fastq
#&gt;       description: FASTQ file.
#&gt;       streamable: no
#&gt;       id: '#fastq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --fastq
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       required: yes
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Result
#&gt;       description: Source FASTQ file with updated metadata.
#&gt;       streamable: no
#&gt;       id: '#result'
#&gt;       outputBinding:
#&gt;         glob: '*.fastq'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     requirements:
#&gt;     - class: CreateFileRequirement
#&gt;       fileDef: []
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0
#&gt;       dockerImageId: ''
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 1000
#&gt;     label: SBG FASTQ Quality Detector
#&gt;     description: FASTQ Quality Scale Detector detects which quality encoding scheme
#&gt;       was used in your reads and automatically enters the proper value in the "Quality
#&gt;       Scale" metadata field.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - python
#&gt;     - /opt/sbg_fastq_quality_scale_detector.py
#&gt;     arguments: []
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 375.3333179
#&gt;     'y': 323.5833156
#&gt;   sbg:x: 300.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#Picard_SortSam'
#&gt;   inputs:
#&gt;   - id: '#Picard_SortSam.validation_stringency'
#&gt;     default: SILENT
#&gt;   - id: '#Picard_SortSam.sort_order'
#&gt;     default: Coordinate
#&gt;   - id: '#Picard_SortSam.quiet'
#&gt;   - id: '#Picard_SortSam.output_type'
#&gt;   - id: '#Picard_SortSam.memory_per_job'
#&gt;   - id: '#Picard_SortSam.max_records_in_ram'
#&gt;   - id: '#Picard_SortSam.input_bam'
#&gt;     source: '#STAR.aligned_reads'
#&gt;   - id: '#Picard_SortSam.create_index'
#&gt;     default: 'True'
#&gt;   - id: '#Picard_SortSam.compression_level'
#&gt;   outputs:
#&gt;   - id: '#Picard_SortSam.sorted_bam'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 2
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 2048
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         sort_order: Coordinate
#&gt;         input_bam:
#&gt;           path: /root/dir/example.tested.bam
#&gt;         memory_per_job: 2048
#&gt;         output_type: ~
#&gt;         create_index: ~
#&gt;     sbg:toolAuthor: Broad Institute
#&gt;     sbg:createdOn: 1450911168
#&gt;     sbg:categories:
#&gt;     - SAM/BAM-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: http://broadinstitute.github.io/picard/index.html
#&gt;       label: Homepage
#&gt;     - id: https://github.com/broadinstitute/picard/releases/tag/1.138
#&gt;       label: Source Code
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Wiki
#&gt;     - id: https://github.com/broadinstitute/picard/zipball/master
#&gt;       label: Download
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/picard-1-140-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: '1.140'
#&gt;     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     sbg:license: MIT License, Apache 2.0 Licence
#&gt;     sbg:revision: 2
#&gt;     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam
#&gt;       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam
#&gt;       SORT_ORDER=coordinate  /root/dir/example.tested.bam
#&gt;     sbg:modifiedOn: 1450911170
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911168
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911169
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911170
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: Picard
#&gt;     id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: validation_stringency
#&gt;         symbols:
#&gt;         - STRICT
#&gt;         - LENIENT
#&gt;         - SILENT
#&gt;         type: enum
#&gt;       label: Validation stringency
#&gt;       description: 'Validation stringency for all SAM files read by this program.
#&gt;         Setting stringency to SILENT can improve performance when processing a BAM
#&gt;         file in which variable-length data (read, qualities, tags) do not otherwise
#&gt;         need to be decoded. This option can be set to ''null'' to clear the default
#&gt;         value. Possible values: {STRICT, LENIENT, SILENT}.'
#&gt;       streamable: no
#&gt;       id: '#validation_stringency'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: VALIDATION_STRINGENCY=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.validation_stringency)
#&gt;               {
#&gt;                 return $job.inputs.validation_stringency
#&gt;               }
#&gt;               else
#&gt;               {
#&gt;                 return "SILENT"
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SILENT
#&gt;       required: no
#&gt;     - type:
#&gt;       - name: sort_order
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - Queryname
#&gt;         - Coordinate
#&gt;         type: enum
#&gt;       label: Sort order
#&gt;       description: 'Sort order of the output file. Possible values: {unsorted, queryname,
#&gt;         coordinate}.'
#&gt;       streamable: no
#&gt;       id: '#sort_order'
#&gt;       inputBinding:
#&gt;         position: 3
#&gt;         prefix: SORT_ORDER=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               p = $job.inputs.sort_order.toLowerCase()
#&gt;               return p
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: Coordinate
#&gt;       sbg:altPrefix: SO
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quiet
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Quiet
#&gt;       description: 'This parameter indicates whether to suppress job-summary info
#&gt;         on System.err. This option can be set to ''null'' to clear the default value.
#&gt;         Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#quiet'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: QUIET=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: output_type
#&gt;         symbols:
#&gt;         - BAM
#&gt;         - SAM
#&gt;         - SAME AS INPUT
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Since Picard tools can output both SAM and BAM files, user can
#&gt;         choose the format of the output file.
#&gt;       streamable: no
#&gt;       id: '#output_type'
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SAME AS INPUT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Memory per job
#&gt;       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for
#&gt;         single threaded jobs.
#&gt;       streamable: no
#&gt;       id: '#memory_per_job'
#&gt;       sbg:toolDefaultValue: '2048'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max records in RAM
#&gt;       description: When writing SAM files that need to be sorted, this parameter will
#&gt;         specify the number of records stored in RAM before spilling to disk. Increasing
#&gt;         this number reduces the number of file handles needed to sort a SAM file,
#&gt;         and increases the amount of RAM needed. This option can be set to 'null' to
#&gt;         clear the default value.
#&gt;       streamable: no
#&gt;       id: '#max_records_in_ram'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: MAX_RECORDS_IN_RAM=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '500000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Input BAM
#&gt;       description: The BAM or SAM file to sort.
#&gt;       streamable: no
#&gt;       id: '#input_bam'
#&gt;       inputBinding:
#&gt;         position: 1
#&gt;         prefix: INPUT=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: File inputs
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;       sbg:altPrefix: I
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: create_index
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Create index
#&gt;       description: 'This parameter indicates whether to create a BAM index when writing
#&gt;         a coordinate-sorted BAM file. This option can be set to ''null'' to clear
#&gt;         the default value. Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#create_index'
#&gt;       inputBinding:
#&gt;         position: 5
#&gt;         prefix: CREATE_INDEX=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Compression level
#&gt;       description: Compression level for all compressed files created (e.g. BAM and
#&gt;         GELI). This option can be set to 'null' to clear the default value.
#&gt;       streamable: no
#&gt;       id: '#compression_level'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: COMPRESSION_LEVEL=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Sorted BAM/SAM
#&gt;       description: Sorted BAM or SAM file.
#&gt;       streamable: no
#&gt;       id: '#sorted_bam'
#&gt;       outputBinding:
#&gt;         glob: '*.sorted.?am'
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;       engineCommand: cwl-engine.js
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
#&gt;       dockerImageId: eab0e70b6629
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n
#&gt;           \ }\n  \treturn 2048\n}"
#&gt;         class: Expression
#&gt;     label: Picard SortSam
#&gt;     description: Picard SortSam sorts the input SAM or BAM. Input and output formats
#&gt;       are determined by the file extension.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - java
#&gt;     - class: Expression
#&gt;       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,
#&gt;         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"
#&gt;       engine: '#cwl-js-engine'
#&gt;     - -jar
#&gt;     - /opt/picard-tools-1.140/picard.jar
#&gt;     - SortSam
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       prefix: OUTPUT=
#&gt;       separate: no
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif
#&gt;           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,
#&gt;           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse
#&gt;           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n}\n}"
#&gt;         class: Expression
#&gt;     - position: 1000
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out
#&gt;           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n
#&gt;           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n
#&gt;           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n
#&gt;           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index
#&gt;           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==
#&gt;           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 773.0831807
#&gt;     'y': 470.9165939
#&gt;   sbg:x: 500.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#STAR'
#&gt;   inputs:
#&gt;   - id: '#STAR.winFlankNbins'
#&gt;   - id: '#STAR.winBinNbits'
#&gt;   - id: '#STAR.winAnchorMultimapNmax'
#&gt;     source: '#winAnchorMultimapNmax'
#&gt;   - id: '#STAR.winAnchorDistNbins'
#&gt;     source: '#winAnchorDistNbins'
#&gt;   - id: '#STAR.twopassMode'
#&gt;   - id: '#STAR.twopass1readsN'
#&gt;   - id: '#STAR.sjdbScore'
#&gt;   - id: '#STAR.sjdbOverhang'
#&gt;     default: 100
#&gt;   - id: '#STAR.sjdbInsertSave'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR.seedSearchStartLmaxOverLread'
#&gt;   - id: '#STAR.seedSearchStartLmax'
#&gt;   - id: '#STAR.seedSearchLmax'
#&gt;   - id: '#STAR.seedPerWindowNmax'
#&gt;   - id: '#STAR.seedPerReadNmax'
#&gt;   - id: '#STAR.seedNoneLociPerWindow'
#&gt;   - id: '#STAR.seedMultimapNmax'
#&gt;   - id: '#STAR.scoreStitchSJshift'
#&gt;   - id: '#STAR.scoreInsOpen'
#&gt;   - id: '#STAR.scoreInsBase'
#&gt;   - id: '#STAR.scoreGenomicLengthLog2scale'
#&gt;   - id: '#STAR.scoreGapNoncan'
#&gt;   - id: '#STAR.scoreGapGCAG'
#&gt;   - id: '#STAR.scoreGapATAC'
#&gt;   - id: '#STAR.scoreGap'
#&gt;   - id: '#STAR.scoreDelOpen'
#&gt;   - id: '#STAR.scoreDelBase'
#&gt;   - id: '#STAR.rg_seq_center'
#&gt;   - id: '#STAR.rg_sample_id'
#&gt;   - id: '#STAR.rg_platform_unit_id'
#&gt;   - id: '#STAR.rg_platform'
#&gt;   - id: '#STAR.rg_mfl'
#&gt;   - id: '#STAR.rg_library_id'
#&gt;   - id: '#STAR.reads'
#&gt;     source: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   - id: '#STAR.readMatesLengthsIn'
#&gt;   - id: '#STAR.readMapNumber'
#&gt;   - id: '#STAR.quantTranscriptomeBan'
#&gt;   - id: '#STAR.quantMode'
#&gt;     default: TranscriptomeSAM
#&gt;   - id: '#STAR.outSortingType'
#&gt;     default: SortedByCoordinate
#&gt;   - id: '#STAR.outSJfilterReads'
#&gt;   - id: '#STAR.outSJfilterOverhangMin'
#&gt;   - id: '#STAR.outSJfilterIntronMaxVsReadN'
#&gt;   - id: '#STAR.outSJfilterDistToOtherSJmin'
#&gt;   - id: '#STAR.outSJfilterCountUniqueMin'
#&gt;   - id: '#STAR.outSJfilterCountTotalMin'
#&gt;   - id: '#STAR.outSAMunmapped'
#&gt;   - id: '#STAR.outSAMtype'
#&gt;     default: BAM
#&gt;   - id: '#STAR.outSAMstrandField'
#&gt;   - id: '#STAR.outSAMreadID'
#&gt;   - id: '#STAR.outSAMprimaryFlag'
#&gt;   - id: '#STAR.outSAMorder'
#&gt;   - id: '#STAR.outSAMmode'
#&gt;   - id: '#STAR.outSAMmapqUnique'
#&gt;   - id: '#STAR.outSAMheaderPG'
#&gt;   - id: '#STAR.outSAMheaderHD'
#&gt;   - id: '#STAR.outSAMflagOR'
#&gt;   - id: '#STAR.outSAMflagAND'
#&gt;   - id: '#STAR.outSAMattributes'
#&gt;   - id: '#STAR.outReadsUnmapped'
#&gt;     default: Fastx
#&gt;   - id: '#STAR.outQSconversionAdd'
#&gt;   - id: '#STAR.outFilterType'
#&gt;   - id: '#STAR.outFilterScoreMinOverLread'
#&gt;   - id: '#STAR.outFilterScoreMin'
#&gt;   - id: '#STAR.outFilterMultimapScoreRange'
#&gt;   - id: '#STAR.outFilterMultimapNmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverReadLmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverLmax'
#&gt;   - id: '#STAR.outFilterMismatchNmax'
#&gt;   - id: '#STAR.outFilterMatchNminOverLread'
#&gt;   - id: '#STAR.outFilterMatchNmin'
#&gt;   - id: '#STAR.outFilterIntronMotifs'
#&gt;   - id: '#STAR.limitSjdbInsertNsj'
#&gt;   - id: '#STAR.limitOutSJoneRead'
#&gt;   - id: '#STAR.limitOutSJcollapsed'
#&gt;   - id: '#STAR.limitBAMsortRAM'
#&gt;   - id: '#STAR.genomeDirName'
#&gt;   - id: '#STAR.genome'
#&gt;     source: '#STAR_Genome_Generate.genome'
#&gt;   - id: '#STAR.clip5pNbases'
#&gt;   - id: '#STAR.clip3pNbases'
#&gt;   - id: '#STAR.clip3pAfterAdapterNbases'
#&gt;   - id: '#STAR.clip3pAdapterSeq'
#&gt;   - id: '#STAR.clip3pAdapterMMp'
#&gt;   - id: '#STAR.chimSegmentMin'
#&gt;   - id: '#STAR.chimScoreSeparation'
#&gt;   - id: '#STAR.chimScoreMin'
#&gt;   - id: '#STAR.chimScoreJunctionNonGTAG'
#&gt;   - id: '#STAR.chimScoreDropMax'
#&gt;   - id: '#STAR.chimOutType'
#&gt;   - id: '#STAR.chimJunctionOverhangMin'
#&gt;   - id: '#STAR.alignWindowsPerReadNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerWindowNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerReadNmax'
#&gt;   - id: '#STAR.alignSplicedMateMapLminOverLmate'
#&gt;   - id: '#STAR.alignSplicedMateMapLmin'
#&gt;   - id: '#STAR.alignSoftClipAtReferenceEnds'
#&gt;   - id: '#STAR.alignSJoverhangMin'
#&gt;   - id: '#STAR.alignSJDBoverhangMin'
#&gt;   - id: '#STAR.alignMatesGapMax'
#&gt;   - id: '#STAR.alignIntronMin'
#&gt;   - id: '#STAR.alignIntronMax'
#&gt;   - id: '#STAR.alignEndsType'
#&gt;   outputs:
#&gt;   - id: '#STAR.unmapped_reads'
#&gt;   - id: '#STAR.transcriptome_aligned_reads'
#&gt;   - id: '#STAR.splice_junctions'
#&gt;   - id: '#STAR.reads_per_gene'
#&gt;   - id: '#STAR.log_files'
#&gt;   - id: '#STAR.intermediate_genome'
#&gt;   - id: '#STAR.chimeric_junctions'
#&gt;   - id: '#STAR.chimeric_alignments'
#&gt;   - id: '#STAR.aligned_reads'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 4
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         alignWindowsPerReadNmax: 0
#&gt;         outSAMheaderPG: outSAMheaderPG
#&gt;         GENOME_DIR_NAME: ''
#&gt;         outFilterMatchNminOverLread: 0
#&gt;         rg_platform_unit_id: rg_platform_unit
#&gt;         alignTranscriptsPerReadNmax: 0
#&gt;         readMapNumber: 0
#&gt;         alignSplicedMateMapLminOverLmate: 0
#&gt;         alignMatesGapMax: 0
#&gt;         outFilterMultimapNmax: 0
#&gt;         clip5pNbases:
#&gt;         - 0
#&gt;         outSAMstrandField: None
#&gt;         readMatesLengthsIn: NotEqual
#&gt;         outSAMattributes: Standard
#&gt;         seedMultimapNmax: 0
#&gt;         rg_mfl: rg_mfl
#&gt;         chimSegmentMin: 0
#&gt;         winAnchorDistNbins: 0
#&gt;         outSortingType: SortedByCoordinate
#&gt;         outFilterMultimapScoreRange: 0
#&gt;         sjdbInsertSave: Basic
#&gt;         clip3pAfterAdapterNbases:
#&gt;         - 0
#&gt;         scoreDelBase: 0
#&gt;         outFilterMatchNmin: 0
#&gt;         twopass1readsN: 0
#&gt;         outSAMunmapped: None
#&gt;         genome:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: genome.ext
#&gt;         sjdbGTFtagExonParentTranscript: ''
#&gt;         limitBAMsortRAM: 0
#&gt;         alignEndsType: Local
#&gt;         seedNoneLociPerWindow: 0
#&gt;         rg_sample_id: rg_sample
#&gt;         sjdbGTFtagExonParentGene: ''
#&gt;         chimScoreMin: 0
#&gt;         outSJfilterIntronMaxVsReadN:
#&gt;         - 0
#&gt;         twopassMode: Basic
#&gt;         alignSplicedMateMapLmin: 0
#&gt;         outSJfilterReads: All
#&gt;         outSAMprimaryFlag: OneBestScore
#&gt;         outSJfilterCountTotalMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         outSAMorder: Paired
#&gt;         outSAMflagAND: 0
#&gt;         chimScoreSeparation: 0
#&gt;         alignSJoverhangMin: 0
#&gt;         outFilterScoreMin: 0
#&gt;         seedSearchStartLmax: 0
#&gt;         scoreGapGCAG: 0
#&gt;         scoreGenomicLengthLog2scale: 0
#&gt;         outFilterIntronMotifs: None
#&gt;         outFilterMismatchNmax: 0
#&gt;         reads:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           metadata:
#&gt;             format: fastq
#&gt;             paired_end: '1'
#&gt;             seq_center: illumina
#&gt;           path: /test-data/mate_1.fastq.bz2
#&gt;         scoreGap: 0
#&gt;         outSJfilterOverhangMin:
#&gt;         - 30
#&gt;         - 12
#&gt;         - 12
#&gt;         - 12
#&gt;         outSAMflagOR: 0
#&gt;         outSAMmode: Full
#&gt;         rg_library_id: ''
#&gt;         chimScoreJunctionNonGTAG: 0
#&gt;         scoreInsOpen: 0
#&gt;         clip3pAdapterSeq:
#&gt;         - clip3pAdapterSeq
#&gt;         chimScoreDropMax: 0
#&gt;         outFilterType: Normal
#&gt;         scoreGapATAC: 0
#&gt;         rg_platform: Ion Torrent PGM
#&gt;         clip3pAdapterMMp:
#&gt;         - 0
#&gt;         sjdbGTFfeatureExon: ''
#&gt;         outQSconversionAdd: 0
#&gt;         quantMode: TranscriptomeSAM
#&gt;         alignIntronMin: 0
#&gt;         scoreInsBase: 0
#&gt;         scoreGapNoncan: 0
#&gt;         seedSearchLmax: 0
#&gt;         outSJfilterDistToOtherSJmin:
#&gt;         - 0
#&gt;         outFilterScoreMinOverLread: 0
#&gt;         alignSJDBoverhangMin: 0
#&gt;         limitOutSJcollapsed: 0
#&gt;         winAnchorMultimapNmax: 0
#&gt;         outFilterMismatchNoverLmax: 0
#&gt;         rg_seq_center: ''
#&gt;         outSAMheaderHD: outSAMheaderHD
#&gt;         chimOutType: Within
#&gt;         quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;         limitOutSJoneRead: 0
#&gt;         alignTranscriptsPerWindowNmax: 0
#&gt;         sjdbOverhang: ~
#&gt;         outReadsUnmapped: Fastx
#&gt;         scoreStitchSJshift: 0
#&gt;         seedPerWindowNmax: 0
#&gt;         outSJfilterCountUniqueMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         scoreDelOpen: 0
#&gt;         sjdbGTFfile:
#&gt;         - path: /demo/test-data/chr20.gtf
#&gt;         clip3pNbases:
#&gt;         - 0
#&gt;         - 3
#&gt;         winBinNbits: 0
#&gt;         sjdbScore: ~
#&gt;         seedSearchStartLmaxOverLread: 0
#&gt;         alignIntronMax: 0
#&gt;         seedPerReadNmax: 0
#&gt;         outFilterMismatchNoverReadLmax: 0
#&gt;         winFlankNbins: 0
#&gt;         sjdbGTFchrPrefix: chrPrefix
#&gt;         alignSoftClipAtReferenceEnds: 'Yes'
#&gt;         outSAMreadID: Standard
#&gt;         outSAMtype: BAM
#&gt;         chimJunctionOverhangMin: 0
#&gt;         limitSjdbInsertNsj: 0
#&gt;         outSAMmapqUnique: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911471
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - ana_d
#&gt;     - bix-demo
#&gt;     - uros_sipetic
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star/4
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 4
#&gt;     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix
#&gt;       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline
#&gt;       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode
#&gt;       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;
#&gt;       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt;     sbg:modifiedOn: 1462889222
#&gt;     sbg:modifiedBy: ana_d
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911471
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911473
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911475
#&gt;       sbg:revision: 2
#&gt;     - sbg:modifiedBy: uros_sipetic
#&gt;       sbg:modifiedOn: 1462878528
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: ana_d
#&gt;       sbg:modifiedOn: 1462889222
#&gt;       sbg:revision: 4
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star/4
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Flanking regions size
#&gt;       description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;         flanking regions for each window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winFlankNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winFlankNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Bin size
#&gt;       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;         each window will occupy an integer number of bins (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '16'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max loci anchors
#&gt;       description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max bins between anchors
#&gt;       description: Max number of bins between two anchors that allows aggregation
#&gt;         of anchors into one window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorDistNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorDistNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '9'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: twopassMode
#&gt;         symbols:
#&gt;         - None
#&gt;         - Basic
#&gt;         type: enum
#&gt;       label: Two-pass mode
#&gt;       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass
#&gt;         mapping, with all 1st pass junctions inserted into the genome indices on the
#&gt;         fly.'
#&gt;       streamable: no
#&gt;       id: '#twopassMode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --twopassMode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to process in 1st step
#&gt;       description: 'Number of reads to process for the 1st step. 0: 1-step only, no
#&gt;         2nd pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#twopass1readsN'
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: sjdbInsertSave
#&gt;         symbols:
#&gt;         - Basic
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: Save junction files
#&gt;       description: 'Which files to save when sjdb junctions are inserted on the fly
#&gt;         at the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;         files; All: all files including big Genome, SA and SAindex. These files are
#&gt;         output as archive.'
#&gt;       streamable: no
#&gt;       id: '#sjdbInsertSave'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts. No need to include
#&gt;         this input, except in case of using "on the fly" annotations.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Search start point normalized
#&gt;       description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;         for paired-end reads).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmaxOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmaxOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1.0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Search start point
#&gt;       description: Defines the search start point through the read - the read is split
#&gt;         into pieces no longer than this value (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seed length
#&gt;       description: Defines the maximum length of the seeds, if =0 max seed length
#&gt;         is infinite (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per window
#&gt;       description: Max number of seeds per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per read
#&gt;       description: Max number of seeds per read (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max one-seed loci per window
#&gt;       description: Max number of one seed loci per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedNoneLociPerWindow'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedNoneLociPerWindow
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Filter pieces for stitching
#&gt;       description: Only pieces that map fewer than this value are utilized in the
#&gt;         stitching procedure (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max score reduction
#&gt;       description: Maximum score reduction while searching for SJ boundaries in the
#&gt;         stitching step.
#&gt;       streamable: no
#&gt;       id: '#scoreStitchSJshift'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreStitchSJshift
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion Open Penalty
#&gt;       description: Insertion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreInsOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion extension penalty
#&gt;       description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreInsBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Log scaled score
#&gt;       description: 'Extra score logarithmically scaled with genomic length of the
#&gt;         alignment: &lt;int&gt;*log2(genomicLength).'
#&gt;       streamable: no
#&gt;       id: '#scoreGenomicLengthLog2scale'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGenomicLengthLog2scale
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-0.25'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-canonical gap open
#&gt;       description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapNoncan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapNoncan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: GC/AG and CT/GC gap open
#&gt;       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapGCAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapGCAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AT/AC and GT/AT gap open
#&gt;       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapATAC'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapATAC
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Gap open penalty
#&gt;       description: Gap open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreGap'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGap
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion open penalty
#&gt;       description: Deletion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreDelOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion extension penalty
#&gt;       description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreDelBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sequencing center
#&gt;       description: Specify the sequencing center for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_seq_center'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sample ID
#&gt;       description: Specify the sample ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_sample_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Platform unit ID
#&gt;       description: Specify the platform unit ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_platform_unit_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: rg_platform
#&gt;         symbols:
#&gt;         - LS 454
#&gt;         - Helicos
#&gt;         - Illumina
#&gt;         - ABI SOLiD
#&gt;         - Ion Torrent PGM
#&gt;         - PacBio
#&gt;         type: enum
#&gt;       label: Platform
#&gt;       description: Specify the version of the technology that was used for sequencing
#&gt;         or assaying.
#&gt;       streamable: no
#&gt;       id: '#rg_platform'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Median fragment length
#&gt;       description: Specify the median fragment length for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_mfl'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Library ID
#&gt;       description: Specify the library ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_library_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Read sequence
#&gt;       description: Read sequence.
#&gt;       streamable: no
#&gt;       id: '#reads'
#&gt;       inputBinding:
#&gt;         position: 10
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp
#&gt;             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n
#&gt;             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =
#&gt;             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n
#&gt;             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end
#&gt;             == 2){\n          right = list[index].path\n        }\n      }\n    }\n
#&gt;             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n
#&gt;             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left
#&gt;             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;
#&gt;             ++index) {\n      \n      if (list[index].metadata != null){\n        if
#&gt;             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n
#&gt;             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n
#&gt;             \       }\n      }\n    }\n    left_join = left.join()\n    right_join
#&gt;             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n
#&gt;             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){
#&gt;             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: readMatesLengthsIn
#&gt;         symbols:
#&gt;         - NotEqual
#&gt;         - Equal
#&gt;         type: enum
#&gt;       label: Reads lengths
#&gt;       description: Equal/Not equal - lengths of names, sequences, qualities for both
#&gt;         mates are the same/not the same. "Not equal" is safe in all situations.
#&gt;       streamable: no
#&gt;       id: '#readMatesLengthsIn'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMatesLengthsIn
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: NotEqual
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to map
#&gt;       description: Number of reads to map from the beginning of the file.
#&gt;       streamable: no
#&gt;       id: '#readMapNumber'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMapNumber
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantTranscriptomeBan
#&gt;         symbols:
#&gt;         - IndelSoftclipSingleend
#&gt;         - Singleend
#&gt;         type: enum
#&gt;       label: Prohibit alignment type
#&gt;       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;         prohibit single-end alignments.'
#&gt;       streamable: no
#&gt;       id: '#quantTranscriptomeBan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --quantTranscriptomeBan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantMode
#&gt;         symbols:
#&gt;         - TranscriptomeSAM
#&gt;         - GeneCounts
#&gt;         type: enum
#&gt;       label: Quantification mode
#&gt;       description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'
#&gt;         option, STAR will count number of reads per gene while mapping.
#&gt;       streamable: no
#&gt;       id: '#quantMode'
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSortingType
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - SortedByCoordinate
#&gt;         - Unsorted SortedByCoordinate
#&gt;         type: enum
#&gt;       label: Output sorting type
#&gt;       description: Type of output sorting.
#&gt;       streamable: no
#&gt;       id: '#outSortingType'
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SortedByCoordinate
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSJfilterReads
#&gt;         symbols:
#&gt;         - All
#&gt;         - Unique
#&gt;         type: enum
#&gt;       label: Collapsed junctions reads
#&gt;       description: 'Which reads to consider for collapsed splice junctions output.
#&gt;         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads
#&gt;         only.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterReads'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterReads
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: All
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min overhang SJ
#&gt;       description: Minimum overhang length for splice junctions on both sides for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterOverhangMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 30 12 12 12
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Max gap allowed
#&gt;       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads
#&gt;         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,
#&gt;         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;         Does not apply to annotated junctions.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterIntronMaxVsReadN'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterIntronMaxVsReadN
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 50000 100000 200000
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min distance to other donor/acceptor
#&gt;       description: Minimum allowed distance to other junctions' donor/acceptor for
#&gt;         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterDistToOtherSJmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterDistToOtherSJmin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 10 0 5 10
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min unique count
#&gt;       description: Minimum uniquely mapping read count per junction for each of the
#&gt;         motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountUniqueMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountUniqueMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min total count
#&gt;       description: Minimum total (multi-mapping+unique) read count per junction for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountTotalMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountTotalMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMunmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Write unmapped in SAM
#&gt;       description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;       streamable: no
#&gt;       id: '#outSAMunmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMunmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMtype
#&gt;         symbols:
#&gt;         - SAM
#&gt;         - BAM
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Format of output alignments.
#&gt;       streamable: no
#&gt;       id: '#outSAMtype'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               SAM_type = $job.inputs.outSAMtype
#&gt;               SORT_type = $job.inputs.outSortingType
#&gt;               if (SAM_type &amp;&amp; SORT_type) {
#&gt;                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SAM
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMstrandField
#&gt;         symbols:
#&gt;         - None
#&gt;         - intronMotif
#&gt;         type: enum
#&gt;       label: Strand field flag
#&gt;       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:
#&gt;         strand derived from the intron motif. Reads with inconsistent and/or non-canonical
#&gt;         introns are filtered out.'
#&gt;       streamable: no
#&gt;       id: '#outSAMstrandField'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMstrandField
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMreadID
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - Number
#&gt;         type: enum
#&gt;       label: Read ID
#&gt;       description: 'Read ID record type. Standard: first word (until space) from the
#&gt;         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)
#&gt;         in the FASTx file.'
#&gt;       streamable: no
#&gt;       id: '#outSAMreadID'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMreadID
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMprimaryFlag
#&gt;         symbols:
#&gt;         - OneBestScore
#&gt;         - AllBestScore
#&gt;         type: enum
#&gt;       label: Primary alignments
#&gt;       description: 'Which alignments are considered primary - all others will be marked
#&gt;         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best
#&gt;         score is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;       streamable: no
#&gt;       id: '#outSAMprimaryFlag'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMprimaryFlag
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: OneBestScore
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMorder
#&gt;         symbols:
#&gt;         - Paired
#&gt;         - PairedKeepInputOrder
#&gt;         type: enum
#&gt;       label: Sorting in SAM
#&gt;       description: 'Type of sorting for the SAM output. Paired: one mate after the
#&gt;         other for all paired alignments; PairedKeepInputOrder: one mate after the
#&gt;         other for all paired alignments, the order is kept the same as in the input
#&gt;         FASTQ files.'
#&gt;       streamable: no
#&gt;       id: '#outSAMorder'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMorder
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Paired
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMmode
#&gt;         symbols:
#&gt;         - Full
#&gt;         - NoQS
#&gt;         type: enum
#&gt;       label: SAM mode
#&gt;       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but
#&gt;         without quality scores.'
#&gt;       streamable: no
#&gt;       id: '#outSAMmode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Full
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: MAPQ value
#&gt;       description: MAPQ value for unique mappers (0 to 255).
#&gt;       streamable: no
#&gt;       id: '#outSAMmapqUnique'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmapqUnique
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '255'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @PG
#&gt;       description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderPG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderPG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @HD
#&gt;       description: '@HD (header) line of the SAM header.'
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderHD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderHD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: OR SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagOR'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagOR
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AND SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagAND'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagAND
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '65535'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMattributes
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - NH
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: SAM attributes
#&gt;       description: 'Desired SAM attributes, in the order desired for the output SAM.
#&gt;         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM
#&gt;         NM MD jM jI; None: no attributes.'
#&gt;       streamable: no
#&gt;       id: '#outSAMattributes'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMattributes
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outReadsUnmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Fastx
#&gt;         type: enum
#&gt;       label: Output unmapped reads
#&gt;       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:
#&gt;         output in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;       streamable: no
#&gt;       id: '#outReadsUnmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outReadsUnmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Quality conversion
#&gt;       description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;         to Sanger, use -31).
#&gt;       streamable: no
#&gt;       id: '#outQSconversionAdd'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outQSconversionAdd
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterType
#&gt;         symbols:
#&gt;         - Normal
#&gt;         - BySJout
#&gt;         type: enum
#&gt;       label: Filtering type
#&gt;       description: 'Type of filtering. Normal: standard filtering using only current
#&gt;         alignment; BySJout: keep only those reads that contain junctions that passed
#&gt;         filtering into SJ.out.tab.'
#&gt;       streamable: no
#&gt;       id: '#outFilterType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: Normal
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min score normalized
#&gt;       description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;         for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMinOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMinOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min score
#&gt;       description: Alignment will be output only if its score is higher than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Multimapping score range
#&gt;       description: The score range below the maximum score for multimapping alignments.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapScoreRange'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapScoreRange
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mappings
#&gt;       description: Read alignments will be output only if the read maps fewer than
#&gt;         this value, otherwise no alignments will be output.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *read* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverReadLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverReadLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *mapped* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mismatches
#&gt;       description: Alignment will be output only if it has fewer mismatches than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min matched bases normalized
#&gt;       description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;         lengths for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNminOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNminOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min matched bases
#&gt;       description: Alignment will be output only if the number of matched bases is
#&gt;         higher than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterIntronMotifs
#&gt;         symbols:
#&gt;         - None
#&gt;         - RemoveNoncanonical
#&gt;         - RemoveNoncanonicalUnannotated
#&gt;         type: enum
#&gt;       label: Motifs filtering
#&gt;       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;         filter out alignments that contain non-canonical unannotated junctions when
#&gt;         using annotated splice junctions database. The annotated non-canonical junctions
#&gt;         will be kept.'
#&gt;       streamable: no
#&gt;       id: '#outFilterIntronMotifs'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterIntronMotifs
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max insert junctions
#&gt;       description: Maximum number of junction to be inserted to the genome on the
#&gt;         fly at the mapping stage, including those from annotations and those detected
#&gt;         in the 1st step of the 2-pass run.
#&gt;       streamable: no
#&gt;       id: '#limitSjdbInsertNsj'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitSjdbInsertNsj
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Junctions max number
#&gt;       description: Max number of junctions for one read (including all multi-mappers).
#&gt;       streamable: no
#&gt;       id: '#limitOutSJoneRead'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJoneRead
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Collapsed junctions max number
#&gt;       description: Max number of collapsed junctions.
#&gt;       streamable: no
#&gt;       id: '#limitOutSJcollapsed'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJcollapsed
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Limit BAM sorting memory
#&gt;       description: Maximum available RAM for sorting BAM. If set to 0, it will be
#&gt;         set to the genome index size.
#&gt;       streamable: no
#&gt;       id: '#limitBAMsortRAM'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitBAMsortRAM
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Genome dir name
#&gt;       description: Name of the directory which contains genome files (when genome.tar
#&gt;         is uncompressed).
#&gt;       streamable: no
#&gt;       id: '#genomeDirName'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeDir
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: $job.inputs.genomeDirName || "genomeDir"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:toolDefaultValue: genomeDir
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome files
#&gt;       description: Genome files created using STAR Genome Generate.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: TAR
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 5p bases
#&gt;       description: Number of bases to clip from 5p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip5pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip5pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p bases
#&gt;       description: Number of bases to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p after adapter seq.
#&gt;       description: Number of bases to clip from 3p of each mate after the adapter
#&gt;         clipping. In case only one value is given, it will be assumed the same for
#&gt;         both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAfterAdapterNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAfterAdapterNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: string
#&gt;         type: array
#&gt;       label: Clip 3p adapter sequence
#&gt;       description: Adapter sequence to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterSeq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterSeq
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: float
#&gt;         type: array
#&gt;       label: Max mismatches proportions
#&gt;       description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;         In case only one value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterMMp'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterMMp
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0.1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min segment length
#&gt;       description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;         (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimSegmentMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimSegmentMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '15'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min separation score
#&gt;       description: Minimum difference (separation) between the best chimeric score
#&gt;         and the next one (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreSeparation'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreSeparation
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min total score
#&gt;       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-GT/AG penalty
#&gt;       description: Penalty for a non-GT/AG chimeric junction.
#&gt;       streamable: no
#&gt;       id: '#chimScoreJunctionNonGTAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreJunctionNonGTAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max drop score
#&gt;       description: Max drop (difference) of chimeric score (the sum of scores of all
#&gt;         chimeric segements) from the read length (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreDropMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreDropMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: chimOutType
#&gt;         symbols:
#&gt;         - SeparateSAMold
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Chimeric output type
#&gt;       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;       streamable: no
#&gt;       id: '#chimOutType'
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: SeparateSAMold
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min junction overhang
#&gt;       description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimJunctionOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimJunctionOverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Max windows per read
#&gt;       description: Max number of windows per read (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignWindowsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignWindowsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per window
#&gt;       description: Max number of transcripts per window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per read
#&gt;       description: Max number of different alignments per read to consider (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min mapped length normalized
#&gt;       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLminOverLmate'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLminOverLmate
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min mapped length
#&gt;       description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignSoftClipAtReferenceEnds
#&gt;         symbols:
#&gt;         - 'Yes'
#&gt;         - 'No'
#&gt;         type: enum
#&gt;       label: Soft clipping
#&gt;       description: 'Option which allows soft clipping of alignments at the reference
#&gt;         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.
#&gt;         Yes: Enables soft clipping; No: Disables soft clipping.'
#&gt;       streamable: no
#&gt;       id: '#alignSoftClipAtReferenceEnds'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSoftClipAtReferenceEnds
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: 'Yes'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min overhang
#&gt;       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: 'Min overhang: annotated'
#&gt;       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced
#&gt;         alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJDBoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJDBoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max mates gap
#&gt;       description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;         by (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignMatesGapMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignMatesGapMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min intron size
#&gt;       description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;       streamable: no
#&gt;       id: '#alignIntronMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '21'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max intron size
#&gt;       description: Maximum intron size, if 0, max intron size will be determined by
#&gt;         (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignIntronMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignEndsType
#&gt;         symbols:
#&gt;         - Local
#&gt;         - EndToEnd
#&gt;         type: enum
#&gt;       label: Alignment type
#&gt;       description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;       streamable: no
#&gt;       id: '#alignEndsType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignEndsType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: Local
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Unmapped reads
#&gt;       description: Output of unmapped reads.
#&gt;       streamable: no
#&gt;       id: '#unmapped_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Unmapped.out*'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Transcriptome alignments
#&gt;       description: Alignments translated into transcript coordinates.
#&gt;       streamable: no
#&gt;       id: '#transcriptome_aligned_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Transcriptome*'
#&gt;       sbg:fileTypes: BAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Splice junctions
#&gt;       description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;         Only junctions supported by uniquely mapping reads are reported.
#&gt;       streamable: no
#&gt;       id: '#splice_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*SJ.out.tab'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Reads per gene
#&gt;       description: File with number of reads per gene. A read is counted if it overlaps
#&gt;         (1nt or more) one and only one gene.
#&gt;       streamable: no
#&gt;       id: '#reads_per_gene'
#&gt;       outputBinding:
#&gt;         glob: '*ReadsPerGene*'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Log files
#&gt;       description: Log files produced during alignment.
#&gt;       streamable: no
#&gt;       id: '#log_files'
#&gt;       outputBinding:
#&gt;         glob: '*Log*.out'
#&gt;       sbg:fileTypes: OUT
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Intermediate genome files
#&gt;       description: Archive with genome files produced when annotations are included
#&gt;         on the fly (in the mapping step).
#&gt;       streamable: no
#&gt;       id: '#intermediate_genome'
#&gt;       outputBinding:
#&gt;         glob: '*_STARgenome.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric junctions
#&gt;       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,
#&gt;         'Chimeric Junctions' won't be output.
#&gt;       streamable: no
#&gt;       id: '#chimeric_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*Chimeric.out.junction'
#&gt;       sbg:fileTypes: JUNCTION
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric alignments
#&gt;       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;         Alignment SAM and Chimeric Junctions outputs.
#&gt;       streamable: no
#&gt;       id: '#chimeric_alignments'
#&gt;       outputBinding:
#&gt;         glob: '*.Chimeric.out.sam'
#&gt;       sbg:fileTypes: SAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Aligned SAM/BAM
#&gt;       description: Aligned sequence in SAM/BAM format.
#&gt;       streamable: no
#&gt;       id: '#aligned_reads'
#&gt;       outputBinding:
#&gt;         glob:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;                 sort_name = '.sortedByCoord'
#&gt;               }
#&gt;               else {
#&gt;                 sort_name = ''
#&gt;               }
#&gt;               if ($job.inputs.outSAMtype == 'BAM') {
#&gt;                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;               }
#&gt;               else {
#&gt;                 sam_name = "*.Aligned.out.sam"
#&gt;               }
#&gt;               return sam_name
#&gt;             }
#&gt;           class: Expression
#&gt;       sbg:fileTypes: SAM, BAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     label: STAR
#&gt;     description: STAR is an ultrafast universal RNA-seq aligner. It has very high
#&gt;       mapping speed, accurate alignment of contiguous and spliced reads, detection
#&gt;       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works
#&gt;       with reads starting from lengths ~15 bases up to ~300 bases. In case of having
#&gt;       longer reads, use of STAR Long is recommended.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - tar
#&gt;     - -xvf
#&gt;     - class: Expression
#&gt;       script: $job.inputs.genome.path
#&gt;       engine: '#cwl-js-engine'
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             file = [].concat($job.inputs.reads)[0].path
#&gt;             extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;             if (extension == "gz") {
#&gt;               return "--readFilesCommand zcat"
#&gt;             } else if (extension == "bz2") {
#&gt;               return "--readFilesCommand bzcat"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n
#&gt;           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",
#&gt;           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)
#&gt;           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n
#&gt;           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",
#&gt;           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)
#&gt;           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)
#&gt;           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)
#&gt;           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n
#&gt;           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"
#&gt;           \")\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.twopassMode == "Basic") {
#&gt;               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.chimOutType == "Within") {
#&gt;               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;             }
#&gt;             else {
#&gt;               return "--chimOutType SeparateSAMold"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n
#&gt;           \   if (value == \"\") {\n      return\n    }\n    else {\n      return
#&gt;           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',
#&gt;           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',
#&gt;           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)
#&gt;           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)
#&gt;           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;           \".concat(param_list.join(\" \"))\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;               return "--quantMode ".concat($job.inputs.quantMode)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 100
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n
#&gt;           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile
#&gt;           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       prefix: --outFileNamePrefix
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;         class: Expression
#&gt;     - position: 101
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"
#&gt;           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,
#&gt;           \" \", mate1fq)\n  }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 624.0
#&gt;     'y': 323
#&gt;   sbg:x: 700.0
#&gt;   sbg:y: 200.0
#&gt; sbg:canvas_zoom: 0.6
#&gt; sbg:canvas_y: -16
#&gt; sbg:canvas_x: -41
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_flow_output</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#log_files"</span>, <span class='st'>"intermediate_genome"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 2
#&gt; sbg:toolAuthor: Seven Bridges Genomics
#&gt; sbg:createdOn: 1463601910
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; - RNA
#&gt; sbg:contributors:
#&gt; - tengfei
#&gt; sbg:project: tengfei/quickstart
#&gt; sbg:createdBy: tengfei
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2
#&gt; sbg:license: Apache License 2.0
#&gt; sbg:revision: 2
#&gt; sbg:modifiedOn: 1463601974
#&gt; sbg:modifiedBy: tengfei
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601910
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601952
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601974
#&gt;   sbg:revision: 2
#&gt; sbg:toolkit: STAR
#&gt; id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'
#&gt; inputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: sjdbGTFfile
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:x: 160.4999759
#&gt;   sbg:y: 195.0833106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: fastq
#&gt;   streamable: no
#&gt;   id: '#fastq'
#&gt;   sbg:x: 164.2499914
#&gt;   sbg:y: 323.7499502
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - File
#&gt;   label: genomeFastaFiles
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   sbg:x: 167.7499601
#&gt;   sbg:y: 469.9999106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 350.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 450.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   source: '#unmapped_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   source: '#transcriptome_aligned_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 233.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   source: '#splice_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 266.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   source: '#reads_per_gene'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 300.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   source: '#log_files'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 333.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   source: '#chimeric_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 366.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   source: '#chimeric_alignments'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 433.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Sorted BAM/SAM
#&gt;   description: Sorted BAM or SAM file.
#&gt;   streamable: no
#&gt;   id: '#sorted_bam'
#&gt;   source: '#sorted_bam'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 466.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Result
#&gt;   description: Source FASTQ file with updated metadata.
#&gt;   streamable: no
#&gt;   id: '#result'
#&gt;   source: '#result'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 533.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; requirements:
#&gt; - class: CreateFileRequirement
#&gt;   fileDef: []
#&gt; hints:
#&gt; - class: sbg:AWSInstanceType
#&gt;   value: c3.8xlarge
#&gt; label: RNA-seq Alignment - STAR
#&gt; description: "Alignment to a reference genome and transcriptome presents the first
#&gt;   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner
#&gt;   capable of mapping full length RNA sequences and detecting de novo canonical junctions,
#&gt;   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian
#&gt;   sequence reads, but fine tuning of its parameters enables customization to satisfy
#&gt;   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).
#&gt;   \ \nSplice junction annotations can optionally be collected from splice junction
#&gt;   databases. Set the \"Overhang length\" parameter to a value larger than zero in
#&gt;   order to use splice junction databases. For constant read length, this value should
#&gt;   (ideally) be equal to mate length decreased by 1; for long reads with non-constant
#&gt;   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ
#&gt;   files reveals read length distribution. STAR can detect chimeric transcripts, but
#&gt;   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted
#&gt;   to a desired minimum chimeric segment length. Aligned reads are reported in BAM
#&gt;   format and can be viewed in a genome browser (such as IGV). A file containing detected
#&gt;   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format
#&gt;   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write
#&gt;   unmapped in SAM\" parameters enable unmapped output type selection."
#&gt; class: Workflow
#&gt; steps:
#&gt; - id: '#STAR_Genome_Generate'
#&gt;   inputs:
#&gt;   - id: '#STAR_Genome_Generate.sjdbScore'
#&gt;   - id: '#STAR_Genome_Generate.sjdbOverhang'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'
#&gt;     source: '#sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'
#&gt;     source: '#sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAsparseD'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAindexNbases'
#&gt;   - id: '#STAR_Genome_Generate.genomeFastaFiles'
#&gt;     source: '#genomeFastaFiles'
#&gt;   - id: '#STAR_Genome_Generate.genomeChrBinNbits'
#&gt;   outputs:
#&gt;   - id: '#STAR_Genome_Generate.genome'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 1
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         sjdbScore: 0
#&gt;         sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;         sjdbOverhang: 0
#&gt;         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;         genomeChrBinNbits: genomeChrBinNbits
#&gt;         genomeSAsparseD: 0
#&gt;         sjdbGTFfile:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /demo/test-files/chr20.gtf
#&gt;         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;         genomeFastaFiles:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /sbgenomics/test-data/chr20.fa
#&gt;         sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;         genomeSAindexNbases: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911469
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 1
#&gt;     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa
#&gt;       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD
#&gt;       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt;     sbg:modifiedOn: 1450911470
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911469
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911470
#&gt;       sbg:revision: 1
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbScore
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbOverhang
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentTranscript
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentGene
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFfeatureExon
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFchrPrefix
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Suffux array sparsity
#&gt;       description: 'Distance between indices: use bigger numbers to decrease needed
#&gt;         RAM at the cost of mapping speed reduction (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#genomeSAsparseD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAsparseD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Pre-indexing string length
#&gt;       description: Length (bases) of the SA pre-indexing string. Typically between
#&gt;         10 and 15. Longer strings will use much more memory, but allow faster searches.
#&gt;         For small genomes, this number needs to be scaled down, with a typical value
#&gt;         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,
#&gt;         this is equal to 9, for 100 kiloBase genome, this is equal to 7.
#&gt;       streamable: no
#&gt;       id: '#genomeSAindexNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAindexNbases
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '14'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome fasta files
#&gt;       description: Reference sequence to which to align the reads.
#&gt;       streamable: no
#&gt;       id: '#genomeFastaFiles'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeFastaFiles
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FA
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Bins size
#&gt;       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins
#&gt;         for genome storage: each chromosome will occupy an integer number of bins.
#&gt;         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,
#&gt;         you may need to reduce this number to reduce RAM consumption. The following
#&gt;         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this
#&gt;         is equal to 15.'
#&gt;       streamable: no
#&gt;       id: '#genomeChrBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeChrBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '18'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Genome Files
#&gt;       description: Genome files comprise binary genome sequence, suffix arrays, text
#&gt;         chromosome names/lengths, splice junctions coordinates, and transcripts/genes
#&gt;         information.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       outputBinding:
#&gt;         glob: '*.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     label: STAR Genome Generate
#&gt;     description: STAR Genome Generate is a tool that generates genome index files.
#&gt;       One set of files should be generated per each genome/annotation combination.
#&gt;       Once produced, these files could be used as long as genome/annotation combination
#&gt;       stays the same. Also, STAR Genome Generate which produced these files and STAR
#&gt;       aligner using them must be the same toolkit version.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - mkdir
#&gt;     - genomeDir
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runMode
#&gt;     - genomeGenerate
#&gt;     - --genomeDir
#&gt;     - ./genomeDir
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 99
#&gt;       separate: yes
#&gt;       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 384.0832266
#&gt;     'y': 446.4998957
#&gt;   sbg:x: 100.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#SBG_FASTQ_Quality_Detector'
#&gt;   inputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.fastq'
#&gt;     source: '#fastq'
#&gt;   outputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 3
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 1000
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         fastq:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /path/to/fastq.ext
#&gt;     sbg:toolAuthor: Seven Bridges Genomics
#&gt;     sbg:createdOn: 1450911312
#&gt;     sbg:categories:
#&gt;     - FASTQ-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:project: bix-demo/sbgtools-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     sbg:license: Apache License 2.0
#&gt;     sbg:revision: 3
#&gt;     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext
#&gt;       /path/to/fastq.ext
#&gt;     sbg:modifiedOn: 1450911314
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911312
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911314
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: SBGTools
#&gt;     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     inputs:
#&gt;     - type:
#&gt;       - File
#&gt;       label: Fastq
#&gt;       description: FASTQ file.
#&gt;       streamable: no
#&gt;       id: '#fastq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --fastq
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       required: yes
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Result
#&gt;       description: Source FASTQ file with updated metadata.
#&gt;       streamable: no
#&gt;       id: '#result'
#&gt;       outputBinding:
#&gt;         glob: '*.fastq'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     requirements:
#&gt;     - class: CreateFileRequirement
#&gt;       fileDef: []
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0
#&gt;       dockerImageId: ''
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 1000
#&gt;     label: SBG FASTQ Quality Detector
#&gt;     description: FASTQ Quality Scale Detector detects which quality encoding scheme
#&gt;       was used in your reads and automatically enters the proper value in the "Quality
#&gt;       Scale" metadata field.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - python
#&gt;     - /opt/sbg_fastq_quality_scale_detector.py
#&gt;     arguments: []
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 375.3333179
#&gt;     'y': 323.5833156
#&gt;   sbg:x: 300.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#Picard_SortSam'
#&gt;   inputs:
#&gt;   - id: '#Picard_SortSam.validation_stringency'
#&gt;     default: SILENT
#&gt;   - id: '#Picard_SortSam.sort_order'
#&gt;     default: Coordinate
#&gt;   - id: '#Picard_SortSam.quiet'
#&gt;   - id: '#Picard_SortSam.output_type'
#&gt;   - id: '#Picard_SortSam.memory_per_job'
#&gt;   - id: '#Picard_SortSam.max_records_in_ram'
#&gt;   - id: '#Picard_SortSam.input_bam'
#&gt;     source: '#STAR.aligned_reads'
#&gt;   - id: '#Picard_SortSam.create_index'
#&gt;     default: 'True'
#&gt;   - id: '#Picard_SortSam.compression_level'
#&gt;   outputs:
#&gt;   - id: '#Picard_SortSam.sorted_bam'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 2
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 2048
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         sort_order: Coordinate
#&gt;         input_bam:
#&gt;           path: /root/dir/example.tested.bam
#&gt;         memory_per_job: 2048
#&gt;         output_type: ~
#&gt;         create_index: ~
#&gt;     sbg:toolAuthor: Broad Institute
#&gt;     sbg:createdOn: 1450911168
#&gt;     sbg:categories:
#&gt;     - SAM/BAM-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: http://broadinstitute.github.io/picard/index.html
#&gt;       label: Homepage
#&gt;     - id: https://github.com/broadinstitute/picard/releases/tag/1.138
#&gt;       label: Source Code
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Wiki
#&gt;     - id: https://github.com/broadinstitute/picard/zipball/master
#&gt;       label: Download
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/picard-1-140-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: '1.140'
#&gt;     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     sbg:license: MIT License, Apache 2.0 Licence
#&gt;     sbg:revision: 2
#&gt;     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam
#&gt;       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam
#&gt;       SORT_ORDER=coordinate  /root/dir/example.tested.bam
#&gt;     sbg:modifiedOn: 1450911170
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911168
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911169
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911170
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: Picard
#&gt;     id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: validation_stringency
#&gt;         symbols:
#&gt;         - STRICT
#&gt;         - LENIENT
#&gt;         - SILENT
#&gt;         type: enum
#&gt;       label: Validation stringency
#&gt;       description: 'Validation stringency for all SAM files read by this program.
#&gt;         Setting stringency to SILENT can improve performance when processing a BAM
#&gt;         file in which variable-length data (read, qualities, tags) do not otherwise
#&gt;         need to be decoded. This option can be set to ''null'' to clear the default
#&gt;         value. Possible values: {STRICT, LENIENT, SILENT}.'
#&gt;       streamable: no
#&gt;       id: '#validation_stringency'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: VALIDATION_STRINGENCY=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.validation_stringency)
#&gt;               {
#&gt;                 return $job.inputs.validation_stringency
#&gt;               }
#&gt;               else
#&gt;               {
#&gt;                 return "SILENT"
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SILENT
#&gt;       required: no
#&gt;     - type:
#&gt;       - name: sort_order
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - Queryname
#&gt;         - Coordinate
#&gt;         type: enum
#&gt;       label: Sort order
#&gt;       description: 'Sort order of the output file. Possible values: {unsorted, queryname,
#&gt;         coordinate}.'
#&gt;       streamable: no
#&gt;       id: '#sort_order'
#&gt;       inputBinding:
#&gt;         position: 3
#&gt;         prefix: SORT_ORDER=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               p = $job.inputs.sort_order.toLowerCase()
#&gt;               return p
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: Coordinate
#&gt;       sbg:altPrefix: SO
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quiet
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Quiet
#&gt;       description: 'This parameter indicates whether to suppress job-summary info
#&gt;         on System.err. This option can be set to ''null'' to clear the default value.
#&gt;         Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#quiet'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: QUIET=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: output_type
#&gt;         symbols:
#&gt;         - BAM
#&gt;         - SAM
#&gt;         - SAME AS INPUT
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Since Picard tools can output both SAM and BAM files, user can
#&gt;         choose the format of the output file.
#&gt;       streamable: no
#&gt;       id: '#output_type'
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SAME AS INPUT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Memory per job
#&gt;       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for
#&gt;         single threaded jobs.
#&gt;       streamable: no
#&gt;       id: '#memory_per_job'
#&gt;       sbg:toolDefaultValue: '2048'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max records in RAM
#&gt;       description: When writing SAM files that need to be sorted, this parameter will
#&gt;         specify the number of records stored in RAM before spilling to disk. Increasing
#&gt;         this number reduces the number of file handles needed to sort a SAM file,
#&gt;         and increases the amount of RAM needed. This option can be set to 'null' to
#&gt;         clear the default value.
#&gt;       streamable: no
#&gt;       id: '#max_records_in_ram'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: MAX_RECORDS_IN_RAM=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '500000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Input BAM
#&gt;       description: The BAM or SAM file to sort.
#&gt;       streamable: no
#&gt;       id: '#input_bam'
#&gt;       inputBinding:
#&gt;         position: 1
#&gt;         prefix: INPUT=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: File inputs
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;       sbg:altPrefix: I
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: create_index
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Create index
#&gt;       description: 'This parameter indicates whether to create a BAM index when writing
#&gt;         a coordinate-sorted BAM file. This option can be set to ''null'' to clear
#&gt;         the default value. Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#create_index'
#&gt;       inputBinding:
#&gt;         position: 5
#&gt;         prefix: CREATE_INDEX=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Compression level
#&gt;       description: Compression level for all compressed files created (e.g. BAM and
#&gt;         GELI). This option can be set to 'null' to clear the default value.
#&gt;       streamable: no
#&gt;       id: '#compression_level'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: COMPRESSION_LEVEL=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Sorted BAM/SAM
#&gt;       description: Sorted BAM or SAM file.
#&gt;       streamable: no
#&gt;       id: '#sorted_bam'
#&gt;       outputBinding:
#&gt;         glob: '*.sorted.?am'
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;       engineCommand: cwl-engine.js
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
#&gt;       dockerImageId: eab0e70b6629
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n
#&gt;           \ }\n  \treturn 2048\n}"
#&gt;         class: Expression
#&gt;     label: Picard SortSam
#&gt;     description: Picard SortSam sorts the input SAM or BAM. Input and output formats
#&gt;       are determined by the file extension.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - java
#&gt;     - class: Expression
#&gt;       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,
#&gt;         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"
#&gt;       engine: '#cwl-js-engine'
#&gt;     - -jar
#&gt;     - /opt/picard-tools-1.140/picard.jar
#&gt;     - SortSam
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       prefix: OUTPUT=
#&gt;       separate: no
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif
#&gt;           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,
#&gt;           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse
#&gt;           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n}\n}"
#&gt;         class: Expression
#&gt;     - position: 1000
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out
#&gt;           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n
#&gt;           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n
#&gt;           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n
#&gt;           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index
#&gt;           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==
#&gt;           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 773.0831807
#&gt;     'y': 470.9165939
#&gt;   sbg:x: 500.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#STAR'
#&gt;   inputs:
#&gt;   - id: '#STAR.winFlankNbins'
#&gt;   - id: '#STAR.winBinNbits'
#&gt;   - id: '#STAR.winAnchorMultimapNmax'
#&gt;     source: '#winAnchorMultimapNmax'
#&gt;   - id: '#STAR.winAnchorDistNbins'
#&gt;     source: '#winAnchorDistNbins'
#&gt;   - id: '#STAR.twopassMode'
#&gt;   - id: '#STAR.twopass1readsN'
#&gt;   - id: '#STAR.sjdbScore'
#&gt;   - id: '#STAR.sjdbOverhang'
#&gt;     default: 100
#&gt;   - id: '#STAR.sjdbInsertSave'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR.seedSearchStartLmaxOverLread'
#&gt;   - id: '#STAR.seedSearchStartLmax'
#&gt;   - id: '#STAR.seedSearchLmax'
#&gt;   - id: '#STAR.seedPerWindowNmax'
#&gt;   - id: '#STAR.seedPerReadNmax'
#&gt;   - id: '#STAR.seedNoneLociPerWindow'
#&gt;   - id: '#STAR.seedMultimapNmax'
#&gt;   - id: '#STAR.scoreStitchSJshift'
#&gt;   - id: '#STAR.scoreInsOpen'
#&gt;   - id: '#STAR.scoreInsBase'
#&gt;   - id: '#STAR.scoreGenomicLengthLog2scale'
#&gt;   - id: '#STAR.scoreGapNoncan'
#&gt;   - id: '#STAR.scoreGapGCAG'
#&gt;   - id: '#STAR.scoreGapATAC'
#&gt;   - id: '#STAR.scoreGap'
#&gt;   - id: '#STAR.scoreDelOpen'
#&gt;   - id: '#STAR.scoreDelBase'
#&gt;   - id: '#STAR.rg_seq_center'
#&gt;   - id: '#STAR.rg_sample_id'
#&gt;   - id: '#STAR.rg_platform_unit_id'
#&gt;   - id: '#STAR.rg_platform'
#&gt;   - id: '#STAR.rg_mfl'
#&gt;   - id: '#STAR.rg_library_id'
#&gt;   - id: '#STAR.reads'
#&gt;     source: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   - id: '#STAR.readMatesLengthsIn'
#&gt;   - id: '#STAR.readMapNumber'
#&gt;   - id: '#STAR.quantTranscriptomeBan'
#&gt;   - id: '#STAR.quantMode'
#&gt;     default: TranscriptomeSAM
#&gt;   - id: '#STAR.outSortingType'
#&gt;     default: SortedByCoordinate
#&gt;   - id: '#STAR.outSJfilterReads'
#&gt;   - id: '#STAR.outSJfilterOverhangMin'
#&gt;   - id: '#STAR.outSJfilterIntronMaxVsReadN'
#&gt;   - id: '#STAR.outSJfilterDistToOtherSJmin'
#&gt;   - id: '#STAR.outSJfilterCountUniqueMin'
#&gt;   - id: '#STAR.outSJfilterCountTotalMin'
#&gt;   - id: '#STAR.outSAMunmapped'
#&gt;   - id: '#STAR.outSAMtype'
#&gt;     default: BAM
#&gt;   - id: '#STAR.outSAMstrandField'
#&gt;   - id: '#STAR.outSAMreadID'
#&gt;   - id: '#STAR.outSAMprimaryFlag'
#&gt;   - id: '#STAR.outSAMorder'
#&gt;   - id: '#STAR.outSAMmode'
#&gt;   - id: '#STAR.outSAMmapqUnique'
#&gt;   - id: '#STAR.outSAMheaderPG'
#&gt;   - id: '#STAR.outSAMheaderHD'
#&gt;   - id: '#STAR.outSAMflagOR'
#&gt;   - id: '#STAR.outSAMflagAND'
#&gt;   - id: '#STAR.outSAMattributes'
#&gt;   - id: '#STAR.outReadsUnmapped'
#&gt;     default: Fastx
#&gt;   - id: '#STAR.outQSconversionAdd'
#&gt;   - id: '#STAR.outFilterType'
#&gt;   - id: '#STAR.outFilterScoreMinOverLread'
#&gt;   - id: '#STAR.outFilterScoreMin'
#&gt;   - id: '#STAR.outFilterMultimapScoreRange'
#&gt;   - id: '#STAR.outFilterMultimapNmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverReadLmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverLmax'
#&gt;   - id: '#STAR.outFilterMismatchNmax'
#&gt;   - id: '#STAR.outFilterMatchNminOverLread'
#&gt;   - id: '#STAR.outFilterMatchNmin'
#&gt;   - id: '#STAR.outFilterIntronMotifs'
#&gt;   - id: '#STAR.limitSjdbInsertNsj'
#&gt;   - id: '#STAR.limitOutSJoneRead'
#&gt;   - id: '#STAR.limitOutSJcollapsed'
#&gt;   - id: '#STAR.limitBAMsortRAM'
#&gt;   - id: '#STAR.genomeDirName'
#&gt;   - id: '#STAR.genome'
#&gt;     source: '#STAR_Genome_Generate.genome'
#&gt;   - id: '#STAR.clip5pNbases'
#&gt;   - id: '#STAR.clip3pNbases'
#&gt;   - id: '#STAR.clip3pAfterAdapterNbases'
#&gt;   - id: '#STAR.clip3pAdapterSeq'
#&gt;   - id: '#STAR.clip3pAdapterMMp'
#&gt;   - id: '#STAR.chimSegmentMin'
#&gt;   - id: '#STAR.chimScoreSeparation'
#&gt;   - id: '#STAR.chimScoreMin'
#&gt;   - id: '#STAR.chimScoreJunctionNonGTAG'
#&gt;   - id: '#STAR.chimScoreDropMax'
#&gt;   - id: '#STAR.chimOutType'
#&gt;   - id: '#STAR.chimJunctionOverhangMin'
#&gt;   - id: '#STAR.alignWindowsPerReadNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerWindowNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerReadNmax'
#&gt;   - id: '#STAR.alignSplicedMateMapLminOverLmate'
#&gt;   - id: '#STAR.alignSplicedMateMapLmin'
#&gt;   - id: '#STAR.alignSoftClipAtReferenceEnds'
#&gt;   - id: '#STAR.alignSJoverhangMin'
#&gt;   - id: '#STAR.alignSJDBoverhangMin'
#&gt;   - id: '#STAR.alignMatesGapMax'
#&gt;   - id: '#STAR.alignIntronMin'
#&gt;   - id: '#STAR.alignIntronMax'
#&gt;   - id: '#STAR.alignEndsType'
#&gt;   outputs:
#&gt;   - id: '#STAR.unmapped_reads'
#&gt;   - id: '#STAR.transcriptome_aligned_reads'
#&gt;   - id: '#STAR.splice_junctions'
#&gt;   - id: '#STAR.reads_per_gene'
#&gt;   - id: '#STAR.log_files'
#&gt;   - id: '#STAR.intermediate_genome'
#&gt;   - id: '#STAR.chimeric_junctions'
#&gt;   - id: '#STAR.chimeric_alignments'
#&gt;   - id: '#STAR.aligned_reads'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 4
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         alignWindowsPerReadNmax: 0
#&gt;         outSAMheaderPG: outSAMheaderPG
#&gt;         GENOME_DIR_NAME: ''
#&gt;         outFilterMatchNminOverLread: 0
#&gt;         rg_platform_unit_id: rg_platform_unit
#&gt;         alignTranscriptsPerReadNmax: 0
#&gt;         readMapNumber: 0
#&gt;         alignSplicedMateMapLminOverLmate: 0
#&gt;         alignMatesGapMax: 0
#&gt;         outFilterMultimapNmax: 0
#&gt;         clip5pNbases:
#&gt;         - 0
#&gt;         outSAMstrandField: None
#&gt;         readMatesLengthsIn: NotEqual
#&gt;         outSAMattributes: Standard
#&gt;         seedMultimapNmax: 0
#&gt;         rg_mfl: rg_mfl
#&gt;         chimSegmentMin: 0
#&gt;         winAnchorDistNbins: 0
#&gt;         outSortingType: SortedByCoordinate
#&gt;         outFilterMultimapScoreRange: 0
#&gt;         sjdbInsertSave: Basic
#&gt;         clip3pAfterAdapterNbases:
#&gt;         - 0
#&gt;         scoreDelBase: 0
#&gt;         outFilterMatchNmin: 0
#&gt;         twopass1readsN: 0
#&gt;         outSAMunmapped: None
#&gt;         genome:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: genome.ext
#&gt;         sjdbGTFtagExonParentTranscript: ''
#&gt;         limitBAMsortRAM: 0
#&gt;         alignEndsType: Local
#&gt;         seedNoneLociPerWindow: 0
#&gt;         rg_sample_id: rg_sample
#&gt;         sjdbGTFtagExonParentGene: ''
#&gt;         chimScoreMin: 0
#&gt;         outSJfilterIntronMaxVsReadN:
#&gt;         - 0
#&gt;         twopassMode: Basic
#&gt;         alignSplicedMateMapLmin: 0
#&gt;         outSJfilterReads: All
#&gt;         outSAMprimaryFlag: OneBestScore
#&gt;         outSJfilterCountTotalMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         outSAMorder: Paired
#&gt;         outSAMflagAND: 0
#&gt;         chimScoreSeparation: 0
#&gt;         alignSJoverhangMin: 0
#&gt;         outFilterScoreMin: 0
#&gt;         seedSearchStartLmax: 0
#&gt;         scoreGapGCAG: 0
#&gt;         scoreGenomicLengthLog2scale: 0
#&gt;         outFilterIntronMotifs: None
#&gt;         outFilterMismatchNmax: 0
#&gt;         reads:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           metadata:
#&gt;             format: fastq
#&gt;             paired_end: '1'
#&gt;             seq_center: illumina
#&gt;           path: /test-data/mate_1.fastq.bz2
#&gt;         scoreGap: 0
#&gt;         outSJfilterOverhangMin:
#&gt;         - 30
#&gt;         - 12
#&gt;         - 12
#&gt;         - 12
#&gt;         outSAMflagOR: 0
#&gt;         outSAMmode: Full
#&gt;         rg_library_id: ''
#&gt;         chimScoreJunctionNonGTAG: 0
#&gt;         scoreInsOpen: 0
#&gt;         clip3pAdapterSeq:
#&gt;         - clip3pAdapterSeq
#&gt;         chimScoreDropMax: 0
#&gt;         outFilterType: Normal
#&gt;         scoreGapATAC: 0
#&gt;         rg_platform: Ion Torrent PGM
#&gt;         clip3pAdapterMMp:
#&gt;         - 0
#&gt;         sjdbGTFfeatureExon: ''
#&gt;         outQSconversionAdd: 0
#&gt;         quantMode: TranscriptomeSAM
#&gt;         alignIntronMin: 0
#&gt;         scoreInsBase: 0
#&gt;         scoreGapNoncan: 0
#&gt;         seedSearchLmax: 0
#&gt;         outSJfilterDistToOtherSJmin:
#&gt;         - 0
#&gt;         outFilterScoreMinOverLread: 0
#&gt;         alignSJDBoverhangMin: 0
#&gt;         limitOutSJcollapsed: 0
#&gt;         winAnchorMultimapNmax: 0
#&gt;         outFilterMismatchNoverLmax: 0
#&gt;         rg_seq_center: ''
#&gt;         outSAMheaderHD: outSAMheaderHD
#&gt;         chimOutType: Within
#&gt;         quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;         limitOutSJoneRead: 0
#&gt;         alignTranscriptsPerWindowNmax: 0
#&gt;         sjdbOverhang: ~
#&gt;         outReadsUnmapped: Fastx
#&gt;         scoreStitchSJshift: 0
#&gt;         seedPerWindowNmax: 0
#&gt;         outSJfilterCountUniqueMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         scoreDelOpen: 0
#&gt;         sjdbGTFfile:
#&gt;         - path: /demo/test-data/chr20.gtf
#&gt;         clip3pNbases:
#&gt;         - 0
#&gt;         - 3
#&gt;         winBinNbits: 0
#&gt;         sjdbScore: ~
#&gt;         seedSearchStartLmaxOverLread: 0
#&gt;         alignIntronMax: 0
#&gt;         seedPerReadNmax: 0
#&gt;         outFilterMismatchNoverReadLmax: 0
#&gt;         winFlankNbins: 0
#&gt;         sjdbGTFchrPrefix: chrPrefix
#&gt;         alignSoftClipAtReferenceEnds: 'Yes'
#&gt;         outSAMreadID: Standard
#&gt;         outSAMtype: BAM
#&gt;         chimJunctionOverhangMin: 0
#&gt;         limitSjdbInsertNsj: 0
#&gt;         outSAMmapqUnique: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911471
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - ana_d
#&gt;     - bix-demo
#&gt;     - uros_sipetic
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star/4
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 4
#&gt;     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix
#&gt;       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline
#&gt;       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode
#&gt;       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;
#&gt;       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt;     sbg:modifiedOn: 1462889222
#&gt;     sbg:modifiedBy: ana_d
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911471
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911473
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911475
#&gt;       sbg:revision: 2
#&gt;     - sbg:modifiedBy: uros_sipetic
#&gt;       sbg:modifiedOn: 1462878528
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: ana_d
#&gt;       sbg:modifiedOn: 1462889222
#&gt;       sbg:revision: 4
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star/4
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Flanking regions size
#&gt;       description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;         flanking regions for each window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winFlankNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winFlankNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Bin size
#&gt;       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;         each window will occupy an integer number of bins (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '16'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max loci anchors
#&gt;       description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max bins between anchors
#&gt;       description: Max number of bins between two anchors that allows aggregation
#&gt;         of anchors into one window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorDistNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorDistNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '9'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: twopassMode
#&gt;         symbols:
#&gt;         - None
#&gt;         - Basic
#&gt;         type: enum
#&gt;       label: Two-pass mode
#&gt;       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass
#&gt;         mapping, with all 1st pass junctions inserted into the genome indices on the
#&gt;         fly.'
#&gt;       streamable: no
#&gt;       id: '#twopassMode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --twopassMode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to process in 1st step
#&gt;       description: 'Number of reads to process for the 1st step. 0: 1-step only, no
#&gt;         2nd pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#twopass1readsN'
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: sjdbInsertSave
#&gt;         symbols:
#&gt;         - Basic
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: Save junction files
#&gt;       description: 'Which files to save when sjdb junctions are inserted on the fly
#&gt;         at the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;         files; All: all files including big Genome, SA and SAindex. These files are
#&gt;         output as archive.'
#&gt;       streamable: no
#&gt;       id: '#sjdbInsertSave'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts. No need to include
#&gt;         this input, except in case of using "on the fly" annotations.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Search start point normalized
#&gt;       description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;         for paired-end reads).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmaxOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmaxOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1.0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Search start point
#&gt;       description: Defines the search start point through the read - the read is split
#&gt;         into pieces no longer than this value (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seed length
#&gt;       description: Defines the maximum length of the seeds, if =0 max seed length
#&gt;         is infinite (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per window
#&gt;       description: Max number of seeds per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per read
#&gt;       description: Max number of seeds per read (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max one-seed loci per window
#&gt;       description: Max number of one seed loci per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedNoneLociPerWindow'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedNoneLociPerWindow
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Filter pieces for stitching
#&gt;       description: Only pieces that map fewer than this value are utilized in the
#&gt;         stitching procedure (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max score reduction
#&gt;       description: Maximum score reduction while searching for SJ boundaries in the
#&gt;         stitching step.
#&gt;       streamable: no
#&gt;       id: '#scoreStitchSJshift'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreStitchSJshift
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion Open Penalty
#&gt;       description: Insertion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreInsOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion extension penalty
#&gt;       description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreInsBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Log scaled score
#&gt;       description: 'Extra score logarithmically scaled with genomic length of the
#&gt;         alignment: &lt;int&gt;*log2(genomicLength).'
#&gt;       streamable: no
#&gt;       id: '#scoreGenomicLengthLog2scale'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGenomicLengthLog2scale
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-0.25'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-canonical gap open
#&gt;       description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapNoncan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapNoncan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: GC/AG and CT/GC gap open
#&gt;       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapGCAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapGCAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AT/AC and GT/AT gap open
#&gt;       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapATAC'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapATAC
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Gap open penalty
#&gt;       description: Gap open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreGap'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGap
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion open penalty
#&gt;       description: Deletion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreDelOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion extension penalty
#&gt;       description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreDelBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sequencing center
#&gt;       description: Specify the sequencing center for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_seq_center'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sample ID
#&gt;       description: Specify the sample ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_sample_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Platform unit ID
#&gt;       description: Specify the platform unit ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_platform_unit_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: rg_platform
#&gt;         symbols:
#&gt;         - LS 454
#&gt;         - Helicos
#&gt;         - Illumina
#&gt;         - ABI SOLiD
#&gt;         - Ion Torrent PGM
#&gt;         - PacBio
#&gt;         type: enum
#&gt;       label: Platform
#&gt;       description: Specify the version of the technology that was used for sequencing
#&gt;         or assaying.
#&gt;       streamable: no
#&gt;       id: '#rg_platform'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Median fragment length
#&gt;       description: Specify the median fragment length for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_mfl'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Library ID
#&gt;       description: Specify the library ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_library_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Read sequence
#&gt;       description: Read sequence.
#&gt;       streamable: no
#&gt;       id: '#reads'
#&gt;       inputBinding:
#&gt;         position: 10
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp
#&gt;             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n
#&gt;             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =
#&gt;             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n
#&gt;             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end
#&gt;             == 2){\n          right = list[index].path\n        }\n      }\n    }\n
#&gt;             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n
#&gt;             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left
#&gt;             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;
#&gt;             ++index) {\n      \n      if (list[index].metadata != null){\n        if
#&gt;             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n
#&gt;             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n
#&gt;             \       }\n      }\n    }\n    left_join = left.join()\n    right_join
#&gt;             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n
#&gt;             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){
#&gt;             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: readMatesLengthsIn
#&gt;         symbols:
#&gt;         - NotEqual
#&gt;         - Equal
#&gt;         type: enum
#&gt;       label: Reads lengths
#&gt;       description: Equal/Not equal - lengths of names, sequences, qualities for both
#&gt;         mates are the same/not the same. "Not equal" is safe in all situations.
#&gt;       streamable: no
#&gt;       id: '#readMatesLengthsIn'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMatesLengthsIn
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: NotEqual
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to map
#&gt;       description: Number of reads to map from the beginning of the file.
#&gt;       streamable: no
#&gt;       id: '#readMapNumber'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMapNumber
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantTranscriptomeBan
#&gt;         symbols:
#&gt;         - IndelSoftclipSingleend
#&gt;         - Singleend
#&gt;         type: enum
#&gt;       label: Prohibit alignment type
#&gt;       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;         prohibit single-end alignments.'
#&gt;       streamable: no
#&gt;       id: '#quantTranscriptomeBan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --quantTranscriptomeBan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantMode
#&gt;         symbols:
#&gt;         - TranscriptomeSAM
#&gt;         - GeneCounts
#&gt;         type: enum
#&gt;       label: Quantification mode
#&gt;       description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'
#&gt;         option, STAR will count number of reads per gene while mapping.
#&gt;       streamable: no
#&gt;       id: '#quantMode'
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSortingType
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - SortedByCoordinate
#&gt;         - Unsorted SortedByCoordinate
#&gt;         type: enum
#&gt;       label: Output sorting type
#&gt;       description: Type of output sorting.
#&gt;       streamable: no
#&gt;       id: '#outSortingType'
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SortedByCoordinate
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSJfilterReads
#&gt;         symbols:
#&gt;         - All
#&gt;         - Unique
#&gt;         type: enum
#&gt;       label: Collapsed junctions reads
#&gt;       description: 'Which reads to consider for collapsed splice junctions output.
#&gt;         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads
#&gt;         only.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterReads'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterReads
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: All
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min overhang SJ
#&gt;       description: Minimum overhang length for splice junctions on both sides for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterOverhangMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 30 12 12 12
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Max gap allowed
#&gt;       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads
#&gt;         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,
#&gt;         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;         Does not apply to annotated junctions.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterIntronMaxVsReadN'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterIntronMaxVsReadN
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 50000 100000 200000
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min distance to other donor/acceptor
#&gt;       description: Minimum allowed distance to other junctions' donor/acceptor for
#&gt;         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterDistToOtherSJmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterDistToOtherSJmin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 10 0 5 10
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min unique count
#&gt;       description: Minimum uniquely mapping read count per junction for each of the
#&gt;         motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountUniqueMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountUniqueMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min total count
#&gt;       description: Minimum total (multi-mapping+unique) read count per junction for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountTotalMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountTotalMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMunmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Write unmapped in SAM
#&gt;       description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;       streamable: no
#&gt;       id: '#outSAMunmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMunmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMtype
#&gt;         symbols:
#&gt;         - SAM
#&gt;         - BAM
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Format of output alignments.
#&gt;       streamable: no
#&gt;       id: '#outSAMtype'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               SAM_type = $job.inputs.outSAMtype
#&gt;               SORT_type = $job.inputs.outSortingType
#&gt;               if (SAM_type &amp;&amp; SORT_type) {
#&gt;                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SAM
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMstrandField
#&gt;         symbols:
#&gt;         - None
#&gt;         - intronMotif
#&gt;         type: enum
#&gt;       label: Strand field flag
#&gt;       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:
#&gt;         strand derived from the intron motif. Reads with inconsistent and/or non-canonical
#&gt;         introns are filtered out.'
#&gt;       streamable: no
#&gt;       id: '#outSAMstrandField'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMstrandField
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMreadID
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - Number
#&gt;         type: enum
#&gt;       label: Read ID
#&gt;       description: 'Read ID record type. Standard: first word (until space) from the
#&gt;         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)
#&gt;         in the FASTx file.'
#&gt;       streamable: no
#&gt;       id: '#outSAMreadID'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMreadID
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMprimaryFlag
#&gt;         symbols:
#&gt;         - OneBestScore
#&gt;         - AllBestScore
#&gt;         type: enum
#&gt;       label: Primary alignments
#&gt;       description: 'Which alignments are considered primary - all others will be marked
#&gt;         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best
#&gt;         score is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;       streamable: no
#&gt;       id: '#outSAMprimaryFlag'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMprimaryFlag
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: OneBestScore
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMorder
#&gt;         symbols:
#&gt;         - Paired
#&gt;         - PairedKeepInputOrder
#&gt;         type: enum
#&gt;       label: Sorting in SAM
#&gt;       description: 'Type of sorting for the SAM output. Paired: one mate after the
#&gt;         other for all paired alignments; PairedKeepInputOrder: one mate after the
#&gt;         other for all paired alignments, the order is kept the same as in the input
#&gt;         FASTQ files.'
#&gt;       streamable: no
#&gt;       id: '#outSAMorder'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMorder
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Paired
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMmode
#&gt;         symbols:
#&gt;         - Full
#&gt;         - NoQS
#&gt;         type: enum
#&gt;       label: SAM mode
#&gt;       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but
#&gt;         without quality scores.'
#&gt;       streamable: no
#&gt;       id: '#outSAMmode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Full
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: MAPQ value
#&gt;       description: MAPQ value for unique mappers (0 to 255).
#&gt;       streamable: no
#&gt;       id: '#outSAMmapqUnique'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmapqUnique
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '255'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @PG
#&gt;       description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderPG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderPG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @HD
#&gt;       description: '@HD (header) line of the SAM header.'
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderHD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderHD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: OR SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagOR'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagOR
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AND SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagAND'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagAND
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '65535'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMattributes
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - NH
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: SAM attributes
#&gt;       description: 'Desired SAM attributes, in the order desired for the output SAM.
#&gt;         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM
#&gt;         NM MD jM jI; None: no attributes.'
#&gt;       streamable: no
#&gt;       id: '#outSAMattributes'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMattributes
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outReadsUnmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Fastx
#&gt;         type: enum
#&gt;       label: Output unmapped reads
#&gt;       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:
#&gt;         output in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;       streamable: no
#&gt;       id: '#outReadsUnmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outReadsUnmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Quality conversion
#&gt;       description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;         to Sanger, use -31).
#&gt;       streamable: no
#&gt;       id: '#outQSconversionAdd'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outQSconversionAdd
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterType
#&gt;         symbols:
#&gt;         - Normal
#&gt;         - BySJout
#&gt;         type: enum
#&gt;       label: Filtering type
#&gt;       description: 'Type of filtering. Normal: standard filtering using only current
#&gt;         alignment; BySJout: keep only those reads that contain junctions that passed
#&gt;         filtering into SJ.out.tab.'
#&gt;       streamable: no
#&gt;       id: '#outFilterType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: Normal
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min score normalized
#&gt;       description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;         for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMinOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMinOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min score
#&gt;       description: Alignment will be output only if its score is higher than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Multimapping score range
#&gt;       description: The score range below the maximum score for multimapping alignments.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapScoreRange'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapScoreRange
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mappings
#&gt;       description: Read alignments will be output only if the read maps fewer than
#&gt;         this value, otherwise no alignments will be output.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *read* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverReadLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverReadLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *mapped* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mismatches
#&gt;       description: Alignment will be output only if it has fewer mismatches than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min matched bases normalized
#&gt;       description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;         lengths for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNminOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNminOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min matched bases
#&gt;       description: Alignment will be output only if the number of matched bases is
#&gt;         higher than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterIntronMotifs
#&gt;         symbols:
#&gt;         - None
#&gt;         - RemoveNoncanonical
#&gt;         - RemoveNoncanonicalUnannotated
#&gt;         type: enum
#&gt;       label: Motifs filtering
#&gt;       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;         filter out alignments that contain non-canonical unannotated junctions when
#&gt;         using annotated splice junctions database. The annotated non-canonical junctions
#&gt;         will be kept.'
#&gt;       streamable: no
#&gt;       id: '#outFilterIntronMotifs'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterIntronMotifs
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max insert junctions
#&gt;       description: Maximum number of junction to be inserted to the genome on the
#&gt;         fly at the mapping stage, including those from annotations and those detected
#&gt;         in the 1st step of the 2-pass run.
#&gt;       streamable: no
#&gt;       id: '#limitSjdbInsertNsj'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitSjdbInsertNsj
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Junctions max number
#&gt;       description: Max number of junctions for one read (including all multi-mappers).
#&gt;       streamable: no
#&gt;       id: '#limitOutSJoneRead'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJoneRead
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Collapsed junctions max number
#&gt;       description: Max number of collapsed junctions.
#&gt;       streamable: no
#&gt;       id: '#limitOutSJcollapsed'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJcollapsed
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Limit BAM sorting memory
#&gt;       description: Maximum available RAM for sorting BAM. If set to 0, it will be
#&gt;         set to the genome index size.
#&gt;       streamable: no
#&gt;       id: '#limitBAMsortRAM'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitBAMsortRAM
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Genome dir name
#&gt;       description: Name of the directory which contains genome files (when genome.tar
#&gt;         is uncompressed).
#&gt;       streamable: no
#&gt;       id: '#genomeDirName'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeDir
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: $job.inputs.genomeDirName || "genomeDir"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:toolDefaultValue: genomeDir
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome files
#&gt;       description: Genome files created using STAR Genome Generate.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: TAR
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 5p bases
#&gt;       description: Number of bases to clip from 5p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip5pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip5pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p bases
#&gt;       description: Number of bases to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p after adapter seq.
#&gt;       description: Number of bases to clip from 3p of each mate after the adapter
#&gt;         clipping. In case only one value is given, it will be assumed the same for
#&gt;         both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAfterAdapterNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAfterAdapterNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: string
#&gt;         type: array
#&gt;       label: Clip 3p adapter sequence
#&gt;       description: Adapter sequence to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterSeq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterSeq
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: float
#&gt;         type: array
#&gt;       label: Max mismatches proportions
#&gt;       description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;         In case only one value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterMMp'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterMMp
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0.1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min segment length
#&gt;       description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;         (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimSegmentMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimSegmentMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '15'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min separation score
#&gt;       description: Minimum difference (separation) between the best chimeric score
#&gt;         and the next one (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreSeparation'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreSeparation
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min total score
#&gt;       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-GT/AG penalty
#&gt;       description: Penalty for a non-GT/AG chimeric junction.
#&gt;       streamable: no
#&gt;       id: '#chimScoreJunctionNonGTAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreJunctionNonGTAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max drop score
#&gt;       description: Max drop (difference) of chimeric score (the sum of scores of all
#&gt;         chimeric segements) from the read length (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreDropMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreDropMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: chimOutType
#&gt;         symbols:
#&gt;         - SeparateSAMold
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Chimeric output type
#&gt;       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;       streamable: no
#&gt;       id: '#chimOutType'
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: SeparateSAMold
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min junction overhang
#&gt;       description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimJunctionOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimJunctionOverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Max windows per read
#&gt;       description: Max number of windows per read (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignWindowsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignWindowsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per window
#&gt;       description: Max number of transcripts per window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per read
#&gt;       description: Max number of different alignments per read to consider (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min mapped length normalized
#&gt;       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLminOverLmate'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLminOverLmate
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min mapped length
#&gt;       description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignSoftClipAtReferenceEnds
#&gt;         symbols:
#&gt;         - 'Yes'
#&gt;         - 'No'
#&gt;         type: enum
#&gt;       label: Soft clipping
#&gt;       description: 'Option which allows soft clipping of alignments at the reference
#&gt;         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.
#&gt;         Yes: Enables soft clipping; No: Disables soft clipping.'
#&gt;       streamable: no
#&gt;       id: '#alignSoftClipAtReferenceEnds'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSoftClipAtReferenceEnds
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: 'Yes'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min overhang
#&gt;       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: 'Min overhang: annotated'
#&gt;       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced
#&gt;         alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJDBoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJDBoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max mates gap
#&gt;       description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;         by (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignMatesGapMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignMatesGapMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min intron size
#&gt;       description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;       streamable: no
#&gt;       id: '#alignIntronMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '21'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max intron size
#&gt;       description: Maximum intron size, if 0, max intron size will be determined by
#&gt;         (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignIntronMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignEndsType
#&gt;         symbols:
#&gt;         - Local
#&gt;         - EndToEnd
#&gt;         type: enum
#&gt;       label: Alignment type
#&gt;       description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;       streamable: no
#&gt;       id: '#alignEndsType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignEndsType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: Local
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Unmapped reads
#&gt;       description: Output of unmapped reads.
#&gt;       streamable: no
#&gt;       id: '#unmapped_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Unmapped.out*'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Transcriptome alignments
#&gt;       description: Alignments translated into transcript coordinates.
#&gt;       streamable: no
#&gt;       id: '#transcriptome_aligned_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Transcriptome*'
#&gt;       sbg:fileTypes: BAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Splice junctions
#&gt;       description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;         Only junctions supported by uniquely mapping reads are reported.
#&gt;       streamable: no
#&gt;       id: '#splice_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*SJ.out.tab'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Reads per gene
#&gt;       description: File with number of reads per gene. A read is counted if it overlaps
#&gt;         (1nt or more) one and only one gene.
#&gt;       streamable: no
#&gt;       id: '#reads_per_gene'
#&gt;       outputBinding:
#&gt;         glob: '*ReadsPerGene*'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Log files
#&gt;       description: Log files produced during alignment.
#&gt;       streamable: no
#&gt;       id: '#log_files'
#&gt;       outputBinding:
#&gt;         glob: '*Log*.out'
#&gt;       sbg:fileTypes: OUT
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Intermediate genome files
#&gt;       description: Archive with genome files produced when annotations are included
#&gt;         on the fly (in the mapping step).
#&gt;       streamable: no
#&gt;       id: '#intermediate_genome'
#&gt;       outputBinding:
#&gt;         glob: '*_STARgenome.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric junctions
#&gt;       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,
#&gt;         'Chimeric Junctions' won't be output.
#&gt;       streamable: no
#&gt;       id: '#chimeric_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*Chimeric.out.junction'
#&gt;       sbg:fileTypes: JUNCTION
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric alignments
#&gt;       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;         Alignment SAM and Chimeric Junctions outputs.
#&gt;       streamable: no
#&gt;       id: '#chimeric_alignments'
#&gt;       outputBinding:
#&gt;         glob: '*.Chimeric.out.sam'
#&gt;       sbg:fileTypes: SAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Aligned SAM/BAM
#&gt;       description: Aligned sequence in SAM/BAM format.
#&gt;       streamable: no
#&gt;       id: '#aligned_reads'
#&gt;       outputBinding:
#&gt;         glob:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;                 sort_name = '.sortedByCoord'
#&gt;               }
#&gt;               else {
#&gt;                 sort_name = ''
#&gt;               }
#&gt;               if ($job.inputs.outSAMtype == 'BAM') {
#&gt;                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;               }
#&gt;               else {
#&gt;                 sam_name = "*.Aligned.out.sam"
#&gt;               }
#&gt;               return sam_name
#&gt;             }
#&gt;           class: Expression
#&gt;       sbg:fileTypes: SAM, BAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     label: STAR
#&gt;     description: STAR is an ultrafast universal RNA-seq aligner. It has very high
#&gt;       mapping speed, accurate alignment of contiguous and spliced reads, detection
#&gt;       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works
#&gt;       with reads starting from lengths ~15 bases up to ~300 bases. In case of having
#&gt;       longer reads, use of STAR Long is recommended.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - tar
#&gt;     - -xvf
#&gt;     - class: Expression
#&gt;       script: $job.inputs.genome.path
#&gt;       engine: '#cwl-js-engine'
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             file = [].concat($job.inputs.reads)[0].path
#&gt;             extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;             if (extension == "gz") {
#&gt;               return "--readFilesCommand zcat"
#&gt;             } else if (extension == "bz2") {
#&gt;               return "--readFilesCommand bzcat"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n
#&gt;           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",
#&gt;           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)
#&gt;           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n
#&gt;           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",
#&gt;           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)
#&gt;           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)
#&gt;           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)
#&gt;           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n
#&gt;           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"
#&gt;           \")\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.twopassMode == "Basic") {
#&gt;               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.chimOutType == "Within") {
#&gt;               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;             }
#&gt;             else {
#&gt;               return "--chimOutType SeparateSAMold"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n
#&gt;           \   if (value == \"\") {\n      return\n    }\n    else {\n      return
#&gt;           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',
#&gt;           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',
#&gt;           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)
#&gt;           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)
#&gt;           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;           \".concat(param_list.join(\" \"))\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;               return "--quantMode ".concat($job.inputs.quantMode)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 100
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n
#&gt;           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile
#&gt;           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       prefix: --outFileNamePrefix
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;         class: Expression
#&gt;     - position: 101
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"
#&gt;           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,
#&gt;           \" \", mate1fq)\n  }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 624.0
#&gt;     'y': 323
#&gt;   sbg:x: 700.0
#&gt;   sbg:y: 200.0
#&gt; sbg:canvas_zoom: 0.6
#&gt; sbg:canvas_y: -16
#&gt; sbg:canvas_x: -41
#&gt; </div><div class='input'><span class='co'># get required node</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_required</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;            fastq genomeFastaFiles 
#&gt;        "File..."           "File" </div><div class='input'><span class='co'># set required node</span>
<span class='va'>f1</span><span class='op'>$</span><span class='va'>steps</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>$</span><span class='va'>run</span><span class='op'>$</span><span class='fu'>set_required</span><span class='op'>(</span><span class='st'>"genomeChrBinNbits"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>not implemented yet!</span></div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_required</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;            fastq genomeFastaFiles 
#&gt;        "File..."           "File" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='va'>steps</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>$</span><span class='va'>run</span><span class='op'>$</span><span class='fu'>set_required</span><span class='op'>(</span><span class='st'>"genomeChrBinNbits"</span>, <span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>not implemented yet!</span></div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_required</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;            fastq genomeFastaFiles 
#&gt;        "File..."           "File" </div><div class='input'><span class='co'># get Tool object from Flow by id and name</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>list_tool</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                        label
#&gt; 1       STAR Genome Generate
#&gt; 2 SBG FASTQ Quality Detector
#&gt; 3             Picard SortSam
#&gt; 4                       STAR
#&gt;                                                   sbgid
#&gt; 1       sevenbridges/public-apps/star-genome-generate/1
#&gt; 2 sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt; 3       sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt; 4                       sevenbridges/public-apps/star/4
#&gt;                            id
#&gt; 1       #STAR_Genome_Generate
#&gt; 2 #SBG_FASTQ_Quality_Detector
#&gt; 3             #Picard_SortSam
#&gt; 4                       #STAR</div><div class='input'><span class='co'># return two</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_tool</span><span class='op'>(</span><span class='st'>"STAR"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [[1]]
#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 1
#&gt; sbg:job:
#&gt;   allocatedResources:
#&gt;     mem: 60000
#&gt;     cpu: 15
#&gt;   inputs:
#&gt;     sjdbScore: 0
#&gt;     sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;     sjdbOverhang: 0
#&gt;     sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;     genomeChrBinNbits: genomeChrBinNbits
#&gt;     genomeSAsparseD: 0
#&gt;     sjdbGTFfile:
#&gt;     - size: 0
#&gt;       secondaryFiles: []
#&gt;       class: File
#&gt;       path: /demo/test-files/chr20.gtf
#&gt;     sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;     genomeFastaFiles:
#&gt;       size: 0
#&gt;       secondaryFiles: []
#&gt;       class: File
#&gt;       path: /sbgenomics/test-data/chr20.fa
#&gt;     sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;     genomeSAindexNbases: 0
#&gt; sbg:toolAuthor: Alexander Dobin/CSHL
#&gt; sbg:createdOn: 1450911469
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; sbg:contributors:
#&gt; - bix-demo
#&gt; sbg:links:
#&gt; - id: https://github.com/alexdobin/STAR
#&gt;   label: Homepage
#&gt; - id: https://github.com/alexdobin/STAR/releases
#&gt;   label: Releases
#&gt; - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;   label: Manual
#&gt; - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;   label: Support
#&gt; - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;   label: Publication
#&gt; sbg:project: bix-demo/star-2-4-2a-demo
#&gt; sbg:createdBy: bix-demo
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt; sbg:license: GNU General Public License v3.0 only
#&gt; sbg:revision: 1
#&gt; sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;   ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa --genomeChrBinNbits
#&gt;   genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD 0 --sjdbGTFfeatureExon
#&gt;   sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;   --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;   0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;   tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt; sbg:modifiedOn: 1450911470
#&gt; sbg:modifiedBy: bix-demo
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911469
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911470
#&gt;   sbg:revision: 1
#&gt; sbg:toolkit: STAR
#&gt; id: sevenbridges/public-apps/star-genome-generate/1
#&gt; inputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Extra alignment score
#&gt;   description: Extra alignment score for alignments that cross database junctions.
#&gt;   streamable: no
#&gt;   id: '#sjdbScore'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --sjdbScore
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: '"Overhang" length'
#&gt;   description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;     ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database is
#&gt;     not used.
#&gt;   streamable: no
#&gt;   id: '#sjdbOverhang'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --sjdbOverhang
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '100'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --sjdbGTFtagExonParentTranscript
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --sjdbGTFtagExonParentGene
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Splice junction file
#&gt;   description: Gene model annotations and/or known transcripts.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: GTF, GFF, TXT
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Set exons feature
#&gt;   description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfeatureExon'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --sjdbGTFfeatureExon
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:toolDefaultValue: exon
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Chromosome names
#&gt;   description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using ENSMEBL
#&gt;     annotations with UCSC geneomes).
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFchrPrefix'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --sjdbGTFchrPrefix
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Suffux array sparsity
#&gt;   description: 'Distance between indices: use bigger numbers to decrease needed RAM
#&gt;     at the cost of mapping speed reduction (int&gt;0).'
#&gt;   streamable: no
#&gt;   id: '#genomeSAsparseD'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --genomeSAsparseD
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Genome generation parameters
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Pre-indexing string length
#&gt;   description: Length (bases) of the SA pre-indexing string. Typically between 10
#&gt;     and 15. Longer strings will use much more memory, but allow faster searches. For
#&gt;     small genomes, this number needs to be scaled down, with a typical value of min(14,
#&gt;     log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome, this is equal to
#&gt;     9, for 100 kiloBase genome, this is equal to 7.
#&gt;   streamable: no
#&gt;   id: '#genomeSAindexNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --genomeSAindexNbases
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Genome generation parameters
#&gt;   sbg:toolDefaultValue: '14'
#&gt;   required: no
#&gt; - type:
#&gt;   - File
#&gt;   label: Genome fasta files
#&gt;   description: Reference sequence to which to align the reads.
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --genomeFastaFiles
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FA
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Bins size
#&gt;   description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins for
#&gt;     genome storage: each chromosome will occupy an integer number of bins. If you
#&gt;     are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds, you may
#&gt;     need to reduce this number to reduce RAM consumption. The following scaling is
#&gt;     recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;     For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this is
#&gt;     equal to 15.'
#&gt;   streamable: no
#&gt;   id: '#genomeChrBinNbits'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --genomeChrBinNbits
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Genome generation parameters
#&gt;   sbg:toolDefaultValue: '18'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Genome Files
#&gt;   description: Genome files comprise binary genome sequence, suffix arrays, text chromosome
#&gt;     names/lengths, splice junctions coordinates, and transcripts/genes information.
#&gt;   streamable: no
#&gt;   id: '#genome'
#&gt;   outputBinding:
#&gt;     glob: '*.tar'
#&gt;   sbg:fileTypes: TAR
#&gt; requirements:
#&gt; - class: ExpressionEngineRequirement
#&gt;   id: '#cwl-js-engine'
#&gt;   requirements:
#&gt;   - class: DockerRequirement
#&gt;     dockerPull: rabix/js-engine
#&gt; hints:
#&gt; - class: DockerRequirement
#&gt;   dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;   dockerImageId: a4b0ad2c3cae
#&gt; - class: sbg:CPURequirement
#&gt;   value: 15
#&gt; - class: sbg:MemRequirement
#&gt;   value: 60000
#&gt; label: STAR Genome Generate
#&gt; description: STAR Genome Generate is a tool that generates genome index files. One
#&gt;   set of files should be generated per each genome/annotation combination. Once produced,
#&gt;   these files could be used as long as genome/annotation combination stays the same.
#&gt;   Also, STAR Genome Generate which produced these files and STAR aligner using them
#&gt;   must be the same toolkit version.
#&gt; class: CommandLineTool
#&gt; baseCommand:
#&gt; - mkdir
#&gt; - genomeDir
#&gt; - '&amp;&amp;'
#&gt; - /opt/STAR
#&gt; - --runMode
#&gt; - genomeGenerate
#&gt; - --genomeDir
#&gt; - ./genomeDir
#&gt; - --runThreadN
#&gt; - class: Expression
#&gt;   script: |-
#&gt;     {
#&gt;       return $job.allocatedResources.cpu
#&gt;     }
#&gt;   engine: '#cwl-js-engine'
#&gt; arguments:
#&gt; - position: 99
#&gt;   separate: yes
#&gt;   valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n  var
#&gt;       list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths =
#&gt;       \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;       \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;       \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext == \"gtf\")
#&gt;       {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n      }\n      if
#&gt;       (ext == \"txt\") {\n        sjFormat = \"True\"\n        return sjFormat\n      }\n
#&gt;       \   })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;       {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;       \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\") {\n
#&gt;       \       return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n  }\n}"
#&gt;     class: Expression
#&gt; stdin: ''
#&gt; stdout: ''
#&gt; successCodes: []
#&gt; temporaryFailCodes: []
#&gt; x: 384.0832266
#&gt; 'y': 446.4998957
#&gt; 
#&gt; 
#&gt; [[2]]
#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 4
#&gt; sbg:job:
#&gt;   allocatedResources:
#&gt;     mem: 60000
#&gt;     cpu: 15
#&gt;   inputs:
#&gt;     alignWindowsPerReadNmax: 0
#&gt;     outSAMheaderPG: outSAMheaderPG
#&gt;     GENOME_DIR_NAME: ''
#&gt;     outFilterMatchNminOverLread: 0
#&gt;     rg_platform_unit_id: rg_platform_unit
#&gt;     alignTranscriptsPerReadNmax: 0
#&gt;     readMapNumber: 0
#&gt;     alignSplicedMateMapLminOverLmate: 0
#&gt;     alignMatesGapMax: 0
#&gt;     outFilterMultimapNmax: 0
#&gt;     clip5pNbases:
#&gt;     - 0
#&gt;     outSAMstrandField: None
#&gt;     readMatesLengthsIn: NotEqual
#&gt;     outSAMattributes: Standard
#&gt;     seedMultimapNmax: 0
#&gt;     rg_mfl: rg_mfl
#&gt;     chimSegmentMin: 0
#&gt;     winAnchorDistNbins: 0
#&gt;     outSortingType: SortedByCoordinate
#&gt;     outFilterMultimapScoreRange: 0
#&gt;     sjdbInsertSave: Basic
#&gt;     clip3pAfterAdapterNbases:
#&gt;     - 0
#&gt;     scoreDelBase: 0
#&gt;     outFilterMatchNmin: 0
#&gt;     twopass1readsN: 0
#&gt;     outSAMunmapped: None
#&gt;     genome:
#&gt;       size: 0
#&gt;       secondaryFiles: []
#&gt;       class: File
#&gt;       path: genome.ext
#&gt;     sjdbGTFtagExonParentTranscript: ''
#&gt;     limitBAMsortRAM: 0
#&gt;     alignEndsType: Local
#&gt;     seedNoneLociPerWindow: 0
#&gt;     rg_sample_id: rg_sample
#&gt;     sjdbGTFtagExonParentGene: ''
#&gt;     chimScoreMin: 0
#&gt;     outSJfilterIntronMaxVsReadN:
#&gt;     - 0
#&gt;     twopassMode: Basic
#&gt;     alignSplicedMateMapLmin: 0
#&gt;     outSJfilterReads: All
#&gt;     outSAMprimaryFlag: OneBestScore
#&gt;     outSJfilterCountTotalMin:
#&gt;     - 3
#&gt;     - 1
#&gt;     - 1
#&gt;     - 1
#&gt;     outSAMorder: Paired
#&gt;     outSAMflagAND: 0
#&gt;     chimScoreSeparation: 0
#&gt;     alignSJoverhangMin: 0
#&gt;     outFilterScoreMin: 0
#&gt;     seedSearchStartLmax: 0
#&gt;     scoreGapGCAG: 0
#&gt;     scoreGenomicLengthLog2scale: 0
#&gt;     outFilterIntronMotifs: None
#&gt;     outFilterMismatchNmax: 0
#&gt;     reads:
#&gt;     - size: 0
#&gt;       secondaryFiles: []
#&gt;       class: File
#&gt;       metadata:
#&gt;         format: fastq
#&gt;         paired_end: '1'
#&gt;         seq_center: illumina
#&gt;       path: /test-data/mate_1.fastq.bz2
#&gt;     scoreGap: 0
#&gt;     outSJfilterOverhangMin:
#&gt;     - 30
#&gt;     - 12
#&gt;     - 12
#&gt;     - 12
#&gt;     outSAMflagOR: 0
#&gt;     outSAMmode: Full
#&gt;     rg_library_id: ''
#&gt;     chimScoreJunctionNonGTAG: 0
#&gt;     scoreInsOpen: 0
#&gt;     clip3pAdapterSeq:
#&gt;     - clip3pAdapterSeq
#&gt;     chimScoreDropMax: 0
#&gt;     outFilterType: Normal
#&gt;     scoreGapATAC: 0
#&gt;     rg_platform: Ion Torrent PGM
#&gt;     clip3pAdapterMMp:
#&gt;     - 0
#&gt;     sjdbGTFfeatureExon: ''
#&gt;     outQSconversionAdd: 0
#&gt;     quantMode: TranscriptomeSAM
#&gt;     alignIntronMin: 0
#&gt;     scoreInsBase: 0
#&gt;     scoreGapNoncan: 0
#&gt;     seedSearchLmax: 0
#&gt;     outSJfilterDistToOtherSJmin:
#&gt;     - 0
#&gt;     outFilterScoreMinOverLread: 0
#&gt;     alignSJDBoverhangMin: 0
#&gt;     limitOutSJcollapsed: 0
#&gt;     winAnchorMultimapNmax: 0
#&gt;     outFilterMismatchNoverLmax: 0
#&gt;     rg_seq_center: ''
#&gt;     outSAMheaderHD: outSAMheaderHD
#&gt;     chimOutType: Within
#&gt;     quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;     limitOutSJoneRead: 0
#&gt;     alignTranscriptsPerWindowNmax: 0
#&gt;     sjdbOverhang: ~
#&gt;     outReadsUnmapped: Fastx
#&gt;     scoreStitchSJshift: 0
#&gt;     seedPerWindowNmax: 0
#&gt;     outSJfilterCountUniqueMin:
#&gt;     - 3
#&gt;     - 1
#&gt;     - 1
#&gt;     - 1
#&gt;     scoreDelOpen: 0
#&gt;     sjdbGTFfile:
#&gt;     - path: /demo/test-data/chr20.gtf
#&gt;     clip3pNbases:
#&gt;     - 0
#&gt;     - 3
#&gt;     winBinNbits: 0
#&gt;     sjdbScore: ~
#&gt;     seedSearchStartLmaxOverLread: 0
#&gt;     alignIntronMax: 0
#&gt;     seedPerReadNmax: 0
#&gt;     outFilterMismatchNoverReadLmax: 0
#&gt;     winFlankNbins: 0
#&gt;     sjdbGTFchrPrefix: chrPrefix
#&gt;     alignSoftClipAtReferenceEnds: 'Yes'
#&gt;     outSAMreadID: Standard
#&gt;     outSAMtype: BAM
#&gt;     chimJunctionOverhangMin: 0
#&gt;     limitSjdbInsertNsj: 0
#&gt;     outSAMmapqUnique: 0
#&gt; sbg:toolAuthor: Alexander Dobin/CSHL
#&gt; sbg:createdOn: 1450911471
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; sbg:contributors:
#&gt; - ana_d
#&gt; - bix-demo
#&gt; - uros_sipetic
#&gt; sbg:links:
#&gt; - id: https://github.com/alexdobin/STAR
#&gt;   label: Homepage
#&gt; - id: https://github.com/alexdobin/STAR/releases
#&gt;   label: Releases
#&gt; - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;   label: Manual
#&gt; - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;   label: Support
#&gt; - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;   label: Publication
#&gt; sbg:project: bix-demo/star-2-4-2a-demo
#&gt; sbg:createdBy: bix-demo
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: sevenbridges/public-apps/star/4
#&gt; sbg:license: GNU General Public License v3.0 only
#&gt; sbg:revision: 4
#&gt; sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;   bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix --sjdbInsertSave
#&gt;   Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline ID:1 CN:illumina
#&gt;   PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode TranscriptomeSAM
#&gt;   --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;   tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp; mv
#&gt;   mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt; sbg:modifiedOn: 1462889222
#&gt; sbg:modifiedBy: ana_d
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911471
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911473
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911475
#&gt;   sbg:revision: 2
#&gt; - sbg:modifiedBy: uros_sipetic
#&gt;   sbg:modifiedOn: 1462878528
#&gt;   sbg:revision: 3
#&gt; - sbg:modifiedBy: ana_d
#&gt;   sbg:modifiedOn: 1462889222
#&gt;   sbg:revision: 4
#&gt; sbg:toolkit: STAR
#&gt; id: sevenbridges/public-apps/star/4
#&gt; inputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Flanking regions size
#&gt;   description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;     flanking regions for each window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winFlankNbins'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winFlankNbins
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '4'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Bin size
#&gt;   description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;     each window will occupy an integer number of bins (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winBinNbits'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winBinNbits
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '16'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winAnchorMultimapNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winAnchorDistNbins
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: twopassMode
#&gt;     symbols:
#&gt;     - None
#&gt;     - Basic
#&gt;     type: enum
#&gt;   label: Two-pass mode
#&gt;   description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass mapping,
#&gt;     with all 1st pass junctions inserted into the genome indices on the fly.'
#&gt;   streamable: no
#&gt;   id: '#twopassMode'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --twopassMode
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 2-pass mapping
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Reads to process in 1st step
#&gt;   description: 'Number of reads to process for the 1st step. 0: 1-step only, no 2nd
#&gt;     pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;   streamable: no
#&gt;   id: '#twopass1readsN'
#&gt;   sbg:category: 2-pass mapping
#&gt;   sbg:toolDefaultValue: '-1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Extra alignment score
#&gt;   description: Extra alignment score for alignments that cross database junctions.
#&gt;   streamable: no
#&gt;   id: '#sjdbScore'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: '2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: '"Overhang" length'
#&gt;   description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;     ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database is
#&gt;     not used.
#&gt;   streamable: no
#&gt;   id: '#sjdbOverhang'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: '100'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: sjdbInsertSave
#&gt;     symbols:
#&gt;     - Basic
#&gt;     - All
#&gt;     - None
#&gt;     type: enum
#&gt;   label: Save junction files
#&gt;   description: 'Which files to save when sjdb junctions are inserted on the fly at
#&gt;     the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;     files; All: all files including big Genome, SA and SAindex. These files are output
#&gt;     as archive.'
#&gt;   streamable: no
#&gt;   id: '#sjdbInsertSave'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Splice junction file
#&gt;   description: Gene model annotations and/or known transcripts. No need to include
#&gt;     this input, except in case of using "on the fly" annotations.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: GTF, GFF, TXT
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Set exons feature
#&gt;   description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfeatureExon'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: exon
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Chromosome names
#&gt;   description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using ENSMEBL
#&gt;     annotations with UCSC geneomes).
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFchrPrefix'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Search start point normalized
#&gt;   description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;     for paired-end reads).
#&gt;   streamable: no
#&gt;   id: '#seedSearchStartLmaxOverLread'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedSearchStartLmaxOverLread
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '1.0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Search start point
#&gt;   description: Defines the search start point through the read - the read is split
#&gt;     into pieces no longer than this value (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#seedSearchStartLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedSearchStartLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max seed length
#&gt;   description: Defines the maximum length of the seeds, if =0 max seed length is infinite
#&gt;     (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedSearchLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedSearchLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max seeds per window
#&gt;   description: Max number of seeds per window (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedPerWindowNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedPerWindowNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max seeds per read
#&gt;   description: Max number of seeds per read (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedPerReadNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedPerReadNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '1000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max one-seed loci per window
#&gt;   description: Max number of one seed loci per window (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedNoneLociPerWindow'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedNoneLociPerWindow
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Filter pieces for stitching
#&gt;   description: Only pieces that map fewer than this value are utilized in the stitching
#&gt;     procedure (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedMultimapNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedMultimapNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max score reduction
#&gt;   description: Maximum score reduction while searching for SJ boundaries in the stitching
#&gt;     step.
#&gt;   streamable: no
#&gt;   id: '#scoreStitchSJshift'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreStitchSJshift
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Insertion Open Penalty
#&gt;   description: Insertion open penalty.
#&gt;   streamable: no
#&gt;   id: '#scoreInsOpen'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreInsOpen
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Insertion extension penalty
#&gt;   description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;   streamable: no
#&gt;   id: '#scoreInsBase'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreInsBase
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Log scaled score
#&gt;   description: 'Extra score logarithmically scaled with genomic length of the alignment:
#&gt;     &lt;int&gt;*log2(genomicLength).'
#&gt;   streamable: no
#&gt;   id: '#scoreGenomicLengthLog2scale'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGenomicLengthLog2scale
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-0.25'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Non-canonical gap open
#&gt;   description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;   streamable: no
#&gt;   id: '#scoreGapNoncan'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGapNoncan
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-8'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: GC/AG and CT/GC gap open
#&gt;   description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;   streamable: no
#&gt;   id: '#scoreGapGCAG'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGapGCAG
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-4'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: AT/AC and GT/AT gap open
#&gt;   description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;   streamable: no
#&gt;   id: '#scoreGapATAC'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGapATAC
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-8'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Gap open penalty
#&gt;   description: Gap open penalty.
#&gt;   streamable: no
#&gt;   id: '#scoreGap'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGap
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Deletion open penalty
#&gt;   description: Deletion open penalty.
#&gt;   streamable: no
#&gt;   id: '#scoreDelOpen'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreDelOpen
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Deletion extension penalty
#&gt;   description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;   streamable: no
#&gt;   id: '#scoreDelBase'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreDelBase
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Sequencing center
#&gt;   description: Specify the sequencing center for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_seq_center'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Sample ID
#&gt;   description: Specify the sample ID for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_sample_id'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Platform unit ID
#&gt;   description: Specify the platform unit ID for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_platform_unit_id'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: rg_platform
#&gt;     symbols:
#&gt;     - LS 454
#&gt;     - Helicos
#&gt;     - Illumina
#&gt;     - ABI SOLiD
#&gt;     - Ion Torrent PGM
#&gt;     - PacBio
#&gt;     type: enum
#&gt;   label: Platform
#&gt;   description: Specify the version of the technology that was used for sequencing
#&gt;     or assaying.
#&gt;   streamable: no
#&gt;   id: '#rg_platform'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Median fragment length
#&gt;   description: Specify the median fragment length for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_mfl'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Library ID
#&gt;   description: Specify the library ID for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_library_id'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Read sequence
#&gt;   description: Read sequence.
#&gt;   streamable: no
#&gt;   id: '#reads'
#&gt;   inputBinding:
#&gt;     position: 10
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n
#&gt;         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if
#&gt;         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n
#&gt;         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata
#&gt;         != null){\n        if (list[index].metadata.paired_end == 1){\n          left
#&gt;         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n
#&gt;         \         right = list[index].path\n        }\n      }\n    }\n    \n    if
#&gt;         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n
#&gt;         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n
#&gt;         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if
#&gt;         (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end
#&gt;         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n
#&gt;         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=
#&gt;         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n
#&gt;         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"
#&gt;         \"))\n  }\n}"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: readMatesLengthsIn
#&gt;     symbols:
#&gt;     - NotEqual
#&gt;     - Equal
#&gt;     type: enum
#&gt;   label: Reads lengths
#&gt;   description: Equal/Not equal - lengths of names, sequences, qualities for both mates
#&gt;     are the same/not the same. "Not equal" is safe in all situations.
#&gt;   streamable: no
#&gt;   id: '#readMatesLengthsIn'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --readMatesLengthsIn
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: NotEqual
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Reads to map
#&gt;   description: Number of reads to map from the beginning of the file.
#&gt;   streamable: no
#&gt;   id: '#readMapNumber'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --readMapNumber
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '-1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: quantTranscriptomeBan
#&gt;     symbols:
#&gt;     - IndelSoftclipSingleend
#&gt;     - Singleend
#&gt;     type: enum
#&gt;   label: Prohibit alignment type
#&gt;   description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;     indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;     prohibit single-end alignments.'
#&gt;   streamable: no
#&gt;   id: '#quantTranscriptomeBan'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --quantTranscriptomeBan
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Quantification of Annotations
#&gt;   sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: quantMode
#&gt;     symbols:
#&gt;     - TranscriptomeSAM
#&gt;     - GeneCounts
#&gt;     type: enum
#&gt;   label: Quantification mode
#&gt;   description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;     SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts' option,
#&gt;     STAR will count number of reads per gene while mapping.
#&gt;   streamable: no
#&gt;   id: '#quantMode'
#&gt;   sbg:category: Quantification of Annotations
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSortingType
#&gt;     symbols:
#&gt;     - Unsorted
#&gt;     - SortedByCoordinate
#&gt;     - Unsorted SortedByCoordinate
#&gt;     type: enum
#&gt;   label: Output sorting type
#&gt;   description: Type of output sorting.
#&gt;   streamable: no
#&gt;   id: '#outSortingType'
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: SortedByCoordinate
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSJfilterReads
#&gt;     symbols:
#&gt;     - All
#&gt;     - Unique
#&gt;     type: enum
#&gt;   label: Collapsed junctions reads
#&gt;   description: 'Which reads to consider for collapsed splice junctions output. All:
#&gt;     all reads, unique- and multi-mappers; Unique: uniquely mapping reads only.'
#&gt;   streamable: no
#&gt;   id: '#outSJfilterReads'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterReads
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: All
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min overhang SJ
#&gt;   description: Minimum overhang length for splice junctions on both sides for each
#&gt;     of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;     field. Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterOverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterOverhangMin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 30 12 12 12
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Max gap allowed
#&gt;   description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads (int
#&gt;     &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b, by
#&gt;     2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;     Does not apply to annotated junctions.'
#&gt;   streamable: no
#&gt;   id: '#outSJfilterIntronMaxVsReadN'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterIntronMaxVsReadN
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 50000 100000 200000
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min distance to other donor/acceptor
#&gt;   description: Minimum allowed distance to other junctions' donor/acceptor for each
#&gt;     of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterDistToOtherSJmin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterDistToOtherSJmin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 10 0 5 10
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min unique count
#&gt;   description: Minimum uniquely mapping read count per junction for each of the motifs.
#&gt;     To set no output for desired motif, assign -1 to the corresponding field. Junctions
#&gt;     are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;     conditions are satisfied. Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterCountUniqueMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterCountUniqueMin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 3 1 1 1
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min total count
#&gt;   description: Minimum total (multi-mapping+unique) read count per junction for each
#&gt;     of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;     field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;     conditions are satisfied. Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterCountTotalMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterCountTotalMin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 3 1 1 1
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMunmapped
#&gt;     symbols:
#&gt;     - None
#&gt;     - Within
#&gt;     type: enum
#&gt;   label: Write unmapped in SAM
#&gt;   description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;     output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;   streamable: no
#&gt;   id: '#outSAMunmapped'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMunmapped
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMtype
#&gt;     symbols:
#&gt;     - SAM
#&gt;     - BAM
#&gt;     type: enum
#&gt;   label: Output format
#&gt;   description: Format of output alignments.
#&gt;   streamable: no
#&gt;   id: '#outSAMtype'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     separate: yes
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: |-
#&gt;         {
#&gt;           SAM_type = $job.inputs.outSAMtype
#&gt;           SORT_type = $job.inputs.outSortingType
#&gt;           if (SAM_type &amp;&amp; SORT_type) {
#&gt;             return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;           }
#&gt;         }
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: SAM
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMstrandField
#&gt;     symbols:
#&gt;     - None
#&gt;     - intronMotif
#&gt;     type: enum
#&gt;   label: Strand field flag
#&gt;   description: 'Cufflinks-like strand field flag. None: not used; intronMotif: strand
#&gt;     derived from the intron motif. Reads with inconsistent and/or non-canonical introns
#&gt;     are filtered out.'
#&gt;   streamable: no
#&gt;   id: '#outSAMstrandField'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMstrandField
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMreadID
#&gt;     symbols:
#&gt;     - Standard
#&gt;     - Number
#&gt;     type: enum
#&gt;   label: Read ID
#&gt;   description: 'Read ID record type. Standard: first word (until space) from the FASTx
#&gt;     read ID line, removing /1,/2 from the end; Number: read number (index) in the
#&gt;     FASTx file.'
#&gt;   streamable: no
#&gt;   id: '#outSAMreadID'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMreadID
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Standard
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMprimaryFlag
#&gt;     symbols:
#&gt;     - OneBestScore
#&gt;     - AllBestScore
#&gt;     type: enum
#&gt;   label: Primary alignments
#&gt;   description: 'Which alignments are considered primary - all others will be marked
#&gt;     with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best score
#&gt;     is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;   streamable: no
#&gt;   id: '#outSAMprimaryFlag'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMprimaryFlag
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: OneBestScore
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMorder
#&gt;     symbols:
#&gt;     - Paired
#&gt;     - PairedKeepInputOrder
#&gt;     type: enum
#&gt;   label: Sorting in SAM
#&gt;   description: 'Type of sorting for the SAM output. Paired: one mate after the other
#&gt;     for all paired alignments; PairedKeepInputOrder: one mate after the other for
#&gt;     all paired alignments, the order is kept the same as in the input FASTQ files.'
#&gt;   streamable: no
#&gt;   id: '#outSAMorder'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMorder
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Paired
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMmode
#&gt;     symbols:
#&gt;     - Full
#&gt;     - NoQS
#&gt;     type: enum
#&gt;   label: SAM mode
#&gt;   description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but without
#&gt;     quality scores.'
#&gt;   streamable: no
#&gt;   id: '#outSAMmode'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMmode
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Full
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: MAPQ value
#&gt;   description: MAPQ value for unique mappers (0 to 255).
#&gt;   streamable: no
#&gt;   id: '#outSAMmapqUnique'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMmapqUnique
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '255'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: SAM header @PG
#&gt;   description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;   streamable: no
#&gt;   id: '#outSAMheaderPG'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMheaderPG
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: SAM header @HD
#&gt;   description: '@HD (header) line of the SAM header.'
#&gt;   streamable: no
#&gt;   id: '#outSAMheaderHD'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMheaderHD
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: OR SAM flag
#&gt;   description: Set specific bits of the SAM FLAG.
#&gt;   streamable: no
#&gt;   id: '#outSAMflagOR'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMflagOR
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: AND SAM flag
#&gt;   description: Set specific bits of the SAM FLAG.
#&gt;   streamable: no
#&gt;   id: '#outSAMflagAND'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMflagAND
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '65535'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMattributes
#&gt;     symbols:
#&gt;     - Standard
#&gt;     - NH
#&gt;     - All
#&gt;     - None
#&gt;     type: enum
#&gt;   label: SAM attributes
#&gt;   description: 'Desired SAM attributes, in the order desired for the output SAM. NH:
#&gt;     any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM NM MD jM
#&gt;     jI; None: no attributes.'
#&gt;   streamable: no
#&gt;   id: '#outSAMattributes'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMattributes
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Standard
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outReadsUnmapped
#&gt;     symbols:
#&gt;     - None
#&gt;     - Fastx
#&gt;     type: enum
#&gt;   label: Output unmapped reads
#&gt;   description: 'Output of unmapped reads (besides SAM). None: no output; Fastx: output
#&gt;     in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;   streamable: no
#&gt;   id: '#outReadsUnmapped'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outReadsUnmapped
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Quality conversion
#&gt;   description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;     to Sanger, use -31).
#&gt;   streamable: no
#&gt;   id: '#outQSconversionAdd'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outQSconversionAdd
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outFilterType
#&gt;     symbols:
#&gt;     - Normal
#&gt;     - BySJout
#&gt;     type: enum
#&gt;   label: Filtering type
#&gt;   description: 'Type of filtering. Normal: standard filtering using only current alignment;
#&gt;     BySJout: keep only those reads that contain junctions that passed filtering into
#&gt;     SJ.out.tab.'
#&gt;   streamable: no
#&gt;   id: '#outFilterType'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterType
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: Normal
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Min score normalized
#&gt;   description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;     for paired-end reads).'
#&gt;   streamable: no
#&gt;   id: '#outFilterScoreMinOverLread'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterScoreMinOverLread
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0.66'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min score
#&gt;   description: Alignment will be output only if its score is higher than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterScoreMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterScoreMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Multimapping score range
#&gt;   description: The score range below the maximum score for multimapping alignments.
#&gt;   streamable: no
#&gt;   id: '#outFilterMultimapScoreRange'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMultimapScoreRange
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max number of mappings
#&gt;   description: Read alignments will be output only if the read maps fewer than this
#&gt;     value, otherwise no alignments will be output.
#&gt;   streamable: no
#&gt;   id: '#outFilterMultimapNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMultimapNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Mismatches to *read* length
#&gt;   description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;     length is less than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMismatchNoverReadLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMismatchNoverReadLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Mismatches to *mapped* length
#&gt;   description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;     length is less than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMismatchNoverLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMismatchNoverLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0.3'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max number of mismatches
#&gt;   description: Alignment will be output only if it has fewer mismatches than this
#&gt;     value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMismatchNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMismatchNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Min matched bases normalized
#&gt;   description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;     lengths for paired-end reads).'
#&gt;   streamable: no
#&gt;   id: '#outFilterMatchNminOverLread'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMatchNminOverLread
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0.66'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min matched bases
#&gt;   description: Alignment will be output only if the number of matched bases is higher
#&gt;     than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMatchNmin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMatchNmin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outFilterIntronMotifs
#&gt;     symbols:
#&gt;     - None
#&gt;     - RemoveNoncanonical
#&gt;     - RemoveNoncanonicalUnannotated
#&gt;     type: enum
#&gt;   label: Motifs filtering
#&gt;   description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;     filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;     filter out alignments that contain non-canonical unannotated junctions when using
#&gt;     annotated splice junctions database. The annotated non-canonical junctions will
#&gt;     be kept.'
#&gt;   streamable: no
#&gt;   id: '#outFilterIntronMotifs'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterIntronMotifs
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max insert junctions
#&gt;   description: Maximum number of junction to be inserted to the genome on the fly
#&gt;     at the mapping stage, including those from annotations and those detected in the
#&gt;     1st step of the 2-pass run.
#&gt;   streamable: no
#&gt;   id: '#limitSjdbInsertNsj'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitSjdbInsertNsj
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '1000000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Junctions max number
#&gt;   description: Max number of junctions for one read (including all multi-mappers).
#&gt;   streamable: no
#&gt;   id: '#limitOutSJoneRead'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitOutSJoneRead
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '1000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Collapsed junctions max number
#&gt;   description: Max number of collapsed junctions.
#&gt;   streamable: no
#&gt;   id: '#limitOutSJcollapsed'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitOutSJcollapsed
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '1000000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Limit BAM sorting memory
#&gt;   description: Maximum available RAM for sorting BAM. If set to 0, it will be set
#&gt;     to the genome index size.
#&gt;   streamable: no
#&gt;   id: '#limitBAMsortRAM'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitBAMsortRAM
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Genome dir name
#&gt;   description: Name of the directory which contains genome files (when genome.tar
#&gt;     is uncompressed).
#&gt;   streamable: no
#&gt;   id: '#genomeDirName'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --genomeDir
#&gt;     separate: yes
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: $job.inputs.genomeDirName || "genomeDir"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:toolDefaultValue: genomeDir
#&gt;   required: no
#&gt; - type:
#&gt;   - File
#&gt;   label: Genome files
#&gt;   description: Genome files created using STAR Genome Generate.
#&gt;   streamable: no
#&gt;   id: '#genome'
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: TAR
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Clip 5p bases
#&gt;   description: Number of bases to clip from 5p of each mate. In case only one value
#&gt;     is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip5pNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip5pNbases
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Clip 3p bases
#&gt;   description: Number of bases to clip from 3p of each mate. In case only one value
#&gt;     is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pNbases
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Clip 3p after adapter seq.
#&gt;   description: Number of bases to clip from 3p of each mate after the adapter clipping.
#&gt;     In case only one value is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pAfterAdapterNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pAfterAdapterNbases
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: string
#&gt;     type: array
#&gt;   label: Clip 3p adapter sequence
#&gt;   description: Adapter sequence to clip from 3p of each mate. In case only one value
#&gt;     is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pAdapterSeq'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pAdapterSeq
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: float
#&gt;     type: array
#&gt;   label: Max mismatches proportions
#&gt;   description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;     In case only one value is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pAdapterMMp'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pAdapterMMp
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0.1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min segment length
#&gt;   description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;     (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimSegmentMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimSegmentMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '15'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min separation score
#&gt;   description: Minimum difference (separation) between the best chimeric score and
#&gt;     the next one (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimScoreSeparation'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreSeparation
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min total score
#&gt;   description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimScoreMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Non-GT/AG penalty
#&gt;   description: Penalty for a non-GT/AG chimeric junction.
#&gt;   streamable: no
#&gt;   id: '#chimScoreJunctionNonGTAG'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreJunctionNonGTAG
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '-1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max drop score
#&gt;   description: Max drop (difference) of chimeric score (the sum of scores of all chimeric
#&gt;     segements) from the read length (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimScoreDropMax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreDropMax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '20'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: chimOutType
#&gt;     symbols:
#&gt;     - SeparateSAMold
#&gt;     - Within
#&gt;     type: enum
#&gt;   label: Chimeric output type
#&gt;   description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;     Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;   streamable: no
#&gt;   id: '#chimOutType'
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: SeparateSAMold
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min junction overhang
#&gt;   description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimJunctionOverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimJunctionOverhangMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '20'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Max windows per read
#&gt;   description: Max number of windows per read (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignWindowsPerReadNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignWindowsPerReadNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max transcripts per window
#&gt;   description: Max number of transcripts per window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignTranscriptsPerWindowNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignTranscriptsPerWindowNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '100'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max transcripts per read
#&gt;   description: Max number of different alignments per read to consider (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignTranscriptsPerReadNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignTranscriptsPerReadNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Min mapped length normalized
#&gt;   description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSplicedMateMapLminOverLmate'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSplicedMateMapLminOverLmate
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0.66'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min mapped length
#&gt;   description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSplicedMateMapLmin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSplicedMateMapLmin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: alignSoftClipAtReferenceEnds
#&gt;     symbols:
#&gt;     - 'Yes'
#&gt;     - 'No'
#&gt;     type: enum
#&gt;   label: Soft clipping
#&gt;   description: 'Option which allows soft clipping of alignments at the reference (chromosome)
#&gt;     ends. Can be disabled for compatibility with Cufflinks/Cuffmerge. Yes: Enables
#&gt;     soft clipping; No: Disables soft clipping.'
#&gt;   streamable: no
#&gt;   id: '#alignSoftClipAtReferenceEnds'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSoftClipAtReferenceEnds
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: 'Yes'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min overhang
#&gt;   description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSJoverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSJoverhangMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '5'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: 'Min overhang: annotated'
#&gt;   description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced alignments
#&gt;     (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSJDBoverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSJDBoverhangMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '3'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max mates gap
#&gt;   description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;     by (2^winBinNbits)*winAnchorDistNbins.
#&gt;   streamable: no
#&gt;   id: '#alignMatesGapMax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignMatesGapMax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min intron size
#&gt;   description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;     &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;   streamable: no
#&gt;   id: '#alignIntronMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignIntronMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '21'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max intron size
#&gt;   description: Maximum intron size, if 0, max intron size will be determined by (2^winBinNbits)*winAnchorDistNbins.
#&gt;   streamable: no
#&gt;   id: '#alignIntronMax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignIntronMax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: alignEndsType
#&gt;     symbols:
#&gt;     - Local
#&gt;     - EndToEnd
#&gt;     type: enum
#&gt;   label: Alignment type
#&gt;   description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;     soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;   streamable: no
#&gt;   id: '#alignEndsType'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignEndsType
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: Local
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   outputBinding:
#&gt;     glob: '*Unmapped.out*'
#&gt;   sbg:fileTypes: FASTQ
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   outputBinding:
#&gt;     glob: '*Transcriptome*'
#&gt;   sbg:fileTypes: BAM
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   outputBinding:
#&gt;     glob: '*SJ.out.tab'
#&gt;   sbg:fileTypes: TAB
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   outputBinding:
#&gt;     glob: '*ReadsPerGene*'
#&gt;   sbg:fileTypes: TAB
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   outputBinding:
#&gt;     glob: '*Log*.out'
#&gt;   sbg:fileTypes: OUT
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   outputBinding:
#&gt;     glob: '*_STARgenome.tar'
#&gt;   sbg:fileTypes: TAR
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   outputBinding:
#&gt;     glob: '*Chimeric.out.junction'
#&gt;   sbg:fileTypes: JUNCTION
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   outputBinding:
#&gt;     glob: '*.Chimeric.out.sam'
#&gt;   sbg:fileTypes: SAM
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Aligned SAM/BAM
#&gt;   description: Aligned sequence in SAM/BAM format.
#&gt;   streamable: no
#&gt;   id: '#aligned_reads'
#&gt;   outputBinding:
#&gt;     glob:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: |-
#&gt;         {
#&gt;           if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;             sort_name = '.sortedByCoord'
#&gt;           }
#&gt;           else {
#&gt;             sort_name = ''
#&gt;           }
#&gt;           if ($job.inputs.outSAMtype == 'BAM') {
#&gt;             sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;           }
#&gt;           else {
#&gt;             sam_name = "*.Aligned.out.sam"
#&gt;           }
#&gt;           return sam_name
#&gt;         }
#&gt;       class: Expression
#&gt;   sbg:fileTypes: SAM, BAM
#&gt; requirements:
#&gt; - class: ExpressionEngineRequirement
#&gt;   id: '#cwl-js-engine'
#&gt;   requirements:
#&gt;   - class: DockerRequirement
#&gt;     dockerPull: rabix/js-engine
#&gt; hints:
#&gt; - class: DockerRequirement
#&gt;   dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;   dockerImageId: a4b0ad2c3cae
#&gt; - class: sbg:MemRequirement
#&gt;   value: 60000
#&gt; - class: sbg:CPURequirement
#&gt;   value: 15
#&gt; label: STAR
#&gt; description: STAR is an ultrafast universal RNA-seq aligner. It has very high mapping
#&gt;   speed, accurate alignment of contiguous and spliced reads, detection of polyA-tails,
#&gt;   non-canonical splices and chimeric (fusion) junctions. It works with reads starting
#&gt;   from lengths ~15 bases up to ~300 bases. In case of having longer reads, use of
#&gt;   STAR Long is recommended.
#&gt; class: CommandLineTool
#&gt; baseCommand:
#&gt; - tar
#&gt; - -xvf
#&gt; - class: Expression
#&gt;   script: $job.inputs.genome.path
#&gt;   engine: '#cwl-js-engine'
#&gt; - '&amp;&amp;'
#&gt; - /opt/STAR
#&gt; - --runThreadN
#&gt; - class: Expression
#&gt;   script: |-
#&gt;     {
#&gt;       return $job.allocatedResources.cpu
#&gt;     }
#&gt;   engine: '#cwl-js-engine'
#&gt; arguments:
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         file = [].concat($job.inputs.reads)[0].path
#&gt;         extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;         if (extension == "gz") {
#&gt;           return "--readFilesCommand zcat"
#&gt;         } else if (extension == "bz2") {
#&gt;           return "--readFilesCommand bzcat"
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n  var
#&gt;       list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths =
#&gt;       \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;       \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;       \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext == \"gtf\")
#&gt;       {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n      }\n      if
#&gt;       (ext == \"txt\") {\n        sjFormat = \"True\"\n        return sjFormat\n      }\n
#&gt;       \   })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;       {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;       \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\") {\n
#&gt;       \       return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n  }\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;       {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n  if
#&gt;       ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\", $job.inputs.sjdbGTFfeatureExon]\n
#&gt;       \ }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript) {\n    c = [\"--sjdbGTFtagExonParentTranscript\",
#&gt;       $job.inputs.sjdbGTFtagExonParentTranscript]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentGene)
#&gt;       {\n    d = [\"--sjdbGTFtagExonParentGene\", $job.inputs.sjdbGTFtagExonParentGene]\n
#&gt;       \ }\n  if ($job.inputs.sjdbOverhang) {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n
#&gt;       \ }\n  if ($job.inputs.sjdbScore) {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n
#&gt;       \ }\n  if ($job.inputs.sjdbInsertSave) {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n
#&gt;       \ }\n  \n  \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;       {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\" \")\n
#&gt;       \ }\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         if ($job.inputs.twopassMode == "Basic") {
#&gt;           return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         if ($job.inputs.chimOutType == "Within") {
#&gt;           return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;         }
#&gt;         else {
#&gt;           return "--chimOutType SeparateSAMold"
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n    if
#&gt;       (value == \"\") {\n      return\n    }\n    else {\n      return param_list.push(key.concat(\":\",
#&gt;       value))\n    }\n  }\n  \n  add_param('ID', \"1\")\n  if ($job.inputs.rg_seq_center)
#&gt;       {\n    add_param('CN', $job.inputs.rg_seq_center)\n  } else if ([].concat($job.inputs.reads)[0].metadata.seq_center)
#&gt;       {\n    add_param('CN', [].concat($job.inputs.reads)[0].metadata.seq_center)\n
#&gt;       \ }\n  if ($job.inputs.rg_library_id) {\n    add_param('LB', $job.inputs.rg_library_id)\n
#&gt;       \ } else if ([].concat($job.inputs.reads)[0].metadata.library_id) {\n    add_param('LB',
#&gt;       [].concat($job.inputs.reads)[0].metadata.library_id)\n  }\n  if ($job.inputs.rg_mfl)
#&gt;       {\n    add_param('PI', $job.inputs.rg_mfl)\n  } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;       {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;       \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;       /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;       {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;       /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;       $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;       {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;       \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;       \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;       [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;       \".concat(param_list.join(\" \"))\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;           return "--quantMode ".concat($job.inputs.quantMode)
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 100
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n
#&gt;       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===
#&gt;       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr
#&gt;       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(
#&gt;       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  intermediate =
#&gt;       common_prefix.replace( /\\-$|\\_$|\\.$/, '' ).concat(\"._STARgenome\")\n  source
#&gt;       = \"./\".concat(intermediate)\n  destination = intermediate.concat(\".tar\")\n
#&gt;       \ if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;       != \"None\") {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n
#&gt;       \ }\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   prefix: --outFileNamePrefix
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n
#&gt;       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===
#&gt;       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr
#&gt;       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(
#&gt;       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;       /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;     class: Expression
#&gt; - position: 101
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n
#&gt;       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===
#&gt;       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr
#&gt;       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(
#&gt;       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  mate1 = common_prefix.replace(
#&gt;       /\\-$|\\_$|\\.$/, '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace(
#&gt;       /\\-$|\\_$|\\.$/, '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;       \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped ==
#&gt;       \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \" \", mate1fq,
#&gt;       \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;       == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \" \",
#&gt;       mate1fq)\n  }\n}"
#&gt;     class: Expression
#&gt; stdin: ''
#&gt; stdout: ''
#&gt; successCodes: []
#&gt; temporaryFailCodes: []
#&gt; x: 624.0
#&gt; 'y': 323
#&gt; 
#&gt; </div><div class='input'><span class='co'># return one</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_tool</span><span class='op'>(</span><span class='st'>"^STAR$"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 4
#&gt; sbg:job:
#&gt;   allocatedResources:
#&gt;     mem: 60000
#&gt;     cpu: 15
#&gt;   inputs:
#&gt;     alignWindowsPerReadNmax: 0
#&gt;     outSAMheaderPG: outSAMheaderPG
#&gt;     GENOME_DIR_NAME: ''
#&gt;     outFilterMatchNminOverLread: 0
#&gt;     rg_platform_unit_id: rg_platform_unit
#&gt;     alignTranscriptsPerReadNmax: 0
#&gt;     readMapNumber: 0
#&gt;     alignSplicedMateMapLminOverLmate: 0
#&gt;     alignMatesGapMax: 0
#&gt;     outFilterMultimapNmax: 0
#&gt;     clip5pNbases:
#&gt;     - 0
#&gt;     outSAMstrandField: None
#&gt;     readMatesLengthsIn: NotEqual
#&gt;     outSAMattributes: Standard
#&gt;     seedMultimapNmax: 0
#&gt;     rg_mfl: rg_mfl
#&gt;     chimSegmentMin: 0
#&gt;     winAnchorDistNbins: 0
#&gt;     outSortingType: SortedByCoordinate
#&gt;     outFilterMultimapScoreRange: 0
#&gt;     sjdbInsertSave: Basic
#&gt;     clip3pAfterAdapterNbases:
#&gt;     - 0
#&gt;     scoreDelBase: 0
#&gt;     outFilterMatchNmin: 0
#&gt;     twopass1readsN: 0
#&gt;     outSAMunmapped: None
#&gt;     genome:
#&gt;       size: 0
#&gt;       secondaryFiles: []
#&gt;       class: File
#&gt;       path: genome.ext
#&gt;     sjdbGTFtagExonParentTranscript: ''
#&gt;     limitBAMsortRAM: 0
#&gt;     alignEndsType: Local
#&gt;     seedNoneLociPerWindow: 0
#&gt;     rg_sample_id: rg_sample
#&gt;     sjdbGTFtagExonParentGene: ''
#&gt;     chimScoreMin: 0
#&gt;     outSJfilterIntronMaxVsReadN:
#&gt;     - 0
#&gt;     twopassMode: Basic
#&gt;     alignSplicedMateMapLmin: 0
#&gt;     outSJfilterReads: All
#&gt;     outSAMprimaryFlag: OneBestScore
#&gt;     outSJfilterCountTotalMin:
#&gt;     - 3
#&gt;     - 1
#&gt;     - 1
#&gt;     - 1
#&gt;     outSAMorder: Paired
#&gt;     outSAMflagAND: 0
#&gt;     chimScoreSeparation: 0
#&gt;     alignSJoverhangMin: 0
#&gt;     outFilterScoreMin: 0
#&gt;     seedSearchStartLmax: 0
#&gt;     scoreGapGCAG: 0
#&gt;     scoreGenomicLengthLog2scale: 0
#&gt;     outFilterIntronMotifs: None
#&gt;     outFilterMismatchNmax: 0
#&gt;     reads:
#&gt;     - size: 0
#&gt;       secondaryFiles: []
#&gt;       class: File
#&gt;       metadata:
#&gt;         format: fastq
#&gt;         paired_end: '1'
#&gt;         seq_center: illumina
#&gt;       path: /test-data/mate_1.fastq.bz2
#&gt;     scoreGap: 0
#&gt;     outSJfilterOverhangMin:
#&gt;     - 30
#&gt;     - 12
#&gt;     - 12
#&gt;     - 12
#&gt;     outSAMflagOR: 0
#&gt;     outSAMmode: Full
#&gt;     rg_library_id: ''
#&gt;     chimScoreJunctionNonGTAG: 0
#&gt;     scoreInsOpen: 0
#&gt;     clip3pAdapterSeq:
#&gt;     - clip3pAdapterSeq
#&gt;     chimScoreDropMax: 0
#&gt;     outFilterType: Normal
#&gt;     scoreGapATAC: 0
#&gt;     rg_platform: Ion Torrent PGM
#&gt;     clip3pAdapterMMp:
#&gt;     - 0
#&gt;     sjdbGTFfeatureExon: ''
#&gt;     outQSconversionAdd: 0
#&gt;     quantMode: TranscriptomeSAM
#&gt;     alignIntronMin: 0
#&gt;     scoreInsBase: 0
#&gt;     scoreGapNoncan: 0
#&gt;     seedSearchLmax: 0
#&gt;     outSJfilterDistToOtherSJmin:
#&gt;     - 0
#&gt;     outFilterScoreMinOverLread: 0
#&gt;     alignSJDBoverhangMin: 0
#&gt;     limitOutSJcollapsed: 0
#&gt;     winAnchorMultimapNmax: 0
#&gt;     outFilterMismatchNoverLmax: 0
#&gt;     rg_seq_center: ''
#&gt;     outSAMheaderHD: outSAMheaderHD
#&gt;     chimOutType: Within
#&gt;     quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;     limitOutSJoneRead: 0
#&gt;     alignTranscriptsPerWindowNmax: 0
#&gt;     sjdbOverhang: ~
#&gt;     outReadsUnmapped: Fastx
#&gt;     scoreStitchSJshift: 0
#&gt;     seedPerWindowNmax: 0
#&gt;     outSJfilterCountUniqueMin:
#&gt;     - 3
#&gt;     - 1
#&gt;     - 1
#&gt;     - 1
#&gt;     scoreDelOpen: 0
#&gt;     sjdbGTFfile:
#&gt;     - path: /demo/test-data/chr20.gtf
#&gt;     clip3pNbases:
#&gt;     - 0
#&gt;     - 3
#&gt;     winBinNbits: 0
#&gt;     sjdbScore: ~
#&gt;     seedSearchStartLmaxOverLread: 0
#&gt;     alignIntronMax: 0
#&gt;     seedPerReadNmax: 0
#&gt;     outFilterMismatchNoverReadLmax: 0
#&gt;     winFlankNbins: 0
#&gt;     sjdbGTFchrPrefix: chrPrefix
#&gt;     alignSoftClipAtReferenceEnds: 'Yes'
#&gt;     outSAMreadID: Standard
#&gt;     outSAMtype: BAM
#&gt;     chimJunctionOverhangMin: 0
#&gt;     limitSjdbInsertNsj: 0
#&gt;     outSAMmapqUnique: 0
#&gt; sbg:toolAuthor: Alexander Dobin/CSHL
#&gt; sbg:createdOn: 1450911471
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; sbg:contributors:
#&gt; - ana_d
#&gt; - bix-demo
#&gt; - uros_sipetic
#&gt; sbg:links:
#&gt; - id: https://github.com/alexdobin/STAR
#&gt;   label: Homepage
#&gt; - id: https://github.com/alexdobin/STAR/releases
#&gt;   label: Releases
#&gt; - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;   label: Manual
#&gt; - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;   label: Support
#&gt; - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;   label: Publication
#&gt; sbg:project: bix-demo/star-2-4-2a-demo
#&gt; sbg:createdBy: bix-demo
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: sevenbridges/public-apps/star/4
#&gt; sbg:license: GNU General Public License v3.0 only
#&gt; sbg:revision: 4
#&gt; sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;   bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix --sjdbInsertSave
#&gt;   Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline ID:1 CN:illumina
#&gt;   PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode TranscriptomeSAM
#&gt;   --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;   tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp; mv
#&gt;   mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt; sbg:modifiedOn: 1462889222
#&gt; sbg:modifiedBy: ana_d
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911471
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911473
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: bix-demo
#&gt;   sbg:modifiedOn: 1450911475
#&gt;   sbg:revision: 2
#&gt; - sbg:modifiedBy: uros_sipetic
#&gt;   sbg:modifiedOn: 1462878528
#&gt;   sbg:revision: 3
#&gt; - sbg:modifiedBy: ana_d
#&gt;   sbg:modifiedOn: 1462889222
#&gt;   sbg:revision: 4
#&gt; sbg:toolkit: STAR
#&gt; id: sevenbridges/public-apps/star/4
#&gt; inputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Flanking regions size
#&gt;   description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;     flanking regions for each window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winFlankNbins'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winFlankNbins
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '4'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Bin size
#&gt;   description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;     each window will occupy an integer number of bins (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winBinNbits'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winBinNbits
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '16'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winAnchorMultimapNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --winAnchorDistNbins
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: twopassMode
#&gt;     symbols:
#&gt;     - None
#&gt;     - Basic
#&gt;     type: enum
#&gt;   label: Two-pass mode
#&gt;   description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass mapping,
#&gt;     with all 1st pass junctions inserted into the genome indices on the fly.'
#&gt;   streamable: no
#&gt;   id: '#twopassMode'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --twopassMode
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 2-pass mapping
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Reads to process in 1st step
#&gt;   description: 'Number of reads to process for the 1st step. 0: 1-step only, no 2nd
#&gt;     pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;   streamable: no
#&gt;   id: '#twopass1readsN'
#&gt;   sbg:category: 2-pass mapping
#&gt;   sbg:toolDefaultValue: '-1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Extra alignment score
#&gt;   description: Extra alignment score for alignments that cross database junctions.
#&gt;   streamable: no
#&gt;   id: '#sjdbScore'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: '2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: '"Overhang" length'
#&gt;   description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;     ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database is
#&gt;     not used.
#&gt;   streamable: no
#&gt;   id: '#sjdbOverhang'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: '100'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: sjdbInsertSave
#&gt;     symbols:
#&gt;     - Basic
#&gt;     - All
#&gt;     - None
#&gt;     type: enum
#&gt;   label: Save junction files
#&gt;   description: 'Which files to save when sjdb junctions are inserted on the fly at
#&gt;     the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;     files; All: all files including big Genome, SA and SAindex. These files are output
#&gt;     as archive.'
#&gt;   streamable: no
#&gt;   id: '#sjdbInsertSave'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Splice junction file
#&gt;   description: Gene model annotations and/or known transcripts. No need to include
#&gt;     this input, except in case of using "on the fly" annotations.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: GTF, GFF, TXT
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Set exons feature
#&gt;   description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfeatureExon'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: exon
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Chromosome names
#&gt;   description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using ENSMEBL
#&gt;     annotations with UCSC geneomes).
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFchrPrefix'
#&gt;   sbg:category: Splice junctions database
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Search start point normalized
#&gt;   description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;     for paired-end reads).
#&gt;   streamable: no
#&gt;   id: '#seedSearchStartLmaxOverLread'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedSearchStartLmaxOverLread
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '1.0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Search start point
#&gt;   description: Defines the search start point through the read - the read is split
#&gt;     into pieces no longer than this value (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#seedSearchStartLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedSearchStartLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max seed length
#&gt;   description: Defines the maximum length of the seeds, if =0 max seed length is infinite
#&gt;     (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedSearchLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedSearchLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max seeds per window
#&gt;   description: Max number of seeds per window (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedPerWindowNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedPerWindowNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max seeds per read
#&gt;   description: Max number of seeds per read (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedPerReadNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedPerReadNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '1000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max one-seed loci per window
#&gt;   description: Max number of one seed loci per window (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedNoneLociPerWindow'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedNoneLociPerWindow
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Filter pieces for stitching
#&gt;   description: Only pieces that map fewer than this value are utilized in the stitching
#&gt;     procedure (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#seedMultimapNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --seedMultimapNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max score reduction
#&gt;   description: Maximum score reduction while searching for SJ boundaries in the stitching
#&gt;     step.
#&gt;   streamable: no
#&gt;   id: '#scoreStitchSJshift'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreStitchSJshift
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Insertion Open Penalty
#&gt;   description: Insertion open penalty.
#&gt;   streamable: no
#&gt;   id: '#scoreInsOpen'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreInsOpen
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Insertion extension penalty
#&gt;   description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;   streamable: no
#&gt;   id: '#scoreInsBase'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreInsBase
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Log scaled score
#&gt;   description: 'Extra score logarithmically scaled with genomic length of the alignment:
#&gt;     &lt;int&gt;*log2(genomicLength).'
#&gt;   streamable: no
#&gt;   id: '#scoreGenomicLengthLog2scale'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGenomicLengthLog2scale
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-0.25'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Non-canonical gap open
#&gt;   description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;   streamable: no
#&gt;   id: '#scoreGapNoncan'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGapNoncan
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-8'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: GC/AG and CT/GC gap open
#&gt;   description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;   streamable: no
#&gt;   id: '#scoreGapGCAG'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGapGCAG
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-4'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: AT/AC and GT/AT gap open
#&gt;   description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;   streamable: no
#&gt;   id: '#scoreGapATAC'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGapATAC
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-8'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Gap open penalty
#&gt;   description: Gap open penalty.
#&gt;   streamable: no
#&gt;   id: '#scoreGap'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreGap
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Deletion open penalty
#&gt;   description: Deletion open penalty.
#&gt;   streamable: no
#&gt;   id: '#scoreDelOpen'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreDelOpen
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Deletion extension penalty
#&gt;   description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;   streamable: no
#&gt;   id: '#scoreDelBase'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --scoreDelBase
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Scoring
#&gt;   sbg:toolDefaultValue: '-2'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Sequencing center
#&gt;   description: Specify the sequencing center for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_seq_center'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Sample ID
#&gt;   description: Specify the sample ID for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_sample_id'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Platform unit ID
#&gt;   description: Specify the platform unit ID for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_platform_unit_id'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: rg_platform
#&gt;     symbols:
#&gt;     - LS 454
#&gt;     - Helicos
#&gt;     - Illumina
#&gt;     - ABI SOLiD
#&gt;     - Ion Torrent PGM
#&gt;     - PacBio
#&gt;     type: enum
#&gt;   label: Platform
#&gt;   description: Specify the version of the technology that was used for sequencing
#&gt;     or assaying.
#&gt;   streamable: no
#&gt;   id: '#rg_platform'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Median fragment length
#&gt;   description: Specify the median fragment length for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_mfl'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Library ID
#&gt;   description: Specify the library ID for RG line.
#&gt;   streamable: no
#&gt;   id: '#rg_library_id'
#&gt;   sbg:category: Read group
#&gt;   sbg:toolDefaultValue: Inferred from metadata
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Read sequence
#&gt;   description: Read sequence.
#&gt;   streamable: no
#&gt;   id: '#reads'
#&gt;   inputBinding:
#&gt;     position: 10
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n
#&gt;         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if
#&gt;         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n
#&gt;         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata
#&gt;         != null){\n        if (list[index].metadata.paired_end == 1){\n          left
#&gt;         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n
#&gt;         \         right = list[index].path\n        }\n      }\n    }\n    \n    if
#&gt;         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n
#&gt;         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n
#&gt;         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if
#&gt;         (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end
#&gt;         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n
#&gt;         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=
#&gt;         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n
#&gt;         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"
#&gt;         \"))\n  }\n}"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: readMatesLengthsIn
#&gt;     symbols:
#&gt;     - NotEqual
#&gt;     - Equal
#&gt;     type: enum
#&gt;   label: Reads lengths
#&gt;   description: Equal/Not equal - lengths of names, sequences, qualities for both mates
#&gt;     are the same/not the same. "Not equal" is safe in all situations.
#&gt;   streamable: no
#&gt;   id: '#readMatesLengthsIn'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --readMatesLengthsIn
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: NotEqual
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Reads to map
#&gt;   description: Number of reads to map from the beginning of the file.
#&gt;   streamable: no
#&gt;   id: '#readMapNumber'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --readMapNumber
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '-1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: quantTranscriptomeBan
#&gt;     symbols:
#&gt;     - IndelSoftclipSingleend
#&gt;     - Singleend
#&gt;     type: enum
#&gt;   label: Prohibit alignment type
#&gt;   description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;     indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;     prohibit single-end alignments.'
#&gt;   streamable: no
#&gt;   id: '#quantTranscriptomeBan'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --quantTranscriptomeBan
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Quantification of Annotations
#&gt;   sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: quantMode
#&gt;     symbols:
#&gt;     - TranscriptomeSAM
#&gt;     - GeneCounts
#&gt;     type: enum
#&gt;   label: Quantification mode
#&gt;   description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;     SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts' option,
#&gt;     STAR will count number of reads per gene while mapping.
#&gt;   streamable: no
#&gt;   id: '#quantMode'
#&gt;   sbg:category: Quantification of Annotations
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSortingType
#&gt;     symbols:
#&gt;     - Unsorted
#&gt;     - SortedByCoordinate
#&gt;     - Unsorted SortedByCoordinate
#&gt;     type: enum
#&gt;   label: Output sorting type
#&gt;   description: Type of output sorting.
#&gt;   streamable: no
#&gt;   id: '#outSortingType'
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: SortedByCoordinate
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSJfilterReads
#&gt;     symbols:
#&gt;     - All
#&gt;     - Unique
#&gt;     type: enum
#&gt;   label: Collapsed junctions reads
#&gt;   description: 'Which reads to consider for collapsed splice junctions output. All:
#&gt;     all reads, unique- and multi-mappers; Unique: uniquely mapping reads only.'
#&gt;   streamable: no
#&gt;   id: '#outSJfilterReads'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterReads
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: All
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min overhang SJ
#&gt;   description: Minimum overhang length for splice junctions on both sides for each
#&gt;     of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;     field. Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterOverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterOverhangMin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 30 12 12 12
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Max gap allowed
#&gt;   description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads (int
#&gt;     &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b, by
#&gt;     2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;     Does not apply to annotated junctions.'
#&gt;   streamable: no
#&gt;   id: '#outSJfilterIntronMaxVsReadN'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterIntronMaxVsReadN
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 50000 100000 200000
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min distance to other donor/acceptor
#&gt;   description: Minimum allowed distance to other junctions' donor/acceptor for each
#&gt;     of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterDistToOtherSJmin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterDistToOtherSJmin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 10 0 5 10
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min unique count
#&gt;   description: Minimum uniquely mapping read count per junction for each of the motifs.
#&gt;     To set no output for desired motif, assign -1 to the corresponding field. Junctions
#&gt;     are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;     conditions are satisfied. Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterCountUniqueMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterCountUniqueMin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 3 1 1 1
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Min total count
#&gt;   description: Minimum total (multi-mapping+unique) read count per junction for each
#&gt;     of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;     field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;     conditions are satisfied. Does not apply to annotated junctions.
#&gt;   streamable: no
#&gt;   id: '#outSJfilterCountTotalMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSJfilterCountTotalMin
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: 'Output filtering: splice junctions'
#&gt;   sbg:toolDefaultValue: 3 1 1 1
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMunmapped
#&gt;     symbols:
#&gt;     - None
#&gt;     - Within
#&gt;     type: enum
#&gt;   label: Write unmapped in SAM
#&gt;   description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;     output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;   streamable: no
#&gt;   id: '#outSAMunmapped'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMunmapped
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMtype
#&gt;     symbols:
#&gt;     - SAM
#&gt;     - BAM
#&gt;     type: enum
#&gt;   label: Output format
#&gt;   description: Format of output alignments.
#&gt;   streamable: no
#&gt;   id: '#outSAMtype'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     separate: yes
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: |-
#&gt;         {
#&gt;           SAM_type = $job.inputs.outSAMtype
#&gt;           SORT_type = $job.inputs.outSortingType
#&gt;           if (SAM_type &amp;&amp; SORT_type) {
#&gt;             return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;           }
#&gt;         }
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: SAM
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMstrandField
#&gt;     symbols:
#&gt;     - None
#&gt;     - intronMotif
#&gt;     type: enum
#&gt;   label: Strand field flag
#&gt;   description: 'Cufflinks-like strand field flag. None: not used; intronMotif: strand
#&gt;     derived from the intron motif. Reads with inconsistent and/or non-canonical introns
#&gt;     are filtered out.'
#&gt;   streamable: no
#&gt;   id: '#outSAMstrandField'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMstrandField
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMreadID
#&gt;     symbols:
#&gt;     - Standard
#&gt;     - Number
#&gt;     type: enum
#&gt;   label: Read ID
#&gt;   description: 'Read ID record type. Standard: first word (until space) from the FASTx
#&gt;     read ID line, removing /1,/2 from the end; Number: read number (index) in the
#&gt;     FASTx file.'
#&gt;   streamable: no
#&gt;   id: '#outSAMreadID'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMreadID
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Standard
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMprimaryFlag
#&gt;     symbols:
#&gt;     - OneBestScore
#&gt;     - AllBestScore
#&gt;     type: enum
#&gt;   label: Primary alignments
#&gt;   description: 'Which alignments are considered primary - all others will be marked
#&gt;     with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best score
#&gt;     is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;   streamable: no
#&gt;   id: '#outSAMprimaryFlag'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMprimaryFlag
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: OneBestScore
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMorder
#&gt;     symbols:
#&gt;     - Paired
#&gt;     - PairedKeepInputOrder
#&gt;     type: enum
#&gt;   label: Sorting in SAM
#&gt;   description: 'Type of sorting for the SAM output. Paired: one mate after the other
#&gt;     for all paired alignments; PairedKeepInputOrder: one mate after the other for
#&gt;     all paired alignments, the order is kept the same as in the input FASTQ files.'
#&gt;   streamable: no
#&gt;   id: '#outSAMorder'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMorder
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Paired
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMmode
#&gt;     symbols:
#&gt;     - Full
#&gt;     - NoQS
#&gt;     type: enum
#&gt;   label: SAM mode
#&gt;   description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but without
#&gt;     quality scores.'
#&gt;   streamable: no
#&gt;   id: '#outSAMmode'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMmode
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Full
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: MAPQ value
#&gt;   description: MAPQ value for unique mappers (0 to 255).
#&gt;   streamable: no
#&gt;   id: '#outSAMmapqUnique'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMmapqUnique
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '255'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: SAM header @PG
#&gt;   description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;   streamable: no
#&gt;   id: '#outSAMheaderPG'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMheaderPG
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: SAM header @HD
#&gt;   description: '@HD (header) line of the SAM header.'
#&gt;   streamable: no
#&gt;   id: '#outSAMheaderHD'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMheaderHD
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: OR SAM flag
#&gt;   description: Set specific bits of the SAM FLAG.
#&gt;   streamable: no
#&gt;   id: '#outSAMflagOR'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMflagOR
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: AND SAM flag
#&gt;   description: Set specific bits of the SAM FLAG.
#&gt;   streamable: no
#&gt;   id: '#outSAMflagAND'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMflagAND
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '65535'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outSAMattributes
#&gt;     symbols:
#&gt;     - Standard
#&gt;     - NH
#&gt;     - All
#&gt;     - None
#&gt;     type: enum
#&gt;   label: SAM attributes
#&gt;   description: 'Desired SAM attributes, in the order desired for the output SAM. NH:
#&gt;     any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM NM MD jM
#&gt;     jI; None: no attributes.'
#&gt;   streamable: no
#&gt;   id: '#outSAMattributes'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outSAMattributes
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: Standard
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outReadsUnmapped
#&gt;     symbols:
#&gt;     - None
#&gt;     - Fastx
#&gt;     type: enum
#&gt;   label: Output unmapped reads
#&gt;   description: 'Output of unmapped reads (besides SAM). None: no output; Fastx: output
#&gt;     in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;   streamable: no
#&gt;   id: '#outReadsUnmapped'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outReadsUnmapped
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Quality conversion
#&gt;   description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;     to Sanger, use -31).
#&gt;   streamable: no
#&gt;   id: '#outQSconversionAdd'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outQSconversionAdd
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outFilterType
#&gt;     symbols:
#&gt;     - Normal
#&gt;     - BySJout
#&gt;     type: enum
#&gt;   label: Filtering type
#&gt;   description: 'Type of filtering. Normal: standard filtering using only current alignment;
#&gt;     BySJout: keep only those reads that contain junctions that passed filtering into
#&gt;     SJ.out.tab.'
#&gt;   streamable: no
#&gt;   id: '#outFilterType'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterType
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: Normal
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Min score normalized
#&gt;   description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;     for paired-end reads).'
#&gt;   streamable: no
#&gt;   id: '#outFilterScoreMinOverLread'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterScoreMinOverLread
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0.66'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min score
#&gt;   description: Alignment will be output only if its score is higher than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterScoreMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterScoreMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Multimapping score range
#&gt;   description: The score range below the maximum score for multimapping alignments.
#&gt;   streamable: no
#&gt;   id: '#outFilterMultimapScoreRange'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMultimapScoreRange
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max number of mappings
#&gt;   description: Read alignments will be output only if the read maps fewer than this
#&gt;     value, otherwise no alignments will be output.
#&gt;   streamable: no
#&gt;   id: '#outFilterMultimapNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMultimapNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Mismatches to *read* length
#&gt;   description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;     length is less than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMismatchNoverReadLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMismatchNoverReadLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Mismatches to *mapped* length
#&gt;   description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;     length is less than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMismatchNoverLmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMismatchNoverLmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0.3'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max number of mismatches
#&gt;   description: Alignment will be output only if it has fewer mismatches than this
#&gt;     value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMismatchNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMismatchNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Min matched bases normalized
#&gt;   description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;     lengths for paired-end reads).'
#&gt;   streamable: no
#&gt;   id: '#outFilterMatchNminOverLread'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMatchNminOverLread
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0.66'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min matched bases
#&gt;   description: Alignment will be output only if the number of matched bases is higher
#&gt;     than this value.
#&gt;   streamable: no
#&gt;   id: '#outFilterMatchNmin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterMatchNmin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: outFilterIntronMotifs
#&gt;     symbols:
#&gt;     - None
#&gt;     - RemoveNoncanonical
#&gt;     - RemoveNoncanonicalUnannotated
#&gt;     type: enum
#&gt;   label: Motifs filtering
#&gt;   description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;     filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;     filter out alignments that contain non-canonical unannotated junctions when using
#&gt;     annotated splice junctions database. The annotated non-canonical junctions will
#&gt;     be kept.'
#&gt;   streamable: no
#&gt;   id: '#outFilterIntronMotifs'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --outFilterIntronMotifs
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Output filtering
#&gt;   sbg:toolDefaultValue: None
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max insert junctions
#&gt;   description: Maximum number of junction to be inserted to the genome on the fly
#&gt;     at the mapping stage, including those from annotations and those detected in the
#&gt;     1st step of the 2-pass run.
#&gt;   streamable: no
#&gt;   id: '#limitSjdbInsertNsj'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitSjdbInsertNsj
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '1000000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Junctions max number
#&gt;   description: Max number of junctions for one read (including all multi-mappers).
#&gt;   streamable: no
#&gt;   id: '#limitOutSJoneRead'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitOutSJoneRead
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '1000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Collapsed junctions max number
#&gt;   description: Max number of collapsed junctions.
#&gt;   streamable: no
#&gt;   id: '#limitOutSJcollapsed'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitOutSJcollapsed
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '1000000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Limit BAM sorting memory
#&gt;   description: Maximum available RAM for sorting BAM. If set to 0, it will be set
#&gt;     to the genome index size.
#&gt;   streamable: no
#&gt;   id: '#limitBAMsortRAM'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --limitBAMsortRAM
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Limits
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Genome dir name
#&gt;   description: Name of the directory which contains genome files (when genome.tar
#&gt;     is uncompressed).
#&gt;   streamable: no
#&gt;   id: '#genomeDirName'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --genomeDir
#&gt;     separate: yes
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: $job.inputs.genomeDirName || "genomeDir"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:toolDefaultValue: genomeDir
#&gt;   required: no
#&gt; - type:
#&gt;   - File
#&gt;   label: Genome files
#&gt;   description: Genome files created using STAR Genome Generate.
#&gt;   streamable: no
#&gt;   id: '#genome'
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: TAR
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Clip 5p bases
#&gt;   description: Number of bases to clip from 5p of each mate. In case only one value
#&gt;     is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip5pNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip5pNbases
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Clip 3p bases
#&gt;   description: Number of bases to clip from 3p of each mate. In case only one value
#&gt;     is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pNbases
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: int
#&gt;     type: array
#&gt;   label: Clip 3p after adapter seq.
#&gt;   description: Number of bases to clip from 3p of each mate after the adapter clipping.
#&gt;     In case only one value is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pAfterAdapterNbases'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pAfterAdapterNbases
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: string
#&gt;     type: array
#&gt;   label: Clip 3p adapter sequence
#&gt;   description: Adapter sequence to clip from 3p of each mate. In case only one value
#&gt;     is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pAdapterSeq'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pAdapterSeq
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '-'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: float
#&gt;     type: array
#&gt;   label: Max mismatches proportions
#&gt;   description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;     In case only one value is given, it will be assumed the same for both mates.
#&gt;   streamable: no
#&gt;   id: '#clip3pAdapterMMp'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --clip3pAdapterMMp
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Read parameters
#&gt;   sbg:toolDefaultValue: '0.1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min segment length
#&gt;   description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;     (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimSegmentMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimSegmentMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '15'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min separation score
#&gt;   description: Minimum difference (separation) between the best chimeric score and
#&gt;     the next one (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimScoreSeparation'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreSeparation
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '10'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min total score
#&gt;   description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimScoreMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Non-GT/AG penalty
#&gt;   description: Penalty for a non-GT/AG chimeric junction.
#&gt;   streamable: no
#&gt;   id: '#chimScoreJunctionNonGTAG'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreJunctionNonGTAG
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '-1'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max drop score
#&gt;   description: Max drop (difference) of chimeric score (the sum of scores of all chimeric
#&gt;     segements) from the read length (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimScoreDropMax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimScoreDropMax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '20'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: chimOutType
#&gt;     symbols:
#&gt;     - SeparateSAMold
#&gt;     - Within
#&gt;     type: enum
#&gt;   label: Chimeric output type
#&gt;   description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;     Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;   streamable: no
#&gt;   id: '#chimOutType'
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: SeparateSAMold
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min junction overhang
#&gt;   description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;   streamable: no
#&gt;   id: '#chimJunctionOverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --chimJunctionOverhangMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Chimeric Alignments
#&gt;   sbg:toolDefaultValue: '20'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Max windows per read
#&gt;   description: Max number of windows per read (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignWindowsPerReadNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignWindowsPerReadNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max transcripts per window
#&gt;   description: Max number of transcripts per window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignTranscriptsPerWindowNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignTranscriptsPerWindowNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '100'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max transcripts per read
#&gt;   description: Max number of different alignments per read to consider (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignTranscriptsPerReadNmax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignTranscriptsPerReadNmax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '10000'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - float
#&gt;   label: Min mapped length normalized
#&gt;   description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSplicedMateMapLminOverLmate'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSplicedMateMapLminOverLmate
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0.66'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min mapped length
#&gt;   description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSplicedMateMapLmin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSplicedMateMapLmin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: alignSoftClipAtReferenceEnds
#&gt;     symbols:
#&gt;     - 'Yes'
#&gt;     - 'No'
#&gt;     type: enum
#&gt;   label: Soft clipping
#&gt;   description: 'Option which allows soft clipping of alignments at the reference (chromosome)
#&gt;     ends. Can be disabled for compatibility with Cufflinks/Cuffmerge. Yes: Enables
#&gt;     soft clipping; No: Disables soft clipping.'
#&gt;   streamable: no
#&gt;   id: '#alignSoftClipAtReferenceEnds'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSoftClipAtReferenceEnds
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: 'Yes'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min overhang
#&gt;   description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSJoverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSJoverhangMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '5'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: 'Min overhang: annotated'
#&gt;   description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced alignments
#&gt;     (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#alignSJDBoverhangMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignSJDBoverhangMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '3'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max mates gap
#&gt;   description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;     by (2^winBinNbits)*winAnchorDistNbins.
#&gt;   streamable: no
#&gt;   id: '#alignMatesGapMax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignMatesGapMax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Min intron size
#&gt;   description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;     &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;   streamable: no
#&gt;   id: '#alignIntronMin'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignIntronMin
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '21'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max intron size
#&gt;   description: Maximum intron size, if 0, max intron size will be determined by (2^winBinNbits)*winAnchorDistNbins.
#&gt;   streamable: no
#&gt;   id: '#alignIntronMax'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignIntronMax
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: '0'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - name: alignEndsType
#&gt;     symbols:
#&gt;     - Local
#&gt;     - EndToEnd
#&gt;     type: enum
#&gt;   label: Alignment type
#&gt;   description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;     soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;   streamable: no
#&gt;   id: '#alignEndsType'
#&gt;   inputBinding:
#&gt;     position: 0
#&gt;     prefix: --alignEndsType
#&gt;     separate: yes
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Alignments and Seeding
#&gt;   sbg:toolDefaultValue: Local
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   outputBinding:
#&gt;     glob: '*Unmapped.out*'
#&gt;   sbg:fileTypes: FASTQ
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   outputBinding:
#&gt;     glob: '*Transcriptome*'
#&gt;   sbg:fileTypes: BAM
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   outputBinding:
#&gt;     glob: '*SJ.out.tab'
#&gt;   sbg:fileTypes: TAB
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   outputBinding:
#&gt;     glob: '*ReadsPerGene*'
#&gt;   sbg:fileTypes: TAB
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   outputBinding:
#&gt;     glob: '*Log*.out'
#&gt;   sbg:fileTypes: OUT
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   outputBinding:
#&gt;     glob: '*_STARgenome.tar'
#&gt;   sbg:fileTypes: TAR
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   outputBinding:
#&gt;     glob: '*Chimeric.out.junction'
#&gt;   sbg:fileTypes: JUNCTION
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   outputBinding:
#&gt;     glob: '*.Chimeric.out.sam'
#&gt;   sbg:fileTypes: SAM
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Aligned SAM/BAM
#&gt;   description: Aligned sequence in SAM/BAM format.
#&gt;   streamable: no
#&gt;   id: '#aligned_reads'
#&gt;   outputBinding:
#&gt;     glob:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: |-
#&gt;         {
#&gt;           if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;             sort_name = '.sortedByCoord'
#&gt;           }
#&gt;           else {
#&gt;             sort_name = ''
#&gt;           }
#&gt;           if ($job.inputs.outSAMtype == 'BAM') {
#&gt;             sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;           }
#&gt;           else {
#&gt;             sam_name = "*.Aligned.out.sam"
#&gt;           }
#&gt;           return sam_name
#&gt;         }
#&gt;       class: Expression
#&gt;   sbg:fileTypes: SAM, BAM
#&gt; requirements:
#&gt; - class: ExpressionEngineRequirement
#&gt;   id: '#cwl-js-engine'
#&gt;   requirements:
#&gt;   - class: DockerRequirement
#&gt;     dockerPull: rabix/js-engine
#&gt; hints:
#&gt; - class: DockerRequirement
#&gt;   dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;   dockerImageId: a4b0ad2c3cae
#&gt; - class: sbg:MemRequirement
#&gt;   value: 60000
#&gt; - class: sbg:CPURequirement
#&gt;   value: 15
#&gt; label: STAR
#&gt; description: STAR is an ultrafast universal RNA-seq aligner. It has very high mapping
#&gt;   speed, accurate alignment of contiguous and spliced reads, detection of polyA-tails,
#&gt;   non-canonical splices and chimeric (fusion) junctions. It works with reads starting
#&gt;   from lengths ~15 bases up to ~300 bases. In case of having longer reads, use of
#&gt;   STAR Long is recommended.
#&gt; class: CommandLineTool
#&gt; baseCommand:
#&gt; - tar
#&gt; - -xvf
#&gt; - class: Expression
#&gt;   script: $job.inputs.genome.path
#&gt;   engine: '#cwl-js-engine'
#&gt; - '&amp;&amp;'
#&gt; - /opt/STAR
#&gt; - --runThreadN
#&gt; - class: Expression
#&gt;   script: |-
#&gt;     {
#&gt;       return $job.allocatedResources.cpu
#&gt;     }
#&gt;   engine: '#cwl-js-engine'
#&gt; arguments:
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         file = [].concat($job.inputs.reads)[0].path
#&gt;         extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;         if (extension == "gz") {
#&gt;           return "--readFilesCommand zcat"
#&gt;         } else if (extension == "bz2") {
#&gt;           return "--readFilesCommand bzcat"
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n  var
#&gt;       list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths =
#&gt;       \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;       \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;       \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext == \"gtf\")
#&gt;       {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n      }\n      if
#&gt;       (ext == \"txt\") {\n        sjFormat = \"True\"\n        return sjFormat\n      }\n
#&gt;       \   })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;       {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;       \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\") {\n
#&gt;       \       return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n  }\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;       {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n  if
#&gt;       ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\", $job.inputs.sjdbGTFfeatureExon]\n
#&gt;       \ }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript) {\n    c = [\"--sjdbGTFtagExonParentTranscript\",
#&gt;       $job.inputs.sjdbGTFtagExonParentTranscript]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentGene)
#&gt;       {\n    d = [\"--sjdbGTFtagExonParentGene\", $job.inputs.sjdbGTFtagExonParentGene]\n
#&gt;       \ }\n  if ($job.inputs.sjdbOverhang) {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n
#&gt;       \ }\n  if ($job.inputs.sjdbScore) {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n
#&gt;       \ }\n  if ($job.inputs.sjdbInsertSave) {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n
#&gt;       \ }\n  \n  \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;       {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\" \")\n
#&gt;       \ }\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         if ($job.inputs.twopassMode == "Basic") {
#&gt;           return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         if ($job.inputs.chimOutType == "Within") {
#&gt;           return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;         }
#&gt;         else {
#&gt;           return "--chimOutType SeparateSAMold"
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n    if
#&gt;       (value == \"\") {\n      return\n    }\n    else {\n      return param_list.push(key.concat(\":\",
#&gt;       value))\n    }\n  }\n  \n  add_param('ID', \"1\")\n  if ($job.inputs.rg_seq_center)
#&gt;       {\n    add_param('CN', $job.inputs.rg_seq_center)\n  } else if ([].concat($job.inputs.reads)[0].metadata.seq_center)
#&gt;       {\n    add_param('CN', [].concat($job.inputs.reads)[0].metadata.seq_center)\n
#&gt;       \ }\n  if ($job.inputs.rg_library_id) {\n    add_param('LB', $job.inputs.rg_library_id)\n
#&gt;       \ } else if ([].concat($job.inputs.reads)[0].metadata.library_id) {\n    add_param('LB',
#&gt;       [].concat($job.inputs.reads)[0].metadata.library_id)\n  }\n  if ($job.inputs.rg_mfl)
#&gt;       {\n    add_param('PI', $job.inputs.rg_mfl)\n  } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;       {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;       \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;       /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;       {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;       /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;       $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;       {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;       \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;       \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;       [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;       \".concat(param_list.join(\" \"))\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: |-
#&gt;       {
#&gt;         if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;           return "--quantMode ".concat($job.inputs.quantMode)
#&gt;         }
#&gt;       }
#&gt;     class: Expression
#&gt; - position: 100
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n
#&gt;       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===
#&gt;       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr
#&gt;       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(
#&gt;       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  intermediate =
#&gt;       common_prefix.replace( /\\-$|\\_$|\\.$/, '' ).concat(\"._STARgenome\")\n  source
#&gt;       = \"./\".concat(intermediate)\n  destination = intermediate.concat(\".tar\")\n
#&gt;       \ if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;       != \"None\") {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n
#&gt;       \ }\n}"
#&gt;     class: Expression
#&gt; - position: 0
#&gt;   prefix: --outFileNamePrefix
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n
#&gt;       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===
#&gt;       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr
#&gt;       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(
#&gt;       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;       /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;     class: Expression
#&gt; - position: 101
#&gt;   separate: yes
#&gt;   valueFrom:
#&gt;     engine: '#cwl-js-engine'
#&gt;     script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(), \n
#&gt;       \     a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp; a1.charAt(i)===
#&gt;       a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n  path_list = []\n  arr
#&gt;       = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return path_list.push(f.path.replace(/\\\\/g,'/').replace(
#&gt;       /.*\\//, '' ))})\n  common_prefix = sharedStart(path_list)\n  mate1 = common_prefix.replace(
#&gt;       /\\-$|\\_$|\\.$/, '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace(
#&gt;       /\\-$|\\_$|\\.$/, '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;       \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped ==
#&gt;       \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \" \", mate1fq,
#&gt;       \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;       == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \" \",
#&gt;       mate1fq)\n  }\n}"
#&gt;     class: Expression
#&gt; stdin: ''
#&gt; stdout: ''
#&gt; successCodes: []
#&gt; temporaryFailCodes: []
#&gt; x: 624.0
#&gt; 'y': 323
#&gt; </div><div class='input'><span class='co'># get included input ports</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input_port</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; #STAR_Genome_Generate #STAR_Genome_Generate                 #STAR 
#&gt;          "#sjdbScore"       "#sjdbOverhang"      "#winFlankNbins" 
#&gt;                 #STAR 
#&gt;        "#winBinNbits" </div><div class='input'><span class='co'># set included input ports</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_input_port</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#STAR.alignSJDBoverhangMin"</span>, <span class='st'>"chimScoreSeparation"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input_port</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   #STAR_Genome_Generate   #STAR_Genome_Generate                   #STAR 
#&gt;            "#sjdbScore"         "#sjdbOverhang"        "#winFlankNbins" 
#&gt;                   #STAR                   #STAR                   #STAR 
#&gt;          "#winBinNbits"  "#chimScoreSeparation" "#alignSJDBoverhangMin" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_input_port</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#STAR.alignSJDBoverhangMin"</span>, <span class='st'>"chimScoreSeparation"</span><span class='op'>)</span>, <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input_port</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; #STAR_Genome_Generate #STAR_Genome_Generate                 #STAR 
#&gt;          "#sjdbScore"       "#sjdbOverhang"      "#winFlankNbins" 
#&gt;                 #STAR 
#&gt;        "#winBinNbits" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input_node</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;         sjdbGTFfile               fastq    genomeFastaFiles 
#&gt;      "#sjdbGTFfile"            "#fastq" "#genomeFastaFiles" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_output_node</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                 unmapped_reads    transcriptome_aligned_reads 
#&gt;              "#unmapped_reads" "#transcriptome_aligned_reads" 
#&gt;               splice_junctions                 reads_per_gene 
#&gt;            "#splice_junctions"              "#reads_per_gene" 
#&gt;                      log_files             chimeric_junctions 
#&gt;                   "#log_files"          "#chimeric_junctions" 
#&gt;            intermediate_genome            chimeric_alignments 
#&gt;         "#intermediate_genome"         "#chimeric_alignments" 
#&gt;                     sorted_bam                         result 
#&gt;                  "#sorted_bam"                      "#result" 
#&gt;            intermediate_genome 
#&gt;         "#intermediate_genome" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>get_input_exposed</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;    sjdbGTFtagExonParentTranscript          sjdbGTFtagExonParentGene 
#&gt; "#sjdbGTFtagExonParentTranscript"       "#sjdbGTFtagExonParentGene" 
#&gt;             winAnchorMultimapNmax                winAnchorDistNbins 
#&gt;          "#winAnchorMultimapNmax"             "#winAnchorDistNbins" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>step_input_id</span><span class='op'>(</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                                    int 
#&gt;                      "#STAR_Genome_Generate.sjdbScore" 
#&gt;                                                    int 
#&gt;                   "#STAR_Genome_Generate.sjdbOverhang" 
#&gt;                                                 string 
#&gt; "#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript" 
#&gt;                                                 string 
#&gt;       "#STAR_Genome_Generate.sjdbGTFtagExonParentGene" 
#&gt;                                                File... 
#&gt;                    "#STAR_Genome_Generate.sjdbGTFfile" 
#&gt;                                                 string 
#&gt;             "#STAR_Genome_Generate.sjdbGTFfeatureExon" 
#&gt;                                                 string 
#&gt;               "#STAR_Genome_Generate.sjdbGTFchrPrefix" 
#&gt;                                                    int 
#&gt;                "#STAR_Genome_Generate.genomeSAsparseD" 
#&gt;                                                    int 
#&gt;            "#STAR_Genome_Generate.genomeSAindexNbases" 
#&gt;                                                   File 
#&gt;               "#STAR_Genome_Generate.genomeFastaFiles" 
#&gt;                                                 string 
#&gt;              "#STAR_Genome_Generate.genomeChrBinNbits" 
#&gt;                                                   File 
#&gt;                    "#SBG_FASTQ_Quality_Detector.fastq" 
#&gt;                                                   enum 
#&gt;                "#Picard_SortSam.validation_stringency" 
#&gt;                                                   enum 
#&gt;                           "#Picard_SortSam.sort_order" 
#&gt;                                                   enum 
#&gt;                                "#Picard_SortSam.quiet" 
#&gt;                                                   enum 
#&gt;                          "#Picard_SortSam.output_type" 
#&gt;                                                    int 
#&gt;                       "#Picard_SortSam.memory_per_job" 
#&gt;                                                    int 
#&gt;                   "#Picard_SortSam.max_records_in_ram" 
#&gt;                                                   File 
#&gt;                            "#Picard_SortSam.input_bam" 
#&gt;                                                   enum 
#&gt;                         "#Picard_SortSam.create_index" 
#&gt;                                                    int 
#&gt;                    "#Picard_SortSam.compression_level" 
#&gt;                                                    int 
#&gt;                                  "#STAR.winFlankNbins" 
#&gt;                                                    int 
#&gt;                                    "#STAR.winBinNbits" 
#&gt;                                                    int 
#&gt;                          "#STAR.winAnchorMultimapNmax" 
#&gt;                                                    int 
#&gt;                             "#STAR.winAnchorDistNbins" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.twopassMode" 
#&gt;                                                    int 
#&gt;                                 "#STAR.twopass1readsN" 
#&gt;                                                    int 
#&gt;                                      "#STAR.sjdbScore" 
#&gt;                                                    int 
#&gt;                                   "#STAR.sjdbOverhang" 
#&gt;                                                   enum 
#&gt;                                 "#STAR.sjdbInsertSave" 
#&gt;                                                 string 
#&gt;                 "#STAR.sjdbGTFtagExonParentTranscript" 
#&gt;                                                 string 
#&gt;                       "#STAR.sjdbGTFtagExonParentGene" 
#&gt;                                                File... 
#&gt;                                    "#STAR.sjdbGTFfile" 
#&gt;                                                 string 
#&gt;                             "#STAR.sjdbGTFfeatureExon" 
#&gt;                                                 string 
#&gt;                               "#STAR.sjdbGTFchrPrefix" 
#&gt;                                                  float 
#&gt;                   "#STAR.seedSearchStartLmaxOverLread" 
#&gt;                                                    int 
#&gt;                            "#STAR.seedSearchStartLmax" 
#&gt;                                                    int 
#&gt;                                 "#STAR.seedSearchLmax" 
#&gt;                                                    int 
#&gt;                              "#STAR.seedPerWindowNmax" 
#&gt;                                                    int 
#&gt;                                "#STAR.seedPerReadNmax" 
#&gt;                                                    int 
#&gt;                          "#STAR.seedNoneLociPerWindow" 
#&gt;                                                    int 
#&gt;                               "#STAR.seedMultimapNmax" 
#&gt;                                                    int 
#&gt;                             "#STAR.scoreStitchSJshift" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreInsOpen" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreInsBase" 
#&gt;                                                  float 
#&gt;                    "#STAR.scoreGenomicLengthLog2scale" 
#&gt;                                                    int 
#&gt;                                 "#STAR.scoreGapNoncan" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreGapGCAG" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreGapATAC" 
#&gt;                                                    int 
#&gt;                                       "#STAR.scoreGap" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreDelOpen" 
#&gt;                                                    int 
#&gt;                                   "#STAR.scoreDelBase" 
#&gt;                                                 string 
#&gt;                                  "#STAR.rg_seq_center" 
#&gt;                                                 string 
#&gt;                                   "#STAR.rg_sample_id" 
#&gt;                                                 string 
#&gt;                            "#STAR.rg_platform_unit_id" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.rg_platform" 
#&gt;                                                 string 
#&gt;                                         "#STAR.rg_mfl" 
#&gt;                                                 string 
#&gt;                                  "#STAR.rg_library_id" 
#&gt;                                                File... 
#&gt;                                          "#STAR.reads" 
#&gt;                                                   enum 
#&gt;                             "#STAR.readMatesLengthsIn" 
#&gt;                                                    int 
#&gt;                                  "#STAR.readMapNumber" 
#&gt;                                                   enum 
#&gt;                          "#STAR.quantTranscriptomeBan" 
#&gt;                                                   enum 
#&gt;                                      "#STAR.quantMode" 
#&gt;                                                   enum 
#&gt;                                 "#STAR.outSortingType" 
#&gt;                                                   enum 
#&gt;                               "#STAR.outSJfilterReads" 
#&gt;                                                 int... 
#&gt;                         "#STAR.outSJfilterOverhangMin" 
#&gt;                                                 int... 
#&gt;                    "#STAR.outSJfilterIntronMaxVsReadN" 
#&gt;                                                 int... 
#&gt;                    "#STAR.outSJfilterDistToOtherSJmin" 
#&gt;                                                 int... 
#&gt;                      "#STAR.outSJfilterCountUniqueMin" 
#&gt;                                                 int... 
#&gt;                       "#STAR.outSJfilterCountTotalMin" 
#&gt;                                                   enum 
#&gt;                                 "#STAR.outSAMunmapped" 
#&gt;                                                   enum 
#&gt;                                     "#STAR.outSAMtype" 
#&gt;                                                   enum 
#&gt;                              "#STAR.outSAMstrandField" 
#&gt;                                                   enum 
#&gt;                                   "#STAR.outSAMreadID" 
#&gt;                                                   enum 
#&gt;                              "#STAR.outSAMprimaryFlag" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.outSAMorder" 
#&gt;                                                   enum 
#&gt;                                     "#STAR.outSAMmode" 
#&gt;                                                    int 
#&gt;                               "#STAR.outSAMmapqUnique" 
#&gt;                                                 string 
#&gt;                                 "#STAR.outSAMheaderPG" 
#&gt;                                                 string 
#&gt;                                 "#STAR.outSAMheaderHD" 
#&gt;                                                    int 
#&gt;                                   "#STAR.outSAMflagOR" 
#&gt;                                                    int 
#&gt;                                  "#STAR.outSAMflagAND" 
#&gt;                                                   enum 
#&gt;                               "#STAR.outSAMattributes" 
#&gt;                                                   enum 
#&gt;                               "#STAR.outReadsUnmapped" 
#&gt;                                                    int 
#&gt;                             "#STAR.outQSconversionAdd" 
#&gt;                                                   enum 
#&gt;                                  "#STAR.outFilterType" 
#&gt;                                                  float 
#&gt;                     "#STAR.outFilterScoreMinOverLread" 
#&gt;                                                    int 
#&gt;                              "#STAR.outFilterScoreMin" 
#&gt;                                                    int 
#&gt;                    "#STAR.outFilterMultimapScoreRange" 
#&gt;                                                    int 
#&gt;                          "#STAR.outFilterMultimapNmax" 
#&gt;                                                  float 
#&gt;                 "#STAR.outFilterMismatchNoverReadLmax" 
#&gt;                                                  float 
#&gt;                     "#STAR.outFilterMismatchNoverLmax" 
#&gt;                                                    int 
#&gt;                          "#STAR.outFilterMismatchNmax" 
#&gt;                                                  float 
#&gt;                    "#STAR.outFilterMatchNminOverLread" 
#&gt;                                                    int 
#&gt;                             "#STAR.outFilterMatchNmin" 
#&gt;                                                   enum 
#&gt;                          "#STAR.outFilterIntronMotifs" 
#&gt;                                                    int 
#&gt;                             "#STAR.limitSjdbInsertNsj" 
#&gt;                                                    int 
#&gt;                              "#STAR.limitOutSJoneRead" 
#&gt;                                                    int 
#&gt;                            "#STAR.limitOutSJcollapsed" 
#&gt;                                                    int 
#&gt;                                "#STAR.limitBAMsortRAM" 
#&gt;                                                 string 
#&gt;                                  "#STAR.genomeDirName" 
#&gt;                                                   File 
#&gt;                                         "#STAR.genome" 
#&gt;                                                 int... 
#&gt;                                   "#STAR.clip5pNbases" 
#&gt;                                                 int... 
#&gt;                                   "#STAR.clip3pNbases" 
#&gt;                                                 int... 
#&gt;                       "#STAR.clip3pAfterAdapterNbases" 
#&gt;                                              string... 
#&gt;                               "#STAR.clip3pAdapterSeq" 
#&gt;                                               float... 
#&gt;                               "#STAR.clip3pAdapterMMp" 
#&gt;                                                    int 
#&gt;                                 "#STAR.chimSegmentMin" 
#&gt;                                                    int 
#&gt;                            "#STAR.chimScoreSeparation" 
#&gt;                                                    int 
#&gt;                                   "#STAR.chimScoreMin" 
#&gt;                                                    int 
#&gt;                       "#STAR.chimScoreJunctionNonGTAG" 
#&gt;                                                    int 
#&gt;                               "#STAR.chimScoreDropMax" 
#&gt;                                                   enum 
#&gt;                                    "#STAR.chimOutType" 
#&gt;                                                    int 
#&gt;                        "#STAR.chimJunctionOverhangMin" 
#&gt;                                                  float 
#&gt;                        "#STAR.alignWindowsPerReadNmax" 
#&gt;                                                    int 
#&gt;                  "#STAR.alignTranscriptsPerWindowNmax" 
#&gt;                                                    int 
#&gt;                    "#STAR.alignTranscriptsPerReadNmax" 
#&gt;                                                  float 
#&gt;               "#STAR.alignSplicedMateMapLminOverLmate" 
#&gt;                                                    int 
#&gt;                        "#STAR.alignSplicedMateMapLmin" 
#&gt;                                                   enum 
#&gt;                   "#STAR.alignSoftClipAtReferenceEnds" 
#&gt;                                                    int 
#&gt;                             "#STAR.alignSJoverhangMin" 
#&gt;                                                    int 
#&gt;                           "#STAR.alignSJDBoverhangMin" 
#&gt;                                                    int 
#&gt;                               "#STAR.alignMatesGapMax" 
#&gt;                                                    int 
#&gt;                                 "#STAR.alignIntronMin" 
#&gt;                                                    int 
#&gt;                                 "#STAR.alignIntronMax" 
#&gt;                                                   enum 
#&gt;                                  "#STAR.alignEndsType" </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "#sjdbGTFfile"                    "#fastq"                         
#&gt; [3] "#genomeFastaFiles"               "#sjdbGTFtagExonParentTranscript"
#&gt; [5] "#sjdbGTFtagExonParentGene"       "#winAnchorMultimapNmax"         
#&gt; [7] "#winAnchorDistNbins"            </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_flow_input</span><span class='op'>(</span><span class='st'>"#STAR.reads"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 2
#&gt; sbg:toolAuthor: Seven Bridges Genomics
#&gt; sbg:createdOn: 1463601910
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; - RNA
#&gt; sbg:contributors:
#&gt; - tengfei
#&gt; sbg:project: tengfei/quickstart
#&gt; sbg:createdBy: tengfei
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2
#&gt; sbg:license: Apache License 2.0
#&gt; sbg:revision: 2
#&gt; sbg:modifiedOn: 1463601974
#&gt; sbg:modifiedBy: tengfei
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601910
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601952
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601974
#&gt;   sbg:revision: 2
#&gt; sbg:toolkit: STAR
#&gt; id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'
#&gt; inputs:
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Read sequence
#&gt;   description: Read sequence.
#&gt;   streamable: no
#&gt;   id: '#reads'
#&gt;   inputBinding:
#&gt;     position: 10
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n
#&gt;         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if
#&gt;         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n
#&gt;         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata
#&gt;         != null){\n        if (list[index].metadata.paired_end == 1){\n          left
#&gt;         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n
#&gt;         \         right = list[index].path\n        }\n      }\n    }\n    \n    if
#&gt;         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n
#&gt;         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n
#&gt;         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if
#&gt;         (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end
#&gt;         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n
#&gt;         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=
#&gt;         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n
#&gt;         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"
#&gt;         \"))\n  }\n}"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: sjdbGTFfile
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:x: 160.4999759
#&gt;   sbg:y: 195.0833106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: fastq
#&gt;   streamable: no
#&gt;   id: '#fastq'
#&gt;   sbg:x: 164.2499914
#&gt;   sbg:y: 323.7499502
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - File
#&gt;   label: genomeFastaFiles
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   sbg:x: 167.7499601
#&gt;   sbg:y: 469.9999106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 350.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 450.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   source: '#unmapped_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   source: '#transcriptome_aligned_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 233.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   source: '#splice_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 266.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   source: '#reads_per_gene'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 300.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   source: '#log_files'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 333.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   source: '#chimeric_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 366.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   source: '#chimeric_alignments'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 433.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Sorted BAM/SAM
#&gt;   description: Sorted BAM or SAM file.
#&gt;   streamable: no
#&gt;   id: '#sorted_bam'
#&gt;   source: '#sorted_bam'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 466.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Result
#&gt;   description: Source FASTQ file with updated metadata.
#&gt;   streamable: no
#&gt;   id: '#result'
#&gt;   source: '#result'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 533.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; requirements:
#&gt; - class: CreateFileRequirement
#&gt;   fileDef: []
#&gt; hints:
#&gt; - class: sbg:AWSInstanceType
#&gt;   value: c3.8xlarge
#&gt; label: RNA-seq Alignment - STAR
#&gt; description: "Alignment to a reference genome and transcriptome presents the first
#&gt;   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner
#&gt;   capable of mapping full length RNA sequences and detecting de novo canonical junctions,
#&gt;   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian
#&gt;   sequence reads, but fine tuning of its parameters enables customization to satisfy
#&gt;   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).
#&gt;   \ \nSplice junction annotations can optionally be collected from splice junction
#&gt;   databases. Set the \"Overhang length\" parameter to a value larger than zero in
#&gt;   order to use splice junction databases. For constant read length, this value should
#&gt;   (ideally) be equal to mate length decreased by 1; for long reads with non-constant
#&gt;   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ
#&gt;   files reveals read length distribution. STAR can detect chimeric transcripts, but
#&gt;   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted
#&gt;   to a desired minimum chimeric segment length. Aligned reads are reported in BAM
#&gt;   format and can be viewed in a genome browser (such as IGV). A file containing detected
#&gt;   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format
#&gt;   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write
#&gt;   unmapped in SAM\" parameters enable unmapped output type selection."
#&gt; class: Workflow
#&gt; steps:
#&gt; - id: '#STAR_Genome_Generate'
#&gt;   inputs:
#&gt;   - id: '#STAR_Genome_Generate.sjdbScore'
#&gt;   - id: '#STAR_Genome_Generate.sjdbOverhang'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'
#&gt;     source: '#sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'
#&gt;     source: '#sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAsparseD'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAindexNbases'
#&gt;   - id: '#STAR_Genome_Generate.genomeFastaFiles'
#&gt;     source: '#genomeFastaFiles'
#&gt;   - id: '#STAR_Genome_Generate.genomeChrBinNbits'
#&gt;   outputs:
#&gt;   - id: '#STAR_Genome_Generate.genome'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 1
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         sjdbScore: 0
#&gt;         sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;         sjdbOverhang: 0
#&gt;         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;         genomeChrBinNbits: genomeChrBinNbits
#&gt;         genomeSAsparseD: 0
#&gt;         sjdbGTFfile:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /demo/test-files/chr20.gtf
#&gt;         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;         genomeFastaFiles:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /sbgenomics/test-data/chr20.fa
#&gt;         sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;         genomeSAindexNbases: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911469
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 1
#&gt;     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa
#&gt;       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD
#&gt;       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt;     sbg:modifiedOn: 1450911470
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911469
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911470
#&gt;       sbg:revision: 1
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbScore
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbOverhang
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentTranscript
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentGene
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFfeatureExon
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFchrPrefix
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Suffux array sparsity
#&gt;       description: 'Distance between indices: use bigger numbers to decrease needed
#&gt;         RAM at the cost of mapping speed reduction (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#genomeSAsparseD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAsparseD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Pre-indexing string length
#&gt;       description: Length (bases) of the SA pre-indexing string. Typically between
#&gt;         10 and 15. Longer strings will use much more memory, but allow faster searches.
#&gt;         For small genomes, this number needs to be scaled down, with a typical value
#&gt;         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,
#&gt;         this is equal to 9, for 100 kiloBase genome, this is equal to 7.
#&gt;       streamable: no
#&gt;       id: '#genomeSAindexNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAindexNbases
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '14'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome fasta files
#&gt;       description: Reference sequence to which to align the reads.
#&gt;       streamable: no
#&gt;       id: '#genomeFastaFiles'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeFastaFiles
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FA
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Bins size
#&gt;       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins
#&gt;         for genome storage: each chromosome will occupy an integer number of bins.
#&gt;         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,
#&gt;         you may need to reduce this number to reduce RAM consumption. The following
#&gt;         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this
#&gt;         is equal to 15.'
#&gt;       streamable: no
#&gt;       id: '#genomeChrBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeChrBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '18'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Genome Files
#&gt;       description: Genome files comprise binary genome sequence, suffix arrays, text
#&gt;         chromosome names/lengths, splice junctions coordinates, and transcripts/genes
#&gt;         information.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       outputBinding:
#&gt;         glob: '*.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     label: STAR Genome Generate
#&gt;     description: STAR Genome Generate is a tool that generates genome index files.
#&gt;       One set of files should be generated per each genome/annotation combination.
#&gt;       Once produced, these files could be used as long as genome/annotation combination
#&gt;       stays the same. Also, STAR Genome Generate which produced these files and STAR
#&gt;       aligner using them must be the same toolkit version.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - mkdir
#&gt;     - genomeDir
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runMode
#&gt;     - genomeGenerate
#&gt;     - --genomeDir
#&gt;     - ./genomeDir
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 99
#&gt;       separate: yes
#&gt;       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 384.0832266
#&gt;     'y': 446.4998957
#&gt;   sbg:x: 100.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#SBG_FASTQ_Quality_Detector'
#&gt;   inputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.fastq'
#&gt;     source: '#fastq'
#&gt;   outputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 3
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 1000
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         fastq:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /path/to/fastq.ext
#&gt;     sbg:toolAuthor: Seven Bridges Genomics
#&gt;     sbg:createdOn: 1450911312
#&gt;     sbg:categories:
#&gt;     - FASTQ-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:project: bix-demo/sbgtools-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     sbg:license: Apache License 2.0
#&gt;     sbg:revision: 3
#&gt;     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext
#&gt;       /path/to/fastq.ext
#&gt;     sbg:modifiedOn: 1450911314
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911312
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911314
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: SBGTools
#&gt;     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     inputs:
#&gt;     - type:
#&gt;       - File
#&gt;       label: Fastq
#&gt;       description: FASTQ file.
#&gt;       streamable: no
#&gt;       id: '#fastq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --fastq
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       required: yes
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Result
#&gt;       description: Source FASTQ file with updated metadata.
#&gt;       streamable: no
#&gt;       id: '#result'
#&gt;       outputBinding:
#&gt;         glob: '*.fastq'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     requirements:
#&gt;     - class: CreateFileRequirement
#&gt;       fileDef: []
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0
#&gt;       dockerImageId: ''
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 1000
#&gt;     label: SBG FASTQ Quality Detector
#&gt;     description: FASTQ Quality Scale Detector detects which quality encoding scheme
#&gt;       was used in your reads and automatically enters the proper value in the "Quality
#&gt;       Scale" metadata field.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - python
#&gt;     - /opt/sbg_fastq_quality_scale_detector.py
#&gt;     arguments: []
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 375.3333179
#&gt;     'y': 323.5833156
#&gt;   sbg:x: 300.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#Picard_SortSam'
#&gt;   inputs:
#&gt;   - id: '#Picard_SortSam.validation_stringency'
#&gt;     default: SILENT
#&gt;   - id: '#Picard_SortSam.sort_order'
#&gt;     default: Coordinate
#&gt;   - id: '#Picard_SortSam.quiet'
#&gt;   - id: '#Picard_SortSam.output_type'
#&gt;   - id: '#Picard_SortSam.memory_per_job'
#&gt;   - id: '#Picard_SortSam.max_records_in_ram'
#&gt;   - id: '#Picard_SortSam.input_bam'
#&gt;     source: '#STAR.aligned_reads'
#&gt;   - id: '#Picard_SortSam.create_index'
#&gt;     default: 'True'
#&gt;   - id: '#Picard_SortSam.compression_level'
#&gt;   outputs:
#&gt;   - id: '#Picard_SortSam.sorted_bam'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 2
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 2048
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         sort_order: Coordinate
#&gt;         input_bam:
#&gt;           path: /root/dir/example.tested.bam
#&gt;         memory_per_job: 2048
#&gt;         output_type: ~
#&gt;         create_index: ~
#&gt;     sbg:toolAuthor: Broad Institute
#&gt;     sbg:createdOn: 1450911168
#&gt;     sbg:categories:
#&gt;     - SAM/BAM-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: http://broadinstitute.github.io/picard/index.html
#&gt;       label: Homepage
#&gt;     - id: https://github.com/broadinstitute/picard/releases/tag/1.138
#&gt;       label: Source Code
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Wiki
#&gt;     - id: https://github.com/broadinstitute/picard/zipball/master
#&gt;       label: Download
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/picard-1-140-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: '1.140'
#&gt;     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     sbg:license: MIT License, Apache 2.0 Licence
#&gt;     sbg:revision: 2
#&gt;     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam
#&gt;       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam
#&gt;       SORT_ORDER=coordinate  /root/dir/example.tested.bam
#&gt;     sbg:modifiedOn: 1450911170
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911168
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911169
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911170
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: Picard
#&gt;     id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: validation_stringency
#&gt;         symbols:
#&gt;         - STRICT
#&gt;         - LENIENT
#&gt;         - SILENT
#&gt;         type: enum
#&gt;       label: Validation stringency
#&gt;       description: 'Validation stringency for all SAM files read by this program.
#&gt;         Setting stringency to SILENT can improve performance when processing a BAM
#&gt;         file in which variable-length data (read, qualities, tags) do not otherwise
#&gt;         need to be decoded. This option can be set to ''null'' to clear the default
#&gt;         value. Possible values: {STRICT, LENIENT, SILENT}.'
#&gt;       streamable: no
#&gt;       id: '#validation_stringency'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: VALIDATION_STRINGENCY=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.validation_stringency)
#&gt;               {
#&gt;                 return $job.inputs.validation_stringency
#&gt;               }
#&gt;               else
#&gt;               {
#&gt;                 return "SILENT"
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SILENT
#&gt;       required: no
#&gt;     - type:
#&gt;       - name: sort_order
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - Queryname
#&gt;         - Coordinate
#&gt;         type: enum
#&gt;       label: Sort order
#&gt;       description: 'Sort order of the output file. Possible values: {unsorted, queryname,
#&gt;         coordinate}.'
#&gt;       streamable: no
#&gt;       id: '#sort_order'
#&gt;       inputBinding:
#&gt;         position: 3
#&gt;         prefix: SORT_ORDER=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               p = $job.inputs.sort_order.toLowerCase()
#&gt;               return p
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: Coordinate
#&gt;       sbg:altPrefix: SO
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quiet
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Quiet
#&gt;       description: 'This parameter indicates whether to suppress job-summary info
#&gt;         on System.err. This option can be set to ''null'' to clear the default value.
#&gt;         Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#quiet'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: QUIET=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: output_type
#&gt;         symbols:
#&gt;         - BAM
#&gt;         - SAM
#&gt;         - SAME AS INPUT
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Since Picard tools can output both SAM and BAM files, user can
#&gt;         choose the format of the output file.
#&gt;       streamable: no
#&gt;       id: '#output_type'
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SAME AS INPUT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Memory per job
#&gt;       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for
#&gt;         single threaded jobs.
#&gt;       streamable: no
#&gt;       id: '#memory_per_job'
#&gt;       sbg:toolDefaultValue: '2048'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max records in RAM
#&gt;       description: When writing SAM files that need to be sorted, this parameter will
#&gt;         specify the number of records stored in RAM before spilling to disk. Increasing
#&gt;         this number reduces the number of file handles needed to sort a SAM file,
#&gt;         and increases the amount of RAM needed. This option can be set to 'null' to
#&gt;         clear the default value.
#&gt;       streamable: no
#&gt;       id: '#max_records_in_ram'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: MAX_RECORDS_IN_RAM=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '500000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Input BAM
#&gt;       description: The BAM or SAM file to sort.
#&gt;       streamable: no
#&gt;       id: '#input_bam'
#&gt;       inputBinding:
#&gt;         position: 1
#&gt;         prefix: INPUT=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: File inputs
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;       sbg:altPrefix: I
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: create_index
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Create index
#&gt;       description: 'This parameter indicates whether to create a BAM index when writing
#&gt;         a coordinate-sorted BAM file. This option can be set to ''null'' to clear
#&gt;         the default value. Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#create_index'
#&gt;       inputBinding:
#&gt;         position: 5
#&gt;         prefix: CREATE_INDEX=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Compression level
#&gt;       description: Compression level for all compressed files created (e.g. BAM and
#&gt;         GELI). This option can be set to 'null' to clear the default value.
#&gt;       streamable: no
#&gt;       id: '#compression_level'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: COMPRESSION_LEVEL=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Sorted BAM/SAM
#&gt;       description: Sorted BAM or SAM file.
#&gt;       streamable: no
#&gt;       id: '#sorted_bam'
#&gt;       outputBinding:
#&gt;         glob: '*.sorted.?am'
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;       engineCommand: cwl-engine.js
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
#&gt;       dockerImageId: eab0e70b6629
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n
#&gt;           \ }\n  \treturn 2048\n}"
#&gt;         class: Expression
#&gt;     label: Picard SortSam
#&gt;     description: Picard SortSam sorts the input SAM or BAM. Input and output formats
#&gt;       are determined by the file extension.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - java
#&gt;     - class: Expression
#&gt;       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,
#&gt;         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"
#&gt;       engine: '#cwl-js-engine'
#&gt;     - -jar
#&gt;     - /opt/picard-tools-1.140/picard.jar
#&gt;     - SortSam
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       prefix: OUTPUT=
#&gt;       separate: no
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif
#&gt;           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,
#&gt;           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse
#&gt;           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n}\n}"
#&gt;         class: Expression
#&gt;     - position: 1000
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out
#&gt;           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n
#&gt;           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n
#&gt;           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n
#&gt;           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index
#&gt;           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==
#&gt;           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 773.0831807
#&gt;     'y': 470.9165939
#&gt;   sbg:x: 500.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#STAR'
#&gt;   inputs:
#&gt;   - id: '#STAR.winFlankNbins'
#&gt;   - id: '#STAR.winBinNbits'
#&gt;   - id: '#STAR.winAnchorMultimapNmax'
#&gt;     source: '#winAnchorMultimapNmax'
#&gt;   - id: '#STAR.winAnchorDistNbins'
#&gt;     source: '#winAnchorDistNbins'
#&gt;   - id: '#STAR.twopassMode'
#&gt;   - id: '#STAR.twopass1readsN'
#&gt;   - id: '#STAR.sjdbScore'
#&gt;   - id: '#STAR.sjdbOverhang'
#&gt;     default: 100
#&gt;   - id: '#STAR.sjdbInsertSave'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR.seedSearchStartLmaxOverLread'
#&gt;   - id: '#STAR.seedSearchStartLmax'
#&gt;   - id: '#STAR.seedSearchLmax'
#&gt;   - id: '#STAR.seedPerWindowNmax'
#&gt;   - id: '#STAR.seedPerReadNmax'
#&gt;   - id: '#STAR.seedNoneLociPerWindow'
#&gt;   - id: '#STAR.seedMultimapNmax'
#&gt;   - id: '#STAR.scoreStitchSJshift'
#&gt;   - id: '#STAR.scoreInsOpen'
#&gt;   - id: '#STAR.scoreInsBase'
#&gt;   - id: '#STAR.scoreGenomicLengthLog2scale'
#&gt;   - id: '#STAR.scoreGapNoncan'
#&gt;   - id: '#STAR.scoreGapGCAG'
#&gt;   - id: '#STAR.scoreGapATAC'
#&gt;   - id: '#STAR.scoreGap'
#&gt;   - id: '#STAR.scoreDelOpen'
#&gt;   - id: '#STAR.scoreDelBase'
#&gt;   - id: '#STAR.rg_seq_center'
#&gt;   - id: '#STAR.rg_sample_id'
#&gt;   - id: '#STAR.rg_platform_unit_id'
#&gt;   - id: '#STAR.rg_platform'
#&gt;   - id: '#STAR.rg_mfl'
#&gt;   - id: '#STAR.rg_library_id'
#&gt;   - id: '#STAR.reads'
#&gt;     source: '#reads'
#&gt;   - id: '#STAR.readMatesLengthsIn'
#&gt;   - id: '#STAR.readMapNumber'
#&gt;   - id: '#STAR.quantTranscriptomeBan'
#&gt;   - id: '#STAR.quantMode'
#&gt;     default: TranscriptomeSAM
#&gt;   - id: '#STAR.outSortingType'
#&gt;     default: SortedByCoordinate
#&gt;   - id: '#STAR.outSJfilterReads'
#&gt;   - id: '#STAR.outSJfilterOverhangMin'
#&gt;   - id: '#STAR.outSJfilterIntronMaxVsReadN'
#&gt;   - id: '#STAR.outSJfilterDistToOtherSJmin'
#&gt;   - id: '#STAR.outSJfilterCountUniqueMin'
#&gt;   - id: '#STAR.outSJfilterCountTotalMin'
#&gt;   - id: '#STAR.outSAMunmapped'
#&gt;   - id: '#STAR.outSAMtype'
#&gt;     default: BAM
#&gt;   - id: '#STAR.outSAMstrandField'
#&gt;   - id: '#STAR.outSAMreadID'
#&gt;   - id: '#STAR.outSAMprimaryFlag'
#&gt;   - id: '#STAR.outSAMorder'
#&gt;   - id: '#STAR.outSAMmode'
#&gt;   - id: '#STAR.outSAMmapqUnique'
#&gt;   - id: '#STAR.outSAMheaderPG'
#&gt;   - id: '#STAR.outSAMheaderHD'
#&gt;   - id: '#STAR.outSAMflagOR'
#&gt;   - id: '#STAR.outSAMflagAND'
#&gt;   - id: '#STAR.outSAMattributes'
#&gt;   - id: '#STAR.outReadsUnmapped'
#&gt;     default: Fastx
#&gt;   - id: '#STAR.outQSconversionAdd'
#&gt;   - id: '#STAR.outFilterType'
#&gt;   - id: '#STAR.outFilterScoreMinOverLread'
#&gt;   - id: '#STAR.outFilterScoreMin'
#&gt;   - id: '#STAR.outFilterMultimapScoreRange'
#&gt;   - id: '#STAR.outFilterMultimapNmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverReadLmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverLmax'
#&gt;   - id: '#STAR.outFilterMismatchNmax'
#&gt;   - id: '#STAR.outFilterMatchNminOverLread'
#&gt;   - id: '#STAR.outFilterMatchNmin'
#&gt;   - id: '#STAR.outFilterIntronMotifs'
#&gt;   - id: '#STAR.limitSjdbInsertNsj'
#&gt;   - id: '#STAR.limitOutSJoneRead'
#&gt;   - id: '#STAR.limitOutSJcollapsed'
#&gt;   - id: '#STAR.limitBAMsortRAM'
#&gt;   - id: '#STAR.genomeDirName'
#&gt;   - id: '#STAR.genome'
#&gt;     source: '#STAR_Genome_Generate.genome'
#&gt;   - id: '#STAR.clip5pNbases'
#&gt;   - id: '#STAR.clip3pNbases'
#&gt;   - id: '#STAR.clip3pAfterAdapterNbases'
#&gt;   - id: '#STAR.clip3pAdapterSeq'
#&gt;   - id: '#STAR.clip3pAdapterMMp'
#&gt;   - id: '#STAR.chimSegmentMin'
#&gt;   - id: '#STAR.chimScoreSeparation'
#&gt;   - id: '#STAR.chimScoreMin'
#&gt;   - id: '#STAR.chimScoreJunctionNonGTAG'
#&gt;   - id: '#STAR.chimScoreDropMax'
#&gt;   - id: '#STAR.chimOutType'
#&gt;   - id: '#STAR.chimJunctionOverhangMin'
#&gt;   - id: '#STAR.alignWindowsPerReadNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerWindowNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerReadNmax'
#&gt;   - id: '#STAR.alignSplicedMateMapLminOverLmate'
#&gt;   - id: '#STAR.alignSplicedMateMapLmin'
#&gt;   - id: '#STAR.alignSoftClipAtReferenceEnds'
#&gt;   - id: '#STAR.alignSJoverhangMin'
#&gt;   - id: '#STAR.alignSJDBoverhangMin'
#&gt;   - id: '#STAR.alignMatesGapMax'
#&gt;   - id: '#STAR.alignIntronMin'
#&gt;   - id: '#STAR.alignIntronMax'
#&gt;   - id: '#STAR.alignEndsType'
#&gt;   outputs:
#&gt;   - id: '#STAR.unmapped_reads'
#&gt;   - id: '#STAR.transcriptome_aligned_reads'
#&gt;   - id: '#STAR.splice_junctions'
#&gt;   - id: '#STAR.reads_per_gene'
#&gt;   - id: '#STAR.log_files'
#&gt;   - id: '#STAR.intermediate_genome'
#&gt;   - id: '#STAR.chimeric_junctions'
#&gt;   - id: '#STAR.chimeric_alignments'
#&gt;   - id: '#STAR.aligned_reads'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 4
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         alignWindowsPerReadNmax: 0
#&gt;         outSAMheaderPG: outSAMheaderPG
#&gt;         GENOME_DIR_NAME: ''
#&gt;         outFilterMatchNminOverLread: 0
#&gt;         rg_platform_unit_id: rg_platform_unit
#&gt;         alignTranscriptsPerReadNmax: 0
#&gt;         readMapNumber: 0
#&gt;         alignSplicedMateMapLminOverLmate: 0
#&gt;         alignMatesGapMax: 0
#&gt;         outFilterMultimapNmax: 0
#&gt;         clip5pNbases:
#&gt;         - 0
#&gt;         outSAMstrandField: None
#&gt;         readMatesLengthsIn: NotEqual
#&gt;         outSAMattributes: Standard
#&gt;         seedMultimapNmax: 0
#&gt;         rg_mfl: rg_mfl
#&gt;         chimSegmentMin: 0
#&gt;         winAnchorDistNbins: 0
#&gt;         outSortingType: SortedByCoordinate
#&gt;         outFilterMultimapScoreRange: 0
#&gt;         sjdbInsertSave: Basic
#&gt;         clip3pAfterAdapterNbases:
#&gt;         - 0
#&gt;         scoreDelBase: 0
#&gt;         outFilterMatchNmin: 0
#&gt;         twopass1readsN: 0
#&gt;         outSAMunmapped: None
#&gt;         genome:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: genome.ext
#&gt;         sjdbGTFtagExonParentTranscript: ''
#&gt;         limitBAMsortRAM: 0
#&gt;         alignEndsType: Local
#&gt;         seedNoneLociPerWindow: 0
#&gt;         rg_sample_id: rg_sample
#&gt;         sjdbGTFtagExonParentGene: ''
#&gt;         chimScoreMin: 0
#&gt;         outSJfilterIntronMaxVsReadN:
#&gt;         - 0
#&gt;         twopassMode: Basic
#&gt;         alignSplicedMateMapLmin: 0
#&gt;         outSJfilterReads: All
#&gt;         outSAMprimaryFlag: OneBestScore
#&gt;         outSJfilterCountTotalMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         outSAMorder: Paired
#&gt;         outSAMflagAND: 0
#&gt;         chimScoreSeparation: 0
#&gt;         alignSJoverhangMin: 0
#&gt;         outFilterScoreMin: 0
#&gt;         seedSearchStartLmax: 0
#&gt;         scoreGapGCAG: 0
#&gt;         scoreGenomicLengthLog2scale: 0
#&gt;         outFilterIntronMotifs: None
#&gt;         outFilterMismatchNmax: 0
#&gt;         reads:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           metadata:
#&gt;             format: fastq
#&gt;             paired_end: '1'
#&gt;             seq_center: illumina
#&gt;           path: /test-data/mate_1.fastq.bz2
#&gt;         scoreGap: 0
#&gt;         outSJfilterOverhangMin:
#&gt;         - 30
#&gt;         - 12
#&gt;         - 12
#&gt;         - 12
#&gt;         outSAMflagOR: 0
#&gt;         outSAMmode: Full
#&gt;         rg_library_id: ''
#&gt;         chimScoreJunctionNonGTAG: 0
#&gt;         scoreInsOpen: 0
#&gt;         clip3pAdapterSeq:
#&gt;         - clip3pAdapterSeq
#&gt;         chimScoreDropMax: 0
#&gt;         outFilterType: Normal
#&gt;         scoreGapATAC: 0
#&gt;         rg_platform: Ion Torrent PGM
#&gt;         clip3pAdapterMMp:
#&gt;         - 0
#&gt;         sjdbGTFfeatureExon: ''
#&gt;         outQSconversionAdd: 0
#&gt;         quantMode: TranscriptomeSAM
#&gt;         alignIntronMin: 0
#&gt;         scoreInsBase: 0
#&gt;         scoreGapNoncan: 0
#&gt;         seedSearchLmax: 0
#&gt;         outSJfilterDistToOtherSJmin:
#&gt;         - 0
#&gt;         outFilterScoreMinOverLread: 0
#&gt;         alignSJDBoverhangMin: 0
#&gt;         limitOutSJcollapsed: 0
#&gt;         winAnchorMultimapNmax: 0
#&gt;         outFilterMismatchNoverLmax: 0
#&gt;         rg_seq_center: ''
#&gt;         outSAMheaderHD: outSAMheaderHD
#&gt;         chimOutType: Within
#&gt;         quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;         limitOutSJoneRead: 0
#&gt;         alignTranscriptsPerWindowNmax: 0
#&gt;         sjdbOverhang: ~
#&gt;         outReadsUnmapped: Fastx
#&gt;         scoreStitchSJshift: 0
#&gt;         seedPerWindowNmax: 0
#&gt;         outSJfilterCountUniqueMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         scoreDelOpen: 0
#&gt;         sjdbGTFfile:
#&gt;         - path: /demo/test-data/chr20.gtf
#&gt;         clip3pNbases:
#&gt;         - 0
#&gt;         - 3
#&gt;         winBinNbits: 0
#&gt;         sjdbScore: ~
#&gt;         seedSearchStartLmaxOverLread: 0
#&gt;         alignIntronMax: 0
#&gt;         seedPerReadNmax: 0
#&gt;         outFilterMismatchNoverReadLmax: 0
#&gt;         winFlankNbins: 0
#&gt;         sjdbGTFchrPrefix: chrPrefix
#&gt;         alignSoftClipAtReferenceEnds: 'Yes'
#&gt;         outSAMreadID: Standard
#&gt;         outSAMtype: BAM
#&gt;         chimJunctionOverhangMin: 0
#&gt;         limitSjdbInsertNsj: 0
#&gt;         outSAMmapqUnique: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911471
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - ana_d
#&gt;     - bix-demo
#&gt;     - uros_sipetic
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star/4
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 4
#&gt;     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix
#&gt;       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline
#&gt;       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode
#&gt;       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;
#&gt;       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt;     sbg:modifiedOn: 1462889222
#&gt;     sbg:modifiedBy: ana_d
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911471
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911473
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911475
#&gt;       sbg:revision: 2
#&gt;     - sbg:modifiedBy: uros_sipetic
#&gt;       sbg:modifiedOn: 1462878528
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: ana_d
#&gt;       sbg:modifiedOn: 1462889222
#&gt;       sbg:revision: 4
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star/4
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Flanking regions size
#&gt;       description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;         flanking regions for each window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winFlankNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winFlankNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Bin size
#&gt;       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;         each window will occupy an integer number of bins (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '16'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max loci anchors
#&gt;       description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max bins between anchors
#&gt;       description: Max number of bins between two anchors that allows aggregation
#&gt;         of anchors into one window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorDistNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorDistNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '9'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: twopassMode
#&gt;         symbols:
#&gt;         - None
#&gt;         - Basic
#&gt;         type: enum
#&gt;       label: Two-pass mode
#&gt;       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass
#&gt;         mapping, with all 1st pass junctions inserted into the genome indices on the
#&gt;         fly.'
#&gt;       streamable: no
#&gt;       id: '#twopassMode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --twopassMode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to process in 1st step
#&gt;       description: 'Number of reads to process for the 1st step. 0: 1-step only, no
#&gt;         2nd pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#twopass1readsN'
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: sjdbInsertSave
#&gt;         symbols:
#&gt;         - Basic
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: Save junction files
#&gt;       description: 'Which files to save when sjdb junctions are inserted on the fly
#&gt;         at the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;         files; All: all files including big Genome, SA and SAindex. These files are
#&gt;         output as archive.'
#&gt;       streamable: no
#&gt;       id: '#sjdbInsertSave'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts. No need to include
#&gt;         this input, except in case of using "on the fly" annotations.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Search start point normalized
#&gt;       description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;         for paired-end reads).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmaxOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmaxOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1.0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Search start point
#&gt;       description: Defines the search start point through the read - the read is split
#&gt;         into pieces no longer than this value (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seed length
#&gt;       description: Defines the maximum length of the seeds, if =0 max seed length
#&gt;         is infinite (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per window
#&gt;       description: Max number of seeds per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per read
#&gt;       description: Max number of seeds per read (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max one-seed loci per window
#&gt;       description: Max number of one seed loci per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedNoneLociPerWindow'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedNoneLociPerWindow
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Filter pieces for stitching
#&gt;       description: Only pieces that map fewer than this value are utilized in the
#&gt;         stitching procedure (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max score reduction
#&gt;       description: Maximum score reduction while searching for SJ boundaries in the
#&gt;         stitching step.
#&gt;       streamable: no
#&gt;       id: '#scoreStitchSJshift'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreStitchSJshift
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion Open Penalty
#&gt;       description: Insertion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreInsOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion extension penalty
#&gt;       description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreInsBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Log scaled score
#&gt;       description: 'Extra score logarithmically scaled with genomic length of the
#&gt;         alignment: &lt;int&gt;*log2(genomicLength).'
#&gt;       streamable: no
#&gt;       id: '#scoreGenomicLengthLog2scale'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGenomicLengthLog2scale
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-0.25'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-canonical gap open
#&gt;       description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapNoncan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapNoncan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: GC/AG and CT/GC gap open
#&gt;       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapGCAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapGCAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AT/AC and GT/AT gap open
#&gt;       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapATAC'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapATAC
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Gap open penalty
#&gt;       description: Gap open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreGap'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGap
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion open penalty
#&gt;       description: Deletion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreDelOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion extension penalty
#&gt;       description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreDelBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sequencing center
#&gt;       description: Specify the sequencing center for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_seq_center'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sample ID
#&gt;       description: Specify the sample ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_sample_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Platform unit ID
#&gt;       description: Specify the platform unit ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_platform_unit_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: rg_platform
#&gt;         symbols:
#&gt;         - LS 454
#&gt;         - Helicos
#&gt;         - Illumina
#&gt;         - ABI SOLiD
#&gt;         - Ion Torrent PGM
#&gt;         - PacBio
#&gt;         type: enum
#&gt;       label: Platform
#&gt;       description: Specify the version of the technology that was used for sequencing
#&gt;         or assaying.
#&gt;       streamable: no
#&gt;       id: '#rg_platform'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Median fragment length
#&gt;       description: Specify the median fragment length for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_mfl'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Library ID
#&gt;       description: Specify the library ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_library_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Read sequence
#&gt;       description: Read sequence.
#&gt;       streamable: no
#&gt;       id: '#reads'
#&gt;       inputBinding:
#&gt;         position: 10
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp
#&gt;             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n
#&gt;             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =
#&gt;             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n
#&gt;             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end
#&gt;             == 2){\n          right = list[index].path\n        }\n      }\n    }\n
#&gt;             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n
#&gt;             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left
#&gt;             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;
#&gt;             ++index) {\n      \n      if (list[index].metadata != null){\n        if
#&gt;             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n
#&gt;             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n
#&gt;             \       }\n      }\n    }\n    left_join = left.join()\n    right_join
#&gt;             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n
#&gt;             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){
#&gt;             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;       sbg:x: 200.0
#&gt;       sbg:y: 200.0
#&gt;       sbg:includeInPorts: yes
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: readMatesLengthsIn
#&gt;         symbols:
#&gt;         - NotEqual
#&gt;         - Equal
#&gt;         type: enum
#&gt;       label: Reads lengths
#&gt;       description: Equal/Not equal - lengths of names, sequences, qualities for both
#&gt;         mates are the same/not the same. "Not equal" is safe in all situations.
#&gt;       streamable: no
#&gt;       id: '#readMatesLengthsIn'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMatesLengthsIn
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: NotEqual
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to map
#&gt;       description: Number of reads to map from the beginning of the file.
#&gt;       streamable: no
#&gt;       id: '#readMapNumber'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMapNumber
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantTranscriptomeBan
#&gt;         symbols:
#&gt;         - IndelSoftclipSingleend
#&gt;         - Singleend
#&gt;         type: enum
#&gt;       label: Prohibit alignment type
#&gt;       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;         prohibit single-end alignments.'
#&gt;       streamable: no
#&gt;       id: '#quantTranscriptomeBan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --quantTranscriptomeBan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantMode
#&gt;         symbols:
#&gt;         - TranscriptomeSAM
#&gt;         - GeneCounts
#&gt;         type: enum
#&gt;       label: Quantification mode
#&gt;       description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'
#&gt;         option, STAR will count number of reads per gene while mapping.
#&gt;       streamable: no
#&gt;       id: '#quantMode'
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSortingType
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - SortedByCoordinate
#&gt;         - Unsorted SortedByCoordinate
#&gt;         type: enum
#&gt;       label: Output sorting type
#&gt;       description: Type of output sorting.
#&gt;       streamable: no
#&gt;       id: '#outSortingType'
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SortedByCoordinate
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSJfilterReads
#&gt;         symbols:
#&gt;         - All
#&gt;         - Unique
#&gt;         type: enum
#&gt;       label: Collapsed junctions reads
#&gt;       description: 'Which reads to consider for collapsed splice junctions output.
#&gt;         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads
#&gt;         only.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterReads'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterReads
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: All
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min overhang SJ
#&gt;       description: Minimum overhang length for splice junctions on both sides for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterOverhangMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 30 12 12 12
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Max gap allowed
#&gt;       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads
#&gt;         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,
#&gt;         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;         Does not apply to annotated junctions.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterIntronMaxVsReadN'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterIntronMaxVsReadN
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 50000 100000 200000
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min distance to other donor/acceptor
#&gt;       description: Minimum allowed distance to other junctions' donor/acceptor for
#&gt;         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterDistToOtherSJmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterDistToOtherSJmin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 10 0 5 10
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min unique count
#&gt;       description: Minimum uniquely mapping read count per junction for each of the
#&gt;         motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountUniqueMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountUniqueMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min total count
#&gt;       description: Minimum total (multi-mapping+unique) read count per junction for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountTotalMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountTotalMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMunmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Write unmapped in SAM
#&gt;       description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;       streamable: no
#&gt;       id: '#outSAMunmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMunmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMtype
#&gt;         symbols:
#&gt;         - SAM
#&gt;         - BAM
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Format of output alignments.
#&gt;       streamable: no
#&gt;       id: '#outSAMtype'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               SAM_type = $job.inputs.outSAMtype
#&gt;               SORT_type = $job.inputs.outSortingType
#&gt;               if (SAM_type &amp;&amp; SORT_type) {
#&gt;                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SAM
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMstrandField
#&gt;         symbols:
#&gt;         - None
#&gt;         - intronMotif
#&gt;         type: enum
#&gt;       label: Strand field flag
#&gt;       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:
#&gt;         strand derived from the intron motif. Reads with inconsistent and/or non-canonical
#&gt;         introns are filtered out.'
#&gt;       streamable: no
#&gt;       id: '#outSAMstrandField'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMstrandField
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMreadID
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - Number
#&gt;         type: enum
#&gt;       label: Read ID
#&gt;       description: 'Read ID record type. Standard: first word (until space) from the
#&gt;         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)
#&gt;         in the FASTx file.'
#&gt;       streamable: no
#&gt;       id: '#outSAMreadID'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMreadID
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMprimaryFlag
#&gt;         symbols:
#&gt;         - OneBestScore
#&gt;         - AllBestScore
#&gt;         type: enum
#&gt;       label: Primary alignments
#&gt;       description: 'Which alignments are considered primary - all others will be marked
#&gt;         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best
#&gt;         score is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;       streamable: no
#&gt;       id: '#outSAMprimaryFlag'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMprimaryFlag
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: OneBestScore
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMorder
#&gt;         symbols:
#&gt;         - Paired
#&gt;         - PairedKeepInputOrder
#&gt;         type: enum
#&gt;       label: Sorting in SAM
#&gt;       description: 'Type of sorting for the SAM output. Paired: one mate after the
#&gt;         other for all paired alignments; PairedKeepInputOrder: one mate after the
#&gt;         other for all paired alignments, the order is kept the same as in the input
#&gt;         FASTQ files.'
#&gt;       streamable: no
#&gt;       id: '#outSAMorder'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMorder
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Paired
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMmode
#&gt;         symbols:
#&gt;         - Full
#&gt;         - NoQS
#&gt;         type: enum
#&gt;       label: SAM mode
#&gt;       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but
#&gt;         without quality scores.'
#&gt;       streamable: no
#&gt;       id: '#outSAMmode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Full
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: MAPQ value
#&gt;       description: MAPQ value for unique mappers (0 to 255).
#&gt;       streamable: no
#&gt;       id: '#outSAMmapqUnique'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmapqUnique
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '255'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @PG
#&gt;       description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderPG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderPG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @HD
#&gt;       description: '@HD (header) line of the SAM header.'
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderHD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderHD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: OR SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagOR'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagOR
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AND SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagAND'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagAND
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '65535'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMattributes
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - NH
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: SAM attributes
#&gt;       description: 'Desired SAM attributes, in the order desired for the output SAM.
#&gt;         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM
#&gt;         NM MD jM jI; None: no attributes.'
#&gt;       streamable: no
#&gt;       id: '#outSAMattributes'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMattributes
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outReadsUnmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Fastx
#&gt;         type: enum
#&gt;       label: Output unmapped reads
#&gt;       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:
#&gt;         output in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;       streamable: no
#&gt;       id: '#outReadsUnmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outReadsUnmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Quality conversion
#&gt;       description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;         to Sanger, use -31).
#&gt;       streamable: no
#&gt;       id: '#outQSconversionAdd'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outQSconversionAdd
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterType
#&gt;         symbols:
#&gt;         - Normal
#&gt;         - BySJout
#&gt;         type: enum
#&gt;       label: Filtering type
#&gt;       description: 'Type of filtering. Normal: standard filtering using only current
#&gt;         alignment; BySJout: keep only those reads that contain junctions that passed
#&gt;         filtering into SJ.out.tab.'
#&gt;       streamable: no
#&gt;       id: '#outFilterType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: Normal
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min score normalized
#&gt;       description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;         for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMinOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMinOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min score
#&gt;       description: Alignment will be output only if its score is higher than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Multimapping score range
#&gt;       description: The score range below the maximum score for multimapping alignments.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapScoreRange'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapScoreRange
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mappings
#&gt;       description: Read alignments will be output only if the read maps fewer than
#&gt;         this value, otherwise no alignments will be output.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *read* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverReadLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverReadLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *mapped* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mismatches
#&gt;       description: Alignment will be output only if it has fewer mismatches than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min matched bases normalized
#&gt;       description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;         lengths for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNminOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNminOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min matched bases
#&gt;       description: Alignment will be output only if the number of matched bases is
#&gt;         higher than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterIntronMotifs
#&gt;         symbols:
#&gt;         - None
#&gt;         - RemoveNoncanonical
#&gt;         - RemoveNoncanonicalUnannotated
#&gt;         type: enum
#&gt;       label: Motifs filtering
#&gt;       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;         filter out alignments that contain non-canonical unannotated junctions when
#&gt;         using annotated splice junctions database. The annotated non-canonical junctions
#&gt;         will be kept.'
#&gt;       streamable: no
#&gt;       id: '#outFilterIntronMotifs'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterIntronMotifs
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max insert junctions
#&gt;       description: Maximum number of junction to be inserted to the genome on the
#&gt;         fly at the mapping stage, including those from annotations and those detected
#&gt;         in the 1st step of the 2-pass run.
#&gt;       streamable: no
#&gt;       id: '#limitSjdbInsertNsj'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitSjdbInsertNsj
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Junctions max number
#&gt;       description: Max number of junctions for one read (including all multi-mappers).
#&gt;       streamable: no
#&gt;       id: '#limitOutSJoneRead'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJoneRead
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Collapsed junctions max number
#&gt;       description: Max number of collapsed junctions.
#&gt;       streamable: no
#&gt;       id: '#limitOutSJcollapsed'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJcollapsed
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Limit BAM sorting memory
#&gt;       description: Maximum available RAM for sorting BAM. If set to 0, it will be
#&gt;         set to the genome index size.
#&gt;       streamable: no
#&gt;       id: '#limitBAMsortRAM'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitBAMsortRAM
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Genome dir name
#&gt;       description: Name of the directory which contains genome files (when genome.tar
#&gt;         is uncompressed).
#&gt;       streamable: no
#&gt;       id: '#genomeDirName'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeDir
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: $job.inputs.genomeDirName || "genomeDir"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:toolDefaultValue: genomeDir
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome files
#&gt;       description: Genome files created using STAR Genome Generate.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: TAR
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 5p bases
#&gt;       description: Number of bases to clip from 5p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip5pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip5pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p bases
#&gt;       description: Number of bases to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p after adapter seq.
#&gt;       description: Number of bases to clip from 3p of each mate after the adapter
#&gt;         clipping. In case only one value is given, it will be assumed the same for
#&gt;         both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAfterAdapterNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAfterAdapterNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: string
#&gt;         type: array
#&gt;       label: Clip 3p adapter sequence
#&gt;       description: Adapter sequence to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterSeq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterSeq
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: float
#&gt;         type: array
#&gt;       label: Max mismatches proportions
#&gt;       description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;         In case only one value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterMMp'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterMMp
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0.1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min segment length
#&gt;       description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;         (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimSegmentMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimSegmentMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '15'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min separation score
#&gt;       description: Minimum difference (separation) between the best chimeric score
#&gt;         and the next one (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreSeparation'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreSeparation
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min total score
#&gt;       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-GT/AG penalty
#&gt;       description: Penalty for a non-GT/AG chimeric junction.
#&gt;       streamable: no
#&gt;       id: '#chimScoreJunctionNonGTAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreJunctionNonGTAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max drop score
#&gt;       description: Max drop (difference) of chimeric score (the sum of scores of all
#&gt;         chimeric segements) from the read length (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreDropMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreDropMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: chimOutType
#&gt;         symbols:
#&gt;         - SeparateSAMold
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Chimeric output type
#&gt;       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;       streamable: no
#&gt;       id: '#chimOutType'
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: SeparateSAMold
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min junction overhang
#&gt;       description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimJunctionOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimJunctionOverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Max windows per read
#&gt;       description: Max number of windows per read (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignWindowsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignWindowsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per window
#&gt;       description: Max number of transcripts per window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per read
#&gt;       description: Max number of different alignments per read to consider (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min mapped length normalized
#&gt;       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLminOverLmate'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLminOverLmate
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min mapped length
#&gt;       description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignSoftClipAtReferenceEnds
#&gt;         symbols:
#&gt;         - 'Yes'
#&gt;         - 'No'
#&gt;         type: enum
#&gt;       label: Soft clipping
#&gt;       description: 'Option which allows soft clipping of alignments at the reference
#&gt;         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.
#&gt;         Yes: Enables soft clipping; No: Disables soft clipping.'
#&gt;       streamable: no
#&gt;       id: '#alignSoftClipAtReferenceEnds'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSoftClipAtReferenceEnds
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: 'Yes'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min overhang
#&gt;       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: 'Min overhang: annotated'
#&gt;       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced
#&gt;         alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJDBoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJDBoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max mates gap
#&gt;       description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;         by (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignMatesGapMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignMatesGapMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min intron size
#&gt;       description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;       streamable: no
#&gt;       id: '#alignIntronMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '21'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max intron size
#&gt;       description: Maximum intron size, if 0, max intron size will be determined by
#&gt;         (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignIntronMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignEndsType
#&gt;         symbols:
#&gt;         - Local
#&gt;         - EndToEnd
#&gt;         type: enum
#&gt;       label: Alignment type
#&gt;       description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;       streamable: no
#&gt;       id: '#alignEndsType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignEndsType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: Local
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Unmapped reads
#&gt;       description: Output of unmapped reads.
#&gt;       streamable: no
#&gt;       id: '#unmapped_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Unmapped.out*'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Transcriptome alignments
#&gt;       description: Alignments translated into transcript coordinates.
#&gt;       streamable: no
#&gt;       id: '#transcriptome_aligned_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Transcriptome*'
#&gt;       sbg:fileTypes: BAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Splice junctions
#&gt;       description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;         Only junctions supported by uniquely mapping reads are reported.
#&gt;       streamable: no
#&gt;       id: '#splice_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*SJ.out.tab'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Reads per gene
#&gt;       description: File with number of reads per gene. A read is counted if it overlaps
#&gt;         (1nt or more) one and only one gene.
#&gt;       streamable: no
#&gt;       id: '#reads_per_gene'
#&gt;       outputBinding:
#&gt;         glob: '*ReadsPerGene*'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Log files
#&gt;       description: Log files produced during alignment.
#&gt;       streamable: no
#&gt;       id: '#log_files'
#&gt;       outputBinding:
#&gt;         glob: '*Log*.out'
#&gt;       sbg:fileTypes: OUT
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Intermediate genome files
#&gt;       description: Archive with genome files produced when annotations are included
#&gt;         on the fly (in the mapping step).
#&gt;       streamable: no
#&gt;       id: '#intermediate_genome'
#&gt;       outputBinding:
#&gt;         glob: '*_STARgenome.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric junctions
#&gt;       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,
#&gt;         'Chimeric Junctions' won't be output.
#&gt;       streamable: no
#&gt;       id: '#chimeric_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*Chimeric.out.junction'
#&gt;       sbg:fileTypes: JUNCTION
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric alignments
#&gt;       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;         Alignment SAM and Chimeric Junctions outputs.
#&gt;       streamable: no
#&gt;       id: '#chimeric_alignments'
#&gt;       outputBinding:
#&gt;         glob: '*.Chimeric.out.sam'
#&gt;       sbg:fileTypes: SAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Aligned SAM/BAM
#&gt;       description: Aligned sequence in SAM/BAM format.
#&gt;       streamable: no
#&gt;       id: '#aligned_reads'
#&gt;       outputBinding:
#&gt;         glob:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;                 sort_name = '.sortedByCoord'
#&gt;               }
#&gt;               else {
#&gt;                 sort_name = ''
#&gt;               }
#&gt;               if ($job.inputs.outSAMtype == 'BAM') {
#&gt;                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;               }
#&gt;               else {
#&gt;                 sam_name = "*.Aligned.out.sam"
#&gt;               }
#&gt;               return sam_name
#&gt;             }
#&gt;           class: Expression
#&gt;       sbg:fileTypes: SAM, BAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     label: STAR
#&gt;     description: STAR is an ultrafast universal RNA-seq aligner. It has very high
#&gt;       mapping speed, accurate alignment of contiguous and spliced reads, detection
#&gt;       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works
#&gt;       with reads starting from lengths ~15 bases up to ~300 bases. In case of having
#&gt;       longer reads, use of STAR Long is recommended.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - tar
#&gt;     - -xvf
#&gt;     - class: Expression
#&gt;       script: $job.inputs.genome.path
#&gt;       engine: '#cwl-js-engine'
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             file = [].concat($job.inputs.reads)[0].path
#&gt;             extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;             if (extension == "gz") {
#&gt;               return "--readFilesCommand zcat"
#&gt;             } else if (extension == "bz2") {
#&gt;               return "--readFilesCommand bzcat"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n
#&gt;           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",
#&gt;           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)
#&gt;           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n
#&gt;           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",
#&gt;           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)
#&gt;           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)
#&gt;           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)
#&gt;           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n
#&gt;           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"
#&gt;           \")\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.twopassMode == "Basic") {
#&gt;               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.chimOutType == "Within") {
#&gt;               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;             }
#&gt;             else {
#&gt;               return "--chimOutType SeparateSAMold"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n
#&gt;           \   if (value == \"\") {\n      return\n    }\n    else {\n      return
#&gt;           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',
#&gt;           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',
#&gt;           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)
#&gt;           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)
#&gt;           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;           \".concat(param_list.join(\" \"))\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;               return "--quantMode ".concat($job.inputs.quantMode)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 100
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n
#&gt;           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile
#&gt;           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       prefix: --outFileNamePrefix
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;         class: Expression
#&gt;     - position: 101
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"
#&gt;           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,
#&gt;           \" \", mate1fq)\n  }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 624.0
#&gt;     'y': 323
#&gt;   sbg:x: 700.0
#&gt;   sbg:y: 200.0
#&gt; sbg:canvas_zoom: 0.6
#&gt; sbg:canvas_y: -16
#&gt; sbg:canvas_x: -41
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>input_id</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "#reads"                          "#sjdbGTFfile"                   
#&gt; [3] "#fastq"                          "#genomeFastaFiles"              
#&gt; [5] "#sjdbGTFtagExonParentTranscript" "#sjdbGTFtagExonParentGene"      
#&gt; [7] "#winAnchorMultimapNmax"          "#winAnchorDistNbins"            </div><div class='input'><span class='co'># batch</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_batch</span><span class='op'>(</span><span class='st'>"sjdbGTFfile"</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"metadata.sample_id"</span>, <span class='st'>"metadata.library_id"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>criteria provided, convert type from ITEM to CRITERIA</span></div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 2
#&gt; sbg:toolAuthor: Seven Bridges Genomics
#&gt; sbg:createdOn: 1463601910
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; - RNA
#&gt; sbg:contributors:
#&gt; - tengfei
#&gt; sbg:project: tengfei/quickstart
#&gt; sbg:createdBy: tengfei
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2
#&gt; sbg:license: Apache License 2.0
#&gt; sbg:revision: 2
#&gt; sbg:modifiedOn: 1463601974
#&gt; sbg:modifiedBy: tengfei
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601910
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601952
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601974
#&gt;   sbg:revision: 2
#&gt; sbg:toolkit: STAR
#&gt; id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'
#&gt; inputs:
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Read sequence
#&gt;   description: Read sequence.
#&gt;   streamable: no
#&gt;   id: '#reads'
#&gt;   inputBinding:
#&gt;     position: 10
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n
#&gt;         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if
#&gt;         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n
#&gt;         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata
#&gt;         != null){\n        if (list[index].metadata.paired_end == 1){\n          left
#&gt;         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n
#&gt;         \         right = list[index].path\n        }\n      }\n    }\n    \n    if
#&gt;         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n
#&gt;         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n
#&gt;         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if
#&gt;         (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end
#&gt;         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n
#&gt;         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=
#&gt;         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n
#&gt;         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"
#&gt;         \"))\n  }\n}"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: sjdbGTFfile
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:x: 160.4999759
#&gt;   sbg:y: 195.0833106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt;   batchType: metadata.library_id
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: fastq
#&gt;   streamable: no
#&gt;   id: '#fastq'
#&gt;   sbg:x: 164.2499914
#&gt;   sbg:y: 323.7499502
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - File
#&gt;   label: genomeFastaFiles
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   sbg:x: 167.7499601
#&gt;   sbg:y: 469.9999106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 350.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 450.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   source: '#unmapped_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   source: '#transcriptome_aligned_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 233.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   source: '#splice_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 266.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   source: '#reads_per_gene'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 300.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   source: '#log_files'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 333.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   source: '#chimeric_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 366.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   source: '#chimeric_alignments'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 433.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Sorted BAM/SAM
#&gt;   description: Sorted BAM or SAM file.
#&gt;   streamable: no
#&gt;   id: '#sorted_bam'
#&gt;   source: '#sorted_bam'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 466.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Result
#&gt;   description: Source FASTQ file with updated metadata.
#&gt;   streamable: no
#&gt;   id: '#result'
#&gt;   source: '#result'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 533.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; requirements:
#&gt; - class: CreateFileRequirement
#&gt;   fileDef: []
#&gt; hints:
#&gt; - class: sbg:AWSInstanceType
#&gt;   value: c3.8xlarge
#&gt; label: RNA-seq Alignment - STAR
#&gt; description: "Alignment to a reference genome and transcriptome presents the first
#&gt;   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner
#&gt;   capable of mapping full length RNA sequences and detecting de novo canonical junctions,
#&gt;   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian
#&gt;   sequence reads, but fine tuning of its parameters enables customization to satisfy
#&gt;   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).
#&gt;   \ \nSplice junction annotations can optionally be collected from splice junction
#&gt;   databases. Set the \"Overhang length\" parameter to a value larger than zero in
#&gt;   order to use splice junction databases. For constant read length, this value should
#&gt;   (ideally) be equal to mate length decreased by 1; for long reads with non-constant
#&gt;   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ
#&gt;   files reveals read length distribution. STAR can detect chimeric transcripts, but
#&gt;   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted
#&gt;   to a desired minimum chimeric segment length. Aligned reads are reported in BAM
#&gt;   format and can be viewed in a genome browser (such as IGV). A file containing detected
#&gt;   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format
#&gt;   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write
#&gt;   unmapped in SAM\" parameters enable unmapped output type selection."
#&gt; class: Workflow
#&gt; steps:
#&gt; - id: '#STAR_Genome_Generate'
#&gt;   inputs:
#&gt;   - id: '#STAR_Genome_Generate.sjdbScore'
#&gt;   - id: '#STAR_Genome_Generate.sjdbOverhang'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'
#&gt;     source: '#sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'
#&gt;     source: '#sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAsparseD'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAindexNbases'
#&gt;   - id: '#STAR_Genome_Generate.genomeFastaFiles'
#&gt;     source: '#genomeFastaFiles'
#&gt;   - id: '#STAR_Genome_Generate.genomeChrBinNbits'
#&gt;   outputs:
#&gt;   - id: '#STAR_Genome_Generate.genome'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 1
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         sjdbScore: 0
#&gt;         sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;         sjdbOverhang: 0
#&gt;         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;         genomeChrBinNbits: genomeChrBinNbits
#&gt;         genomeSAsparseD: 0
#&gt;         sjdbGTFfile:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /demo/test-files/chr20.gtf
#&gt;         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;         genomeFastaFiles:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /sbgenomics/test-data/chr20.fa
#&gt;         sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;         genomeSAindexNbases: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911469
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 1
#&gt;     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa
#&gt;       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD
#&gt;       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt;     sbg:modifiedOn: 1450911470
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911469
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911470
#&gt;       sbg:revision: 1
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbScore
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbOverhang
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentTranscript
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentGene
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFfeatureExon
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFchrPrefix
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Suffux array sparsity
#&gt;       description: 'Distance between indices: use bigger numbers to decrease needed
#&gt;         RAM at the cost of mapping speed reduction (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#genomeSAsparseD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAsparseD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Pre-indexing string length
#&gt;       description: Length (bases) of the SA pre-indexing string. Typically between
#&gt;         10 and 15. Longer strings will use much more memory, but allow faster searches.
#&gt;         For small genomes, this number needs to be scaled down, with a typical value
#&gt;         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,
#&gt;         this is equal to 9, for 100 kiloBase genome, this is equal to 7.
#&gt;       streamable: no
#&gt;       id: '#genomeSAindexNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAindexNbases
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '14'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome fasta files
#&gt;       description: Reference sequence to which to align the reads.
#&gt;       streamable: no
#&gt;       id: '#genomeFastaFiles'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeFastaFiles
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FA
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Bins size
#&gt;       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins
#&gt;         for genome storage: each chromosome will occupy an integer number of bins.
#&gt;         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,
#&gt;         you may need to reduce this number to reduce RAM consumption. The following
#&gt;         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this
#&gt;         is equal to 15.'
#&gt;       streamable: no
#&gt;       id: '#genomeChrBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeChrBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '18'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Genome Files
#&gt;       description: Genome files comprise binary genome sequence, suffix arrays, text
#&gt;         chromosome names/lengths, splice junctions coordinates, and transcripts/genes
#&gt;         information.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       outputBinding:
#&gt;         glob: '*.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     label: STAR Genome Generate
#&gt;     description: STAR Genome Generate is a tool that generates genome index files.
#&gt;       One set of files should be generated per each genome/annotation combination.
#&gt;       Once produced, these files could be used as long as genome/annotation combination
#&gt;       stays the same. Also, STAR Genome Generate which produced these files and STAR
#&gt;       aligner using them must be the same toolkit version.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - mkdir
#&gt;     - genomeDir
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runMode
#&gt;     - genomeGenerate
#&gt;     - --genomeDir
#&gt;     - ./genomeDir
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 99
#&gt;       separate: yes
#&gt;       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 384.0832266
#&gt;     'y': 446.4998957
#&gt;   sbg:x: 100.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#SBG_FASTQ_Quality_Detector'
#&gt;   inputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.fastq'
#&gt;     source: '#fastq'
#&gt;   outputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 3
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 1000
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         fastq:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /path/to/fastq.ext
#&gt;     sbg:toolAuthor: Seven Bridges Genomics
#&gt;     sbg:createdOn: 1450911312
#&gt;     sbg:categories:
#&gt;     - FASTQ-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:project: bix-demo/sbgtools-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     sbg:license: Apache License 2.0
#&gt;     sbg:revision: 3
#&gt;     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext
#&gt;       /path/to/fastq.ext
#&gt;     sbg:modifiedOn: 1450911314
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911312
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911314
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: SBGTools
#&gt;     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     inputs:
#&gt;     - type:
#&gt;       - File
#&gt;       label: Fastq
#&gt;       description: FASTQ file.
#&gt;       streamable: no
#&gt;       id: '#fastq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --fastq
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       required: yes
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Result
#&gt;       description: Source FASTQ file with updated metadata.
#&gt;       streamable: no
#&gt;       id: '#result'
#&gt;       outputBinding:
#&gt;         glob: '*.fastq'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     requirements:
#&gt;     - class: CreateFileRequirement
#&gt;       fileDef: []
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0
#&gt;       dockerImageId: ''
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 1000
#&gt;     label: SBG FASTQ Quality Detector
#&gt;     description: FASTQ Quality Scale Detector detects which quality encoding scheme
#&gt;       was used in your reads and automatically enters the proper value in the "Quality
#&gt;       Scale" metadata field.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - python
#&gt;     - /opt/sbg_fastq_quality_scale_detector.py
#&gt;     arguments: []
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 375.3333179
#&gt;     'y': 323.5833156
#&gt;   sbg:x: 300.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#Picard_SortSam'
#&gt;   inputs:
#&gt;   - id: '#Picard_SortSam.validation_stringency'
#&gt;     default: SILENT
#&gt;   - id: '#Picard_SortSam.sort_order'
#&gt;     default: Coordinate
#&gt;   - id: '#Picard_SortSam.quiet'
#&gt;   - id: '#Picard_SortSam.output_type'
#&gt;   - id: '#Picard_SortSam.memory_per_job'
#&gt;   - id: '#Picard_SortSam.max_records_in_ram'
#&gt;   - id: '#Picard_SortSam.input_bam'
#&gt;     source: '#STAR.aligned_reads'
#&gt;   - id: '#Picard_SortSam.create_index'
#&gt;     default: 'True'
#&gt;   - id: '#Picard_SortSam.compression_level'
#&gt;   outputs:
#&gt;   - id: '#Picard_SortSam.sorted_bam'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 2
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 2048
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         sort_order: Coordinate
#&gt;         input_bam:
#&gt;           path: /root/dir/example.tested.bam
#&gt;         memory_per_job: 2048
#&gt;         output_type: ~
#&gt;         create_index: ~
#&gt;     sbg:toolAuthor: Broad Institute
#&gt;     sbg:createdOn: 1450911168
#&gt;     sbg:categories:
#&gt;     - SAM/BAM-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: http://broadinstitute.github.io/picard/index.html
#&gt;       label: Homepage
#&gt;     - id: https://github.com/broadinstitute/picard/releases/tag/1.138
#&gt;       label: Source Code
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Wiki
#&gt;     - id: https://github.com/broadinstitute/picard/zipball/master
#&gt;       label: Download
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/picard-1-140-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: '1.140'
#&gt;     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     sbg:license: MIT License, Apache 2.0 Licence
#&gt;     sbg:revision: 2
#&gt;     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam
#&gt;       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam
#&gt;       SORT_ORDER=coordinate  /root/dir/example.tested.bam
#&gt;     sbg:modifiedOn: 1450911170
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911168
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911169
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911170
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: Picard
#&gt;     id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: validation_stringency
#&gt;         symbols:
#&gt;         - STRICT
#&gt;         - LENIENT
#&gt;         - SILENT
#&gt;         type: enum
#&gt;       label: Validation stringency
#&gt;       description: 'Validation stringency for all SAM files read by this program.
#&gt;         Setting stringency to SILENT can improve performance when processing a BAM
#&gt;         file in which variable-length data (read, qualities, tags) do not otherwise
#&gt;         need to be decoded. This option can be set to ''null'' to clear the default
#&gt;         value. Possible values: {STRICT, LENIENT, SILENT}.'
#&gt;       streamable: no
#&gt;       id: '#validation_stringency'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: VALIDATION_STRINGENCY=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.validation_stringency)
#&gt;               {
#&gt;                 return $job.inputs.validation_stringency
#&gt;               }
#&gt;               else
#&gt;               {
#&gt;                 return "SILENT"
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SILENT
#&gt;       required: no
#&gt;     - type:
#&gt;       - name: sort_order
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - Queryname
#&gt;         - Coordinate
#&gt;         type: enum
#&gt;       label: Sort order
#&gt;       description: 'Sort order of the output file. Possible values: {unsorted, queryname,
#&gt;         coordinate}.'
#&gt;       streamable: no
#&gt;       id: '#sort_order'
#&gt;       inputBinding:
#&gt;         position: 3
#&gt;         prefix: SORT_ORDER=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               p = $job.inputs.sort_order.toLowerCase()
#&gt;               return p
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: Coordinate
#&gt;       sbg:altPrefix: SO
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quiet
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Quiet
#&gt;       description: 'This parameter indicates whether to suppress job-summary info
#&gt;         on System.err. This option can be set to ''null'' to clear the default value.
#&gt;         Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#quiet'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: QUIET=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: output_type
#&gt;         symbols:
#&gt;         - BAM
#&gt;         - SAM
#&gt;         - SAME AS INPUT
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Since Picard tools can output both SAM and BAM files, user can
#&gt;         choose the format of the output file.
#&gt;       streamable: no
#&gt;       id: '#output_type'
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SAME AS INPUT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Memory per job
#&gt;       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for
#&gt;         single threaded jobs.
#&gt;       streamable: no
#&gt;       id: '#memory_per_job'
#&gt;       sbg:toolDefaultValue: '2048'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max records in RAM
#&gt;       description: When writing SAM files that need to be sorted, this parameter will
#&gt;         specify the number of records stored in RAM before spilling to disk. Increasing
#&gt;         this number reduces the number of file handles needed to sort a SAM file,
#&gt;         and increases the amount of RAM needed. This option can be set to 'null' to
#&gt;         clear the default value.
#&gt;       streamable: no
#&gt;       id: '#max_records_in_ram'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: MAX_RECORDS_IN_RAM=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '500000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Input BAM
#&gt;       description: The BAM or SAM file to sort.
#&gt;       streamable: no
#&gt;       id: '#input_bam'
#&gt;       inputBinding:
#&gt;         position: 1
#&gt;         prefix: INPUT=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: File inputs
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;       sbg:altPrefix: I
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: create_index
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Create index
#&gt;       description: 'This parameter indicates whether to create a BAM index when writing
#&gt;         a coordinate-sorted BAM file. This option can be set to ''null'' to clear
#&gt;         the default value. Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#create_index'
#&gt;       inputBinding:
#&gt;         position: 5
#&gt;         prefix: CREATE_INDEX=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Compression level
#&gt;       description: Compression level for all compressed files created (e.g. BAM and
#&gt;         GELI). This option can be set to 'null' to clear the default value.
#&gt;       streamable: no
#&gt;       id: '#compression_level'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: COMPRESSION_LEVEL=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Sorted BAM/SAM
#&gt;       description: Sorted BAM or SAM file.
#&gt;       streamable: no
#&gt;       id: '#sorted_bam'
#&gt;       outputBinding:
#&gt;         glob: '*.sorted.?am'
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;       engineCommand: cwl-engine.js
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
#&gt;       dockerImageId: eab0e70b6629
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n
#&gt;           \ }\n  \treturn 2048\n}"
#&gt;         class: Expression
#&gt;     label: Picard SortSam
#&gt;     description: Picard SortSam sorts the input SAM or BAM. Input and output formats
#&gt;       are determined by the file extension.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - java
#&gt;     - class: Expression
#&gt;       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,
#&gt;         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"
#&gt;       engine: '#cwl-js-engine'
#&gt;     - -jar
#&gt;     - /opt/picard-tools-1.140/picard.jar
#&gt;     - SortSam
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       prefix: OUTPUT=
#&gt;       separate: no
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif
#&gt;           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,
#&gt;           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse
#&gt;           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n}\n}"
#&gt;         class: Expression
#&gt;     - position: 1000
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out
#&gt;           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n
#&gt;           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n
#&gt;           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n
#&gt;           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index
#&gt;           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==
#&gt;           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 773.0831807
#&gt;     'y': 470.9165939
#&gt;   sbg:x: 500.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#STAR'
#&gt;   inputs:
#&gt;   - id: '#STAR.winFlankNbins'
#&gt;   - id: '#STAR.winBinNbits'
#&gt;   - id: '#STAR.winAnchorMultimapNmax'
#&gt;     source: '#winAnchorMultimapNmax'
#&gt;   - id: '#STAR.winAnchorDistNbins'
#&gt;     source: '#winAnchorDistNbins'
#&gt;   - id: '#STAR.twopassMode'
#&gt;   - id: '#STAR.twopass1readsN'
#&gt;   - id: '#STAR.sjdbScore'
#&gt;   - id: '#STAR.sjdbOverhang'
#&gt;     default: 100
#&gt;   - id: '#STAR.sjdbInsertSave'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR.seedSearchStartLmaxOverLread'
#&gt;   - id: '#STAR.seedSearchStartLmax'
#&gt;   - id: '#STAR.seedSearchLmax'
#&gt;   - id: '#STAR.seedPerWindowNmax'
#&gt;   - id: '#STAR.seedPerReadNmax'
#&gt;   - id: '#STAR.seedNoneLociPerWindow'
#&gt;   - id: '#STAR.seedMultimapNmax'
#&gt;   - id: '#STAR.scoreStitchSJshift'
#&gt;   - id: '#STAR.scoreInsOpen'
#&gt;   - id: '#STAR.scoreInsBase'
#&gt;   - id: '#STAR.scoreGenomicLengthLog2scale'
#&gt;   - id: '#STAR.scoreGapNoncan'
#&gt;   - id: '#STAR.scoreGapGCAG'
#&gt;   - id: '#STAR.scoreGapATAC'
#&gt;   - id: '#STAR.scoreGap'
#&gt;   - id: '#STAR.scoreDelOpen'
#&gt;   - id: '#STAR.scoreDelBase'
#&gt;   - id: '#STAR.rg_seq_center'
#&gt;   - id: '#STAR.rg_sample_id'
#&gt;   - id: '#STAR.rg_platform_unit_id'
#&gt;   - id: '#STAR.rg_platform'
#&gt;   - id: '#STAR.rg_mfl'
#&gt;   - id: '#STAR.rg_library_id'
#&gt;   - id: '#STAR.reads'
#&gt;     source: '#reads'
#&gt;   - id: '#STAR.readMatesLengthsIn'
#&gt;   - id: '#STAR.readMapNumber'
#&gt;   - id: '#STAR.quantTranscriptomeBan'
#&gt;   - id: '#STAR.quantMode'
#&gt;     default: TranscriptomeSAM
#&gt;   - id: '#STAR.outSortingType'
#&gt;     default: SortedByCoordinate
#&gt;   - id: '#STAR.outSJfilterReads'
#&gt;   - id: '#STAR.outSJfilterOverhangMin'
#&gt;   - id: '#STAR.outSJfilterIntronMaxVsReadN'
#&gt;   - id: '#STAR.outSJfilterDistToOtherSJmin'
#&gt;   - id: '#STAR.outSJfilterCountUniqueMin'
#&gt;   - id: '#STAR.outSJfilterCountTotalMin'
#&gt;   - id: '#STAR.outSAMunmapped'
#&gt;   - id: '#STAR.outSAMtype'
#&gt;     default: BAM
#&gt;   - id: '#STAR.outSAMstrandField'
#&gt;   - id: '#STAR.outSAMreadID'
#&gt;   - id: '#STAR.outSAMprimaryFlag'
#&gt;   - id: '#STAR.outSAMorder'
#&gt;   - id: '#STAR.outSAMmode'
#&gt;   - id: '#STAR.outSAMmapqUnique'
#&gt;   - id: '#STAR.outSAMheaderPG'
#&gt;   - id: '#STAR.outSAMheaderHD'
#&gt;   - id: '#STAR.outSAMflagOR'
#&gt;   - id: '#STAR.outSAMflagAND'
#&gt;   - id: '#STAR.outSAMattributes'
#&gt;   - id: '#STAR.outReadsUnmapped'
#&gt;     default: Fastx
#&gt;   - id: '#STAR.outQSconversionAdd'
#&gt;   - id: '#STAR.outFilterType'
#&gt;   - id: '#STAR.outFilterScoreMinOverLread'
#&gt;   - id: '#STAR.outFilterScoreMin'
#&gt;   - id: '#STAR.outFilterMultimapScoreRange'
#&gt;   - id: '#STAR.outFilterMultimapNmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverReadLmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverLmax'
#&gt;   - id: '#STAR.outFilterMismatchNmax'
#&gt;   - id: '#STAR.outFilterMatchNminOverLread'
#&gt;   - id: '#STAR.outFilterMatchNmin'
#&gt;   - id: '#STAR.outFilterIntronMotifs'
#&gt;   - id: '#STAR.limitSjdbInsertNsj'
#&gt;   - id: '#STAR.limitOutSJoneRead'
#&gt;   - id: '#STAR.limitOutSJcollapsed'
#&gt;   - id: '#STAR.limitBAMsortRAM'
#&gt;   - id: '#STAR.genomeDirName'
#&gt;   - id: '#STAR.genome'
#&gt;     source: '#STAR_Genome_Generate.genome'
#&gt;   - id: '#STAR.clip5pNbases'
#&gt;   - id: '#STAR.clip3pNbases'
#&gt;   - id: '#STAR.clip3pAfterAdapterNbases'
#&gt;   - id: '#STAR.clip3pAdapterSeq'
#&gt;   - id: '#STAR.clip3pAdapterMMp'
#&gt;   - id: '#STAR.chimSegmentMin'
#&gt;   - id: '#STAR.chimScoreSeparation'
#&gt;   - id: '#STAR.chimScoreMin'
#&gt;   - id: '#STAR.chimScoreJunctionNonGTAG'
#&gt;   - id: '#STAR.chimScoreDropMax'
#&gt;   - id: '#STAR.chimOutType'
#&gt;   - id: '#STAR.chimJunctionOverhangMin'
#&gt;   - id: '#STAR.alignWindowsPerReadNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerWindowNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerReadNmax'
#&gt;   - id: '#STAR.alignSplicedMateMapLminOverLmate'
#&gt;   - id: '#STAR.alignSplicedMateMapLmin'
#&gt;   - id: '#STAR.alignSoftClipAtReferenceEnds'
#&gt;   - id: '#STAR.alignSJoverhangMin'
#&gt;   - id: '#STAR.alignSJDBoverhangMin'
#&gt;   - id: '#STAR.alignMatesGapMax'
#&gt;   - id: '#STAR.alignIntronMin'
#&gt;   - id: '#STAR.alignIntronMax'
#&gt;   - id: '#STAR.alignEndsType'
#&gt;   outputs:
#&gt;   - id: '#STAR.unmapped_reads'
#&gt;   - id: '#STAR.transcriptome_aligned_reads'
#&gt;   - id: '#STAR.splice_junctions'
#&gt;   - id: '#STAR.reads_per_gene'
#&gt;   - id: '#STAR.log_files'
#&gt;   - id: '#STAR.intermediate_genome'
#&gt;   - id: '#STAR.chimeric_junctions'
#&gt;   - id: '#STAR.chimeric_alignments'
#&gt;   - id: '#STAR.aligned_reads'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 4
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         alignWindowsPerReadNmax: 0
#&gt;         outSAMheaderPG: outSAMheaderPG
#&gt;         GENOME_DIR_NAME: ''
#&gt;         outFilterMatchNminOverLread: 0
#&gt;         rg_platform_unit_id: rg_platform_unit
#&gt;         alignTranscriptsPerReadNmax: 0
#&gt;         readMapNumber: 0
#&gt;         alignSplicedMateMapLminOverLmate: 0
#&gt;         alignMatesGapMax: 0
#&gt;         outFilterMultimapNmax: 0
#&gt;         clip5pNbases:
#&gt;         - 0
#&gt;         outSAMstrandField: None
#&gt;         readMatesLengthsIn: NotEqual
#&gt;         outSAMattributes: Standard
#&gt;         seedMultimapNmax: 0
#&gt;         rg_mfl: rg_mfl
#&gt;         chimSegmentMin: 0
#&gt;         winAnchorDistNbins: 0
#&gt;         outSortingType: SortedByCoordinate
#&gt;         outFilterMultimapScoreRange: 0
#&gt;         sjdbInsertSave: Basic
#&gt;         clip3pAfterAdapterNbases:
#&gt;         - 0
#&gt;         scoreDelBase: 0
#&gt;         outFilterMatchNmin: 0
#&gt;         twopass1readsN: 0
#&gt;         outSAMunmapped: None
#&gt;         genome:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: genome.ext
#&gt;         sjdbGTFtagExonParentTranscript: ''
#&gt;         limitBAMsortRAM: 0
#&gt;         alignEndsType: Local
#&gt;         seedNoneLociPerWindow: 0
#&gt;         rg_sample_id: rg_sample
#&gt;         sjdbGTFtagExonParentGene: ''
#&gt;         chimScoreMin: 0
#&gt;         outSJfilterIntronMaxVsReadN:
#&gt;         - 0
#&gt;         twopassMode: Basic
#&gt;         alignSplicedMateMapLmin: 0
#&gt;         outSJfilterReads: All
#&gt;         outSAMprimaryFlag: OneBestScore
#&gt;         outSJfilterCountTotalMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         outSAMorder: Paired
#&gt;         outSAMflagAND: 0
#&gt;         chimScoreSeparation: 0
#&gt;         alignSJoverhangMin: 0
#&gt;         outFilterScoreMin: 0
#&gt;         seedSearchStartLmax: 0
#&gt;         scoreGapGCAG: 0
#&gt;         scoreGenomicLengthLog2scale: 0
#&gt;         outFilterIntronMotifs: None
#&gt;         outFilterMismatchNmax: 0
#&gt;         reads:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           metadata:
#&gt;             format: fastq
#&gt;             paired_end: '1'
#&gt;             seq_center: illumina
#&gt;           path: /test-data/mate_1.fastq.bz2
#&gt;         scoreGap: 0
#&gt;         outSJfilterOverhangMin:
#&gt;         - 30
#&gt;         - 12
#&gt;         - 12
#&gt;         - 12
#&gt;         outSAMflagOR: 0
#&gt;         outSAMmode: Full
#&gt;         rg_library_id: ''
#&gt;         chimScoreJunctionNonGTAG: 0
#&gt;         scoreInsOpen: 0
#&gt;         clip3pAdapterSeq:
#&gt;         - clip3pAdapterSeq
#&gt;         chimScoreDropMax: 0
#&gt;         outFilterType: Normal
#&gt;         scoreGapATAC: 0
#&gt;         rg_platform: Ion Torrent PGM
#&gt;         clip3pAdapterMMp:
#&gt;         - 0
#&gt;         sjdbGTFfeatureExon: ''
#&gt;         outQSconversionAdd: 0
#&gt;         quantMode: TranscriptomeSAM
#&gt;         alignIntronMin: 0
#&gt;         scoreInsBase: 0
#&gt;         scoreGapNoncan: 0
#&gt;         seedSearchLmax: 0
#&gt;         outSJfilterDistToOtherSJmin:
#&gt;         - 0
#&gt;         outFilterScoreMinOverLread: 0
#&gt;         alignSJDBoverhangMin: 0
#&gt;         limitOutSJcollapsed: 0
#&gt;         winAnchorMultimapNmax: 0
#&gt;         outFilterMismatchNoverLmax: 0
#&gt;         rg_seq_center: ''
#&gt;         outSAMheaderHD: outSAMheaderHD
#&gt;         chimOutType: Within
#&gt;         quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;         limitOutSJoneRead: 0
#&gt;         alignTranscriptsPerWindowNmax: 0
#&gt;         sjdbOverhang: ~
#&gt;         outReadsUnmapped: Fastx
#&gt;         scoreStitchSJshift: 0
#&gt;         seedPerWindowNmax: 0
#&gt;         outSJfilterCountUniqueMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         scoreDelOpen: 0
#&gt;         sjdbGTFfile:
#&gt;         - path: /demo/test-data/chr20.gtf
#&gt;         clip3pNbases:
#&gt;         - 0
#&gt;         - 3
#&gt;         winBinNbits: 0
#&gt;         sjdbScore: ~
#&gt;         seedSearchStartLmaxOverLread: 0
#&gt;         alignIntronMax: 0
#&gt;         seedPerReadNmax: 0
#&gt;         outFilterMismatchNoverReadLmax: 0
#&gt;         winFlankNbins: 0
#&gt;         sjdbGTFchrPrefix: chrPrefix
#&gt;         alignSoftClipAtReferenceEnds: 'Yes'
#&gt;         outSAMreadID: Standard
#&gt;         outSAMtype: BAM
#&gt;         chimJunctionOverhangMin: 0
#&gt;         limitSjdbInsertNsj: 0
#&gt;         outSAMmapqUnique: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911471
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - ana_d
#&gt;     - bix-demo
#&gt;     - uros_sipetic
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star/4
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 4
#&gt;     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix
#&gt;       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline
#&gt;       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode
#&gt;       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;
#&gt;       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt;     sbg:modifiedOn: 1462889222
#&gt;     sbg:modifiedBy: ana_d
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911471
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911473
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911475
#&gt;       sbg:revision: 2
#&gt;     - sbg:modifiedBy: uros_sipetic
#&gt;       sbg:modifiedOn: 1462878528
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: ana_d
#&gt;       sbg:modifiedOn: 1462889222
#&gt;       sbg:revision: 4
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star/4
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Flanking regions size
#&gt;       description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;         flanking regions for each window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winFlankNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winFlankNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Bin size
#&gt;       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;         each window will occupy an integer number of bins (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '16'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max loci anchors
#&gt;       description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max bins between anchors
#&gt;       description: Max number of bins between two anchors that allows aggregation
#&gt;         of anchors into one window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorDistNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorDistNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '9'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: twopassMode
#&gt;         symbols:
#&gt;         - None
#&gt;         - Basic
#&gt;         type: enum
#&gt;       label: Two-pass mode
#&gt;       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass
#&gt;         mapping, with all 1st pass junctions inserted into the genome indices on the
#&gt;         fly.'
#&gt;       streamable: no
#&gt;       id: '#twopassMode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --twopassMode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to process in 1st step
#&gt;       description: 'Number of reads to process for the 1st step. 0: 1-step only, no
#&gt;         2nd pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#twopass1readsN'
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: sjdbInsertSave
#&gt;         symbols:
#&gt;         - Basic
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: Save junction files
#&gt;       description: 'Which files to save when sjdb junctions are inserted on the fly
#&gt;         at the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;         files; All: all files including big Genome, SA and SAindex. These files are
#&gt;         output as archive.'
#&gt;       streamable: no
#&gt;       id: '#sjdbInsertSave'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts. No need to include
#&gt;         this input, except in case of using "on the fly" annotations.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Search start point normalized
#&gt;       description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;         for paired-end reads).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmaxOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmaxOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1.0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Search start point
#&gt;       description: Defines the search start point through the read - the read is split
#&gt;         into pieces no longer than this value (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seed length
#&gt;       description: Defines the maximum length of the seeds, if =0 max seed length
#&gt;         is infinite (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per window
#&gt;       description: Max number of seeds per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per read
#&gt;       description: Max number of seeds per read (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max one-seed loci per window
#&gt;       description: Max number of one seed loci per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedNoneLociPerWindow'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedNoneLociPerWindow
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Filter pieces for stitching
#&gt;       description: Only pieces that map fewer than this value are utilized in the
#&gt;         stitching procedure (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max score reduction
#&gt;       description: Maximum score reduction while searching for SJ boundaries in the
#&gt;         stitching step.
#&gt;       streamable: no
#&gt;       id: '#scoreStitchSJshift'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreStitchSJshift
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion Open Penalty
#&gt;       description: Insertion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreInsOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion extension penalty
#&gt;       description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreInsBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Log scaled score
#&gt;       description: 'Extra score logarithmically scaled with genomic length of the
#&gt;         alignment: &lt;int&gt;*log2(genomicLength).'
#&gt;       streamable: no
#&gt;       id: '#scoreGenomicLengthLog2scale'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGenomicLengthLog2scale
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-0.25'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-canonical gap open
#&gt;       description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapNoncan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapNoncan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: GC/AG and CT/GC gap open
#&gt;       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapGCAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapGCAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AT/AC and GT/AT gap open
#&gt;       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapATAC'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapATAC
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Gap open penalty
#&gt;       description: Gap open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreGap'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGap
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion open penalty
#&gt;       description: Deletion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreDelOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion extension penalty
#&gt;       description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreDelBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sequencing center
#&gt;       description: Specify the sequencing center for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_seq_center'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sample ID
#&gt;       description: Specify the sample ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_sample_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Platform unit ID
#&gt;       description: Specify the platform unit ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_platform_unit_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: rg_platform
#&gt;         symbols:
#&gt;         - LS 454
#&gt;         - Helicos
#&gt;         - Illumina
#&gt;         - ABI SOLiD
#&gt;         - Ion Torrent PGM
#&gt;         - PacBio
#&gt;         type: enum
#&gt;       label: Platform
#&gt;       description: Specify the version of the technology that was used for sequencing
#&gt;         or assaying.
#&gt;       streamable: no
#&gt;       id: '#rg_platform'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Median fragment length
#&gt;       description: Specify the median fragment length for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_mfl'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Library ID
#&gt;       description: Specify the library ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_library_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Read sequence
#&gt;       description: Read sequence.
#&gt;       streamable: no
#&gt;       id: '#reads'
#&gt;       inputBinding:
#&gt;         position: 10
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp
#&gt;             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n
#&gt;             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =
#&gt;             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n
#&gt;             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end
#&gt;             == 2){\n          right = list[index].path\n        }\n      }\n    }\n
#&gt;             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n
#&gt;             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left
#&gt;             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;
#&gt;             ++index) {\n      \n      if (list[index].metadata != null){\n        if
#&gt;             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n
#&gt;             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n
#&gt;             \       }\n      }\n    }\n    left_join = left.join()\n    right_join
#&gt;             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n
#&gt;             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){
#&gt;             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;       sbg:x: 200.0
#&gt;       sbg:y: 200.0
#&gt;       sbg:includeInPorts: yes
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: readMatesLengthsIn
#&gt;         symbols:
#&gt;         - NotEqual
#&gt;         - Equal
#&gt;         type: enum
#&gt;       label: Reads lengths
#&gt;       description: Equal/Not equal - lengths of names, sequences, qualities for both
#&gt;         mates are the same/not the same. "Not equal" is safe in all situations.
#&gt;       streamable: no
#&gt;       id: '#readMatesLengthsIn'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMatesLengthsIn
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: NotEqual
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to map
#&gt;       description: Number of reads to map from the beginning of the file.
#&gt;       streamable: no
#&gt;       id: '#readMapNumber'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMapNumber
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantTranscriptomeBan
#&gt;         symbols:
#&gt;         - IndelSoftclipSingleend
#&gt;         - Singleend
#&gt;         type: enum
#&gt;       label: Prohibit alignment type
#&gt;       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;         prohibit single-end alignments.'
#&gt;       streamable: no
#&gt;       id: '#quantTranscriptomeBan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --quantTranscriptomeBan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantMode
#&gt;         symbols:
#&gt;         - TranscriptomeSAM
#&gt;         - GeneCounts
#&gt;         type: enum
#&gt;       label: Quantification mode
#&gt;       description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'
#&gt;         option, STAR will count number of reads per gene while mapping.
#&gt;       streamable: no
#&gt;       id: '#quantMode'
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSortingType
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - SortedByCoordinate
#&gt;         - Unsorted SortedByCoordinate
#&gt;         type: enum
#&gt;       label: Output sorting type
#&gt;       description: Type of output sorting.
#&gt;       streamable: no
#&gt;       id: '#outSortingType'
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SortedByCoordinate
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSJfilterReads
#&gt;         symbols:
#&gt;         - All
#&gt;         - Unique
#&gt;         type: enum
#&gt;       label: Collapsed junctions reads
#&gt;       description: 'Which reads to consider for collapsed splice junctions output.
#&gt;         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads
#&gt;         only.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterReads'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterReads
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: All
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min overhang SJ
#&gt;       description: Minimum overhang length for splice junctions on both sides for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterOverhangMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 30 12 12 12
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Max gap allowed
#&gt;       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads
#&gt;         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,
#&gt;         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;         Does not apply to annotated junctions.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterIntronMaxVsReadN'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterIntronMaxVsReadN
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 50000 100000 200000
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min distance to other donor/acceptor
#&gt;       description: Minimum allowed distance to other junctions' donor/acceptor for
#&gt;         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterDistToOtherSJmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterDistToOtherSJmin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 10 0 5 10
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min unique count
#&gt;       description: Minimum uniquely mapping read count per junction for each of the
#&gt;         motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountUniqueMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountUniqueMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min total count
#&gt;       description: Minimum total (multi-mapping+unique) read count per junction for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountTotalMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountTotalMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMunmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Write unmapped in SAM
#&gt;       description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;       streamable: no
#&gt;       id: '#outSAMunmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMunmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMtype
#&gt;         symbols:
#&gt;         - SAM
#&gt;         - BAM
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Format of output alignments.
#&gt;       streamable: no
#&gt;       id: '#outSAMtype'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               SAM_type = $job.inputs.outSAMtype
#&gt;               SORT_type = $job.inputs.outSortingType
#&gt;               if (SAM_type &amp;&amp; SORT_type) {
#&gt;                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SAM
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMstrandField
#&gt;         symbols:
#&gt;         - None
#&gt;         - intronMotif
#&gt;         type: enum
#&gt;       label: Strand field flag
#&gt;       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:
#&gt;         strand derived from the intron motif. Reads with inconsistent and/or non-canonical
#&gt;         introns are filtered out.'
#&gt;       streamable: no
#&gt;       id: '#outSAMstrandField'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMstrandField
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMreadID
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - Number
#&gt;         type: enum
#&gt;       label: Read ID
#&gt;       description: 'Read ID record type. Standard: first word (until space) from the
#&gt;         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)
#&gt;         in the FASTx file.'
#&gt;       streamable: no
#&gt;       id: '#outSAMreadID'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMreadID
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMprimaryFlag
#&gt;         symbols:
#&gt;         - OneBestScore
#&gt;         - AllBestScore
#&gt;         type: enum
#&gt;       label: Primary alignments
#&gt;       description: 'Which alignments are considered primary - all others will be marked
#&gt;         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best
#&gt;         score is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;       streamable: no
#&gt;       id: '#outSAMprimaryFlag'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMprimaryFlag
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: OneBestScore
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMorder
#&gt;         symbols:
#&gt;         - Paired
#&gt;         - PairedKeepInputOrder
#&gt;         type: enum
#&gt;       label: Sorting in SAM
#&gt;       description: 'Type of sorting for the SAM output. Paired: one mate after the
#&gt;         other for all paired alignments; PairedKeepInputOrder: one mate after the
#&gt;         other for all paired alignments, the order is kept the same as in the input
#&gt;         FASTQ files.'
#&gt;       streamable: no
#&gt;       id: '#outSAMorder'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMorder
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Paired
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMmode
#&gt;         symbols:
#&gt;         - Full
#&gt;         - NoQS
#&gt;         type: enum
#&gt;       label: SAM mode
#&gt;       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but
#&gt;         without quality scores.'
#&gt;       streamable: no
#&gt;       id: '#outSAMmode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Full
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: MAPQ value
#&gt;       description: MAPQ value for unique mappers (0 to 255).
#&gt;       streamable: no
#&gt;       id: '#outSAMmapqUnique'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmapqUnique
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '255'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @PG
#&gt;       description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderPG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderPG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @HD
#&gt;       description: '@HD (header) line of the SAM header.'
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderHD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderHD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: OR SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagOR'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagOR
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AND SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagAND'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagAND
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '65535'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMattributes
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - NH
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: SAM attributes
#&gt;       description: 'Desired SAM attributes, in the order desired for the output SAM.
#&gt;         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM
#&gt;         NM MD jM jI; None: no attributes.'
#&gt;       streamable: no
#&gt;       id: '#outSAMattributes'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMattributes
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outReadsUnmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Fastx
#&gt;         type: enum
#&gt;       label: Output unmapped reads
#&gt;       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:
#&gt;         output in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;       streamable: no
#&gt;       id: '#outReadsUnmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outReadsUnmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Quality conversion
#&gt;       description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;         to Sanger, use -31).
#&gt;       streamable: no
#&gt;       id: '#outQSconversionAdd'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outQSconversionAdd
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterType
#&gt;         symbols:
#&gt;         - Normal
#&gt;         - BySJout
#&gt;         type: enum
#&gt;       label: Filtering type
#&gt;       description: 'Type of filtering. Normal: standard filtering using only current
#&gt;         alignment; BySJout: keep only those reads that contain junctions that passed
#&gt;         filtering into SJ.out.tab.'
#&gt;       streamable: no
#&gt;       id: '#outFilterType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: Normal
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min score normalized
#&gt;       description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;         for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMinOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMinOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min score
#&gt;       description: Alignment will be output only if its score is higher than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Multimapping score range
#&gt;       description: The score range below the maximum score for multimapping alignments.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapScoreRange'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapScoreRange
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mappings
#&gt;       description: Read alignments will be output only if the read maps fewer than
#&gt;         this value, otherwise no alignments will be output.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *read* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverReadLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverReadLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *mapped* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mismatches
#&gt;       description: Alignment will be output only if it has fewer mismatches than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min matched bases normalized
#&gt;       description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;         lengths for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNminOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNminOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min matched bases
#&gt;       description: Alignment will be output only if the number of matched bases is
#&gt;         higher than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterIntronMotifs
#&gt;         symbols:
#&gt;         - None
#&gt;         - RemoveNoncanonical
#&gt;         - RemoveNoncanonicalUnannotated
#&gt;         type: enum
#&gt;       label: Motifs filtering
#&gt;       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;         filter out alignments that contain non-canonical unannotated junctions when
#&gt;         using annotated splice junctions database. The annotated non-canonical junctions
#&gt;         will be kept.'
#&gt;       streamable: no
#&gt;       id: '#outFilterIntronMotifs'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterIntronMotifs
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max insert junctions
#&gt;       description: Maximum number of junction to be inserted to the genome on the
#&gt;         fly at the mapping stage, including those from annotations and those detected
#&gt;         in the 1st step of the 2-pass run.
#&gt;       streamable: no
#&gt;       id: '#limitSjdbInsertNsj'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitSjdbInsertNsj
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Junctions max number
#&gt;       description: Max number of junctions for one read (including all multi-mappers).
#&gt;       streamable: no
#&gt;       id: '#limitOutSJoneRead'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJoneRead
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Collapsed junctions max number
#&gt;       description: Max number of collapsed junctions.
#&gt;       streamable: no
#&gt;       id: '#limitOutSJcollapsed'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJcollapsed
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Limit BAM sorting memory
#&gt;       description: Maximum available RAM for sorting BAM. If set to 0, it will be
#&gt;         set to the genome index size.
#&gt;       streamable: no
#&gt;       id: '#limitBAMsortRAM'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitBAMsortRAM
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Genome dir name
#&gt;       description: Name of the directory which contains genome files (when genome.tar
#&gt;         is uncompressed).
#&gt;       streamable: no
#&gt;       id: '#genomeDirName'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeDir
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: $job.inputs.genomeDirName || "genomeDir"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:toolDefaultValue: genomeDir
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome files
#&gt;       description: Genome files created using STAR Genome Generate.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: TAR
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 5p bases
#&gt;       description: Number of bases to clip from 5p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip5pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip5pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p bases
#&gt;       description: Number of bases to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p after adapter seq.
#&gt;       description: Number of bases to clip from 3p of each mate after the adapter
#&gt;         clipping. In case only one value is given, it will be assumed the same for
#&gt;         both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAfterAdapterNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAfterAdapterNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: string
#&gt;         type: array
#&gt;       label: Clip 3p adapter sequence
#&gt;       description: Adapter sequence to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterSeq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterSeq
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: float
#&gt;         type: array
#&gt;       label: Max mismatches proportions
#&gt;       description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;         In case only one value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterMMp'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterMMp
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0.1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min segment length
#&gt;       description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;         (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimSegmentMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimSegmentMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '15'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min separation score
#&gt;       description: Minimum difference (separation) between the best chimeric score
#&gt;         and the next one (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreSeparation'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreSeparation
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min total score
#&gt;       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-GT/AG penalty
#&gt;       description: Penalty for a non-GT/AG chimeric junction.
#&gt;       streamable: no
#&gt;       id: '#chimScoreJunctionNonGTAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreJunctionNonGTAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max drop score
#&gt;       description: Max drop (difference) of chimeric score (the sum of scores of all
#&gt;         chimeric segements) from the read length (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreDropMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreDropMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: chimOutType
#&gt;         symbols:
#&gt;         - SeparateSAMold
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Chimeric output type
#&gt;       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;       streamable: no
#&gt;       id: '#chimOutType'
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: SeparateSAMold
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min junction overhang
#&gt;       description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimJunctionOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimJunctionOverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Max windows per read
#&gt;       description: Max number of windows per read (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignWindowsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignWindowsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per window
#&gt;       description: Max number of transcripts per window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per read
#&gt;       description: Max number of different alignments per read to consider (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min mapped length normalized
#&gt;       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLminOverLmate'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLminOverLmate
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min mapped length
#&gt;       description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignSoftClipAtReferenceEnds
#&gt;         symbols:
#&gt;         - 'Yes'
#&gt;         - 'No'
#&gt;         type: enum
#&gt;       label: Soft clipping
#&gt;       description: 'Option which allows soft clipping of alignments at the reference
#&gt;         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.
#&gt;         Yes: Enables soft clipping; No: Disables soft clipping.'
#&gt;       streamable: no
#&gt;       id: '#alignSoftClipAtReferenceEnds'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSoftClipAtReferenceEnds
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: 'Yes'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min overhang
#&gt;       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: 'Min overhang: annotated'
#&gt;       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced
#&gt;         alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJDBoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJDBoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max mates gap
#&gt;       description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;         by (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignMatesGapMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignMatesGapMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min intron size
#&gt;       description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;       streamable: no
#&gt;       id: '#alignIntronMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '21'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max intron size
#&gt;       description: Maximum intron size, if 0, max intron size will be determined by
#&gt;         (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignIntronMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignEndsType
#&gt;         symbols:
#&gt;         - Local
#&gt;         - EndToEnd
#&gt;         type: enum
#&gt;       label: Alignment type
#&gt;       description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;       streamable: no
#&gt;       id: '#alignEndsType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignEndsType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: Local
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Unmapped reads
#&gt;       description: Output of unmapped reads.
#&gt;       streamable: no
#&gt;       id: '#unmapped_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Unmapped.out*'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Transcriptome alignments
#&gt;       description: Alignments translated into transcript coordinates.
#&gt;       streamable: no
#&gt;       id: '#transcriptome_aligned_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Transcriptome*'
#&gt;       sbg:fileTypes: BAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Splice junctions
#&gt;       description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;         Only junctions supported by uniquely mapping reads are reported.
#&gt;       streamable: no
#&gt;       id: '#splice_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*SJ.out.tab'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Reads per gene
#&gt;       description: File with number of reads per gene. A read is counted if it overlaps
#&gt;         (1nt or more) one and only one gene.
#&gt;       streamable: no
#&gt;       id: '#reads_per_gene'
#&gt;       outputBinding:
#&gt;         glob: '*ReadsPerGene*'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Log files
#&gt;       description: Log files produced during alignment.
#&gt;       streamable: no
#&gt;       id: '#log_files'
#&gt;       outputBinding:
#&gt;         glob: '*Log*.out'
#&gt;       sbg:fileTypes: OUT
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Intermediate genome files
#&gt;       description: Archive with genome files produced when annotations are included
#&gt;         on the fly (in the mapping step).
#&gt;       streamable: no
#&gt;       id: '#intermediate_genome'
#&gt;       outputBinding:
#&gt;         glob: '*_STARgenome.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric junctions
#&gt;       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,
#&gt;         'Chimeric Junctions' won't be output.
#&gt;       streamable: no
#&gt;       id: '#chimeric_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*Chimeric.out.junction'
#&gt;       sbg:fileTypes: JUNCTION
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric alignments
#&gt;       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;         Alignment SAM and Chimeric Junctions outputs.
#&gt;       streamable: no
#&gt;       id: '#chimeric_alignments'
#&gt;       outputBinding:
#&gt;         glob: '*.Chimeric.out.sam'
#&gt;       sbg:fileTypes: SAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Aligned SAM/BAM
#&gt;       description: Aligned sequence in SAM/BAM format.
#&gt;       streamable: no
#&gt;       id: '#aligned_reads'
#&gt;       outputBinding:
#&gt;         glob:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;                 sort_name = '.sortedByCoord'
#&gt;               }
#&gt;               else {
#&gt;                 sort_name = ''
#&gt;               }
#&gt;               if ($job.inputs.outSAMtype == 'BAM') {
#&gt;                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;               }
#&gt;               else {
#&gt;                 sam_name = "*.Aligned.out.sam"
#&gt;               }
#&gt;               return sam_name
#&gt;             }
#&gt;           class: Expression
#&gt;       sbg:fileTypes: SAM, BAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     label: STAR
#&gt;     description: STAR is an ultrafast universal RNA-seq aligner. It has very high
#&gt;       mapping speed, accurate alignment of contiguous and spliced reads, detection
#&gt;       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works
#&gt;       with reads starting from lengths ~15 bases up to ~300 bases. In case of having
#&gt;       longer reads, use of STAR Long is recommended.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - tar
#&gt;     - -xvf
#&gt;     - class: Expression
#&gt;       script: $job.inputs.genome.path
#&gt;       engine: '#cwl-js-engine'
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             file = [].concat($job.inputs.reads)[0].path
#&gt;             extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;             if (extension == "gz") {
#&gt;               return "--readFilesCommand zcat"
#&gt;             } else if (extension == "bz2") {
#&gt;               return "--readFilesCommand bzcat"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n
#&gt;           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",
#&gt;           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)
#&gt;           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n
#&gt;           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",
#&gt;           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)
#&gt;           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)
#&gt;           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)
#&gt;           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n
#&gt;           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"
#&gt;           \")\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.twopassMode == "Basic") {
#&gt;               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.chimOutType == "Within") {
#&gt;               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;             }
#&gt;             else {
#&gt;               return "--chimOutType SeparateSAMold"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n
#&gt;           \   if (value == \"\") {\n      return\n    }\n    else {\n      return
#&gt;           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',
#&gt;           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',
#&gt;           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)
#&gt;           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)
#&gt;           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;           \".concat(param_list.join(\" \"))\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;               return "--quantMode ".concat($job.inputs.quantMode)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 100
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n
#&gt;           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile
#&gt;           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       prefix: --outFileNamePrefix
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;         class: Expression
#&gt;     - position: 101
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"
#&gt;           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,
#&gt;           \" \", mate1fq)\n  }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 624.0
#&gt;     'y': 323
#&gt;   sbg:x: 700.0
#&gt;   sbg:y: 200.0
#&gt; sbg:canvas_zoom: 0.6
#&gt; sbg:canvas_y: -16
#&gt; sbg:canvas_x: -41
#&gt; sbg:batchInput: '#sjdbGTFfile'
#&gt; sbg:batchBy:
#&gt;   type: criteria
#&gt;   criteria:
#&gt;   - metadata.sample_id
#&gt;   - metadata.library_id
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>set_batch</span><span class='op'>(</span><span class='st'>"sjdbGTFfile"</span>, type <span class='op'>=</span> <span class='st'>"ITEM"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 2
#&gt; sbg:toolAuthor: Seven Bridges Genomics
#&gt; sbg:createdOn: 1463601910
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; - RNA
#&gt; sbg:contributors:
#&gt; - tengfei
#&gt; sbg:project: tengfei/quickstart
#&gt; sbg:createdBy: tengfei
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2
#&gt; sbg:license: Apache License 2.0
#&gt; sbg:revision: 2
#&gt; sbg:modifiedOn: 1463601974
#&gt; sbg:modifiedBy: tengfei
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601910
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601952
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601974
#&gt;   sbg:revision: 2
#&gt; sbg:toolkit: STAR
#&gt; id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'
#&gt; inputs:
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Read sequence
#&gt;   description: Read sequence.
#&gt;   streamable: no
#&gt;   id: '#reads'
#&gt;   inputBinding:
#&gt;     position: 10
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n
#&gt;         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if
#&gt;         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n
#&gt;         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata
#&gt;         != null){\n        if (list[index].metadata.paired_end == 1){\n          left
#&gt;         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n
#&gt;         \         right = list[index].path\n        }\n      }\n    }\n    \n    if
#&gt;         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n
#&gt;         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n
#&gt;         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if
#&gt;         (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end
#&gt;         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n
#&gt;         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=
#&gt;         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n
#&gt;         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"
#&gt;         \"))\n  }\n}"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: sjdbGTFfile
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:x: 160.4999759
#&gt;   sbg:y: 195.0833106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: fastq
#&gt;   streamable: no
#&gt;   id: '#fastq'
#&gt;   sbg:x: 164.2499914
#&gt;   sbg:y: 323.7499502
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - File
#&gt;   label: genomeFastaFiles
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   sbg:x: 167.7499601
#&gt;   sbg:y: 469.9999106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 350.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 450.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   source: '#unmapped_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   source: '#transcriptome_aligned_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 233.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   source: '#splice_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 266.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   source: '#reads_per_gene'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 300.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   source: '#log_files'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 333.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   source: '#chimeric_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 366.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   source: '#chimeric_alignments'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 433.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Sorted BAM/SAM
#&gt;   description: Sorted BAM or SAM file.
#&gt;   streamable: no
#&gt;   id: '#sorted_bam'
#&gt;   source: '#sorted_bam'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 466.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Result
#&gt;   description: Source FASTQ file with updated metadata.
#&gt;   streamable: no
#&gt;   id: '#result'
#&gt;   source: '#result'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 533.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; requirements:
#&gt; - class: CreateFileRequirement
#&gt;   fileDef: []
#&gt; hints:
#&gt; - class: sbg:AWSInstanceType
#&gt;   value: c3.8xlarge
#&gt; label: RNA-seq Alignment - STAR
#&gt; description: "Alignment to a reference genome and transcriptome presents the first
#&gt;   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner
#&gt;   capable of mapping full length RNA sequences and detecting de novo canonical junctions,
#&gt;   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian
#&gt;   sequence reads, but fine tuning of its parameters enables customization to satisfy
#&gt;   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).
#&gt;   \ \nSplice junction annotations can optionally be collected from splice junction
#&gt;   databases. Set the \"Overhang length\" parameter to a value larger than zero in
#&gt;   order to use splice junction databases. For constant read length, this value should
#&gt;   (ideally) be equal to mate length decreased by 1; for long reads with non-constant
#&gt;   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ
#&gt;   files reveals read length distribution. STAR can detect chimeric transcripts, but
#&gt;   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted
#&gt;   to a desired minimum chimeric segment length. Aligned reads are reported in BAM
#&gt;   format and can be viewed in a genome browser (such as IGV). A file containing detected
#&gt;   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format
#&gt;   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write
#&gt;   unmapped in SAM\" parameters enable unmapped output type selection."
#&gt; class: Workflow
#&gt; steps:
#&gt; - id: '#STAR_Genome_Generate'
#&gt;   inputs:
#&gt;   - id: '#STAR_Genome_Generate.sjdbScore'
#&gt;   - id: '#STAR_Genome_Generate.sjdbOverhang'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'
#&gt;     source: '#sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'
#&gt;     source: '#sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAsparseD'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAindexNbases'
#&gt;   - id: '#STAR_Genome_Generate.genomeFastaFiles'
#&gt;     source: '#genomeFastaFiles'
#&gt;   - id: '#STAR_Genome_Generate.genomeChrBinNbits'
#&gt;   outputs:
#&gt;   - id: '#STAR_Genome_Generate.genome'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 1
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         sjdbScore: 0
#&gt;         sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;         sjdbOverhang: 0
#&gt;         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;         genomeChrBinNbits: genomeChrBinNbits
#&gt;         genomeSAsparseD: 0
#&gt;         sjdbGTFfile:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /demo/test-files/chr20.gtf
#&gt;         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;         genomeFastaFiles:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /sbgenomics/test-data/chr20.fa
#&gt;         sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;         genomeSAindexNbases: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911469
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 1
#&gt;     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa
#&gt;       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD
#&gt;       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt;     sbg:modifiedOn: 1450911470
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911469
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911470
#&gt;       sbg:revision: 1
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbScore
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbOverhang
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentTranscript
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentGene
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFfeatureExon
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFchrPrefix
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Suffux array sparsity
#&gt;       description: 'Distance between indices: use bigger numbers to decrease needed
#&gt;         RAM at the cost of mapping speed reduction (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#genomeSAsparseD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAsparseD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Pre-indexing string length
#&gt;       description: Length (bases) of the SA pre-indexing string. Typically between
#&gt;         10 and 15. Longer strings will use much more memory, but allow faster searches.
#&gt;         For small genomes, this number needs to be scaled down, with a typical value
#&gt;         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,
#&gt;         this is equal to 9, for 100 kiloBase genome, this is equal to 7.
#&gt;       streamable: no
#&gt;       id: '#genomeSAindexNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAindexNbases
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '14'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome fasta files
#&gt;       description: Reference sequence to which to align the reads.
#&gt;       streamable: no
#&gt;       id: '#genomeFastaFiles'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeFastaFiles
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FA
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Bins size
#&gt;       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins
#&gt;         for genome storage: each chromosome will occupy an integer number of bins.
#&gt;         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,
#&gt;         you may need to reduce this number to reduce RAM consumption. The following
#&gt;         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this
#&gt;         is equal to 15.'
#&gt;       streamable: no
#&gt;       id: '#genomeChrBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeChrBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '18'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Genome Files
#&gt;       description: Genome files comprise binary genome sequence, suffix arrays, text
#&gt;         chromosome names/lengths, splice junctions coordinates, and transcripts/genes
#&gt;         information.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       outputBinding:
#&gt;         glob: '*.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     label: STAR Genome Generate
#&gt;     description: STAR Genome Generate is a tool that generates genome index files.
#&gt;       One set of files should be generated per each genome/annotation combination.
#&gt;       Once produced, these files could be used as long as genome/annotation combination
#&gt;       stays the same. Also, STAR Genome Generate which produced these files and STAR
#&gt;       aligner using them must be the same toolkit version.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - mkdir
#&gt;     - genomeDir
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runMode
#&gt;     - genomeGenerate
#&gt;     - --genomeDir
#&gt;     - ./genomeDir
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 99
#&gt;       separate: yes
#&gt;       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 384.0832266
#&gt;     'y': 446.4998957
#&gt;   sbg:x: 100.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#SBG_FASTQ_Quality_Detector'
#&gt;   inputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.fastq'
#&gt;     source: '#fastq'
#&gt;   outputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 3
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 1000
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         fastq:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /path/to/fastq.ext
#&gt;     sbg:toolAuthor: Seven Bridges Genomics
#&gt;     sbg:createdOn: 1450911312
#&gt;     sbg:categories:
#&gt;     - FASTQ-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:project: bix-demo/sbgtools-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     sbg:license: Apache License 2.0
#&gt;     sbg:revision: 3
#&gt;     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext
#&gt;       /path/to/fastq.ext
#&gt;     sbg:modifiedOn: 1450911314
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911312
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911314
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: SBGTools
#&gt;     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     inputs:
#&gt;     - type:
#&gt;       - File
#&gt;       label: Fastq
#&gt;       description: FASTQ file.
#&gt;       streamable: no
#&gt;       id: '#fastq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --fastq
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       required: yes
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Result
#&gt;       description: Source FASTQ file with updated metadata.
#&gt;       streamable: no
#&gt;       id: '#result'
#&gt;       outputBinding:
#&gt;         glob: '*.fastq'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     requirements:
#&gt;     - class: CreateFileRequirement
#&gt;       fileDef: []
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0
#&gt;       dockerImageId: ''
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 1000
#&gt;     label: SBG FASTQ Quality Detector
#&gt;     description: FASTQ Quality Scale Detector detects which quality encoding scheme
#&gt;       was used in your reads and automatically enters the proper value in the "Quality
#&gt;       Scale" metadata field.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - python
#&gt;     - /opt/sbg_fastq_quality_scale_detector.py
#&gt;     arguments: []
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 375.3333179
#&gt;     'y': 323.5833156
#&gt;   sbg:x: 300.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#Picard_SortSam'
#&gt;   inputs:
#&gt;   - id: '#Picard_SortSam.validation_stringency'
#&gt;     default: SILENT
#&gt;   - id: '#Picard_SortSam.sort_order'
#&gt;     default: Coordinate
#&gt;   - id: '#Picard_SortSam.quiet'
#&gt;   - id: '#Picard_SortSam.output_type'
#&gt;   - id: '#Picard_SortSam.memory_per_job'
#&gt;   - id: '#Picard_SortSam.max_records_in_ram'
#&gt;   - id: '#Picard_SortSam.input_bam'
#&gt;     source: '#STAR.aligned_reads'
#&gt;   - id: '#Picard_SortSam.create_index'
#&gt;     default: 'True'
#&gt;   - id: '#Picard_SortSam.compression_level'
#&gt;   outputs:
#&gt;   - id: '#Picard_SortSam.sorted_bam'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 2
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 2048
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         sort_order: Coordinate
#&gt;         input_bam:
#&gt;           path: /root/dir/example.tested.bam
#&gt;         memory_per_job: 2048
#&gt;         output_type: ~
#&gt;         create_index: ~
#&gt;     sbg:toolAuthor: Broad Institute
#&gt;     sbg:createdOn: 1450911168
#&gt;     sbg:categories:
#&gt;     - SAM/BAM-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: http://broadinstitute.github.io/picard/index.html
#&gt;       label: Homepage
#&gt;     - id: https://github.com/broadinstitute/picard/releases/tag/1.138
#&gt;       label: Source Code
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Wiki
#&gt;     - id: https://github.com/broadinstitute/picard/zipball/master
#&gt;       label: Download
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/picard-1-140-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: '1.140'
#&gt;     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     sbg:license: MIT License, Apache 2.0 Licence
#&gt;     sbg:revision: 2
#&gt;     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam
#&gt;       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam
#&gt;       SORT_ORDER=coordinate  /root/dir/example.tested.bam
#&gt;     sbg:modifiedOn: 1450911170
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911168
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911169
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911170
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: Picard
#&gt;     id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: validation_stringency
#&gt;         symbols:
#&gt;         - STRICT
#&gt;         - LENIENT
#&gt;         - SILENT
#&gt;         type: enum
#&gt;       label: Validation stringency
#&gt;       description: 'Validation stringency for all SAM files read by this program.
#&gt;         Setting stringency to SILENT can improve performance when processing a BAM
#&gt;         file in which variable-length data (read, qualities, tags) do not otherwise
#&gt;         need to be decoded. This option can be set to ''null'' to clear the default
#&gt;         value. Possible values: {STRICT, LENIENT, SILENT}.'
#&gt;       streamable: no
#&gt;       id: '#validation_stringency'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: VALIDATION_STRINGENCY=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.validation_stringency)
#&gt;               {
#&gt;                 return $job.inputs.validation_stringency
#&gt;               }
#&gt;               else
#&gt;               {
#&gt;                 return "SILENT"
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SILENT
#&gt;       required: no
#&gt;     - type:
#&gt;       - name: sort_order
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - Queryname
#&gt;         - Coordinate
#&gt;         type: enum
#&gt;       label: Sort order
#&gt;       description: 'Sort order of the output file. Possible values: {unsorted, queryname,
#&gt;         coordinate}.'
#&gt;       streamable: no
#&gt;       id: '#sort_order'
#&gt;       inputBinding:
#&gt;         position: 3
#&gt;         prefix: SORT_ORDER=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               p = $job.inputs.sort_order.toLowerCase()
#&gt;               return p
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: Coordinate
#&gt;       sbg:altPrefix: SO
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quiet
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Quiet
#&gt;       description: 'This parameter indicates whether to suppress job-summary info
#&gt;         on System.err. This option can be set to ''null'' to clear the default value.
#&gt;         Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#quiet'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: QUIET=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: output_type
#&gt;         symbols:
#&gt;         - BAM
#&gt;         - SAM
#&gt;         - SAME AS INPUT
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Since Picard tools can output both SAM and BAM files, user can
#&gt;         choose the format of the output file.
#&gt;       streamable: no
#&gt;       id: '#output_type'
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SAME AS INPUT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Memory per job
#&gt;       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for
#&gt;         single threaded jobs.
#&gt;       streamable: no
#&gt;       id: '#memory_per_job'
#&gt;       sbg:toolDefaultValue: '2048'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max records in RAM
#&gt;       description: When writing SAM files that need to be sorted, this parameter will
#&gt;         specify the number of records stored in RAM before spilling to disk. Increasing
#&gt;         this number reduces the number of file handles needed to sort a SAM file,
#&gt;         and increases the amount of RAM needed. This option can be set to 'null' to
#&gt;         clear the default value.
#&gt;       streamable: no
#&gt;       id: '#max_records_in_ram'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: MAX_RECORDS_IN_RAM=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '500000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Input BAM
#&gt;       description: The BAM or SAM file to sort.
#&gt;       streamable: no
#&gt;       id: '#input_bam'
#&gt;       inputBinding:
#&gt;         position: 1
#&gt;         prefix: INPUT=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: File inputs
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;       sbg:altPrefix: I
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: create_index
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Create index
#&gt;       description: 'This parameter indicates whether to create a BAM index when writing
#&gt;         a coordinate-sorted BAM file. This option can be set to ''null'' to clear
#&gt;         the default value. Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#create_index'
#&gt;       inputBinding:
#&gt;         position: 5
#&gt;         prefix: CREATE_INDEX=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Compression level
#&gt;       description: Compression level for all compressed files created (e.g. BAM and
#&gt;         GELI). This option can be set to 'null' to clear the default value.
#&gt;       streamable: no
#&gt;       id: '#compression_level'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: COMPRESSION_LEVEL=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Sorted BAM/SAM
#&gt;       description: Sorted BAM or SAM file.
#&gt;       streamable: no
#&gt;       id: '#sorted_bam'
#&gt;       outputBinding:
#&gt;         glob: '*.sorted.?am'
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;       engineCommand: cwl-engine.js
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
#&gt;       dockerImageId: eab0e70b6629
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n
#&gt;           \ }\n  \treturn 2048\n}"
#&gt;         class: Expression
#&gt;     label: Picard SortSam
#&gt;     description: Picard SortSam sorts the input SAM or BAM. Input and output formats
#&gt;       are determined by the file extension.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - java
#&gt;     - class: Expression
#&gt;       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,
#&gt;         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"
#&gt;       engine: '#cwl-js-engine'
#&gt;     - -jar
#&gt;     - /opt/picard-tools-1.140/picard.jar
#&gt;     - SortSam
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       prefix: OUTPUT=
#&gt;       separate: no
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif
#&gt;           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,
#&gt;           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse
#&gt;           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n}\n}"
#&gt;         class: Expression
#&gt;     - position: 1000
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out
#&gt;           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n
#&gt;           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n
#&gt;           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n
#&gt;           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index
#&gt;           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==
#&gt;           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 773.0831807
#&gt;     'y': 470.9165939
#&gt;   sbg:x: 500.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#STAR'
#&gt;   inputs:
#&gt;   - id: '#STAR.winFlankNbins'
#&gt;   - id: '#STAR.winBinNbits'
#&gt;   - id: '#STAR.winAnchorMultimapNmax'
#&gt;     source: '#winAnchorMultimapNmax'
#&gt;   - id: '#STAR.winAnchorDistNbins'
#&gt;     source: '#winAnchorDistNbins'
#&gt;   - id: '#STAR.twopassMode'
#&gt;   - id: '#STAR.twopass1readsN'
#&gt;   - id: '#STAR.sjdbScore'
#&gt;   - id: '#STAR.sjdbOverhang'
#&gt;     default: 100
#&gt;   - id: '#STAR.sjdbInsertSave'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR.seedSearchStartLmaxOverLread'
#&gt;   - id: '#STAR.seedSearchStartLmax'
#&gt;   - id: '#STAR.seedSearchLmax'
#&gt;   - id: '#STAR.seedPerWindowNmax'
#&gt;   - id: '#STAR.seedPerReadNmax'
#&gt;   - id: '#STAR.seedNoneLociPerWindow'
#&gt;   - id: '#STAR.seedMultimapNmax'
#&gt;   - id: '#STAR.scoreStitchSJshift'
#&gt;   - id: '#STAR.scoreInsOpen'
#&gt;   - id: '#STAR.scoreInsBase'
#&gt;   - id: '#STAR.scoreGenomicLengthLog2scale'
#&gt;   - id: '#STAR.scoreGapNoncan'
#&gt;   - id: '#STAR.scoreGapGCAG'
#&gt;   - id: '#STAR.scoreGapATAC'
#&gt;   - id: '#STAR.scoreGap'
#&gt;   - id: '#STAR.scoreDelOpen'
#&gt;   - id: '#STAR.scoreDelBase'
#&gt;   - id: '#STAR.rg_seq_center'
#&gt;   - id: '#STAR.rg_sample_id'
#&gt;   - id: '#STAR.rg_platform_unit_id'
#&gt;   - id: '#STAR.rg_platform'
#&gt;   - id: '#STAR.rg_mfl'
#&gt;   - id: '#STAR.rg_library_id'
#&gt;   - id: '#STAR.reads'
#&gt;     source: '#reads'
#&gt;   - id: '#STAR.readMatesLengthsIn'
#&gt;   - id: '#STAR.readMapNumber'
#&gt;   - id: '#STAR.quantTranscriptomeBan'
#&gt;   - id: '#STAR.quantMode'
#&gt;     default: TranscriptomeSAM
#&gt;   - id: '#STAR.outSortingType'
#&gt;     default: SortedByCoordinate
#&gt;   - id: '#STAR.outSJfilterReads'
#&gt;   - id: '#STAR.outSJfilterOverhangMin'
#&gt;   - id: '#STAR.outSJfilterIntronMaxVsReadN'
#&gt;   - id: '#STAR.outSJfilterDistToOtherSJmin'
#&gt;   - id: '#STAR.outSJfilterCountUniqueMin'
#&gt;   - id: '#STAR.outSJfilterCountTotalMin'
#&gt;   - id: '#STAR.outSAMunmapped'
#&gt;   - id: '#STAR.outSAMtype'
#&gt;     default: BAM
#&gt;   - id: '#STAR.outSAMstrandField'
#&gt;   - id: '#STAR.outSAMreadID'
#&gt;   - id: '#STAR.outSAMprimaryFlag'
#&gt;   - id: '#STAR.outSAMorder'
#&gt;   - id: '#STAR.outSAMmode'
#&gt;   - id: '#STAR.outSAMmapqUnique'
#&gt;   - id: '#STAR.outSAMheaderPG'
#&gt;   - id: '#STAR.outSAMheaderHD'
#&gt;   - id: '#STAR.outSAMflagOR'
#&gt;   - id: '#STAR.outSAMflagAND'
#&gt;   - id: '#STAR.outSAMattributes'
#&gt;   - id: '#STAR.outReadsUnmapped'
#&gt;     default: Fastx
#&gt;   - id: '#STAR.outQSconversionAdd'
#&gt;   - id: '#STAR.outFilterType'
#&gt;   - id: '#STAR.outFilterScoreMinOverLread'
#&gt;   - id: '#STAR.outFilterScoreMin'
#&gt;   - id: '#STAR.outFilterMultimapScoreRange'
#&gt;   - id: '#STAR.outFilterMultimapNmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverReadLmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverLmax'
#&gt;   - id: '#STAR.outFilterMismatchNmax'
#&gt;   - id: '#STAR.outFilterMatchNminOverLread'
#&gt;   - id: '#STAR.outFilterMatchNmin'
#&gt;   - id: '#STAR.outFilterIntronMotifs'
#&gt;   - id: '#STAR.limitSjdbInsertNsj'
#&gt;   - id: '#STAR.limitOutSJoneRead'
#&gt;   - id: '#STAR.limitOutSJcollapsed'
#&gt;   - id: '#STAR.limitBAMsortRAM'
#&gt;   - id: '#STAR.genomeDirName'
#&gt;   - id: '#STAR.genome'
#&gt;     source: '#STAR_Genome_Generate.genome'
#&gt;   - id: '#STAR.clip5pNbases'
#&gt;   - id: '#STAR.clip3pNbases'
#&gt;   - id: '#STAR.clip3pAfterAdapterNbases'
#&gt;   - id: '#STAR.clip3pAdapterSeq'
#&gt;   - id: '#STAR.clip3pAdapterMMp'
#&gt;   - id: '#STAR.chimSegmentMin'
#&gt;   - id: '#STAR.chimScoreSeparation'
#&gt;   - id: '#STAR.chimScoreMin'
#&gt;   - id: '#STAR.chimScoreJunctionNonGTAG'
#&gt;   - id: '#STAR.chimScoreDropMax'
#&gt;   - id: '#STAR.chimOutType'
#&gt;   - id: '#STAR.chimJunctionOverhangMin'
#&gt;   - id: '#STAR.alignWindowsPerReadNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerWindowNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerReadNmax'
#&gt;   - id: '#STAR.alignSplicedMateMapLminOverLmate'
#&gt;   - id: '#STAR.alignSplicedMateMapLmin'
#&gt;   - id: '#STAR.alignSoftClipAtReferenceEnds'
#&gt;   - id: '#STAR.alignSJoverhangMin'
#&gt;   - id: '#STAR.alignSJDBoverhangMin'
#&gt;   - id: '#STAR.alignMatesGapMax'
#&gt;   - id: '#STAR.alignIntronMin'
#&gt;   - id: '#STAR.alignIntronMax'
#&gt;   - id: '#STAR.alignEndsType'
#&gt;   outputs:
#&gt;   - id: '#STAR.unmapped_reads'
#&gt;   - id: '#STAR.transcriptome_aligned_reads'
#&gt;   - id: '#STAR.splice_junctions'
#&gt;   - id: '#STAR.reads_per_gene'
#&gt;   - id: '#STAR.log_files'
#&gt;   - id: '#STAR.intermediate_genome'
#&gt;   - id: '#STAR.chimeric_junctions'
#&gt;   - id: '#STAR.chimeric_alignments'
#&gt;   - id: '#STAR.aligned_reads'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 4
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         alignWindowsPerReadNmax: 0
#&gt;         outSAMheaderPG: outSAMheaderPG
#&gt;         GENOME_DIR_NAME: ''
#&gt;         outFilterMatchNminOverLread: 0
#&gt;         rg_platform_unit_id: rg_platform_unit
#&gt;         alignTranscriptsPerReadNmax: 0
#&gt;         readMapNumber: 0
#&gt;         alignSplicedMateMapLminOverLmate: 0
#&gt;         alignMatesGapMax: 0
#&gt;         outFilterMultimapNmax: 0
#&gt;         clip5pNbases:
#&gt;         - 0
#&gt;         outSAMstrandField: None
#&gt;         readMatesLengthsIn: NotEqual
#&gt;         outSAMattributes: Standard
#&gt;         seedMultimapNmax: 0
#&gt;         rg_mfl: rg_mfl
#&gt;         chimSegmentMin: 0
#&gt;         winAnchorDistNbins: 0
#&gt;         outSortingType: SortedByCoordinate
#&gt;         outFilterMultimapScoreRange: 0
#&gt;         sjdbInsertSave: Basic
#&gt;         clip3pAfterAdapterNbases:
#&gt;         - 0
#&gt;         scoreDelBase: 0
#&gt;         outFilterMatchNmin: 0
#&gt;         twopass1readsN: 0
#&gt;         outSAMunmapped: None
#&gt;         genome:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: genome.ext
#&gt;         sjdbGTFtagExonParentTranscript: ''
#&gt;         limitBAMsortRAM: 0
#&gt;         alignEndsType: Local
#&gt;         seedNoneLociPerWindow: 0
#&gt;         rg_sample_id: rg_sample
#&gt;         sjdbGTFtagExonParentGene: ''
#&gt;         chimScoreMin: 0
#&gt;         outSJfilterIntronMaxVsReadN:
#&gt;         - 0
#&gt;         twopassMode: Basic
#&gt;         alignSplicedMateMapLmin: 0
#&gt;         outSJfilterReads: All
#&gt;         outSAMprimaryFlag: OneBestScore
#&gt;         outSJfilterCountTotalMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         outSAMorder: Paired
#&gt;         outSAMflagAND: 0
#&gt;         chimScoreSeparation: 0
#&gt;         alignSJoverhangMin: 0
#&gt;         outFilterScoreMin: 0
#&gt;         seedSearchStartLmax: 0
#&gt;         scoreGapGCAG: 0
#&gt;         scoreGenomicLengthLog2scale: 0
#&gt;         outFilterIntronMotifs: None
#&gt;         outFilterMismatchNmax: 0
#&gt;         reads:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           metadata:
#&gt;             format: fastq
#&gt;             paired_end: '1'
#&gt;             seq_center: illumina
#&gt;           path: /test-data/mate_1.fastq.bz2
#&gt;         scoreGap: 0
#&gt;         outSJfilterOverhangMin:
#&gt;         - 30
#&gt;         - 12
#&gt;         - 12
#&gt;         - 12
#&gt;         outSAMflagOR: 0
#&gt;         outSAMmode: Full
#&gt;         rg_library_id: ''
#&gt;         chimScoreJunctionNonGTAG: 0
#&gt;         scoreInsOpen: 0
#&gt;         clip3pAdapterSeq:
#&gt;         - clip3pAdapterSeq
#&gt;         chimScoreDropMax: 0
#&gt;         outFilterType: Normal
#&gt;         scoreGapATAC: 0
#&gt;         rg_platform: Ion Torrent PGM
#&gt;         clip3pAdapterMMp:
#&gt;         - 0
#&gt;         sjdbGTFfeatureExon: ''
#&gt;         outQSconversionAdd: 0
#&gt;         quantMode: TranscriptomeSAM
#&gt;         alignIntronMin: 0
#&gt;         scoreInsBase: 0
#&gt;         scoreGapNoncan: 0
#&gt;         seedSearchLmax: 0
#&gt;         outSJfilterDistToOtherSJmin:
#&gt;         - 0
#&gt;         outFilterScoreMinOverLread: 0
#&gt;         alignSJDBoverhangMin: 0
#&gt;         limitOutSJcollapsed: 0
#&gt;         winAnchorMultimapNmax: 0
#&gt;         outFilterMismatchNoverLmax: 0
#&gt;         rg_seq_center: ''
#&gt;         outSAMheaderHD: outSAMheaderHD
#&gt;         chimOutType: Within
#&gt;         quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;         limitOutSJoneRead: 0
#&gt;         alignTranscriptsPerWindowNmax: 0
#&gt;         sjdbOverhang: ~
#&gt;         outReadsUnmapped: Fastx
#&gt;         scoreStitchSJshift: 0
#&gt;         seedPerWindowNmax: 0
#&gt;         outSJfilterCountUniqueMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         scoreDelOpen: 0
#&gt;         sjdbGTFfile:
#&gt;         - path: /demo/test-data/chr20.gtf
#&gt;         clip3pNbases:
#&gt;         - 0
#&gt;         - 3
#&gt;         winBinNbits: 0
#&gt;         sjdbScore: ~
#&gt;         seedSearchStartLmaxOverLread: 0
#&gt;         alignIntronMax: 0
#&gt;         seedPerReadNmax: 0
#&gt;         outFilterMismatchNoverReadLmax: 0
#&gt;         winFlankNbins: 0
#&gt;         sjdbGTFchrPrefix: chrPrefix
#&gt;         alignSoftClipAtReferenceEnds: 'Yes'
#&gt;         outSAMreadID: Standard
#&gt;         outSAMtype: BAM
#&gt;         chimJunctionOverhangMin: 0
#&gt;         limitSjdbInsertNsj: 0
#&gt;         outSAMmapqUnique: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911471
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - ana_d
#&gt;     - bix-demo
#&gt;     - uros_sipetic
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star/4
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 4
#&gt;     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix
#&gt;       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline
#&gt;       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode
#&gt;       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;
#&gt;       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt;     sbg:modifiedOn: 1462889222
#&gt;     sbg:modifiedBy: ana_d
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911471
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911473
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911475
#&gt;       sbg:revision: 2
#&gt;     - sbg:modifiedBy: uros_sipetic
#&gt;       sbg:modifiedOn: 1462878528
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: ana_d
#&gt;       sbg:modifiedOn: 1462889222
#&gt;       sbg:revision: 4
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star/4
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Flanking regions size
#&gt;       description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;         flanking regions for each window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winFlankNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winFlankNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Bin size
#&gt;       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;         each window will occupy an integer number of bins (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '16'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max loci anchors
#&gt;       description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max bins between anchors
#&gt;       description: Max number of bins between two anchors that allows aggregation
#&gt;         of anchors into one window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorDistNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorDistNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '9'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: twopassMode
#&gt;         symbols:
#&gt;         - None
#&gt;         - Basic
#&gt;         type: enum
#&gt;       label: Two-pass mode
#&gt;       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass
#&gt;         mapping, with all 1st pass junctions inserted into the genome indices on the
#&gt;         fly.'
#&gt;       streamable: no
#&gt;       id: '#twopassMode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --twopassMode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to process in 1st step
#&gt;       description: 'Number of reads to process for the 1st step. 0: 1-step only, no
#&gt;         2nd pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#twopass1readsN'
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: sjdbInsertSave
#&gt;         symbols:
#&gt;         - Basic
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: Save junction files
#&gt;       description: 'Which files to save when sjdb junctions are inserted on the fly
#&gt;         at the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;         files; All: all files including big Genome, SA and SAindex. These files are
#&gt;         output as archive.'
#&gt;       streamable: no
#&gt;       id: '#sjdbInsertSave'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts. No need to include
#&gt;         this input, except in case of using "on the fly" annotations.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Search start point normalized
#&gt;       description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;         for paired-end reads).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmaxOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmaxOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1.0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Search start point
#&gt;       description: Defines the search start point through the read - the read is split
#&gt;         into pieces no longer than this value (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seed length
#&gt;       description: Defines the maximum length of the seeds, if =0 max seed length
#&gt;         is infinite (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per window
#&gt;       description: Max number of seeds per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per read
#&gt;       description: Max number of seeds per read (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max one-seed loci per window
#&gt;       description: Max number of one seed loci per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedNoneLociPerWindow'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedNoneLociPerWindow
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Filter pieces for stitching
#&gt;       description: Only pieces that map fewer than this value are utilized in the
#&gt;         stitching procedure (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max score reduction
#&gt;       description: Maximum score reduction while searching for SJ boundaries in the
#&gt;         stitching step.
#&gt;       streamable: no
#&gt;       id: '#scoreStitchSJshift'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreStitchSJshift
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion Open Penalty
#&gt;       description: Insertion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreInsOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion extension penalty
#&gt;       description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreInsBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Log scaled score
#&gt;       description: 'Extra score logarithmically scaled with genomic length of the
#&gt;         alignment: &lt;int&gt;*log2(genomicLength).'
#&gt;       streamable: no
#&gt;       id: '#scoreGenomicLengthLog2scale'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGenomicLengthLog2scale
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-0.25'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-canonical gap open
#&gt;       description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapNoncan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapNoncan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: GC/AG and CT/GC gap open
#&gt;       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapGCAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapGCAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AT/AC and GT/AT gap open
#&gt;       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapATAC'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapATAC
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Gap open penalty
#&gt;       description: Gap open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreGap'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGap
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion open penalty
#&gt;       description: Deletion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreDelOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion extension penalty
#&gt;       description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreDelBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sequencing center
#&gt;       description: Specify the sequencing center for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_seq_center'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sample ID
#&gt;       description: Specify the sample ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_sample_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Platform unit ID
#&gt;       description: Specify the platform unit ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_platform_unit_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: rg_platform
#&gt;         symbols:
#&gt;         - LS 454
#&gt;         - Helicos
#&gt;         - Illumina
#&gt;         - ABI SOLiD
#&gt;         - Ion Torrent PGM
#&gt;         - PacBio
#&gt;         type: enum
#&gt;       label: Platform
#&gt;       description: Specify the version of the technology that was used for sequencing
#&gt;         or assaying.
#&gt;       streamable: no
#&gt;       id: '#rg_platform'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Median fragment length
#&gt;       description: Specify the median fragment length for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_mfl'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Library ID
#&gt;       description: Specify the library ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_library_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Read sequence
#&gt;       description: Read sequence.
#&gt;       streamable: no
#&gt;       id: '#reads'
#&gt;       inputBinding:
#&gt;         position: 10
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp
#&gt;             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n
#&gt;             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =
#&gt;             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n
#&gt;             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end
#&gt;             == 2){\n          right = list[index].path\n        }\n      }\n    }\n
#&gt;             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n
#&gt;             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left
#&gt;             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;
#&gt;             ++index) {\n      \n      if (list[index].metadata != null){\n        if
#&gt;             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n
#&gt;             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n
#&gt;             \       }\n      }\n    }\n    left_join = left.join()\n    right_join
#&gt;             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n
#&gt;             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){
#&gt;             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;       sbg:x: 200.0
#&gt;       sbg:y: 200.0
#&gt;       sbg:includeInPorts: yes
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: readMatesLengthsIn
#&gt;         symbols:
#&gt;         - NotEqual
#&gt;         - Equal
#&gt;         type: enum
#&gt;       label: Reads lengths
#&gt;       description: Equal/Not equal - lengths of names, sequences, qualities for both
#&gt;         mates are the same/not the same. "Not equal" is safe in all situations.
#&gt;       streamable: no
#&gt;       id: '#readMatesLengthsIn'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMatesLengthsIn
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: NotEqual
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to map
#&gt;       description: Number of reads to map from the beginning of the file.
#&gt;       streamable: no
#&gt;       id: '#readMapNumber'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMapNumber
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantTranscriptomeBan
#&gt;         symbols:
#&gt;         - IndelSoftclipSingleend
#&gt;         - Singleend
#&gt;         type: enum
#&gt;       label: Prohibit alignment type
#&gt;       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;         prohibit single-end alignments.'
#&gt;       streamable: no
#&gt;       id: '#quantTranscriptomeBan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --quantTranscriptomeBan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantMode
#&gt;         symbols:
#&gt;         - TranscriptomeSAM
#&gt;         - GeneCounts
#&gt;         type: enum
#&gt;       label: Quantification mode
#&gt;       description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'
#&gt;         option, STAR will count number of reads per gene while mapping.
#&gt;       streamable: no
#&gt;       id: '#quantMode'
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSortingType
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - SortedByCoordinate
#&gt;         - Unsorted SortedByCoordinate
#&gt;         type: enum
#&gt;       label: Output sorting type
#&gt;       description: Type of output sorting.
#&gt;       streamable: no
#&gt;       id: '#outSortingType'
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SortedByCoordinate
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSJfilterReads
#&gt;         symbols:
#&gt;         - All
#&gt;         - Unique
#&gt;         type: enum
#&gt;       label: Collapsed junctions reads
#&gt;       description: 'Which reads to consider for collapsed splice junctions output.
#&gt;         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads
#&gt;         only.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterReads'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterReads
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: All
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min overhang SJ
#&gt;       description: Minimum overhang length for splice junctions on both sides for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterOverhangMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 30 12 12 12
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Max gap allowed
#&gt;       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads
#&gt;         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,
#&gt;         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;         Does not apply to annotated junctions.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterIntronMaxVsReadN'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterIntronMaxVsReadN
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 50000 100000 200000
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min distance to other donor/acceptor
#&gt;       description: Minimum allowed distance to other junctions' donor/acceptor for
#&gt;         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterDistToOtherSJmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterDistToOtherSJmin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 10 0 5 10
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min unique count
#&gt;       description: Minimum uniquely mapping read count per junction for each of the
#&gt;         motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountUniqueMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountUniqueMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min total count
#&gt;       description: Minimum total (multi-mapping+unique) read count per junction for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountTotalMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountTotalMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMunmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Write unmapped in SAM
#&gt;       description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;       streamable: no
#&gt;       id: '#outSAMunmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMunmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMtype
#&gt;         symbols:
#&gt;         - SAM
#&gt;         - BAM
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Format of output alignments.
#&gt;       streamable: no
#&gt;       id: '#outSAMtype'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               SAM_type = $job.inputs.outSAMtype
#&gt;               SORT_type = $job.inputs.outSortingType
#&gt;               if (SAM_type &amp;&amp; SORT_type) {
#&gt;                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SAM
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMstrandField
#&gt;         symbols:
#&gt;         - None
#&gt;         - intronMotif
#&gt;         type: enum
#&gt;       label: Strand field flag
#&gt;       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:
#&gt;         strand derived from the intron motif. Reads with inconsistent and/or non-canonical
#&gt;         introns are filtered out.'
#&gt;       streamable: no
#&gt;       id: '#outSAMstrandField'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMstrandField
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMreadID
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - Number
#&gt;         type: enum
#&gt;       label: Read ID
#&gt;       description: 'Read ID record type. Standard: first word (until space) from the
#&gt;         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)
#&gt;         in the FASTx file.'
#&gt;       streamable: no
#&gt;       id: '#outSAMreadID'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMreadID
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMprimaryFlag
#&gt;         symbols:
#&gt;         - OneBestScore
#&gt;         - AllBestScore
#&gt;         type: enum
#&gt;       label: Primary alignments
#&gt;       description: 'Which alignments are considered primary - all others will be marked
#&gt;         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best
#&gt;         score is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;       streamable: no
#&gt;       id: '#outSAMprimaryFlag'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMprimaryFlag
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: OneBestScore
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMorder
#&gt;         symbols:
#&gt;         - Paired
#&gt;         - PairedKeepInputOrder
#&gt;         type: enum
#&gt;       label: Sorting in SAM
#&gt;       description: 'Type of sorting for the SAM output. Paired: one mate after the
#&gt;         other for all paired alignments; PairedKeepInputOrder: one mate after the
#&gt;         other for all paired alignments, the order is kept the same as in the input
#&gt;         FASTQ files.'
#&gt;       streamable: no
#&gt;       id: '#outSAMorder'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMorder
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Paired
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMmode
#&gt;         symbols:
#&gt;         - Full
#&gt;         - NoQS
#&gt;         type: enum
#&gt;       label: SAM mode
#&gt;       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but
#&gt;         without quality scores.'
#&gt;       streamable: no
#&gt;       id: '#outSAMmode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Full
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: MAPQ value
#&gt;       description: MAPQ value for unique mappers (0 to 255).
#&gt;       streamable: no
#&gt;       id: '#outSAMmapqUnique'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmapqUnique
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '255'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @PG
#&gt;       description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderPG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderPG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @HD
#&gt;       description: '@HD (header) line of the SAM header.'
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderHD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderHD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: OR SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagOR'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagOR
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AND SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagAND'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagAND
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '65535'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMattributes
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - NH
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: SAM attributes
#&gt;       description: 'Desired SAM attributes, in the order desired for the output SAM.
#&gt;         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM
#&gt;         NM MD jM jI; None: no attributes.'
#&gt;       streamable: no
#&gt;       id: '#outSAMattributes'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMattributes
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outReadsUnmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Fastx
#&gt;         type: enum
#&gt;       label: Output unmapped reads
#&gt;       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:
#&gt;         output in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;       streamable: no
#&gt;       id: '#outReadsUnmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outReadsUnmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Quality conversion
#&gt;       description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;         to Sanger, use -31).
#&gt;       streamable: no
#&gt;       id: '#outQSconversionAdd'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outQSconversionAdd
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterType
#&gt;         symbols:
#&gt;         - Normal
#&gt;         - BySJout
#&gt;         type: enum
#&gt;       label: Filtering type
#&gt;       description: 'Type of filtering. Normal: standard filtering using only current
#&gt;         alignment; BySJout: keep only those reads that contain junctions that passed
#&gt;         filtering into SJ.out.tab.'
#&gt;       streamable: no
#&gt;       id: '#outFilterType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: Normal
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min score normalized
#&gt;       description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;         for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMinOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMinOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min score
#&gt;       description: Alignment will be output only if its score is higher than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Multimapping score range
#&gt;       description: The score range below the maximum score for multimapping alignments.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapScoreRange'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapScoreRange
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mappings
#&gt;       description: Read alignments will be output only if the read maps fewer than
#&gt;         this value, otherwise no alignments will be output.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *read* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverReadLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverReadLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *mapped* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mismatches
#&gt;       description: Alignment will be output only if it has fewer mismatches than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min matched bases normalized
#&gt;       description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;         lengths for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNminOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNminOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min matched bases
#&gt;       description: Alignment will be output only if the number of matched bases is
#&gt;         higher than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterIntronMotifs
#&gt;         symbols:
#&gt;         - None
#&gt;         - RemoveNoncanonical
#&gt;         - RemoveNoncanonicalUnannotated
#&gt;         type: enum
#&gt;       label: Motifs filtering
#&gt;       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;         filter out alignments that contain non-canonical unannotated junctions when
#&gt;         using annotated splice junctions database. The annotated non-canonical junctions
#&gt;         will be kept.'
#&gt;       streamable: no
#&gt;       id: '#outFilterIntronMotifs'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterIntronMotifs
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max insert junctions
#&gt;       description: Maximum number of junction to be inserted to the genome on the
#&gt;         fly at the mapping stage, including those from annotations and those detected
#&gt;         in the 1st step of the 2-pass run.
#&gt;       streamable: no
#&gt;       id: '#limitSjdbInsertNsj'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitSjdbInsertNsj
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Junctions max number
#&gt;       description: Max number of junctions for one read (including all multi-mappers).
#&gt;       streamable: no
#&gt;       id: '#limitOutSJoneRead'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJoneRead
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Collapsed junctions max number
#&gt;       description: Max number of collapsed junctions.
#&gt;       streamable: no
#&gt;       id: '#limitOutSJcollapsed'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJcollapsed
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Limit BAM sorting memory
#&gt;       description: Maximum available RAM for sorting BAM. If set to 0, it will be
#&gt;         set to the genome index size.
#&gt;       streamable: no
#&gt;       id: '#limitBAMsortRAM'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitBAMsortRAM
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Genome dir name
#&gt;       description: Name of the directory which contains genome files (when genome.tar
#&gt;         is uncompressed).
#&gt;       streamable: no
#&gt;       id: '#genomeDirName'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeDir
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: $job.inputs.genomeDirName || "genomeDir"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:toolDefaultValue: genomeDir
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome files
#&gt;       description: Genome files created using STAR Genome Generate.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: TAR
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 5p bases
#&gt;       description: Number of bases to clip from 5p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip5pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip5pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p bases
#&gt;       description: Number of bases to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p after adapter seq.
#&gt;       description: Number of bases to clip from 3p of each mate after the adapter
#&gt;         clipping. In case only one value is given, it will be assumed the same for
#&gt;         both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAfterAdapterNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAfterAdapterNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: string
#&gt;         type: array
#&gt;       label: Clip 3p adapter sequence
#&gt;       description: Adapter sequence to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterSeq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterSeq
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: float
#&gt;         type: array
#&gt;       label: Max mismatches proportions
#&gt;       description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;         In case only one value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterMMp'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterMMp
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0.1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min segment length
#&gt;       description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;         (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimSegmentMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimSegmentMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '15'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min separation score
#&gt;       description: Minimum difference (separation) between the best chimeric score
#&gt;         and the next one (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreSeparation'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreSeparation
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min total score
#&gt;       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-GT/AG penalty
#&gt;       description: Penalty for a non-GT/AG chimeric junction.
#&gt;       streamable: no
#&gt;       id: '#chimScoreJunctionNonGTAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreJunctionNonGTAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max drop score
#&gt;       description: Max drop (difference) of chimeric score (the sum of scores of all
#&gt;         chimeric segements) from the read length (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreDropMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreDropMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: chimOutType
#&gt;         symbols:
#&gt;         - SeparateSAMold
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Chimeric output type
#&gt;       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;       streamable: no
#&gt;       id: '#chimOutType'
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: SeparateSAMold
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min junction overhang
#&gt;       description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimJunctionOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimJunctionOverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Max windows per read
#&gt;       description: Max number of windows per read (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignWindowsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignWindowsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per window
#&gt;       description: Max number of transcripts per window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per read
#&gt;       description: Max number of different alignments per read to consider (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min mapped length normalized
#&gt;       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLminOverLmate'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLminOverLmate
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min mapped length
#&gt;       description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignSoftClipAtReferenceEnds
#&gt;         symbols:
#&gt;         - 'Yes'
#&gt;         - 'No'
#&gt;         type: enum
#&gt;       label: Soft clipping
#&gt;       description: 'Option which allows soft clipping of alignments at the reference
#&gt;         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.
#&gt;         Yes: Enables soft clipping; No: Disables soft clipping.'
#&gt;       streamable: no
#&gt;       id: '#alignSoftClipAtReferenceEnds'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSoftClipAtReferenceEnds
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: 'Yes'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min overhang
#&gt;       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: 'Min overhang: annotated'
#&gt;       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced
#&gt;         alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJDBoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJDBoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max mates gap
#&gt;       description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;         by (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignMatesGapMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignMatesGapMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min intron size
#&gt;       description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;       streamable: no
#&gt;       id: '#alignIntronMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '21'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max intron size
#&gt;       description: Maximum intron size, if 0, max intron size will be determined by
#&gt;         (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignIntronMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignEndsType
#&gt;         symbols:
#&gt;         - Local
#&gt;         - EndToEnd
#&gt;         type: enum
#&gt;       label: Alignment type
#&gt;       description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;       streamable: no
#&gt;       id: '#alignEndsType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignEndsType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: Local
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Unmapped reads
#&gt;       description: Output of unmapped reads.
#&gt;       streamable: no
#&gt;       id: '#unmapped_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Unmapped.out*'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Transcriptome alignments
#&gt;       description: Alignments translated into transcript coordinates.
#&gt;       streamable: no
#&gt;       id: '#transcriptome_aligned_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Transcriptome*'
#&gt;       sbg:fileTypes: BAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Splice junctions
#&gt;       description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;         Only junctions supported by uniquely mapping reads are reported.
#&gt;       streamable: no
#&gt;       id: '#splice_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*SJ.out.tab'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Reads per gene
#&gt;       description: File with number of reads per gene. A read is counted if it overlaps
#&gt;         (1nt or more) one and only one gene.
#&gt;       streamable: no
#&gt;       id: '#reads_per_gene'
#&gt;       outputBinding:
#&gt;         glob: '*ReadsPerGene*'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Log files
#&gt;       description: Log files produced during alignment.
#&gt;       streamable: no
#&gt;       id: '#log_files'
#&gt;       outputBinding:
#&gt;         glob: '*Log*.out'
#&gt;       sbg:fileTypes: OUT
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Intermediate genome files
#&gt;       description: Archive with genome files produced when annotations are included
#&gt;         on the fly (in the mapping step).
#&gt;       streamable: no
#&gt;       id: '#intermediate_genome'
#&gt;       outputBinding:
#&gt;         glob: '*_STARgenome.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric junctions
#&gt;       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,
#&gt;         'Chimeric Junctions' won't be output.
#&gt;       streamable: no
#&gt;       id: '#chimeric_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*Chimeric.out.junction'
#&gt;       sbg:fileTypes: JUNCTION
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric alignments
#&gt;       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;         Alignment SAM and Chimeric Junctions outputs.
#&gt;       streamable: no
#&gt;       id: '#chimeric_alignments'
#&gt;       outputBinding:
#&gt;         glob: '*.Chimeric.out.sam'
#&gt;       sbg:fileTypes: SAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Aligned SAM/BAM
#&gt;       description: Aligned sequence in SAM/BAM format.
#&gt;       streamable: no
#&gt;       id: '#aligned_reads'
#&gt;       outputBinding:
#&gt;         glob:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;                 sort_name = '.sortedByCoord'
#&gt;               }
#&gt;               else {
#&gt;                 sort_name = ''
#&gt;               }
#&gt;               if ($job.inputs.outSAMtype == 'BAM') {
#&gt;                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;               }
#&gt;               else {
#&gt;                 sam_name = "*.Aligned.out.sam"
#&gt;               }
#&gt;               return sam_name
#&gt;             }
#&gt;           class: Expression
#&gt;       sbg:fileTypes: SAM, BAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     label: STAR
#&gt;     description: STAR is an ultrafast universal RNA-seq aligner. It has very high
#&gt;       mapping speed, accurate alignment of contiguous and spliced reads, detection
#&gt;       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works
#&gt;       with reads starting from lengths ~15 bases up to ~300 bases. In case of having
#&gt;       longer reads, use of STAR Long is recommended.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - tar
#&gt;     - -xvf
#&gt;     - class: Expression
#&gt;       script: $job.inputs.genome.path
#&gt;       engine: '#cwl-js-engine'
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             file = [].concat($job.inputs.reads)[0].path
#&gt;             extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;             if (extension == "gz") {
#&gt;               return "--readFilesCommand zcat"
#&gt;             } else if (extension == "bz2") {
#&gt;               return "--readFilesCommand bzcat"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n
#&gt;           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",
#&gt;           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)
#&gt;           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n
#&gt;           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",
#&gt;           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)
#&gt;           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)
#&gt;           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)
#&gt;           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n
#&gt;           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"
#&gt;           \")\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.twopassMode == "Basic") {
#&gt;               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.chimOutType == "Within") {
#&gt;               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;             }
#&gt;             else {
#&gt;               return "--chimOutType SeparateSAMold"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n
#&gt;           \   if (value == \"\") {\n      return\n    }\n    else {\n      return
#&gt;           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',
#&gt;           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',
#&gt;           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)
#&gt;           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)
#&gt;           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;           \".concat(param_list.join(\" \"))\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;               return "--quantMode ".concat($job.inputs.quantMode)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 100
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n
#&gt;           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile
#&gt;           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       prefix: --outFileNamePrefix
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;         class: Expression
#&gt;     - position: 101
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"
#&gt;           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,
#&gt;           \" \", mate1fq)\n  }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 624.0
#&gt;     'y': 323
#&gt;   sbg:x: 700.0
#&gt;   sbg:y: 200.0
#&gt; sbg:canvas_zoom: 0.6
#&gt; sbg:canvas_y: -16
#&gt; sbg:canvas_x: -41
#&gt; sbg:batchInput: '#sjdbGTFfile'
#&gt; sbg:batchBy:
#&gt;   type: item
#&gt; </div><div class='input'><span class='co'># add source to id</span>
<span class='va'>f1</span><span class='op'>$</span><span class='fu'>link_map</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                                      id
#&gt; 1  #STAR_Genome_Generate.sjdbGTFtagExonParentTranscript
#&gt; 2        #STAR_Genome_Generate.sjdbGTFtagExonParentGene
#&gt; 3                     #STAR_Genome_Generate.sjdbGTFfile
#&gt; 4                #STAR_Genome_Generate.genomeFastaFiles
#&gt; 5                     #SBG_FASTQ_Quality_Detector.fastq
#&gt; 6                             #Picard_SortSam.input_bam
#&gt; 7                           #STAR.winAnchorMultimapNmax
#&gt; 8                              #STAR.winAnchorDistNbins
#&gt; 9                                     #STAR.sjdbGTFfile
#&gt; 10                                          #STAR.reads
#&gt; 11                                         #STAR.genome
#&gt; 12                                      #unmapped_reads
#&gt; 13                         #transcriptome_aligned_reads
#&gt; 14                                    #splice_junctions
#&gt; 15                                      #reads_per_gene
#&gt; 16                                           #log_files
#&gt; 17                                  #chimeric_junctions
#&gt; 18                                 #intermediate_genome
#&gt; 19                                 #intermediate_genome
#&gt; 20                                 #chimeric_alignments
#&gt; 21                                          #sorted_bam
#&gt; 22                                              #result
#&gt; 23                                 #intermediate_genome
#&gt; 24                                 #intermediate_genome
#&gt;                             source   type
#&gt; 1  #sjdbGTFtagExonParentTranscript  input
#&gt; 2        #sjdbGTFtagExonParentGene  input
#&gt; 3                     #sjdbGTFfile  input
#&gt; 4                #genomeFastaFiles  input
#&gt; 5                           #fastq  input
#&gt; 6              #STAR.aligned_reads  input
#&gt; 7           #winAnchorMultimapNmax  input
#&gt; 8              #winAnchorDistNbins  input
#&gt; 9                     #sjdbGTFfile  input
#&gt; 10                          #reads  input
#&gt; 11    #STAR_Genome_Generate.genome  input
#&gt; 12                 #unmapped_reads output
#&gt; 13    #transcriptome_aligned_reads output
#&gt; 14               #splice_junctions output
#&gt; 15                 #reads_per_gene output
#&gt; 16                      #log_files output
#&gt; 17             #chimeric_junctions output
#&gt; 18            #intermediate_genome output
#&gt; 19             intermediate_genome output
#&gt; 20            #chimeric_alignments output
#&gt; 21                     #sorted_bam output
#&gt; 22                         #result output
#&gt; 23            #intermediate_genome output
#&gt; 24             intermediate_genome output</div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>add_source_to_id</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"test1"</span>, <span class='st'>"test2"</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"#STAR.genome"</span>, <span class='st'>"#STAR.reads"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; sbg:validationErrors: []
#&gt; sbg:sbgMaintained: no
#&gt; sbg:latestRevision: 2
#&gt; sbg:toolAuthor: Seven Bridges Genomics
#&gt; sbg:createdOn: 1463601910
#&gt; sbg:categories:
#&gt; - Alignment
#&gt; - RNA
#&gt; sbg:contributors:
#&gt; - tengfei
#&gt; sbg:project: tengfei/quickstart
#&gt; sbg:createdBy: tengfei
#&gt; sbg:toolkitVersion: 2.4.2a
#&gt; sbg:id: tengfei/quickstart/rna-seq-alignment-star-demo/2
#&gt; sbg:license: Apache License 2.0
#&gt; sbg:revision: 2
#&gt; sbg:modifiedOn: 1463601974
#&gt; sbg:modifiedBy: tengfei
#&gt; sbg:revisionsInfo:
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601910
#&gt;   sbg:revision: 0
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601952
#&gt;   sbg:revision: 1
#&gt; - sbg:modifiedBy: tengfei
#&gt;   sbg:modifiedOn: 1463601974
#&gt;   sbg:revision: 2
#&gt; sbg:toolkit: STAR
#&gt; id: '#tengfei/quickstart/rna-seq-alignment-star-demo/2'
#&gt; inputs:
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Read sequence
#&gt;   description: Read sequence.
#&gt;   streamable: no
#&gt;   id: '#reads'
#&gt;   inputBinding:
#&gt;     position: 10
#&gt;     separate: yes
#&gt;     itemSeparator: ' '
#&gt;     valueFrom:
#&gt;       engine: '#cwl-js-engine'
#&gt;       script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp = []\n
#&gt;         \ \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n  }else if
#&gt;         (list.length == 2){    \n    \n    left = \"\"\n    right = \"\"\n      \n
#&gt;         \   for (index = 0; index &lt; list.length; ++index) {\n      \n      if (list[index].metadata
#&gt;         != null){\n        if (list[index].metadata.paired_end == 1){\n          left
#&gt;         = list[index].path\n        }else if (list[index].metadata.paired_end == 2){\n
#&gt;         \         right = list[index].path\n        }\n      }\n    }\n    \n    if
#&gt;         (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n      resp.push(right)\n
#&gt;         \   }\n  }\n  else if (list.length &gt; 2){\n    left = []\n    right = []\n
#&gt;         \     \n    for (index = 0; index &lt; list.length; ++index) {\n      \n      if
#&gt;         (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;         == 1){\n          left.push(list[index].path)\n        }else if (list[index].metadata.paired_end
#&gt;         == 2){\n          right.push(list[index].path)\n        }\n      }\n    }\n
#&gt;         \   left_join = left.join()\n    right_join = right.join()\n    if (left !=
#&gt;         [] &amp;&amp; right != []){      \n      resp.push(left_join)\n      resp.push(right_join)\n
#&gt;         \   }\t\n  }\n  \n  if(resp.length &gt; 0){    \n    return \"--readFilesIn \".concat(resp.join(\"
#&gt;         \"))\n  }\n}"
#&gt;       class: Expression
#&gt;     shellQuote: no
#&gt;     sbg:cmdInclude: yes
#&gt;     streamable: no
#&gt;     separator: ' '
#&gt;   sbg:category: Basic
#&gt;   sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: sjdbGTFfile
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFfile'
#&gt;   sbg:x: 160.4999759
#&gt;   sbg:y: 195.0833106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: fastq
#&gt;   streamable: no
#&gt;   id: '#fastq'
#&gt;   sbg:x: 164.2499914
#&gt;   sbg:y: 323.7499502
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - File
#&gt;   label: genomeFastaFiles
#&gt;   streamable: no
#&gt;   id: '#genomeFastaFiles'
#&gt;   sbg:x: 167.7499601
#&gt;   sbg:y: 469.9999106
#&gt;   sbg:includeInPorts: yes
#&gt;   required: yes
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Exons' parents name
#&gt;   description: Tag name to be used as exons' transcript-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentTranscript'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 350.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: transcript_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - string
#&gt;   label: Gene name
#&gt;   description: Tag name to be used as exons' gene-parents.
#&gt;   streamable: no
#&gt;   id: '#sjdbGTFtagExonParentGene'
#&gt;   sbg:category: Splice junctions db parameters
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: gene_id
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max loci anchors
#&gt;   description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorMultimapNmax'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 450.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '50'
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - int
#&gt;   label: Max bins between anchors
#&gt;   description: Max number of bins between two anchors that allows aggregation of anchors
#&gt;     into one window (int&gt;0).
#&gt;   streamable: no
#&gt;   id: '#winAnchorDistNbins'
#&gt;   sbg:category: Windows, Anchors, Binning
#&gt;   sbg:x: 200.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   sbg:toolDefaultValue: '9'
#&gt;   required: no
#&gt; outputs:
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Unmapped reads
#&gt;   description: Output of unmapped reads.
#&gt;   streamable: no
#&gt;   id: '#unmapped_reads'
#&gt;   source: '#unmapped_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 200.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Transcriptome alignments
#&gt;   description: Alignments translated into transcript coordinates.
#&gt;   streamable: no
#&gt;   id: '#transcriptome_aligned_reads'
#&gt;   source: '#transcriptome_aligned_reads'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 233.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Splice junctions
#&gt;   description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;     Only junctions supported by uniquely mapping reads are reported.
#&gt;   streamable: no
#&gt;   id: '#splice_junctions'
#&gt;   source: '#splice_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 266.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Reads per gene
#&gt;   description: File with number of reads per gene. A read is counted if it overlaps
#&gt;     (1nt or more) one and only one gene.
#&gt;   streamable: no
#&gt;   id: '#reads_per_gene'
#&gt;   source: '#reads_per_gene'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 300.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - items: File
#&gt;     type: array
#&gt;   label: Log files
#&gt;   description: Log files produced during alignment.
#&gt;   streamable: no
#&gt;   id: '#log_files'
#&gt;   source: '#log_files'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 333.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric junctions
#&gt;   description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0, 'Chimeric
#&gt;     Junctions' won't be output.
#&gt;   streamable: no
#&gt;   id: '#chimeric_junctions'
#&gt;   source: '#chimeric_junctions'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 366.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 400.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Chimeric alignments
#&gt;   description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;     Alignment SAM and Chimeric Junctions outputs.
#&gt;   streamable: no
#&gt;   id: '#chimeric_alignments'
#&gt;   source: '#chimeric_alignments'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 433.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Sorted BAM/SAM
#&gt;   description: Sorted BAM or SAM file.
#&gt;   streamable: no
#&gt;   id: '#sorted_bam'
#&gt;   source: '#sorted_bam'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 466.6666667
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Result
#&gt;   description: Source FASTQ file with updated metadata.
#&gt;   streamable: no
#&gt;   id: '#result'
#&gt;   source: '#result'
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 500.0
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; - type:
#&gt;   - 'null'
#&gt;   - File
#&gt;   label: Intermediate genome files
#&gt;   description: Archive with genome files produced when annotations are included on
#&gt;     the fly (in the mapping step).
#&gt;   streamable: no
#&gt;   id: '#intermediate_genome'
#&gt;   source:
#&gt;   - '#intermediate_genome'
#&gt;   - intermediate_genome
#&gt;   sbg:x: 800.0
#&gt;   sbg:y: 533.3333333
#&gt;   sbg:includeInPorts: yes
#&gt;   required: no
#&gt; requirements:
#&gt; - class: CreateFileRequirement
#&gt;   fileDef: []
#&gt; hints:
#&gt; - class: sbg:AWSInstanceType
#&gt;   value: c3.8xlarge
#&gt; label: RNA-seq Alignment - STAR
#&gt; description: "Alignment to a reference genome and transcriptome presents the first
#&gt;   step of RNA-Seq analysis. This pipeline uses STAR, an ultrafast RNA-seq aligner
#&gt;   capable of mapping full length RNA sequences and detecting de novo canonical junctions,
#&gt;   non-canonical splices, and chimeric (fusion) transcripts. It is optimized for mammalian
#&gt;   sequence reads, but fine tuning of its parameters enables customization to satisfy
#&gt;   unique needs.\n\nSTAR accepts one file per sample (or two files for paired-end data).
#&gt;   \ \nSplice junction annotations can optionally be collected from splice junction
#&gt;   databases. Set the \"Overhang length\" parameter to a value larger than zero in
#&gt;   order to use splice junction databases. For constant read length, this value should
#&gt;   (ideally) be equal to mate length decreased by 1; for long reads with non-constant
#&gt;   length, this value should be 100 (pipeline default). \nFastQC Analysis on FASTQ
#&gt;   files reveals read length distribution. STAR can detect chimeric transcripts, but
#&gt;   parameter \"Min segment length\" in \"Chimeric Alignments\" category must be adjusted
#&gt;   to a desired minimum chimeric segment length. Aligned reads are reported in BAM
#&gt;   format and can be viewed in a genome browser (such as IGV). A file containing detected
#&gt;   splice junctions is also produced.\n\nUnmapped reads are reported in FASTQ format
#&gt;   and can be included in an output BAM file. The \"Output unmapped reads\" and \"Write
#&gt;   unmapped in SAM\" parameters enable unmapped output type selection."
#&gt; class: Workflow
#&gt; steps:
#&gt; - id: '#STAR_Genome_Generate'
#&gt;   inputs:
#&gt;   - id: '#STAR_Genome_Generate.sjdbScore'
#&gt;   - id: '#STAR_Genome_Generate.sjdbOverhang'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentTranscript'
#&gt;     source: '#sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFtagExonParentGene'
#&gt;     source: '#sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR_Genome_Generate.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAsparseD'
#&gt;   - id: '#STAR_Genome_Generate.genomeSAindexNbases'
#&gt;   - id: '#STAR_Genome_Generate.genomeFastaFiles'
#&gt;     source: '#genomeFastaFiles'
#&gt;   - id: '#STAR_Genome_Generate.genomeChrBinNbits'
#&gt;   outputs:
#&gt;   - id: '#STAR_Genome_Generate.genome'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 1
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         sjdbScore: 0
#&gt;         sjdbGTFfeatureExon: sjdbGTFfeatureExon
#&gt;         sjdbOverhang: 0
#&gt;         sjdbGTFtagExonParentTranscript: sjdbGTFtagExonParentTranscript
#&gt;         genomeChrBinNbits: genomeChrBinNbits
#&gt;         genomeSAsparseD: 0
#&gt;         sjdbGTFfile:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /demo/test-files/chr20.gtf
#&gt;         sjdbGTFtagExonParentGene: sjdbGTFtagExonParentGene
#&gt;         genomeFastaFiles:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /sbgenomics/test-data/chr20.fa
#&gt;         sjdbGTFchrPrefix: sjdbGTFchrPrefix
#&gt;         genomeSAindexNbases: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911469
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 1
#&gt;     sbg:cmdPreview: mkdir genomeDir &amp;&amp; /opt/STAR --runMode genomeGenerate --genomeDir
#&gt;       ./genomeDir --runThreadN 15 --genomeFastaFiles /sbgenomics/test-data/chr20.fa
#&gt;       --genomeChrBinNbits genomeChrBinNbits --genomeSAindexNbases 0 --genomeSAsparseD
#&gt;       0 --sjdbGTFfeatureExon sjdbGTFfeatureExon --sjdbGTFtagExonParentTranscript sjdbGTFtagExonParentTranscript
#&gt;       --sjdbGTFtagExonParentGene sjdbGTFtagExonParentGene --sjdbOverhang 0 --sjdbScore
#&gt;       0 --sjdbGTFchrPrefix sjdbGTFchrPrefix  --sjdbGTFfile /demo/test-files/chr20.gtf  &amp;&amp;
#&gt;       tar -vcf genome.tar ./genomeDir /sbgenomics/test-data/chr20.fa
#&gt;     sbg:modifiedOn: 1450911470
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911469
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911470
#&gt;       sbg:revision: 1
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star-genome-generate/1
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbScore
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbOverhang
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentTranscript
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFtagExonParentGene
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFfeatureExon
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --sjdbGTFchrPrefix
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Splice junctions db parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Suffux array sparsity
#&gt;       description: 'Distance between indices: use bigger numbers to decrease needed
#&gt;         RAM at the cost of mapping speed reduction (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#genomeSAsparseD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAsparseD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Pre-indexing string length
#&gt;       description: Length (bases) of the SA pre-indexing string. Typically between
#&gt;         10 and 15. Longer strings will use much more memory, but allow faster searches.
#&gt;         For small genomes, this number needs to be scaled down, with a typical value
#&gt;         of min(14, log2(GenomeLength)/2 - 1). For example, for 1 megaBase genome,
#&gt;         this is equal to 9, for 100 kiloBase genome, this is equal to 7.
#&gt;       streamable: no
#&gt;       id: '#genomeSAindexNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeSAindexNbases
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '14'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome fasta files
#&gt;       description: Reference sequence to which to align the reads.
#&gt;       streamable: no
#&gt;       id: '#genomeFastaFiles'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeFastaFiles
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FA
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Bins size
#&gt;       description: 'Set log2(chrBin), where chrBin is the size (bits) of the bins
#&gt;         for genome storage: each chromosome will occupy an integer number of bins.
#&gt;         If you are using a genome with a large (&gt;5,000) number of chrosomes/scaffolds,
#&gt;         you may need to reduce this number to reduce RAM consumption. The following
#&gt;         scaling is recomended: genomeChrBinNbits = min(18, log2(GenomeLength/NumberOfReferences)).
#&gt;         For example, for 3 gigaBase genome with 100,000 chromosomes/scaffolds, this
#&gt;         is equal to 15.'
#&gt;       streamable: no
#&gt;       id: '#genomeChrBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeChrBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Genome generation parameters
#&gt;       sbg:toolDefaultValue: '18'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Genome Files
#&gt;       description: Genome files comprise binary genome sequence, suffix arrays, text
#&gt;         chromosome names/lengths, splice junctions coordinates, and transcripts/genes
#&gt;         information.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       outputBinding:
#&gt;         glob: '*.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     label: STAR Genome Generate
#&gt;     description: STAR Genome Generate is a tool that generates genome index files.
#&gt;       One set of files should be generated per each genome/annotation combination.
#&gt;       Once produced, these files could be used as long as genome/annotation combination
#&gt;       stays the same. Also, STAR Genome Generate which produced these files and STAR
#&gt;       aligner using them must be the same toolkit version.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - mkdir
#&gt;     - genomeDir
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runMode
#&gt;     - genomeGenerate
#&gt;     - --genomeDir
#&gt;     - ./genomeDir
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 99
#&gt;       separate: yes
#&gt;       valueFrom: '&amp;&amp; tar -vcf genome.tar ./genomeDir'
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 384.0832266
#&gt;     'y': 446.4998957
#&gt;   sbg:x: 100.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#SBG_FASTQ_Quality_Detector'
#&gt;   inputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.fastq'
#&gt;     source: '#fastq'
#&gt;   outputs:
#&gt;   - id: '#SBG_FASTQ_Quality_Detector.result'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 3
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 1000
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         fastq:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: /path/to/fastq.ext
#&gt;     sbg:toolAuthor: Seven Bridges Genomics
#&gt;     sbg:createdOn: 1450911312
#&gt;     sbg:categories:
#&gt;     - FASTQ-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:project: bix-demo/sbgtools-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     sbg:license: Apache License 2.0
#&gt;     sbg:revision: 3
#&gt;     sbg:cmdPreview: python /opt/sbg_fastq_quality_scale_detector.py --fastq /path/to/fastq.ext
#&gt;       /path/to/fastq.ext
#&gt;     sbg:modifiedOn: 1450911314
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911312
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911314
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911313
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: SBGTools
#&gt;     id: sevenbridges/public-apps/sbg-fastq-quality-detector/3
#&gt;     inputs:
#&gt;     - type:
#&gt;       - File
#&gt;       label: Fastq
#&gt;       description: FASTQ file.
#&gt;       streamable: no
#&gt;       id: '#fastq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --fastq
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       required: yes
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Result
#&gt;       description: Source FASTQ file with updated metadata.
#&gt;       streamable: no
#&gt;       id: '#result'
#&gt;       outputBinding:
#&gt;         glob: '*.fastq'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     requirements:
#&gt;     - class: CreateFileRequirement
#&gt;       fileDef: []
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/tziotas/sbg_fastq_quality_scale_detector:1.0
#&gt;       dockerImageId: ''
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 1000
#&gt;     label: SBG FASTQ Quality Detector
#&gt;     description: FASTQ Quality Scale Detector detects which quality encoding scheme
#&gt;       was used in your reads and automatically enters the proper value in the "Quality
#&gt;       Scale" metadata field.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - python
#&gt;     - /opt/sbg_fastq_quality_scale_detector.py
#&gt;     arguments: []
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 375.3333179
#&gt;     'y': 323.5833156
#&gt;   sbg:x: 300.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#Picard_SortSam'
#&gt;   inputs:
#&gt;   - id: '#Picard_SortSam.validation_stringency'
#&gt;     default: SILENT
#&gt;   - id: '#Picard_SortSam.sort_order'
#&gt;     default: Coordinate
#&gt;   - id: '#Picard_SortSam.quiet'
#&gt;   - id: '#Picard_SortSam.output_type'
#&gt;   - id: '#Picard_SortSam.memory_per_job'
#&gt;   - id: '#Picard_SortSam.max_records_in_ram'
#&gt;   - id: '#Picard_SortSam.input_bam'
#&gt;     source: '#STAR.aligned_reads'
#&gt;   - id: '#Picard_SortSam.create_index'
#&gt;     default: 'True'
#&gt;   - id: '#Picard_SortSam.compression_level'
#&gt;   outputs:
#&gt;   - id: '#Picard_SortSam.sorted_bam'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 2
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 2048
#&gt;         cpu: 1
#&gt;       inputs:
#&gt;         sort_order: Coordinate
#&gt;         input_bam:
#&gt;           path: /root/dir/example.tested.bam
#&gt;         memory_per_job: 2048
#&gt;         output_type: ~
#&gt;         create_index: ~
#&gt;     sbg:toolAuthor: Broad Institute
#&gt;     sbg:createdOn: 1450911168
#&gt;     sbg:categories:
#&gt;     - SAM/BAM-Processing
#&gt;     sbg:contributors:
#&gt;     - bix-demo
#&gt;     sbg:links:
#&gt;     - id: http://broadinstitute.github.io/picard/index.html
#&gt;       label: Homepage
#&gt;     - id: https://github.com/broadinstitute/picard/releases/tag/1.138
#&gt;       label: Source Code
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Wiki
#&gt;     - id: https://github.com/broadinstitute/picard/zipball/master
#&gt;       label: Download
#&gt;     - id: http://broadinstitute.github.io/picard/
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/picard-1-140-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: '1.140'
#&gt;     sbg:id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     sbg:license: MIT License, Apache 2.0 Licence
#&gt;     sbg:revision: 2
#&gt;     sbg:cmdPreview: java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam
#&gt;       OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate   INPUT=/root/dir/example.tested.bam
#&gt;       SORT_ORDER=coordinate  /root/dir/example.tested.bam
#&gt;     sbg:modifiedOn: 1450911170
#&gt;     sbg:modifiedBy: bix-demo
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911168
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911169
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911170
#&gt;       sbg:revision: 2
#&gt;     sbg:toolkit: Picard
#&gt;     id: sevenbridges/public-apps/picard-sortsam-1-140/2
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: validation_stringency
#&gt;         symbols:
#&gt;         - STRICT
#&gt;         - LENIENT
#&gt;         - SILENT
#&gt;         type: enum
#&gt;       label: Validation stringency
#&gt;       description: 'Validation stringency for all SAM files read by this program.
#&gt;         Setting stringency to SILENT can improve performance when processing a BAM
#&gt;         file in which variable-length data (read, qualities, tags) do not otherwise
#&gt;         need to be decoded. This option can be set to ''null'' to clear the default
#&gt;         value. Possible values: {STRICT, LENIENT, SILENT}.'
#&gt;       streamable: no
#&gt;       id: '#validation_stringency'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: VALIDATION_STRINGENCY=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.validation_stringency)
#&gt;               {
#&gt;                 return $job.inputs.validation_stringency
#&gt;               }
#&gt;               else
#&gt;               {
#&gt;                 return "SILENT"
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SILENT
#&gt;       required: no
#&gt;     - type:
#&gt;       - name: sort_order
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - Queryname
#&gt;         - Coordinate
#&gt;         type: enum
#&gt;       label: Sort order
#&gt;       description: 'Sort order of the output file. Possible values: {unsorted, queryname,
#&gt;         coordinate}.'
#&gt;       streamable: no
#&gt;       id: '#sort_order'
#&gt;       inputBinding:
#&gt;         position: 3
#&gt;         prefix: SORT_ORDER=
#&gt;         separate: no
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               p = $job.inputs.sort_order.toLowerCase()
#&gt;               return p
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: Coordinate
#&gt;       sbg:altPrefix: SO
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quiet
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Quiet
#&gt;       description: 'This parameter indicates whether to suppress job-summary info
#&gt;         on System.err. This option can be set to ''null'' to clear the default value.
#&gt;         Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#quiet'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: QUIET=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: output_type
#&gt;         symbols:
#&gt;         - BAM
#&gt;         - SAM
#&gt;         - SAME AS INPUT
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Since Picard tools can output both SAM and BAM files, user can
#&gt;         choose the format of the output file.
#&gt;       streamable: no
#&gt;       id: '#output_type'
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: SAME AS INPUT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Memory per job
#&gt;       description: Amount of RAM memory to be used per job. Defaults to 2048 MB for
#&gt;         single threaded jobs.
#&gt;       streamable: no
#&gt;       id: '#memory_per_job'
#&gt;       sbg:toolDefaultValue: '2048'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max records in RAM
#&gt;       description: When writing SAM files that need to be sorted, this parameter will
#&gt;         specify the number of records stored in RAM before spilling to disk. Increasing
#&gt;         this number reduces the number of file handles needed to sort a SAM file,
#&gt;         and increases the amount of RAM needed. This option can be set to 'null' to
#&gt;         clear the default value.
#&gt;       streamable: no
#&gt;       id: '#max_records_in_ram'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: MAX_RECORDS_IN_RAM=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '500000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Input BAM
#&gt;       description: The BAM or SAM file to sort.
#&gt;       streamable: no
#&gt;       id: '#input_bam'
#&gt;       inputBinding:
#&gt;         position: 1
#&gt;         prefix: INPUT=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: File inputs
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;       sbg:altPrefix: I
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: create_index
#&gt;         symbols:
#&gt;         - 'True'
#&gt;         - 'False'
#&gt;         type: enum
#&gt;       label: Create index
#&gt;       description: 'This parameter indicates whether to create a BAM index when writing
#&gt;         a coordinate-sorted BAM file. This option can be set to ''null'' to clear
#&gt;         the default value. Possible values: {true, false}.'
#&gt;       streamable: no
#&gt;       id: '#create_index'
#&gt;       inputBinding:
#&gt;         position: 5
#&gt;         prefix: CREATE_INDEX=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: 'False'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Compression level
#&gt;       description: Compression level for all compressed files created (e.g. BAM and
#&gt;         GELI). This option can be set to 'null' to clear the default value.
#&gt;       streamable: no
#&gt;       id: '#compression_level'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: COMPRESSION_LEVEL=
#&gt;         separate: no
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Other input types
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Sorted BAM/SAM
#&gt;       description: Sorted BAM or SAM file.
#&gt;       streamable: no
#&gt;       id: '#sorted_bam'
#&gt;       outputBinding:
#&gt;         glob: '*.sorted.?am'
#&gt;       sbg:fileTypes: BAM, SAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;       engineCommand: cwl-engine.js
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
#&gt;       dockerImageId: eab0e70b6629
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 1
#&gt;     - class: sbg:MemRequirement
#&gt;       value:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  if($job.inputs.memory_per_job){\n  \treturn $job.inputs.memory_per_job\n
#&gt;           \ }\n  \treturn 2048\n}"
#&gt;         class: Expression
#&gt;     label: Picard SortSam
#&gt;     description: Picard SortSam sorts the input SAM or BAM. Input and output formats
#&gt;       are determined by the file extension.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - java
#&gt;     - class: Expression
#&gt;       script: "{   \n  if($job.inputs.memory_per_job){\n    return '-Xmx'.concat($job.inputs.memory_per_job,
#&gt;         'M')\n  }   \n  \treturn '-Xmx2048M'\n}"
#&gt;       engine: '#cwl-js-engine'
#&gt;     - -jar
#&gt;     - /opt/picard-tools-1.140/picard.jar
#&gt;     - SortSam
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       prefix: OUTPUT=
#&gt;       separate: no
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  ext = $job.inputs.output_type\n\nif
#&gt;           (ext === \"BAM\")\n{\n    return filename.split('.').slice(0, -1).concat(\"sorted.bam\").join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    }\n\nelse if (ext === \"SAM\")\n{\n    return filename.split('.').slice(0,
#&gt;           -1).concat(\"sorted.sam\").join('.').replace(/^.*[\\\\\\/]/, '')\n}\n\nelse
#&gt;           \n{\n\treturn filename.split('.').slice(0, -1).concat(\"sorted.\"+filename.split('.').slice(-1)[0]).join(\".\").replace(/^.*[\\\\\\/]/,
#&gt;           '')\n}\n}"
#&gt;         class: Expression
#&gt;     - position: 1000
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  filename = $job.inputs.input_bam.path\n  \n  /* figuring out
#&gt;           output file type */\n  ext = $job.inputs.output_type\n  if (ext === \"BAM\")\n
#&gt;           \ {\n    out_extension = \"BAM\"\n  }\n  else if (ext === \"SAM\")\n  {\n
#&gt;           \   out_extension = \"SAM\"\n  }\n  else \n  {\n\tout_extension = filename.split('.').slice(-1)[0].toUpperCase()\n
#&gt;           \ }  \n  \n  /* if exist moving .bai in bam.bai */\n  if ($job.inputs.create_index
#&gt;           === 'True' &amp;&amp; $job.inputs.sort_order === 'Coordinate' &amp;&amp; out_extension ==
#&gt;           \"BAM\")\n  {\n    \n    old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\\\\/]/,
#&gt;           '')\n    return \"; mv \" + \" \" + old_name + \" \" + new_name\n  }\n\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 773.0831807
#&gt;     'y': 470.9165939
#&gt;   sbg:x: 500.0
#&gt;   sbg:y: 200.0
#&gt; - id: '#STAR'
#&gt;   inputs:
#&gt;   - id: '#STAR.winFlankNbins'
#&gt;   - id: '#STAR.winBinNbits'
#&gt;   - id: '#STAR.winAnchorMultimapNmax'
#&gt;     source: '#winAnchorMultimapNmax'
#&gt;   - id: '#STAR.winAnchorDistNbins'
#&gt;     source: '#winAnchorDistNbins'
#&gt;   - id: '#STAR.twopassMode'
#&gt;   - id: '#STAR.twopass1readsN'
#&gt;   - id: '#STAR.sjdbScore'
#&gt;   - id: '#STAR.sjdbOverhang'
#&gt;     default: 100
#&gt;   - id: '#STAR.sjdbInsertSave'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentTranscript'
#&gt;   - id: '#STAR.sjdbGTFtagExonParentGene'
#&gt;   - id: '#STAR.sjdbGTFfile'
#&gt;     source: '#sjdbGTFfile'
#&gt;   - id: '#STAR.sjdbGTFfeatureExon'
#&gt;   - id: '#STAR.sjdbGTFchrPrefix'
#&gt;   - id: '#STAR.seedSearchStartLmaxOverLread'
#&gt;   - id: '#STAR.seedSearchStartLmax'
#&gt;   - id: '#STAR.seedSearchLmax'
#&gt;   - id: '#STAR.seedPerWindowNmax'
#&gt;   - id: '#STAR.seedPerReadNmax'
#&gt;   - id: '#STAR.seedNoneLociPerWindow'
#&gt;   - id: '#STAR.seedMultimapNmax'
#&gt;   - id: '#STAR.scoreStitchSJshift'
#&gt;   - id: '#STAR.scoreInsOpen'
#&gt;   - id: '#STAR.scoreInsBase'
#&gt;   - id: '#STAR.scoreGenomicLengthLog2scale'
#&gt;   - id: '#STAR.scoreGapNoncan'
#&gt;   - id: '#STAR.scoreGapGCAG'
#&gt;   - id: '#STAR.scoreGapATAC'
#&gt;   - id: '#STAR.scoreGap'
#&gt;   - id: '#STAR.scoreDelOpen'
#&gt;   - id: '#STAR.scoreDelBase'
#&gt;   - id: '#STAR.rg_seq_center'
#&gt;   - id: '#STAR.rg_sample_id'
#&gt;   - id: '#STAR.rg_platform_unit_id'
#&gt;   - id: '#STAR.rg_platform'
#&gt;   - id: '#STAR.rg_mfl'
#&gt;   - id: '#STAR.rg_library_id'
#&gt;   - id: '#STAR.reads'
#&gt;     source:
#&gt;     - '#reads'
#&gt;     - test2
#&gt;   - id: '#STAR.readMatesLengthsIn'
#&gt;   - id: '#STAR.readMapNumber'
#&gt;   - id: '#STAR.quantTranscriptomeBan'
#&gt;   - id: '#STAR.quantMode'
#&gt;     default: TranscriptomeSAM
#&gt;   - id: '#STAR.outSortingType'
#&gt;     default: SortedByCoordinate
#&gt;   - id: '#STAR.outSJfilterReads'
#&gt;   - id: '#STAR.outSJfilterOverhangMin'
#&gt;   - id: '#STAR.outSJfilterIntronMaxVsReadN'
#&gt;   - id: '#STAR.outSJfilterDistToOtherSJmin'
#&gt;   - id: '#STAR.outSJfilterCountUniqueMin'
#&gt;   - id: '#STAR.outSJfilterCountTotalMin'
#&gt;   - id: '#STAR.outSAMunmapped'
#&gt;   - id: '#STAR.outSAMtype'
#&gt;     default: BAM
#&gt;   - id: '#STAR.outSAMstrandField'
#&gt;   - id: '#STAR.outSAMreadID'
#&gt;   - id: '#STAR.outSAMprimaryFlag'
#&gt;   - id: '#STAR.outSAMorder'
#&gt;   - id: '#STAR.outSAMmode'
#&gt;   - id: '#STAR.outSAMmapqUnique'
#&gt;   - id: '#STAR.outSAMheaderPG'
#&gt;   - id: '#STAR.outSAMheaderHD'
#&gt;   - id: '#STAR.outSAMflagOR'
#&gt;   - id: '#STAR.outSAMflagAND'
#&gt;   - id: '#STAR.outSAMattributes'
#&gt;   - id: '#STAR.outReadsUnmapped'
#&gt;     default: Fastx
#&gt;   - id: '#STAR.outQSconversionAdd'
#&gt;   - id: '#STAR.outFilterType'
#&gt;   - id: '#STAR.outFilterScoreMinOverLread'
#&gt;   - id: '#STAR.outFilterScoreMin'
#&gt;   - id: '#STAR.outFilterMultimapScoreRange'
#&gt;   - id: '#STAR.outFilterMultimapNmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverReadLmax'
#&gt;   - id: '#STAR.outFilterMismatchNoverLmax'
#&gt;   - id: '#STAR.outFilterMismatchNmax'
#&gt;   - id: '#STAR.outFilterMatchNminOverLread'
#&gt;   - id: '#STAR.outFilterMatchNmin'
#&gt;   - id: '#STAR.outFilterIntronMotifs'
#&gt;   - id: '#STAR.limitSjdbInsertNsj'
#&gt;   - id: '#STAR.limitOutSJoneRead'
#&gt;   - id: '#STAR.limitOutSJcollapsed'
#&gt;   - id: '#STAR.limitBAMsortRAM'
#&gt;   - id: '#STAR.genomeDirName'
#&gt;   - id: '#STAR.genome'
#&gt;     source:
#&gt;     - '#STAR_Genome_Generate.genome'
#&gt;     - test1
#&gt;   - id: '#STAR.clip5pNbases'
#&gt;   - id: '#STAR.clip3pNbases'
#&gt;   - id: '#STAR.clip3pAfterAdapterNbases'
#&gt;   - id: '#STAR.clip3pAdapterSeq'
#&gt;   - id: '#STAR.clip3pAdapterMMp'
#&gt;   - id: '#STAR.chimSegmentMin'
#&gt;   - id: '#STAR.chimScoreSeparation'
#&gt;   - id: '#STAR.chimScoreMin'
#&gt;   - id: '#STAR.chimScoreJunctionNonGTAG'
#&gt;   - id: '#STAR.chimScoreDropMax'
#&gt;   - id: '#STAR.chimOutType'
#&gt;   - id: '#STAR.chimJunctionOverhangMin'
#&gt;   - id: '#STAR.alignWindowsPerReadNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerWindowNmax'
#&gt;   - id: '#STAR.alignTranscriptsPerReadNmax'
#&gt;   - id: '#STAR.alignSplicedMateMapLminOverLmate'
#&gt;   - id: '#STAR.alignSplicedMateMapLmin'
#&gt;   - id: '#STAR.alignSoftClipAtReferenceEnds'
#&gt;   - id: '#STAR.alignSJoverhangMin'
#&gt;   - id: '#STAR.alignSJDBoverhangMin'
#&gt;   - id: '#STAR.alignMatesGapMax'
#&gt;   - id: '#STAR.alignIntronMin'
#&gt;   - id: '#STAR.alignIntronMax'
#&gt;   - id: '#STAR.alignEndsType'
#&gt;   outputs:
#&gt;   - id: '#STAR.unmapped_reads'
#&gt;   - id: '#STAR.transcriptome_aligned_reads'
#&gt;   - id: '#STAR.splice_junctions'
#&gt;   - id: '#STAR.reads_per_gene'
#&gt;   - id: '#STAR.log_files'
#&gt;   - id: '#STAR.intermediate_genome'
#&gt;   - id: '#STAR.chimeric_junctions'
#&gt;   - id: '#STAR.chimeric_alignments'
#&gt;   - id: '#STAR.aligned_reads'
#&gt;   hints: []
#&gt;   run:
#&gt;     sbg:validationErrors: []
#&gt;     sbg:sbgMaintained: no
#&gt;     sbg:latestRevision: 4
#&gt;     sbg:job:
#&gt;       allocatedResources:
#&gt;         mem: 60000
#&gt;         cpu: 15
#&gt;       inputs:
#&gt;         alignWindowsPerReadNmax: 0
#&gt;         outSAMheaderPG: outSAMheaderPG
#&gt;         GENOME_DIR_NAME: ''
#&gt;         outFilterMatchNminOverLread: 0
#&gt;         rg_platform_unit_id: rg_platform_unit
#&gt;         alignTranscriptsPerReadNmax: 0
#&gt;         readMapNumber: 0
#&gt;         alignSplicedMateMapLminOverLmate: 0
#&gt;         alignMatesGapMax: 0
#&gt;         outFilterMultimapNmax: 0
#&gt;         clip5pNbases:
#&gt;         - 0
#&gt;         outSAMstrandField: None
#&gt;         readMatesLengthsIn: NotEqual
#&gt;         outSAMattributes: Standard
#&gt;         seedMultimapNmax: 0
#&gt;         rg_mfl: rg_mfl
#&gt;         chimSegmentMin: 0
#&gt;         winAnchorDistNbins: 0
#&gt;         outSortingType: SortedByCoordinate
#&gt;         outFilterMultimapScoreRange: 0
#&gt;         sjdbInsertSave: Basic
#&gt;         clip3pAfterAdapterNbases:
#&gt;         - 0
#&gt;         scoreDelBase: 0
#&gt;         outFilterMatchNmin: 0
#&gt;         twopass1readsN: 0
#&gt;         outSAMunmapped: None
#&gt;         genome:
#&gt;           size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           path: genome.ext
#&gt;         sjdbGTFtagExonParentTranscript: ''
#&gt;         limitBAMsortRAM: 0
#&gt;         alignEndsType: Local
#&gt;         seedNoneLociPerWindow: 0
#&gt;         rg_sample_id: rg_sample
#&gt;         sjdbGTFtagExonParentGene: ''
#&gt;         chimScoreMin: 0
#&gt;         outSJfilterIntronMaxVsReadN:
#&gt;         - 0
#&gt;         twopassMode: Basic
#&gt;         alignSplicedMateMapLmin: 0
#&gt;         outSJfilterReads: All
#&gt;         outSAMprimaryFlag: OneBestScore
#&gt;         outSJfilterCountTotalMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         outSAMorder: Paired
#&gt;         outSAMflagAND: 0
#&gt;         chimScoreSeparation: 0
#&gt;         alignSJoverhangMin: 0
#&gt;         outFilterScoreMin: 0
#&gt;         seedSearchStartLmax: 0
#&gt;         scoreGapGCAG: 0
#&gt;         scoreGenomicLengthLog2scale: 0
#&gt;         outFilterIntronMotifs: None
#&gt;         outFilterMismatchNmax: 0
#&gt;         reads:
#&gt;         - size: 0
#&gt;           secondaryFiles: []
#&gt;           class: File
#&gt;           metadata:
#&gt;             format: fastq
#&gt;             paired_end: '1'
#&gt;             seq_center: illumina
#&gt;           path: /test-data/mate_1.fastq.bz2
#&gt;         scoreGap: 0
#&gt;         outSJfilterOverhangMin:
#&gt;         - 30
#&gt;         - 12
#&gt;         - 12
#&gt;         - 12
#&gt;         outSAMflagOR: 0
#&gt;         outSAMmode: Full
#&gt;         rg_library_id: ''
#&gt;         chimScoreJunctionNonGTAG: 0
#&gt;         scoreInsOpen: 0
#&gt;         clip3pAdapterSeq:
#&gt;         - clip3pAdapterSeq
#&gt;         chimScoreDropMax: 0
#&gt;         outFilterType: Normal
#&gt;         scoreGapATAC: 0
#&gt;         rg_platform: Ion Torrent PGM
#&gt;         clip3pAdapterMMp:
#&gt;         - 0
#&gt;         sjdbGTFfeatureExon: ''
#&gt;         outQSconversionAdd: 0
#&gt;         quantMode: TranscriptomeSAM
#&gt;         alignIntronMin: 0
#&gt;         scoreInsBase: 0
#&gt;         scoreGapNoncan: 0
#&gt;         seedSearchLmax: 0
#&gt;         outSJfilterDistToOtherSJmin:
#&gt;         - 0
#&gt;         outFilterScoreMinOverLread: 0
#&gt;         alignSJDBoverhangMin: 0
#&gt;         limitOutSJcollapsed: 0
#&gt;         winAnchorMultimapNmax: 0
#&gt;         outFilterMismatchNoverLmax: 0
#&gt;         rg_seq_center: ''
#&gt;         outSAMheaderHD: outSAMheaderHD
#&gt;         chimOutType: Within
#&gt;         quantTranscriptomeBan: IndelSoftclipSingleend
#&gt;         limitOutSJoneRead: 0
#&gt;         alignTranscriptsPerWindowNmax: 0
#&gt;         sjdbOverhang: ~
#&gt;         outReadsUnmapped: Fastx
#&gt;         scoreStitchSJshift: 0
#&gt;         seedPerWindowNmax: 0
#&gt;         outSJfilterCountUniqueMin:
#&gt;         - 3
#&gt;         - 1
#&gt;         - 1
#&gt;         - 1
#&gt;         scoreDelOpen: 0
#&gt;         sjdbGTFfile:
#&gt;         - path: /demo/test-data/chr20.gtf
#&gt;         clip3pNbases:
#&gt;         - 0
#&gt;         - 3
#&gt;         winBinNbits: 0
#&gt;         sjdbScore: ~
#&gt;         seedSearchStartLmaxOverLread: 0
#&gt;         alignIntronMax: 0
#&gt;         seedPerReadNmax: 0
#&gt;         outFilterMismatchNoverReadLmax: 0
#&gt;         winFlankNbins: 0
#&gt;         sjdbGTFchrPrefix: chrPrefix
#&gt;         alignSoftClipAtReferenceEnds: 'Yes'
#&gt;         outSAMreadID: Standard
#&gt;         outSAMtype: BAM
#&gt;         chimJunctionOverhangMin: 0
#&gt;         limitSjdbInsertNsj: 0
#&gt;         outSAMmapqUnique: 0
#&gt;     sbg:toolAuthor: Alexander Dobin/CSHL
#&gt;     sbg:createdOn: 1450911471
#&gt;     sbg:categories:
#&gt;     - Alignment
#&gt;     sbg:contributors:
#&gt;     - ana_d
#&gt;     - bix-demo
#&gt;     - uros_sipetic
#&gt;     sbg:links:
#&gt;     - id: https://github.com/alexdobin/STAR
#&gt;       label: Homepage
#&gt;     - id: https://github.com/alexdobin/STAR/releases
#&gt;       label: Releases
#&gt;     - id: https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf
#&gt;       label: Manual
#&gt;     - id: https://groups.google.com/forum/#!forum/rna-star
#&gt;       label: Support
#&gt;     - id: http://www.ncbi.nlm.nih.gov/pubmed/23104886
#&gt;       label: Publication
#&gt;     sbg:project: bix-demo/star-2-4-2a-demo
#&gt;     sbg:createdBy: bix-demo
#&gt;     sbg:toolkitVersion: 2.4.2a
#&gt;     sbg:id: sevenbridges/public-apps/star/4
#&gt;     sbg:license: GNU General Public License v3.0 only
#&gt;     sbg:revision: 4
#&gt;     sbg:cmdPreview: tar -xvf genome.ext &amp;&amp; /opt/STAR --runThreadN 15  --readFilesCommand
#&gt;       bzcat  --sjdbGTFfile /demo/test-data/chr20.gtf  --sjdbGTFchrPrefix chrPrefix
#&gt;       --sjdbInsertSave Basic  --twopass1readsN 0  --chimOutType WithinBAM  --outSAMattrRGline
#&gt;       ID:1 CN:illumina PI:rg_mfl PL:Ion_Torrent_PGM PU:rg_platform_unit SM:rg_sample  --quantMode
#&gt;       TranscriptomeSAM --outFileNamePrefix ./mate_1.fastq.bz2.  --readFilesIn /test-data/mate_1.fastq.bz2  &amp;&amp;
#&gt;       tar -vcf mate_1.fastq.bz2._STARgenome.tar ./mate_1.fastq.bz2._STARgenome  &amp;&amp;
#&gt;       mv mate_1.fastq.bz2.Unmapped.out.mate1 mate_1.fastq.bz2.Unmapped.out.mate1.fastq
#&gt;     sbg:modifiedOn: 1462889222
#&gt;     sbg:modifiedBy: ana_d
#&gt;     sbg:revisionsInfo:
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911471
#&gt;       sbg:revision: 0
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911473
#&gt;       sbg:revision: 1
#&gt;     - sbg:modifiedBy: bix-demo
#&gt;       sbg:modifiedOn: 1450911475
#&gt;       sbg:revision: 2
#&gt;     - sbg:modifiedBy: uros_sipetic
#&gt;       sbg:modifiedOn: 1462878528
#&gt;       sbg:revision: 3
#&gt;     - sbg:modifiedBy: ana_d
#&gt;       sbg:modifiedOn: 1462889222
#&gt;       sbg:revision: 4
#&gt;     sbg:toolkit: STAR
#&gt;     id: sevenbridges/public-apps/star/4
#&gt;     inputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Flanking regions size
#&gt;       description: =log2(winFlank), where win Flank is the size of the left and right
#&gt;         flanking regions for each window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winFlankNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winFlankNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Bin size
#&gt;       description: =log2(winBin), where winBin is the size of the bin for the windows/clustering,
#&gt;         each window will occupy an integer number of bins (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winBinNbits'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winBinNbits
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:includeInPorts: yes
#&gt;       sbg:toolDefaultValue: '16'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max loci anchors
#&gt;       description: Max number of loci anchors are allowed to map to (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max bins between anchors
#&gt;       description: Max number of bins between two anchors that allows aggregation
#&gt;         of anchors into one window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#winAnchorDistNbins'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --winAnchorDistNbins
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Windows, Anchors, Binning
#&gt;       sbg:toolDefaultValue: '9'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: twopassMode
#&gt;         symbols:
#&gt;         - None
#&gt;         - Basic
#&gt;         type: enum
#&gt;       label: Two-pass mode
#&gt;       description: '2-pass mapping mode. None: 1-pass mapping; Basic: basic 2-pass
#&gt;         mapping, with all 1st pass junctions inserted into the genome indices on the
#&gt;         fly.'
#&gt;       streamable: no
#&gt;       id: '#twopassMode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --twopassMode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to process in 1st step
#&gt;       description: 'Number of reads to process for the 1st step. 0: 1-step only, no
#&gt;         2nd pass; use very large number to map all reads in the first step (int&gt;0).'
#&gt;       streamable: no
#&gt;       id: '#twopass1readsN'
#&gt;       sbg:category: 2-pass mapping
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Extra alignment score
#&gt;       description: Extra alignment score for alignments that cross database junctions.
#&gt;       streamable: no
#&gt;       id: '#sjdbScore'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: '"Overhang" length'
#&gt;       description: Length of the donor/acceptor sequence on each side of the junctions,
#&gt;         ideally = (mate_length - 1) (int &gt;= 0), if int = 0, splice junction database
#&gt;         is not used.
#&gt;       streamable: no
#&gt;       id: '#sjdbOverhang'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: sjdbInsertSave
#&gt;         symbols:
#&gt;         - Basic
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: Save junction files
#&gt;       description: 'Which files to save when sjdb junctions are inserted on the fly
#&gt;         at the mapping step. None: not saving files at all; Basic: only small junction/transcript
#&gt;         files; All: all files including big Genome, SA and SAindex. These files are
#&gt;         output as archive.'
#&gt;       streamable: no
#&gt;       id: '#sjdbInsertSave'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Exons' parents name
#&gt;       description: Tag name to be used as exons' transcript-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentTranscript'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: transcript_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Gene name
#&gt;       description: Tag name to be used as exons' gene-parents.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFtagExonParentGene'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: gene_id
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Splice junction file
#&gt;       description: Gene model annotations and/or known transcripts. No need to include
#&gt;         this input, except in case of using "on the fly" annotations.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfile'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: GTF, GFF, TXT
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Set exons feature
#&gt;       description: Feature type in GTF file to be used as exons for building transcripts.
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFfeatureExon'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: exon
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Chromosome names
#&gt;       description: Prefix for chromosome names in a GTF file (e.g. 'chr' for using
#&gt;         ENSMEBL annotations with UCSC geneomes).
#&gt;       streamable: no
#&gt;       id: '#sjdbGTFchrPrefix'
#&gt;       sbg:category: Splice junctions database
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Search start point normalized
#&gt;       description: seedSearchStartLmax normalized to read length (sum of mates' lengths
#&gt;         for paired-end reads).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmaxOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmaxOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1.0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Search start point
#&gt;       description: Defines the search start point through the read - the read is split
#&gt;         into pieces no longer than this value (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchStartLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchStartLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seed length
#&gt;       description: Defines the maximum length of the seeds, if =0 max seed length
#&gt;         is infinite (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedSearchLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedSearchLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per window
#&gt;       description: Max number of seeds per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '50'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max seeds per read
#&gt;       description: Max number of seeds per read (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max one-seed loci per window
#&gt;       description: Max number of one seed loci per window (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedNoneLociPerWindow'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedNoneLociPerWindow
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Filter pieces for stitching
#&gt;       description: Only pieces that map fewer than this value are utilized in the
#&gt;         stitching procedure (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#seedMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --seedMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max score reduction
#&gt;       description: Maximum score reduction while searching for SJ boundaries in the
#&gt;         stitching step.
#&gt;       streamable: no
#&gt;       id: '#scoreStitchSJshift'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreStitchSJshift
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion Open Penalty
#&gt;       description: Insertion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreInsOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Insertion extension penalty
#&gt;       description: Insertion extension penalty per base (in addition to --scoreInsOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreInsBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreInsBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Log scaled score
#&gt;       description: 'Extra score logarithmically scaled with genomic length of the
#&gt;         alignment: &lt;int&gt;*log2(genomicLength).'
#&gt;       streamable: no
#&gt;       id: '#scoreGenomicLengthLog2scale'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGenomicLengthLog2scale
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-0.25'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-canonical gap open
#&gt;       description: Non-canonical gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapNoncan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapNoncan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: GC/AG and CT/GC gap open
#&gt;       description: GC/AG and CT/GC gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapGCAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapGCAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-4'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AT/AC and GT/AT gap open
#&gt;       description: AT/AC and GT/AT gap open penalty (in addition to --scoreGap).
#&gt;       streamable: no
#&gt;       id: '#scoreGapATAC'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGapATAC
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-8'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Gap open penalty
#&gt;       description: Gap open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreGap'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreGap
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion open penalty
#&gt;       description: Deletion open penalty.
#&gt;       streamable: no
#&gt;       id: '#scoreDelOpen'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelOpen
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Deletion extension penalty
#&gt;       description: Deletion extension penalty per base (in addition to --scoreDelOpen).
#&gt;       streamable: no
#&gt;       id: '#scoreDelBase'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --scoreDelBase
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Scoring
#&gt;       sbg:toolDefaultValue: '-2'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sequencing center
#&gt;       description: Specify the sequencing center for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_seq_center'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Sample ID
#&gt;       description: Specify the sample ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_sample_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Platform unit ID
#&gt;       description: Specify the platform unit ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_platform_unit_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: rg_platform
#&gt;         symbols:
#&gt;         - LS 454
#&gt;         - Helicos
#&gt;         - Illumina
#&gt;         - ABI SOLiD
#&gt;         - Ion Torrent PGM
#&gt;         - PacBio
#&gt;         type: enum
#&gt;       label: Platform
#&gt;       description: Specify the version of the technology that was used for sequencing
#&gt;         or assaying.
#&gt;       streamable: no
#&gt;       id: '#rg_platform'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Median fragment length
#&gt;       description: Specify the median fragment length for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_mfl'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Library ID
#&gt;       description: Specify the library ID for RG line.
#&gt;       streamable: no
#&gt;       id: '#rg_library_id'
#&gt;       sbg:category: Read group
#&gt;       sbg:toolDefaultValue: Inferred from metadata
#&gt;       required: no
#&gt;     - type:
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Read sequence
#&gt;       description: Read sequence.
#&gt;       streamable: no
#&gt;       id: '#reads'
#&gt;       inputBinding:
#&gt;         position: 10
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: "{\t\n  var list = [].concat($job.inputs.reads)\n  \n  var resp
#&gt;             = []\n  \n  if (list.length == 1){\n    resp.push(list[0].path)\n    \n
#&gt;             \ }else if (list.length == 2){    \n    \n    left = \"\"\n    right =
#&gt;             \"\"\n      \n    for (index = 0; index &lt; list.length; ++index) {\n      \n
#&gt;             \     if (list[index].metadata != null){\n        if (list[index].metadata.paired_end
#&gt;             == 1){\n          left = list[index].path\n        }else if (list[index].metadata.paired_end
#&gt;             == 2){\n          right = list[index].path\n        }\n      }\n    }\n
#&gt;             \   \n    if (left != \"\" &amp;&amp; right != \"\"){      \n      resp.push(left)\n
#&gt;             \     resp.push(right)\n    }\n  }\n  else if (list.length &gt; 2){\n    left
#&gt;             = []\n    right = []\n      \n    for (index = 0; index &lt; list.length;
#&gt;             ++index) {\n      \n      if (list[index].metadata != null){\n        if
#&gt;             (list[index].metadata.paired_end == 1){\n          left.push(list[index].path)\n
#&gt;             \       }else if (list[index].metadata.paired_end == 2){\n          right.push(list[index].path)\n
#&gt;             \       }\n      }\n    }\n    left_join = left.join()\n    right_join
#&gt;             = right.join()\n    if (left != [] &amp;&amp; right != []){      \n      resp.push(left_join)\n
#&gt;             \     resp.push(right_join)\n    }\t\n  }\n  \n  if(resp.length &gt; 0){
#&gt;             \   \n    return \"--readFilesIn \".concat(resp.join(\" \"))\n  }\n}"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
#&gt;       sbg:x: 200.0
#&gt;       sbg:y: 200.0
#&gt;       sbg:includeInPorts: yes
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: readMatesLengthsIn
#&gt;         symbols:
#&gt;         - NotEqual
#&gt;         - Equal
#&gt;         type: enum
#&gt;       label: Reads lengths
#&gt;       description: Equal/Not equal - lengths of names, sequences, qualities for both
#&gt;         mates are the same/not the same. "Not equal" is safe in all situations.
#&gt;       streamable: no
#&gt;       id: '#readMatesLengthsIn'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMatesLengthsIn
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: NotEqual
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Reads to map
#&gt;       description: Number of reads to map from the beginning of the file.
#&gt;       streamable: no
#&gt;       id: '#readMapNumber'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --readMapNumber
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantTranscriptomeBan
#&gt;         symbols:
#&gt;         - IndelSoftclipSingleend
#&gt;         - Singleend
#&gt;         type: enum
#&gt;       label: Prohibit alignment type
#&gt;       description: 'Prohibit various alignment type. IndelSoftclipSingleend: prohibit
#&gt;         indels, soft clipping and single-end alignments - compatible with RSEM; Singleend:
#&gt;         prohibit single-end alignments.'
#&gt;       streamable: no
#&gt;       id: '#quantTranscriptomeBan'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --quantTranscriptomeBan
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: IndelSoftclipSingleend
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: quantMode
#&gt;         symbols:
#&gt;         - TranscriptomeSAM
#&gt;         - GeneCounts
#&gt;         type: enum
#&gt;       label: Quantification mode
#&gt;       description: Types of quantification requested. 'TranscriptomeSAM' option outputs
#&gt;         SAM/BAM alignments to transcriptome into a separate file. With 'GeneCounts'
#&gt;         option, STAR will count number of reads per gene while mapping.
#&gt;       streamable: no
#&gt;       id: '#quantMode'
#&gt;       sbg:category: Quantification of Annotations
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSortingType
#&gt;         symbols:
#&gt;         - Unsorted
#&gt;         - SortedByCoordinate
#&gt;         - Unsorted SortedByCoordinate
#&gt;         type: enum
#&gt;       label: Output sorting type
#&gt;       description: Type of output sorting.
#&gt;       streamable: no
#&gt;       id: '#outSortingType'
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SortedByCoordinate
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSJfilterReads
#&gt;         symbols:
#&gt;         - All
#&gt;         - Unique
#&gt;         type: enum
#&gt;       label: Collapsed junctions reads
#&gt;       description: 'Which reads to consider for collapsed splice junctions output.
#&gt;         All: all reads, unique- and multi-mappers; Unique: uniquely mapping reads
#&gt;         only.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterReads'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterReads
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: All
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min overhang SJ
#&gt;       description: Minimum overhang length for splice junctions on both sides for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterOverhangMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 30 12 12 12
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Max gap allowed
#&gt;       description: 'Maximum gap allowed for junctions supported by 1,2,3...N reads
#&gt;         (int &gt;= 0) i.e. by default junctions supported by 1 read can have gaps &lt;=50000b,
#&gt;         by 2 reads: &lt;=100000b, by 3 reads: &lt;=200000. By 4 or more reads: any gap &lt;=alignIntronMax.
#&gt;         Does not apply to annotated junctions.'
#&gt;       streamable: no
#&gt;       id: '#outSJfilterIntronMaxVsReadN'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterIntronMaxVsReadN
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 50000 100000 200000
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min distance to other donor/acceptor
#&gt;       description: Minimum allowed distance to other junctions' donor/acceptor for
#&gt;         each of the motifs (int &gt;= 0). Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterDistToOtherSJmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterDistToOtherSJmin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 10 0 5 10
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min unique count
#&gt;       description: Minimum uniquely mapping read count per junction for each of the
#&gt;         motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountUniqueMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountUniqueMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Min total count
#&gt;       description: Minimum total (multi-mapping+unique) read count per junction for
#&gt;         each of the motifs. To set no output for desired motif, assign -1 to the corresponding
#&gt;         field. Junctions are output if one of --outSJfilterCountUniqueMin OR --outSJfilterCountTotalMin
#&gt;         conditions are satisfied. Does not apply to annotated junctions.
#&gt;       streamable: no
#&gt;       id: '#outSJfilterCountTotalMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSJfilterCountTotalMin
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: 'Output filtering: splice junctions'
#&gt;       sbg:toolDefaultValue: 3 1 1 1
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMunmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Write unmapped in SAM
#&gt;       description: 'Output of unmapped reads in the SAM format. None: no output Within:
#&gt;         output unmapped reads within the main SAM file (i.e. Aligned.out.sam).'
#&gt;       streamable: no
#&gt;       id: '#outSAMunmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMunmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMtype
#&gt;         symbols:
#&gt;         - SAM
#&gt;         - BAM
#&gt;         type: enum
#&gt;       label: Output format
#&gt;       description: Format of output alignments.
#&gt;       streamable: no
#&gt;       id: '#outSAMtype'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               SAM_type = $job.inputs.outSAMtype
#&gt;               SORT_type = $job.inputs.outSortingType
#&gt;               if (SAM_type &amp;&amp; SORT_type) {
#&gt;                 return "--outSAMtype ".concat(SAM_type, " ", SORT_type)
#&gt;               }
#&gt;             }
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: SAM
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMstrandField
#&gt;         symbols:
#&gt;         - None
#&gt;         - intronMotif
#&gt;         type: enum
#&gt;       label: Strand field flag
#&gt;       description: 'Cufflinks-like strand field flag. None: not used; intronMotif:
#&gt;         strand derived from the intron motif. Reads with inconsistent and/or non-canonical
#&gt;         introns are filtered out.'
#&gt;       streamable: no
#&gt;       id: '#outSAMstrandField'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMstrandField
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMreadID
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - Number
#&gt;         type: enum
#&gt;       label: Read ID
#&gt;       description: 'Read ID record type. Standard: first word (until space) from the
#&gt;         FASTx read ID line, removing /1,/2 from the end; Number: read number (index)
#&gt;         in the FASTx file.'
#&gt;       streamable: no
#&gt;       id: '#outSAMreadID'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMreadID
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMprimaryFlag
#&gt;         symbols:
#&gt;         - OneBestScore
#&gt;         - AllBestScore
#&gt;         type: enum
#&gt;       label: Primary alignments
#&gt;       description: 'Which alignments are considered primary - all others will be marked
#&gt;         with 0x100 bit in the FLAG. OneBestScore: only one alignment with the best
#&gt;         score is primary; AllBestScore: all alignments with the best score are primary.'
#&gt;       streamable: no
#&gt;       id: '#outSAMprimaryFlag'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMprimaryFlag
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: OneBestScore
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMorder
#&gt;         symbols:
#&gt;         - Paired
#&gt;         - PairedKeepInputOrder
#&gt;         type: enum
#&gt;       label: Sorting in SAM
#&gt;       description: 'Type of sorting for the SAM output. Paired: one mate after the
#&gt;         other for all paired alignments; PairedKeepInputOrder: one mate after the
#&gt;         other for all paired alignments, the order is kept the same as in the input
#&gt;         FASTQ files.'
#&gt;       streamable: no
#&gt;       id: '#outSAMorder'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMorder
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Paired
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMmode
#&gt;         symbols:
#&gt;         - Full
#&gt;         - NoQS
#&gt;         type: enum
#&gt;       label: SAM mode
#&gt;       description: 'Mode of SAM output. Full: full SAM output; NoQS: full SAM but
#&gt;         without quality scores.'
#&gt;       streamable: no
#&gt;       id: '#outSAMmode'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmode
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Full
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: MAPQ value
#&gt;       description: MAPQ value for unique mappers (0 to 255).
#&gt;       streamable: no
#&gt;       id: '#outSAMmapqUnique'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMmapqUnique
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '255'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @PG
#&gt;       description: Extra @PG (software) line of the SAM header (in addition to STAR).
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderPG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderPG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: SAM header @HD
#&gt;       description: '@HD (header) line of the SAM header.'
#&gt;       streamable: no
#&gt;       id: '#outSAMheaderHD'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMheaderHD
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: OR SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagOR'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagOR
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: AND SAM flag
#&gt;       description: Set specific bits of the SAM FLAG.
#&gt;       streamable: no
#&gt;       id: '#outSAMflagAND'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMflagAND
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '65535'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outSAMattributes
#&gt;         symbols:
#&gt;         - Standard
#&gt;         - NH
#&gt;         - All
#&gt;         - None
#&gt;         type: enum
#&gt;       label: SAM attributes
#&gt;       description: 'Desired SAM attributes, in the order desired for the output SAM.
#&gt;         NH: any combination in any order; Standard: NH HI AS nM; All: NH HI AS nM
#&gt;         NM MD jM jI; None: no attributes.'
#&gt;       streamable: no
#&gt;       id: '#outSAMattributes'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outSAMattributes
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: Standard
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outReadsUnmapped
#&gt;         symbols:
#&gt;         - None
#&gt;         - Fastx
#&gt;         type: enum
#&gt;       label: Output unmapped reads
#&gt;       description: 'Output of unmapped reads (besides SAM). None: no output; Fastx:
#&gt;         output in separate fasta/fastq files, Unmapped.out.mate1/2.'
#&gt;       streamable: no
#&gt;       id: '#outReadsUnmapped'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outReadsUnmapped
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Quality conversion
#&gt;       description: Add this number to the quality score (e.g. to convert from Illumina
#&gt;         to Sanger, use -31).
#&gt;       streamable: no
#&gt;       id: '#outQSconversionAdd'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outQSconversionAdd
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterType
#&gt;         symbols:
#&gt;         - Normal
#&gt;         - BySJout
#&gt;         type: enum
#&gt;       label: Filtering type
#&gt;       description: 'Type of filtering. Normal: standard filtering using only current
#&gt;         alignment; BySJout: keep only those reads that contain junctions that passed
#&gt;         filtering into SJ.out.tab.'
#&gt;       streamable: no
#&gt;       id: '#outFilterType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: Normal
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min score normalized
#&gt;       description: '''Minimum score'' normalized to read length (sum of mates'' lengths
#&gt;         for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMinOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMinOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min score
#&gt;       description: Alignment will be output only if its score is higher than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Multimapping score range
#&gt;       description: The score range below the maximum score for multimapping alignments.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapScoreRange'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapScoreRange
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mappings
#&gt;       description: Read alignments will be output only if the read maps fewer than
#&gt;         this value, otherwise no alignments will be output.
#&gt;       streamable: no
#&gt;       id: '#outFilterMultimapNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMultimapNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *read* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *read*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverReadLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverReadLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Mismatches to *mapped* length
#&gt;       description: Alignment will be output only if its ratio of mismatches to *mapped*
#&gt;         length is less than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNoverLmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNoverLmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max number of mismatches
#&gt;       description: Alignment will be output only if it has fewer mismatches than this
#&gt;         value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMismatchNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMismatchNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min matched bases normalized
#&gt;       description: '''Minimum matched bases'' normalized to read length (sum of mates
#&gt;         lengths for paired-end reads).'
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNminOverLread'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNminOverLread
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min matched bases
#&gt;       description: Alignment will be output only if the number of matched bases is
#&gt;         higher than this value.
#&gt;       streamable: no
#&gt;       id: '#outFilterMatchNmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterMatchNmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: outFilterIntronMotifs
#&gt;         symbols:
#&gt;         - None
#&gt;         - RemoveNoncanonical
#&gt;         - RemoveNoncanonicalUnannotated
#&gt;         type: enum
#&gt;       label: Motifs filtering
#&gt;       description: 'Filter alignment using their motifs. None: no filtering; RemoveNoncanonical:
#&gt;         filter out alignments that contain non-canonical junctions; RemoveNoncanonicalUnannotated:
#&gt;         filter out alignments that contain non-canonical unannotated junctions when
#&gt;         using annotated splice junctions database. The annotated non-canonical junctions
#&gt;         will be kept.'
#&gt;       streamable: no
#&gt;       id: '#outFilterIntronMotifs'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --outFilterIntronMotifs
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Output filtering
#&gt;       sbg:toolDefaultValue: None
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max insert junctions
#&gt;       description: Maximum number of junction to be inserted to the genome on the
#&gt;         fly at the mapping stage, including those from annotations and those detected
#&gt;         in the 1st step of the 2-pass run.
#&gt;       streamable: no
#&gt;       id: '#limitSjdbInsertNsj'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitSjdbInsertNsj
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Junctions max number
#&gt;       description: Max number of junctions for one read (including all multi-mappers).
#&gt;       streamable: no
#&gt;       id: '#limitOutSJoneRead'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJoneRead
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Collapsed junctions max number
#&gt;       description: Max number of collapsed junctions.
#&gt;       streamable: no
#&gt;       id: '#limitOutSJcollapsed'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitOutSJcollapsed
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '1000000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Limit BAM sorting memory
#&gt;       description: Maximum available RAM for sorting BAM. If set to 0, it will be
#&gt;         set to the genome index size.
#&gt;       streamable: no
#&gt;       id: '#limitBAMsortRAM'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --limitBAMsortRAM
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Limits
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - string
#&gt;       label: Genome dir name
#&gt;       description: Name of the directory which contains genome files (when genome.tar
#&gt;         is uncompressed).
#&gt;       streamable: no
#&gt;       id: '#genomeDirName'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --genomeDir
#&gt;         separate: yes
#&gt;         valueFrom:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: $job.inputs.genomeDirName || "genomeDir"
#&gt;           class: Expression
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Basic
#&gt;       sbg:toolDefaultValue: genomeDir
#&gt;       required: no
#&gt;     - type:
#&gt;       - File
#&gt;       label: Genome files
#&gt;       description: Genome files created using STAR Genome Generate.
#&gt;       streamable: no
#&gt;       id: '#genome'
#&gt;       sbg:category: Basic
#&gt;       sbg:fileTypes: TAR
#&gt;       required: yes
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 5p bases
#&gt;       description: Number of bases to clip from 5p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip5pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip5pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p bases
#&gt;       description: Number of bases to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: int
#&gt;         type: array
#&gt;       label: Clip 3p after adapter seq.
#&gt;       description: Number of bases to clip from 3p of each mate after the adapter
#&gt;         clipping. In case only one value is given, it will be assumed the same for
#&gt;         both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAfterAdapterNbases'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAfterAdapterNbases
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: string
#&gt;         type: array
#&gt;       label: Clip 3p adapter sequence
#&gt;       description: Adapter sequence to clip from 3p of each mate. In case only one
#&gt;         value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterSeq'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterSeq
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '-'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: float
#&gt;         type: array
#&gt;       label: Max mismatches proportions
#&gt;       description: Max proportion of mismatches for 3p adapter clipping for each mate.
#&gt;         In case only one value is given, it will be assumed the same for both mates.
#&gt;       streamable: no
#&gt;       id: '#clip3pAdapterMMp'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --clip3pAdapterMMp
#&gt;         separate: yes
#&gt;         itemSeparator: ' '
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Read parameters
#&gt;       sbg:toolDefaultValue: '0.1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min segment length
#&gt;       description: Minimum length of chimeric segment length, if =0, no chimeric output
#&gt;         (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimSegmentMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimSegmentMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '15'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min separation score
#&gt;       description: Minimum difference (separation) between the best chimeric score
#&gt;         and the next one (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreSeparation'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreSeparation
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '10'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min total score
#&gt;       description: Minimum total (summed) score of the chimeric segments (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Non-GT/AG penalty
#&gt;       description: Penalty for a non-GT/AG chimeric junction.
#&gt;       streamable: no
#&gt;       id: '#chimScoreJunctionNonGTAG'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreJunctionNonGTAG
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '-1'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max drop score
#&gt;       description: Max drop (difference) of chimeric score (the sum of scores of all
#&gt;         chimeric segements) from the read length (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimScoreDropMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimScoreDropMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: chimOutType
#&gt;         symbols:
#&gt;         - SeparateSAMold
#&gt;         - Within
#&gt;         type: enum
#&gt;       label: Chimeric output type
#&gt;       description: 'Type of chimeric output. SeparateSAMold: output old SAM into separate
#&gt;         Chimeric.out.sam file; Within: output into main aligned SAM/BAM files.'
#&gt;       streamable: no
#&gt;       id: '#chimOutType'
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: SeparateSAMold
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min junction overhang
#&gt;       description: Minimum overhang for a chimeric junction (int&gt;=0).
#&gt;       streamable: no
#&gt;       id: '#chimJunctionOverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --chimJunctionOverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Chimeric Alignments
#&gt;       sbg:toolDefaultValue: '20'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Max windows per read
#&gt;       description: Max number of windows per read (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignWindowsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignWindowsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per window
#&gt;       description: Max number of transcripts per window (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerWindowNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerWindowNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '100'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max transcripts per read
#&gt;       description: Max number of different alignments per read to consider (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignTranscriptsPerReadNmax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignTranscriptsPerReadNmax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '10000'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - float
#&gt;       label: Min mapped length normalized
#&gt;       description: alignSplicedMateMapLmin normalized to mate length (float&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLminOverLmate'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLminOverLmate
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0.66'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min mapped length
#&gt;       description: Minimum mapped length for a read mate that is spliced (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSplicedMateMapLmin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSplicedMateMapLmin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignSoftClipAtReferenceEnds
#&gt;         symbols:
#&gt;         - 'Yes'
#&gt;         - 'No'
#&gt;         type: enum
#&gt;       label: Soft clipping
#&gt;       description: 'Option which allows soft clipping of alignments at the reference
#&gt;         (chromosome) ends. Can be disabled for compatibility with Cufflinks/Cuffmerge.
#&gt;         Yes: Enables soft clipping; No: Disables soft clipping.'
#&gt;       streamable: no
#&gt;       id: '#alignSoftClipAtReferenceEnds'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSoftClipAtReferenceEnds
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: 'Yes'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min overhang
#&gt;       description: Minimum overhang (i.e. block size) for spliced alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '5'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: 'Min overhang: annotated'
#&gt;       description: Minimum overhang (i.e. block size) for annotated (sjdb) spliced
#&gt;         alignments (int&gt;0).
#&gt;       streamable: no
#&gt;       id: '#alignSJDBoverhangMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignSJDBoverhangMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:includeInPorts: no
#&gt;       sbg:toolDefaultValue: '3'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max mates gap
#&gt;       description: Maximum gap between two mates, if 0, max intron gap will be determined
#&gt;         by (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignMatesGapMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignMatesGapMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Min intron size
#&gt;       description: 'Minimum intron size: genomic gap is considered intron if its length
#&gt;         &gt;= alignIntronMin, otherwise it is considered Deletion (int&gt;=0).'
#&gt;       streamable: no
#&gt;       id: '#alignIntronMin'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMin
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '21'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - int
#&gt;       label: Max intron size
#&gt;       description: Maximum intron size, if 0, max intron size will be determined by
#&gt;         (2^winBinNbits)*winAnchorDistNbins.
#&gt;       streamable: no
#&gt;       id: '#alignIntronMax'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignIntronMax
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: '0'
#&gt;       required: no
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - name: alignEndsType
#&gt;         symbols:
#&gt;         - Local
#&gt;         - EndToEnd
#&gt;         type: enum
#&gt;       label: Alignment type
#&gt;       description: 'Type of read ends alignment. Local: standard local alignment with
#&gt;         soft-clipping allowed. EndToEnd: force end to end read alignment, do not soft-clip.'
#&gt;       streamable: no
#&gt;       id: '#alignEndsType'
#&gt;       inputBinding:
#&gt;         position: 0
#&gt;         prefix: --alignEndsType
#&gt;         separate: yes
#&gt;         shellQuote: no
#&gt;         sbg:cmdInclude: yes
#&gt;         streamable: no
#&gt;         separator: ' '
#&gt;       sbg:category: Alignments and Seeding
#&gt;       sbg:toolDefaultValue: Local
#&gt;       required: no
#&gt;     outputs:
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Unmapped reads
#&gt;       description: Output of unmapped reads.
#&gt;       streamable: no
#&gt;       id: '#unmapped_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Unmapped.out*'
#&gt;       sbg:fileTypes: FASTQ
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Transcriptome alignments
#&gt;       description: Alignments translated into transcript coordinates.
#&gt;       streamable: no
#&gt;       id: '#transcriptome_aligned_reads'
#&gt;       outputBinding:
#&gt;         glob: '*Transcriptome*'
#&gt;       sbg:fileTypes: BAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Splice junctions
#&gt;       description: High confidence collapsed splice junctions in tab-delimited format.
#&gt;         Only junctions supported by uniquely mapping reads are reported.
#&gt;       streamable: no
#&gt;       id: '#splice_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*SJ.out.tab'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Reads per gene
#&gt;       description: File with number of reads per gene. A read is counted if it overlaps
#&gt;         (1nt or more) one and only one gene.
#&gt;       streamable: no
#&gt;       id: '#reads_per_gene'
#&gt;       outputBinding:
#&gt;         glob: '*ReadsPerGene*'
#&gt;       sbg:fileTypes: TAB
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - items: File
#&gt;         type: array
#&gt;       label: Log files
#&gt;       description: Log files produced during alignment.
#&gt;       streamable: no
#&gt;       id: '#log_files'
#&gt;       outputBinding:
#&gt;         glob: '*Log*.out'
#&gt;       sbg:fileTypes: OUT
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Intermediate genome files
#&gt;       description: Archive with genome files produced when annotations are included
#&gt;         on the fly (in the mapping step).
#&gt;       streamable: no
#&gt;       id: '#intermediate_genome'
#&gt;       outputBinding:
#&gt;         glob: '*_STARgenome.tar'
#&gt;       sbg:fileTypes: TAR
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric junctions
#&gt;       description: If chimSegmentMin in 'Chimeric Alignments' section is set to 0,
#&gt;         'Chimeric Junctions' won't be output.
#&gt;       streamable: no
#&gt;       id: '#chimeric_junctions'
#&gt;       outputBinding:
#&gt;         glob: '*Chimeric.out.junction'
#&gt;       sbg:fileTypes: JUNCTION
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Chimeric alignments
#&gt;       description: Aligned Chimeric sequences SAM - if chimSegmentMin = 0, no Chimeric
#&gt;         Alignment SAM and Chimeric Junctions outputs.
#&gt;       streamable: no
#&gt;       id: '#chimeric_alignments'
#&gt;       outputBinding:
#&gt;         glob: '*.Chimeric.out.sam'
#&gt;       sbg:fileTypes: SAM
#&gt;     - type:
#&gt;       - 'null'
#&gt;       - File
#&gt;       label: Aligned SAM/BAM
#&gt;       description: Aligned sequence in SAM/BAM format.
#&gt;       streamable: no
#&gt;       id: '#aligned_reads'
#&gt;       outputBinding:
#&gt;         glob:
#&gt;           engine: '#cwl-js-engine'
#&gt;           script: |-
#&gt;             {
#&gt;               if ($job.inputs.outSortingType == 'SortedByCoordinate') {
#&gt;                 sort_name = '.sortedByCoord'
#&gt;               }
#&gt;               else {
#&gt;                 sort_name = ''
#&gt;               }
#&gt;               if ($job.inputs.outSAMtype == 'BAM') {
#&gt;                 sam_name = "*.Aligned".concat( sort_name, '.out.bam')
#&gt;               }
#&gt;               else {
#&gt;                 sam_name = "*.Aligned.out.sam"
#&gt;               }
#&gt;               return sam_name
#&gt;             }
#&gt;           class: Expression
#&gt;       sbg:fileTypes: SAM, BAM
#&gt;     requirements:
#&gt;     - class: ExpressionEngineRequirement
#&gt;       id: '#cwl-js-engine'
#&gt;       requirements:
#&gt;       - class: DockerRequirement
#&gt;         dockerPull: rabix/js-engine
#&gt;     hints:
#&gt;     - class: DockerRequirement
#&gt;       dockerPull: images.sbgenomics.com/ana_d/star:2.4.2a
#&gt;       dockerImageId: a4b0ad2c3cae
#&gt;     - class: sbg:MemRequirement
#&gt;       value: 60000
#&gt;     - class: sbg:CPURequirement
#&gt;       value: 15
#&gt;     label: STAR
#&gt;     description: STAR is an ultrafast universal RNA-seq aligner. It has very high
#&gt;       mapping speed, accurate alignment of contiguous and spliced reads, detection
#&gt;       of polyA-tails, non-canonical splices and chimeric (fusion) junctions. It works
#&gt;       with reads starting from lengths ~15 bases up to ~300 bases. In case of having
#&gt;       longer reads, use of STAR Long is recommended.
#&gt;     class: CommandLineTool
#&gt;     baseCommand:
#&gt;     - tar
#&gt;     - -xvf
#&gt;     - class: Expression
#&gt;       script: $job.inputs.genome.path
#&gt;       engine: '#cwl-js-engine'
#&gt;     - '&amp;&amp;'
#&gt;     - /opt/STAR
#&gt;     - --runThreadN
#&gt;     - class: Expression
#&gt;       script: |-
#&gt;         {
#&gt;           return $job.allocatedResources.cpu
#&gt;         }
#&gt;       engine: '#cwl-js-engine'
#&gt;     arguments:
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             file = [].concat($job.inputs.reads)[0].path
#&gt;             extension = /(?:\.([^.]+))?$/.exec(file)[1]
#&gt;             if (extension == "gz") {
#&gt;               return "--readFilesCommand zcat"
#&gt;             } else if (extension == "bz2") {
#&gt;               return "--readFilesCommand bzcat"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\t\n  var sjFormat = \"False\"\n  var gtfgffFormat = \"False\"\n
#&gt;           \ var list = $job.inputs.sjdbGTFfile\n  var paths_list = []\n  var joined_paths
#&gt;           = \"\"\n  \n  if (list) {\n    list.forEach(function(f){return paths_list.push(f.path)})\n
#&gt;           \   joined_paths = paths_list.join(\" \")\n\n\n    paths_list.forEach(function(f){\n
#&gt;           \     ext = f.replace(/^.*\\./, '')\n      if (ext == \"gff\" || ext ==
#&gt;           \"gtf\") {\n        gtfgffFormat = \"True\"\n        return gtfgffFormat\n
#&gt;           \     }\n      if (ext == \"txt\") {\n        sjFormat = \"True\"\n        return
#&gt;           sjFormat\n      }\n    })\n\n    if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.sjdbInsertSave
#&gt;           != \"None\") {\n      if (sjFormat == \"True\") {\n        return \"--sjdbFileChrStartEnd
#&gt;           \".concat(joined_paths)\n      }\n      else if (gtfgffFormat == \"True\")
#&gt;           {\n        return \"--sjdbGTFfile \".concat(joined_paths)\n      }\n    }\n
#&gt;           \ }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  a = b = c = d = e = f = g = []\n  if ($job.inputs.sjdbGTFchrPrefix)
#&gt;           {\n    a = [\"--sjdbGTFchrPrefix\", $job.inputs.sjdbGTFchrPrefix]\n  }\n
#&gt;           \ if ($job.inputs.sjdbGTFfeatureExon) {\n    b = [\"--sjdbGTFfeatureExon\",
#&gt;           $job.inputs.sjdbGTFfeatureExon]\n  }\n  if ($job.inputs.sjdbGTFtagExonParentTranscript)
#&gt;           {\n    c = [\"--sjdbGTFtagExonParentTranscript\", $job.inputs.sjdbGTFtagExonParentTranscript]\n
#&gt;           \ }\n  if ($job.inputs.sjdbGTFtagExonParentGene) {\n    d = [\"--sjdbGTFtagExonParentGene\",
#&gt;           $job.inputs.sjdbGTFtagExonParentGene]\n  }\n  if ($job.inputs.sjdbOverhang)
#&gt;           {\n    e = [\"--sjdbOverhang\", $job.inputs.sjdbOverhang]\n  }\n  if ($job.inputs.sjdbScore)
#&gt;           {\n    f = [\"--sjdbScore\", $job.inputs.sjdbScore]\n  }\n  if ($job.inputs.sjdbInsertSave)
#&gt;           {\n    g = [\"--sjdbInsertSave\", $job.inputs.sjdbInsertSave]\n  }\n  \n
#&gt;           \ \n  \n  if ($job.inputs.sjdbInsertSave != \"None\" &amp;&amp; $job.inputs.sjdbGTFfile)
#&gt;           {\n    new_list = a.concat(b, c, d, e, f, g)\n    return new_list.join(\"
#&gt;           \")\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.twopassMode == "Basic") {
#&gt;               return "--twopass1readsN ".concat($job.inputs.twopass1readsN)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.chimOutType == "Within") {
#&gt;               return "--chimOutType ".concat("Within", $job.inputs.outSAMtype)
#&gt;             }
#&gt;             else {
#&gt;               return "--chimOutType SeparateSAMold"
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  var param_list = []\n  \n  function add_param(key, value){\n
#&gt;           \   if (value == \"\") {\n      return\n    }\n    else {\n      return
#&gt;           param_list.push(key.concat(\":\", value))\n    }\n  }\n  \n  add_param('ID',
#&gt;           \"1\")\n  if ($job.inputs.rg_seq_center) {\n    add_param('CN', $job.inputs.rg_seq_center)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.seq_center) {\n    add_param('CN',
#&gt;           [].concat($job.inputs.reads)[0].metadata.seq_center)\n  }\n  if ($job.inputs.rg_library_id)
#&gt;           {\n    add_param('LB', $job.inputs.rg_library_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.library_id)
#&gt;           {\n    add_param('LB', [].concat($job.inputs.reads)[0].metadata.library_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_mfl) {\n    add_param('PI', $job.inputs.rg_mfl)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.median_fragment_length)
#&gt;           {\n    add_param('PI', [].concat($job.inputs.reads)[0].metadata.median_fragment_length)\n
#&gt;           \ }\n  if ($job.inputs.rg_platform) {\n    add_param('PL', $job.inputs.rg_platform.replace(/
#&gt;           /g,\"_\"))\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform)
#&gt;           {\n    add_param('PL', [].concat($job.inputs.reads)[0].metadata.platform.replace(/
#&gt;           /g,\"_\"))\n  }\n  if ($job.inputs.rg_platform_unit_id) {\n    add_param('PU',
#&gt;           $job.inputs.rg_platform_unit_id)\n  } else if ([].concat($job.inputs.reads)[0].metadata.platform_unit_id)
#&gt;           {\n    add_param('PU', [].concat($job.inputs.reads)[0].metadata.platform_unit_id)\n
#&gt;           \ }\n  if ($job.inputs.rg_sample_id) {\n    add_param('SM', $job.inputs.rg_sample_id)\n
#&gt;           \ } else if ([].concat($job.inputs.reads)[0].metadata.sample_id) {\n    add_param('SM',
#&gt;           [].concat($job.inputs.reads)[0].metadata.sample_id)\n  }\n  return \"--outSAMattrRGline
#&gt;           \".concat(param_list.join(\" \"))\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: |-
#&gt;           {
#&gt;             if ($job.inputs.sjdbGTFfile &amp;&amp; $job.inputs.quantMode) {
#&gt;               return "--quantMode ".concat($job.inputs.quantMode)
#&gt;             }
#&gt;           }
#&gt;         class: Expression
#&gt;     - position: 100
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  intermediate = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\"._STARgenome\")\n  source = \"./\".concat(intermediate)\n
#&gt;           \ destination = intermediate.concat(\".tar\")\n  if ($job.inputs.sjdbGTFfile
#&gt;           &amp;&amp; $job.inputs.sjdbInsertSave &amp;&amp; $job.inputs.sjdbInsertSave != \"None\")
#&gt;           {\n    return \"&amp;&amp; tar -vcf \".concat(destination, \" \", source)\n  }\n}"
#&gt;         class: Expression
#&gt;     - position: 0
#&gt;       prefix: --outFileNamePrefix
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  return \"./\".concat(common_prefix.replace(
#&gt;           /\\-$|\\_$|\\.$/, '' ), \".\")\n}"
#&gt;         class: Expression
#&gt;     - position: 101
#&gt;       separate: yes
#&gt;       valueFrom:
#&gt;         engine: '#cwl-js-engine'
#&gt;         script: "{\n  function sharedStart(array){\n  var A= array.concat().sort(),
#&gt;           \n      a1= A[0], a2= A[A.length-1], L= a1.length, i= 0;\n  while(i&lt;L &amp;&amp;
#&gt;           a1.charAt(i)=== a2.charAt(i)) i++;\n  return a1.substring(0, i);\n  }\n
#&gt;           \ path_list = []\n  arr = [].concat($job.inputs.reads)\n  arr.forEach(function(f){return
#&gt;           path_list.push(f.path.replace(/\\\\/g,'/').replace( /.*\\//, '' ))})\n  common_prefix
#&gt;           = sharedStart(path_list)\n  mate1 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate1\")\n  mate2 = common_prefix.replace( /\\-$|\\_$|\\.$/,
#&gt;           '' ).concat(\".Unmapped.out.mate2\")\n  mate1fq = mate1.concat(\".fastq\")\n
#&gt;           \ mate2fq = mate2.concat(\".fastq\")\n  if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length &gt; 1) {\n    return \"&amp;&amp; mv \".concat(mate1, \"
#&gt;           \", mate1fq, \" &amp;&amp; mv \", mate2, \" \", mate2fq)\n  }\n  else if ($job.inputs.outReadsUnmapped
#&gt;           == \"Fastx\" &amp;&amp; arr.length == 1) {\n    return \"&amp;&amp; mv \".concat(mate1,
#&gt;           \" \", mate1fq)\n  }\n}"
#&gt;         class: Expression
#&gt;     stdin: ''
#&gt;     stdout: ''
#&gt;     successCodes: []
#&gt;     temporaryFailCodes: []
#&gt;     x: 624.0
#&gt;     'y': 323
#&gt;   sbg:x: 700.0
#&gt;   sbg:y: 200.0
#&gt; sbg:canvas_zoom: 0.6
#&gt; sbg:canvas_y: -16
#&gt; sbg:canvas_x: -41
#&gt; sbg:batchInput: '#sjdbGTFfile'
#&gt; sbg:batchBy:
#&gt;   type: item
#&gt; </div><div class='input'><span class='va'>f1</span><span class='op'>$</span><span class='fu'>link_map</span><span class='op'>(</span><span class='op'>)</span>
</div><div class='output co'>#&gt;                                                      id
#&gt; 1  #STAR_Genome_Generate.sjdbGTFtagExonParentTranscript
#&gt; 2        #STAR_Genome_Generate.sjdbGTFtagExonParentGene
#&gt; 3                     #STAR_Genome_Generate.sjdbGTFfile
#&gt; 4                #STAR_Genome_Generate.genomeFastaFiles
#&gt; 5                     #SBG_FASTQ_Quality_Detector.fastq
#&gt; 6                             #Picard_SortSam.input_bam
#&gt; 7                           #STAR.winAnchorMultimapNmax
#&gt; 8                              #STAR.winAnchorDistNbins
#&gt; 9                                     #STAR.sjdbGTFfile
#&gt; 10                                          #STAR.reads
#&gt; 11                                          #STAR.reads
#&gt; 12                                         #STAR.genome
#&gt; 13                                         #STAR.genome
#&gt; 14                                      #unmapped_reads
#&gt; 15                         #transcriptome_aligned_reads
#&gt; 16                                    #splice_junctions
#&gt; 17                                      #reads_per_gene
#&gt; 18                                           #log_files
#&gt; 19                                  #chimeric_junctions
#&gt; 20                                 #intermediate_genome
#&gt; 21                                 #intermediate_genome
#&gt; 22                                 #chimeric_alignments
#&gt; 23                                          #sorted_bam
#&gt; 24                                              #result
#&gt; 25                                 #intermediate_genome
#&gt; 26                                 #intermediate_genome
#&gt;                             source   type
#&gt; 1  #sjdbGTFtagExonParentTranscript  input
#&gt; 2        #sjdbGTFtagExonParentGene  input
#&gt; 3                     #sjdbGTFfile  input
#&gt; 4                #genomeFastaFiles  input
#&gt; 5                           #fastq  input
#&gt; 6              #STAR.aligned_reads  input
#&gt; 7           #winAnchorMultimapNmax  input
#&gt; 8              #winAnchorDistNbins  input
#&gt; 9                     #sjdbGTFfile  input
#&gt; 10                          #reads  input
#&gt; 11                           test2  input
#&gt; 12    #STAR_Genome_Generate.genome  input
#&gt; 13                           test1  input
#&gt; 14                 #unmapped_reads output
#&gt; 15    #transcriptome_aligned_reads output
#&gt; 16               #splice_junctions output
#&gt; 17                 #reads_per_gene output
#&gt; 18                      #log_files output
#&gt; 19             #chimeric_junctions output
#&gt; 20            #intermediate_genome output
#&gt; 21             intermediate_genome output
#&gt; 22            #chimeric_alignments output
#&gt; 23                     #sorted_bam output
#&gt; 24                         #result output
#&gt; 25            #intermediate_genome output
#&gt; 26             intermediate_genome output</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Soner Koc, Nan Xiao, Tengfei Yin, Seven Bridges Genomics.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


